---
title: "code_nx_source_code_figs_v10"
output: html_document
date: "2024-04-21"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r}
#load packages
install.packages("pacman")

if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install("ComplexHeatmap", force = TRUE)

library(pacman)
pacman::p_load(data.table,tidyverse,stringr, tools, smoothmest, ggforce,directlabels,RColorBrewer, Rmisc, tools, lemon, gridExtra, ggbeeswarm, raster, writexl,ggpubr, RColorBrewer, UpSetR, grid, circlize, ComplexUpset, Hmisc,corrplot, heatmaply,devtools, sjmisc, patchwork,ggalt,janitor,gplots, fuzzyjoin, R.utils, TreeTools, TreeDist, word2vec, ComplexHeatmap,scales,httr,drawProteins,ggrepel, plotrix, rstatix,ggtext, dendextend,dendsort,ggh4x)

```

# data dirs
```{r}
inputdir = "~/Documents/EMBL/Nx/Nx_paper/input/"
plotdir =  "~/Documents/EMBL/Nx/Nx_paper/plots/"
outputdir = "~/Documents/EMBL/Nx/Nx_paper/output/"

sourcedatadir = "~/Documents/EMBL/Nx/Nx_paper/source_data/"
supptablesdir = "~/Documents/EMBL/Nx/Nx_paper/SuppTables/"
```

#plotting function 2DTPP

```{r}
plot2dTppFcHeatmap <- function(df, name, 
                               drug_name = "",
                               midpoint = 1){
  clustername <- temperature <- conc <- 
    rel_value <- fc <- NULL
  
  heat_df <- df %>% 
    dplyr::filter(clustername == name) %>% 
    dplyr::select(clustername, temperature, 
                  conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(
      temperature, levels = rev(sort(unique(temperature)))),
      conc = as.factor(conc)) %>% 
    dplyr::group_by(clustername, temperature, conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup()
  
  ggplot(heat_df, aes(conc, temperature)) +
    geom_tile(aes(fill = fc)) +
    scale_fill_gradient2(
      "Fold\nchange", 
      low = "firebrick", mid = "khaki", 
      high = "darkgreen", midpoint = midpoint) +
    labs(x = paste(drug_name, "(µg/mL)"),
         y = expression("Temperature ("*~degree*C*")")) +
    ggtitle(name) +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5),
          axis.text = element_text(size = 10),
          axis.title = element_text(size = 12),
          legend.key.size = unit(1,"line"),
          legend.title = element_text(size=10),
          legend.text = element_text(size=10))
}

capitalize_first_and_last <- function(string) {
  first_letter <- toupper(substring(string, 1, 1))
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string) - 1))
  return(paste(first_letter, middle_letters, last_letter, sep = ""))
}


```

## mean MIC to write in text

### per species
```{r}
dilut = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  mutate(Method=rep("Microdilution"), nrow(.)) %>%
  pivot_longer(cols = c(3:13),names_to = "MIC",values_to = "n_strains") %>%
  group_by(MIC) %>%
  summarise(totstrain=sum(n_strains)) %>%
  ungroup() %>%
  mutate(MIC = as.numeric(MIC),
         micstrain = MIC*totstrain,
         meanmic = sum(micstrain/sum(totstrain)))
  
disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(Species != "NA") %>%
  mutate(Method=rep("Disk diffusion"), nrow(.)) %>%
  pivot_longer(cols = c(3:27),names_to = "MIC",values_to = "n_strains") %>%
  group_by(MIC) %>%
  summarise(totstrain=sum(n_strains)) %>%
  ungroup() %>%
  mutate(MIC = as.numeric(MIC),
         micstrain = MIC*totstrain,
         meanmic = sum(micstrain/sum(totstrain)))


agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(strain != "NA") %>%
  mutate(Method=rep("Agar dilution"), nrow(.)) %>%
  pivot_longer(cols = c(3:9),names_to = "MIC",values_to = "n_strains") %>%
  group_by(MIC) %>%
  summarise(totstrain=sum(n_strains)) %>%
  ungroup() %>%
  mutate(MIC = as.numeric(MIC),
         micstrain = MIC*totstrain,
         meanmic = sum(micstrain/sum(totstrain)))

sum(agar$totstrain)

mean(agar$meanmic)


### check which families have lower MIC than e. coli


library(forcats)
agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
  dplyr::mutate(Species = fct_relevel(Species, rev(species_order))) %>%
  pivot_longer(cols=3:13,names_to = "MIC", values_to = "no_strains") %>%
  mutate(MIC=as.numeric(MIC))
          
toplot <-  agar %>%
  rowwise() %>%
  mutate(repeated_values = list(rep(MIC, no_strains))) %>%
  unnest(repeated_values)


# median in broth
toplot %>% # group_by(Species) %>%
  summarise(meanMIC=mean(MIC),
            medianMIC=median(MIC))

# median per species in broth
toplot= toplot %>% group_by(Species) %>%
  summarise(meanMIC=mean(MIC),
            medianMIC=median(MIC))


#median in agar
agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
  filter(strain != "NA") %>%
  mutate(Method=rep("Agar dilution"), nrow(.)) %>%
  pivot_longer(cols = c(3:9),names_to = "MIC",values_to = "no_strains") %>%
  mutate(MIC=as.numeric(MIC))
          
toplot <-  agar %>%
  rowwise() %>%
  mutate(repeated_values = list(rep(MIC, no_strains))) %>%
  unnest(repeated_values)

toplot %>% # group_by(Species) %>%
  summarise(meanMIC=mean(MIC),
            medianMIC=median(MIC))

```

### per family

```{r}
agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023.xlsx"), sheet = 1)  %>%
  pivot_longer(cols=3:13,names_to = "MIC", values_to = "no_strains") %>%
  mutate(MIC=as.numeric(MIC))

toplot <-  agar %>%
  rowwise() %>%
  mutate(repeated_values = list(rep(MIC, no_strains))) %>%
  unnest(repeated_values)

test = toplot %>%filter(Species == "Klebsiella pneumoniae")
mean(test$MIC)

ggboxplot(test$repeated_values, add="jitter")

# get average value MIC to cite in text
View(toplot %>% group_by(Species) %>%
  summarise(meanMIC=mean(MIC),
            medianmic=median(MIC)))

toplot %>%summarise(meanMIC=mean(MIC))

# per family
toplot=toplot %>%
  mutate(family = case_when(Species %in% c("Neisseria gonorrhoeae") ~ "Neisseriaceae",
            Species %in% c("Achromobacter xylosoxidans","Burkholderia cenocepacia") ~ "Burkholderiaceae",
            Species %in% c("Stenotrophomonas maltophilia") ~ "Xantomonadaceae",
            Species %in% c("Pseudomonas aeruginosa","Pseudomonas putida") ~ "Pseudomonadaceae",
            Species %in% c("Acinetobacter baumannii","Acinetobacter johnsonii", "Acinetobacter junii", "Acinetobacter lwoffii","Acinetobacter pittii","Acinetobacter ursingii","Moraxella catarrhalis","Moraxella osloensis") ~ "Moraxellaceae",
            TRUE ~ "Enterobacteriaceae"))

View(toplot %>% group_by(family) %>%
  summarise(meanMIC=mean(MIC),
            medianmic= median(MIC)))


```

## Fig. 1b - how many species across 3 methods and in EUCAST vs us

EUCAST has this file with only 5 species! https://www.eucast.org/mic_and_zone_distributions_and_ecoffs
In the end used the full distributions, also available - we overcome them anyway.


```{r}
# file with full distributions
eucast = readxl::read_xlsx(paste0(inputdir,"EUCAST_MIC.xlsx"), sheet = 1) %>%
  #remove Gram+
  filter(!(Organism %in% c("Staphylococcus aureus", "Staphylococcus epidermidis","Staphylococcus saprophyticus","Coagulase-negative staphylococci", "Enterococcus faecalis","Enterococcus faecium"))) %>%
    dplyr::mutate(Species = case_when(#Organism %in% c("Proteus mirabilis", "Proteus vulgaris", "Proteus spp. indole-positive") ~ "Proteus spp.",
                                      Organism %in% c("Acinetobacter baumannii", "Acinetobacter beijerinckii", "Acinetobacter calcoaceticus","Acinetobacter guillouiae","Acinetobacter haemolyticus","Acinetobacter johnsonnii","Acinetobacter junii","Acinetobacter lwoffi","Acinetobacter pittii","Acinetobacter radioresistens","Acinetobacter septicus", "Acinetobacter ursingii") ~ "Acinetobacter spp.",
                                      #Organism =="Enterobacter cloacae" ~ "Enterobacter spp.",
                                      TRUE ~ Organism),
                  Method = rep("EUCAST",nrow(.))) %>%
  distinct(Species,Method)
  
  
dilut = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  mutate(Method=rep("Microdilution"), nrow(.)) %>%
  select(Species, Method)
disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(Species != "NA") %>%
  mutate(Method=rep("Disk diffusion"), nrow(.)) %>%
  select(Species, Method)
agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(strain != "NA") %>%
  mutate(Method=rep("Agar dilution"), nrow(.)) %>%
  select(strain, Method) %>%
  rename(Species=strain)

us = rbind.data.frame(dilut,disk,agar) %>%
#  filter(!(Species %in% c("Klebsiella aerogenes","Moraxella osloensis"))) %>%
    dplyr::mutate(Species = case_when(Species %in% c("Proteus spp. indole-positive") ~ "Proteus spp.",
                                      Species %in% c("Serratia marcescens") ~ "Serratia spp.",
                                      Species %in% c("Acinetobacter baumannii","Acinetobacter beijerinckii", "Acinetobacter calcoaceticus","Acinetobacter guillouiae","Acinetobacter haemolyticus","Acinetobacter johnsonnii","Acinetobacter johnsonii","Acinetobacter junii","Acinetobacter lwoffi","Acinetobacter lwoffii","Acinetobacter pittii","Acinetobacter radioresistens","Acinetobacter septicus", "Acinetobacter ursingii") ~ "Acinetobacter spp.",
                                      Species %in% c("Citrobacter freundii", "Citrobacter koseri") ~ "Citrobacter spp.",
                                      Species %in% c("Pseudomonas putida", "Pseudomonas aeruginosa") ~ "Pseudomonas spp.",
                                      Species %in% c("Klebsiella aerogenes", "Klebsiella oxytoca", "Klebsiella pneumoniae") ~ "Klebsiella spp.",
                                      TRUE ~ Species)) %>%
                    distinct(Species,Method)



test = rbind.data.frame(us,eucast)#%>%
  # mutate(study = case_when(Method == "EUCAST" ~ "EUCAST",
  #                          TRUE ~ "This study"))
        
# test = test %>%
#   dplyr::mutate(Spec_study = interaction(Species, study, sep = "-")) %>%
#   dplyr::select(Spec_study, Method)
        
# test = test %>%
#   dplyr::select(Species, Method)
  
test = table(test)
test[test>0] = 1

test = as.data.frame(test)

test = test %>%
  pivot_wider(names_from = Method, values_from = Freq) %>%
  as.data.frame()

test = test[apply(test[,-1], 1, function(x) !all(x==0)),]

test1 = dplyr::mutate(test, A = ifelse(`Agar dilution`, "Agar dilution", NA),
       D = ifelse(`Disk diffusion`, "Disk diffusion", NA),
       M = ifelse(`Microdilution`, "Microdilution", NA),
       R = ifelse(`Microdilution`, "EUCAST", NA))

test1 = tidyr::unite(test1, "Methods", A:M, na.rm=T, sep='&')
# test1 = test1 %>%
#   dplyr::mutate(Species = sapply(strsplit(as.character(Spec_study), "-"), '[', 1),
#                 Study = sapply(strsplit(as.character(Spec_study), "-"), '[', 2)) %>%
#   dplyr::select(-Spec_study)

# 
# 
# test1 = tidyr::unite(test1, "Methods", A:R, na.rm=T, sep='&')
# 
# names(upset_themes)

pdf(paste0(plotdir, "susceptibility.pdf"), width = 6, height = 4.5)
ComplexUpset::upset(test1,c("Agar dilution", "Disk diffusion", "Microdilution", "EUCAST"),
  stripes = "white",
  sort_intersections=FALSE,
  intersections=list("EUCAST",
                     c("Microdilution", "Disk diffusion","Agar dilution","EUCAST"), 
                     c("Microdilution", "Disk diffusion","Agar dilution"),
                    "Disk diffusion"),
  base_annotations=list('Intersection size'=intersection_size(
    #counts=FALSE,
      text = list(size=5.5),
      
      text_colors = c(on_background = "black", on_bar = "white"))),
  set_sizes=(upset_set_size(
    geom = geom_bar(width = 0.5),
    position = "right") +
    geom_text(aes(label=..count..), hjust=1.2, stat='count', size =5.5, color ="white", fontface="bold") +
      expand_limits(y=30) +
      theme(axis.text.x=element_text(size = 8))),
  matrix=intersection_matrix(
    geom = geom_point(size = 3),
    segment = geom_segment(),
    outline_color = list(active = "black", inactive = "grey70")
  ),
  themes = upset_modify_themes(list('Intersection size'=theme(axis.line.x = element_blank(),
                                                              axis.ticks.x = element_blank(),
                                                              axis.line.y = element_line(),
                                                              axis.text = element_text(size=15),
                                                              #text = element_text(size=10),
                                                              legend.position = 'none',
                                                              legend.text = element_text(size=10),
                                                              axis.title=element_text(size=15),
                                                              axis.ticks.y = element_line(),
                                                              plot.background = element_blank(),
                                                              panel.grid = element_blank()),
                                    'intersections_matrix'=theme(axis.line.x = element_blank(),
                                                                 axis.ticks.x = element_blank(),
                                                                 axis.title = element_blank(),
                                                                 axis.text = element_text(size=15),
                                                                #text = element_text(size=10),
                                                                plot.background = element_blank(),
                                                                 panel.grid = element_blank()),
                                    'overall_sizes'=theme(axis.line.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          axis.line.x = element_line(),
                                                          axis.ticks.x = element_line(),
                                                          axis.text.x.bottom = element_text(size=15),
                                                          axis.title = element_text(size=15),
                                                          plot.background = element_blank(),
                                                          text = element_text(size=15),
                                                          panel.grid = element_blank()))),
  width_ratio=0.7,
  height_ratio = 0.6,
sort_sets = F)
dev.off()


towrite = rbind.data.frame(us,eucast)

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.1b.xlsx"))

### fig 1b right part - species resolution


us = rbind.data.frame(dilut,disk,agar) %>%
 filter(!(Species %in% c("Salmonella spp.","Shigella spp.", "Burkholderia cenocepacia", "Psychrobacter spp.")))

toadd = readxl::read_xlsx(paste0(inputdir,"species_to_add.xlsx"))

resol = rbind.data.frame(us,eucast, toadd) %>%
  distinct(Species, Method)
test = table(resol)
test[test>0] = 1

test = as.data.frame(test)

test = test %>%
  pivot_wider(names_from = Method, values_from = Freq) %>%
  as.data.frame()

test = test[apply(test[,-1], 1, function(x) !all(x==0)),]

test1 = dplyr::mutate(test, A = ifelse(`Agar dilution`, "Agar dilution", NA),
       D = ifelse(`Disk diffusion`, "Disk diffusion", NA),
       M = ifelse(`Microdilution`, "Microdilution", NA),
       R = ifelse(`Microdilution`, "EUCAST", NA))

test1 = tidyr::unite(test1, "Methods", A:M, na.rm=T, sep='&')


pdf(paste0(plotdir,"Fig.1b_new_240310_species_resolved.pdf"), width = 6, height = 4.5)
ComplexUpset::upset(test1,c("Agar dilution", "Disk diffusion", "Microdilution", "EUCAST"),
  stripes = "white",
  sort_intersections=FALSE,
  intersections=list("EUCAST",
                     c("Microdilution", "Disk diffusion","Agar dilution","EUCAST"), 
                     c("Microdilution", "Disk diffusion","Agar dilution"),
                    "Disk diffusion"),
  base_annotations=list('Intersection size'=intersection_size(
    #counts=FALSE,
      text = list(size=5.5),
      
      text_colors = c(on_background = "black", on_bar = "white"))),
  set_sizes=(upset_set_size(
    geom = geom_bar(width = 0.5),
    position = "right") +
    geom_text(aes(label=..count..), hjust=1.2, stat='count', size =5.5, color ="white", fontface="bold") +
      expand_limits(y=30) +
      theme(axis.text.x=element_text(size = 8))),
  matrix=intersection_matrix(
    geom = geom_point(size = 3),
    segment = geom_segment(),
    outline_color = list(active = "black", inactive = "grey70")
  ),
  themes = upset_modify_themes(list('Intersection size'=theme(axis.line.x = element_blank(),
                                                              axis.ticks.x = element_blank(),
                                                              axis.line.y = element_line(),
                                                              axis.text = element_text(size=15),
                                                              #text = element_text(size=10),
                                                              legend.position = 'none',
                                                              legend.text = element_text(size=10),
                                                              axis.title=element_text(size=15),
                                                              axis.ticks.y = element_line(),
                                                              plot.background = element_blank(),
                                                              panel.grid = element_blank()),
                                    'intersections_matrix'=theme(axis.line.x = element_blank(),
                                                                 axis.ticks.x = element_blank(),
                                                                 axis.title = element_blank(),
                                                                 axis.text = element_text(size=15),
                                                                #text = element_text(size=10),
                                                                plot.background = element_blank(),
                                                                 panel.grid = element_blank()),
                                    'overall_sizes'=theme(axis.line.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          axis.line.x = element_line(),
                                                          axis.ticks.x = element_line(),
                                                          axis.text.x.bottom = element_text(size=15),
                                                          axis.title = element_text(size=15),
                                                          plot.background = element_blank(),
                                                          text = element_text(size=15),
                                                          panel.grid = element_blank()))),
  width_ratio=0.7,
  height_ratio = 0.6,
sort_sets = F)

dev.off()


resol = rbind.data.frame(us,eucast, toadd)

write_xlsx(resol,paste0(sourcedatadir,"source_Fig.1b.xlsx"))
```

## Fig. 1c - MIC in broth and phylogeny

#### Left: tree from gdtnb

```{r}
install.packages("castor")
library(castor)


#tree <- ape::read.tree("~/Documents/EMBL/Nx/phylogeny_forMICs/Askarbek_analysis/tree_for_Eli/Eli.species.tree")

tree <- ape::read.tree("~/Documents/EMBL/Nx/Nx_paper/phylogeny_tree/tree_for_Eli/Eli.species1.tree")
all_distances <- get_all_distances_to_root(tree, as_edge_count=FALSE)

node_distances = all_distances[(100+1):(100+tree$Nnode)]

plot(tree)

pdf(paste0(plotdir,"gtdb_tree_complete.pdf"), width = 10, height=22)
plot(tree)
dev.off()


#hist(node_distances, xlab="distance to root", ylab="# nodes", prob=FALSE);

# prune tree 
species = c("Acinetobacter baumannii",
            #"Acinetobacter beijerinckii",
            #"Acinetobacter calcoaceticus",
            #"Acinetobacter guillouiae",
            #"Acinetobacter haemolyticus",
            "Acinetobacter johnsonii",
            "Acinetobacter junii",
            "Acinetobacter lwoffii",
            "Acinetobacter pittii",
            #"Acinetobacter radioresistens",
            "Acinetobacter ursingii",
            "Achromobacter xylosoxidans",
            "Burkholderia cenocepacia",
            "Citrobacter freundii",
            "Citrobacter_Bkoseri",
            "Escherichia coli",
            "Enterobacter cloacae",
            "Hafnia alvei",
            "Klebsiella oxytoca", #previously "Klebsiella oxytoca"
            "Klebsiella pneumoniae",
            #"Klebsiella aerogenes",
            "Moraxella catarrhalis",
            #"Moraxella_Aosloensis",
            "Morganella morganii",
            "Neisseria gonorrhoeae",
            "Pseudomonas aeruginosa",
            "Pseudomonas_Eputida",
            "Proteus mirabilis",
            "Proteus vulgaris",
            "Stenotrophomonas maltophilia",
            "Serratia marcescens",
            "Salmonella enterica")

sort(tree$tip.label)

species = gsub(species, pattern= " ", replacement="")
species = paste("s__", species, sep="")

pruned.tree<-drop.tip(tree,tree$tip.label[-match(species, tree$tip.label)])

ape::write.tree(pruned.tree,"~/Documents/EMBL/Nx/Nx_paper/phylogeny_tree/tree_for_Eli/Eli.species1_pruned.tree")

pdf(paste0(plotdir,"gtdb_tree.pdf"), width = 10, height=15)
plot(pruned.tree)
dev.off()


library(ggtree)


pdf(paste0(plotdir,"tree_MICs.pdf"), width=10, height=10)
ggtree(pruned.tree) + 
  geom_tiplab(size = 5,offset = 0.001,align = TRUE) + 
  theme_tree2()+
  xlim(0, 1.6)+                       # set the x-axis limits of our tree
  #xlab("genetic distance (0.001 = 4 nucleotides difference)")+                    # add a label to the x-axis 
  theme(
    axis.title.x = element_text(size = 15),
   # axis.title.y = element_blank(),
    legend.title = element_text(face = "bold", size = 12),
    legend.text = element_text(face = "bold", size = 10),
    plot.title = element_text(size = 12, face = "bold"))
dev.off()


# get order of species

species_order_old= c("Neisseria gonorrhoeae",
                "Burkholderia cenocepacia",
                "Achromobacter xylosoxidans",
                "Stenotrophomonas maltophilia",
                "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Moraxella catarrhalis",
                #"Moraxella osloensis",
                "Acinetobacter junii",
                 "Acinetobacter beijerinckii",
                 "Acinetobacter haemolyticus",
                 "Acinetobacter calcoaceticus",
                 "Acinetobacter pittii",
                 "Acinetobacter baumannii",
                 "Acinetobacter ursingii",
                 "Acinetobacter lwoffii",
                 "Acinetobacter johnsonii",
                 "Acinetobacter guillouiae",
                 "Acinetobacter radioresistens",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                #"Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                "Neisseria gonorrhoeae",
                  "Serratia marcescens",
                 "Hafnia alvei")


species_order= c("Acinetobacter junii",
                 "Acinetobacter beijerinckii",
                 "Acinetobacter haemolyticus",
                 "Acinetobacter calcoaceticus",
                 "Acinetobacter pittii",
                 "Acinetobacter baumannii",
                 "Acinetobacter ursingii",
                 "Acinetobacter lwoffii",
                 "Acinetobacter johnsonii",
                 "Acinetobacter guillouiae",
                 "Acinetobacter radioresistens",
                #  "Moraxella catarrhalis",
                "Moraxella osloensis",
                 "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                "Achromobacter xylosoxidans",
                "Burkholderia cenocepacia",
                 "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                #"Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                "Neisseria gonorrhoeae",
                  "Serratia marcescens",
                 "Hafnia alvei")

```

### Right: species total MIC heatmap with GTDB dendro

#### new version - number strains 
```{r}
library(forcats)

species_order= c("Acinetobacter junii",
                 "Acinetobacter beijerinckii",
                 "Acinetobacter haemolyticus",
                 "Acinetobacter calcoaceticus",
                 "Acinetobacter pittii",
                 "Acinetobacter baumannii",
                 "Acinetobacter ursingii",
                 "Acinetobacter lwoffii",
                 "Acinetobacter johnsonii",
                 "Acinetobacter guillouiae",
                 "Acinetobacter radioresistens",
                #  "Moraxella catarrhalis",
                "Moraxella osloensis",
                 "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                "Achromobacter xylosoxidans",
                "Burkholderia cenocepacia",
                 "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                #"Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                "Neisseria gonorrhoeae",
                  "Serratia marcescens",
                 "Hafnia alvei")

agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
#agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_11.02.2024_.xlsx"), sheet = 1) %>%
  dplyr::mutate(Species = fct_relevel(Species, rev(species_order)))

agar = agar %>%
  pivot_longer(cols=3:13,names_to = "MIC", values_to = "no_strains") %>%
  mutate(Species = gsub(Species, pattern = "cenocepacia",replacement="cenocepacia complex"),
         Species_italics = case_when(Species == "Salmonella spp." ~ "*Salmonella* spp.",
                                     Species == "Shigella spp." ~ "*Shigella* spp.",
                                    TRUE ~ paste0("*", Species, "* ")),
         MIC=as.numeric(MIC),
         Species_label = paste0(Species_italics, "(n=", `n=`, ")")) %>%
  mutate(Species_label=as.character(Species_label)) 

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  # "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")

#create order with the n annotation
legend = agar %>% 
  dplyr::select(Species, Species_label) %>%
  distinct() %>%
  arrange(factor(Species, levels=species_order))

toplot= agar %>%
  mutate(Species_label=as.character(Species_label))

toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  
          
toplot = toplot %>% 
  filter(MIC >= 0.25)


pdf(paste0(plotdir,"Fig.1c_alternative_numberstrains.pdf"), width =8.181818, height=10)
ggplot(toplot, aes (x = as.factor(MIC), Species_label)) +
    geom_tile(aes(fill = no_strains, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=no_strains), size =5) +
  scale_fill_gradient(name = "Number of strains\nwith MIC value", low="white", high="grey40") +
  labs(y="", x =  "MIC (µg/ml)") +
 scale_fill_gradientn(colors = c("white","grey70", "grey60", "grey40"),
                      values = scales::rescale(c(0,5,20, max(toplot$no_strains)))) +
   theme_classic() +
  #theme_bw() +
  theme(axis.text.y = element_markdown(color = "black", size = 15),
        axis.text.x=element_text(size=13, angle =60, hjust = 1),
        legend.position="right",
        axis.title=element_text(size=18))

 dev.off()
 
pdf(paste0(plotdir,"Fig.1c_alternative_numberstrains_legend.pdf"), width =8.181818, height=10)
ggplot(toplot, aes (x = as.factor(MIC), Species_label)) +
    geom_tile(aes(fill = no_strains, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=no_strains), size =5) +
  scale_fill_gradient(name = "Number of strains\nwith MIC value", low="white", high="grey40") +
  labs(y="", x =  "MIC (µg/ml)") +
 scale_fill_gradientn(colors = c("white","grey70", "grey60", "grey40"),
                      values = scales::rescale(c(0,5,20, max(toplot$no_strains)))) +
   theme_classic() +
  #theme_bw() +
  theme(axis.text.y = element_markdown(color = "black", size = 15),
        axis.text.x=element_text(size=13, angle =60, hjust = 1),
        legend.position="right",
        axis.title=element_text(size=18))

 dev.off()


```


```{r}
library(forcats)
agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
#agar = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_11.02.2024_.xlsx"), sheet = 1) %>%
  dplyr::mutate(Species = fct_relevel(Species, rev(species_order)))

agar = agar %>%
  pivot_longer(cols=3:13,names_to = "MIC", values_to = "no_strains") %>%
  mutate(Species = gsub(Species, pattern = "cenocepacia",replacement="cenocepacia complex"),
         Species_italics = paste0("*", Species, "* "),
         MIC=as.numeric(MIC),
         Species_label = paste0(Species_italics, "(n=", `n=`, ")")) %>%
  mutate(Species_label=as.character(Species_label)) 

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  # "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")

#create order with the n annotation
legend = agar %>% 
  dplyr::select(Species, Species_label) %>%
  distinct() %>%
  arrange(factor(Species, levels=species_order))

toplot= agar %>%
  mutate(Species_label=as.character(Species_label))

toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  

# toplot = toplot %>%
#   dplyr::select(-repeated_values)

library(ggtext)

pdf(paste0(plotdir,"Fig.1c_alternative_numberstrains.pdf"), width = 10, height=10)
ggplot(agar, aes (x = as.factor(MIC), reorder(Species, Species_label))) +
    geom_tile(aes(fill = no_strains, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=no_strains)) +
    #scale_fill_gradientn(colours = c("white","darkblue"), limits = c(0,max(allCB$ODnorm_bug_rob))) +
   #scale_fill_gradient(name = "Number of strains", low="#ffffff", high="darkred") +
  scale_fill_gradient(name = "Number of strains\nwith MIC value", low="white", high="grey40") +
  labs(y="Species", x = "MIC (mg/ml") +
   # scale_fill_gradientn(colors = c("white", "white", "darkblue"),
 #                      values = scales::rescale(c(0,0.005, 0.095, max(df$OD_no1st)))) + 
   theme_classic()
dev.off()

# another possibility is average of MIC boxplots with tree on side
toplot <-  agar %>%
  rowwise() %>%
  mutate(repeated_values = list(rep(MIC, no_strains))) %>%
  unnest(repeated_values)

# get average value MIC to cite in text
#View(toplot %>% group_by(Species) %>%
#  summarise(meanMIC=mean(MIC)))

toplot %>%summarise(meanMIC=mean(MIC))

toplot =toplot%>%
  mutate(Species_label = gsub(Species_label, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
  mutate(Species = gsub(Species, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
   mutate(Species_label=as.character(Species_label))

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  # "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")

#create order with the n annotation
legend = toplot %>% 
  dplyr::select(Species, Species_label) %>%
  distinct() %>%
  arrange(factor(Species, levels=species_order))

toplot= toplot %>%
  mutate(Species_label=as.character(Species_label))

toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  

toplot = toplot %>%
  dplyr::select(-repeated_values)

library(ggtext)

pdf(paste0(plotdir,"Fig.1b_MIC_boxplot.pdf"), width=6.5, height=6)

ggplot(toplot,aes(MIC, reorder(Species_label, Species))) +
  geom_boxplot(width =0.5) +
 scale_x_continuous(trans="log2",breaks = c(0.0625,0.125,0.25,0.5,1,2,4,8,16,32,64),
                labels = function(b) paste0(signif(b, 3))) +
  labs(y="", x = "MIC [µg/ml]") +
  theme_bw() +
  theme(axis.text.y = element_markdown(color = "black"),
        axis.text.x=element_text(size=11))
dev.off()


ggboxplot(toplot%>% filter(Species == "Klebsiella pneumoniae"), x= "Species",y="MIC")

# ggerrorplot(toplot %>%
#             #  mutate(MIC = as.numeric(round(MIC, digits=2))) %>%
#               filter(!(Species %in% c("Moraxella catarrhalis", "Klebsiella aerogenes"))), x = "MIC", y = "Species_label", numeric.x.axis = TRUE, error.plot = "errorbar", add="se") +
#   stat_summary(geom = "point",size = 6,shape=108,col = "black",fun = "mean") +
#   geom_point(aes(x=MIC, color="black"),alpha = 0.2,size =1,position = position_jitterdodge(0.5, dodge.width = .05, jitter.height = 0.1)) +
#   scale_color_manual(values="black") +
#   scale_x_continuous(trans="log2",
#                      breaks = c(0.0625,0.125,0.25,0.5,1,2,4,8,16,32,64),
#                   labels = function(b) paste0(signif(b, 3))) +
#   labs(y="", x = "MIC [µg/ml]") +
#   theme_bw() +
#   theme(axis.text.y = element_markdown(color = "black"))

towrite= toplot %>%
  dplyr::rename(tot_number_of_strains = `n=`) %>%
  group_by(Species,tot_number_of_strains, MIC) %>%
  dplyr::summarise(number_strains_at_MIC= n()) %>%
  mutate(note = case_when(Species == "Salmonella spp." ~ "includes Salmonella enterica subsp. enterica serovar Enteritidis and serovar Typhi",
                          Species == "Shigella spp." ~ "includes Shigella sonnei and Shigella flexneri",
                          Species == "Burkholderia cenocepacia complex" ~ "includes Burkholderia cenocepacia, Burkholderia multivorans, and Burkholderia vietnamensis",
                          TRUE ~ ""))


write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.1c.xlsx"))


```

## Fig. 1d - killing Abau
```{r}
killing = readxl::read_xlsx(paste0(inputdir,"Time-kill_ACBA_ATCC19606.xlsx")) %>%
  janitor::clean_names() %>%
  filter(time_h != "bla") %>%
  pivot_longer(names_to="condition", values_to = "CFU", cols = c(2:29), names_prefix = "^x*_") %>%
  mutate(condition = sapply(strsplit(as.character(condition),"_"),'[',1),
         condition = gsub(condition,pattern="growth",replacement=0),
         condition = gsub(condition,pattern="x",replacement=""),
         condition=as.numeric(condition),
         time_h=as.numeric(time_h)) %>%
  group_by(condition, time_h) %>%
  mutate(meanCFU=mean(CFU,na.rm = TRUE),
         sdCFU = sd(CFU, na.rm=TRUE),
         seCFU=sd(CFU, na.rm=TRUE)/sqrt(length(na.omit(CFU)))) %>%
  filter(condition != 0)


pdf(paste0(plotdir,"Abau_killing.pdf"), width = 5.8, height=5)

killing$condition <- factor(killing$condition, levels=c('1','2', '4', '8', '16', '32'))

my_palette = brewer.pal("Oranges",n = 9)[3:9]


ggplot(killing, aes(as.factor(time_h),meanCFU, group = condition, color = condition)) + 
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)), 
                breaks =trans_breaks(log10, function(x) 10^x, 5)) +
  scale_x_discrete(breaks=c(0,2,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=killing[!is.na(killing$meanCFU),]) +
  geom_errorbar(aes(ymin = meanCFU-seCFU, ymax = meanCFU+sdCFU), width = 1,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(µg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("A. baumannii"))) +
       #y= expression(log[10](CFU/ml)), title=expression(italic("A. baumannii"))) +
  #expression(Blah[1*d])
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()

# for source data see ED Fig. 2a
# towrite = killing %>%
#   dplyr::rename(mean=meanCFU,
#                 se=seCFU,
#                 `Nitroxoline concentration (µg/ml)`= condition,
#                 `Time (h)` = time_h) %>%
#   dplyr::select(-sdCFU)
# 
# 
# #write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.1d.xlsx"))


```

## Fig. 1f - Salmonella intracellular

```{r}
library(rstatix)

strain_2601 = readxl::read_xlsx(paste0(inputdir,"IntracellularKilling_Vorlage_11.02.2024_MTSG_.xlsx"), skip = 2,sheet=2) 
strain_5878 = readxl::read_xlsx(paste0(inputdir,"IntracellularKilling_Vorlage_11.02.2024_MTSG_.xlsx"), skip = 2,sheet=1) 

killing = gdata::combine(strain_5878, strain_2601) %>%
  pivot_longer(cols=c(2:4),names_to="condition",values_to = "CFU/ml") %>%
  #na.omit(.) %>%
  mutate(source = plyr::mapvalues(source, from = c("strain_5878","strain_2601"),
                                         to = c("italic('S. ')~Typhi~'5878'","italic('S. ')~Typhi~'2601'"))) %>%
  #drop_na(`CFU/ml`) %>%
  clean_names() %>%
  dplyr::group_by(date, condition, source) %>%
  mutate(cfu_ml = log10(cfu_ml)) %>%
  reframe(meancfu_ml = mean(cfu_ml, na.rm=TRUE),
            sd = sd(cfu_ml, na.rm = TRUE))

                       
stat.test <- killing %>%
  distinct(source, meancfu_ml,condition) %>%
  dplyr::group_by(source) %>%
  t_test(meancfu_ml ~ condition, var.equal = TRUE) %>%
  #adjust_pvalue(method = "BH") %>%
  add_significance("p") %>%
  add_xy_position(fun = "mean_sd", x = "condition", dodge = 0.8) %>%
  filter(!(group1 =="DMSO" & group2 =="preNx"))

                 
library(stringi)      
stat.test = stat.test %>%
  mutate(xmin=case_when(xmin == 1 ~ 2,
                        xmin == 2 ~ 3,
                        TRUE ~ 1),
         xmax=case_when(xmin == 1 ~ 2,
                        xmin == 2 ~ 3,
                        TRUE ~ 1))
                                                

# decrease of how much?

killing %>%
  group_by(source, condition) %>%
  summarise(mean=mean(meancfu_ml))


killing = killing %>%
  dplyr::group_by(source,condition) %>%
  mutate(meanratio=mean(meancfu_ml),
         sdratio=sd(meancfu_ml),
         seratio=sqrt(sum((meancfu_ml-mean(meancfu_ml))^2/(length(meancfu_ml)-1)))/sqrt(length(meancfu_ml))) %>%
         #seratio=std.error(ratio)) %>%
  ungroup()


pdf(paste0(plotdir, "Salmonella_intracell_killing_230212.pdf"), width=7.5, height=5.5)
killing$condition <- factor(killing$condition, levels=c("preNx", "DMSO", "Nx"))

killing = killing %>% distinct(condition, meanratio, meancfu_ml, sdratio, source)
ggplot(killing, aes(x = condition, y = meanratio)) +
  geom_bar(stat = "identity", position = position_dodge(0.8), width = 0.5,fill="grey50") +
  geom_errorbar(aes(ymin = meanratio - sdratio, ymax = meanratio + sdratio),width = 0.15, color="grey50") +
  geom_jitter(aes(x = condition,y=meancfu_ml), width = 0.1,height = 0.1, 
              size = 3, shape = 21, alpha = 1, fill="grey90", color="grey20") +
   stat_pvalue_manual(stat.test, tip.length = 0.01,#bracket.nudge.y = 0.5,
                     label = "p.signif",  y.position = 5, step.increase = 0.1, step.group.by = "source",size = 5) +
  facet_wrap(~ source, labeller = label_parsed)+
  scale_y_continuous(#trans = "log10", 
                     breaks = c(0,2,4), labels= math_format(10^.x)) +
  #ylim(0,5.5) +
  scale_x_discrete(labels=c("before\ntreatment","solvent\ncontrol","nitroxoline\n(5 µg/ml)")) +
  labs(x="", y= expression(CFU/ml)) +
  theme_bw() +
  guides(color="none") +
  theme(strip.text = element_text(size=16),
        axis.text=element_text(size=16),
        axis.title=element_text(size=16))
dev.off()



ggbarplot(killing, x= "condition", y = "log10cfu", add=c("mean_se","jitter")) +
    stat_compare_means(label = "p.signif", method = "t.test",size = 4, ref.group = "Nx", label.y=5, bracket.size = 5) +
  facet_wrap(~ source, labeller = label_parsed)  



towrite = killing %>%
  dplyr::rename(mean=meanratio,
                sd=sdratio) %>%
  mutate(condition = gsub(condition,pattern="preNx",replacement="before treatment"),
         condition = gsub(condition,pattern="Nx",replacement="nitroxoline (5 µg/ml)"),
         strain = case_when(source == "italic('S. ')~Typhi~'5878'" ~ "S. Typhi 5878",
                            TRUE ~ "S. Typhi 2601")) %>%
  dplyr::select(strain,condition, cfu_ml, log10cfu, mean, sd)


write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.1f.xlsx"))


```



# Fig. 2 combinations and resensitization of colistin mutants


## Fig. 2a - Bliss scores




### recalculate Bliss scores with new supp files


```{r}
#toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.3a.xlsx"))
towrite = readxl::read_xlsx(paste0(inputdir,"source_Fig.2a_old.xlsx")) %>%
  select(-c(cumulative_eps_quartiles,eps.med,eps.q1,eps.q3)) %>%
    mutate(type = case_when(horizontal_drug %in% c("Fosfomycin", "Gentamicin") ~ "antagonism",
                            horizontal_drug %in% c("Colistin") ~ "synergy",
                            horizontal_drug %in% c("Fosfomycin", "8−Hydroxyquinoline","CCCP","Nitrofurantoin", "Clarithromycin", "Piperacillin") ~ "neutral",
                            TRUE ~ type))

write_xlsx(towrite,paste0(sourcedatadir,"source_EDFig.3a.xlsx"))

toplot = readxl::read_xlsx(paste0(inputdir,"source_Fig.2a_old.xlsx")) %>%
  dplyr::select(-c(cumulative_eps_quartiles, eps.med, eps.q1, eps.q3, type))

  hzt_alone = toplot%>% subset(nitroxoline_concentration == 0.00) %>%
    dplyr::select(c("horizontal_drug","horizontal_concentration", "norm_fitness")) %>%
    dplyr::rename("fit_hzt"= "norm_fitness")
  
  vert_alone = toplot %>% subset(horizontal_concentration == 0.00) %>%
    dplyr::select(c("horizontal_drug", "nitroxoline_concentration", "norm_fitness")) %>%
    dplyr::rename("fit_vert"= "norm_fitness")
  
  out = inner_join(hzt_alone, toplot)
  out = inner_join(vert_alone, out)
  
  out$fit_comb = out$norm_fitness
  
  out = mutate(out, exp_fit = fit_vert * fit_hzt,
               bliss = fit_comb - exp_fit)
  
out = out %>% filter(!(horizontal_drug == "Colistin" & horizontal_concentration > 0.2))

#try to assign interaction sign using first and third quartiles
#assign interaction sign: synergy if 1st quartile < -0.1, antag if 3rd > 0.1

out = out %>% dplyr::group_by(horizontal_drug) %>%
  dplyr::mutate(eps.med = median(bliss),
         eps.q1 = quantile(bliss, 0.25),
         eps.q3 = quantile(bliss, 0.75)) %>%
     dplyr::mutate(type = ifelse(eps.q1 < -0.1 & eps.med < -0.06, "synergy",
                        ifelse(eps.q3 > 0.1 & eps.med > 0.04, "antagonism","neutral")),
                  cumulative_eps_quartiles = ifelse(type == "synergy", eps.q1,
                                  ifelse(type == "antagonism", eps.q3, eps.med))) %>%
  select(horizontal_drug,bliss,horizontal_concentration, nitroxoline_concentration,replicate,median_norm_fitness, cumulative_eps_quartiles, eps.med,eps.q1,eps.q3, type) %>%
  distinct()

out = inner_join(out, toplot)
rm(vert_alone)
rm(hzt_alone)
rm(toplot)


out = out %>%
  mutate(Drugclass= case_when(horizontal_drug %in% c("Fosfomycin", "Aztreonam", "Mecillinam","Temocillin","Piperacillin","Ceftazidime", "Cefepime", "Imipenem", "Meropenem", "Ertapenem", "Vancomycin", "Colistin") ~ "Cell wall and outer membrane",
                              horizontal_drug %in% c("Ciprofloxacin", "Levofloxacin", "Novobiocin","Trimethoprim") ~ "DNA",
                              horizontal_drug %in% c("8−Hydroxyquinoline", "Clioquinol", "Thiolutin") ~ "Other metal chelators",
                              horizontal_drug %in% c("Amikacin", "Gentamicin", "Tobramycin","Azithromycin", "Clarithromycin", "Erythromycin","Tetracycline","Minocycline", "Doxycycline", "Tigecycline") ~ "Protein synthesis",
                              TRUE ~ "Other"))
out = out %>%
  mutate(cumulative_eps_quartiles=round(cumulative_eps_quartiles, digits =2))

out = distinct(out)

write_xlsx(out, paste0(sourcedatadir,"source_Fig.2a.xlsx"))

# plot
colours = data.frame("type" = c("synergy", "antagonism", "neutral"), "values" = c("turquoise4", "#d9a520", "grey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$type

drugclass = "Cell wall and outer membrane"

p1 = ggplot (filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Fosfomycin", "Aztreonam", "Mecillinam","Temocillin","Piperacillin","Ceftazidime", "Cefepime", "Imipenem", "Meropenem", "Ertapenem", "Vancomycin", "Colistin"))),
                                                       x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  #geom_point(alpha = 0.3, position=position_jitter(height=.2, width=.1)) +
  labs (x = "Interaction score", y="") +
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  #annotate('rect', xmin=-0.1, xmax=0.1, ymin=0, ymax=15, alpha=.2, fill='grey') +
  facet_grid(~ Drugclass, scales =  "free_y", space = "free") +
  #scale_x_discrete(expand = c(-1, +1)) +
  xlim(-1, 1)+
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "DNA"

p2 = ggplot(filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Ciprofloxacin", "Levofloxacin", "Novobiocin","Trimethoprim"))), x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  #geom_point(alpha = 0.3, position=position_jitter(height=.2, width=.1)) +
  labs (x = "Interaction score", y="") +
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  xlim(-1, 1)+
  #scale_x_discrete(expand = c(-1, +1)) +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "Other metal chelators"

p3 = ggplot(filter(out, Drugclass == drugclass), aes(y= horizontal_drug, x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  #geom_point(alpha = 0.3, position=position_jitter(height=.2, width=.1)) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  #scale_x_discrete(expand = c(-1, +1)) +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "Protein synthesis"

p4 = ggplot(filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Amikacin", "Gentamicin", "Tobramycin","Azithromycin", "Clarithromycin", "Erythromycin","Tetracycline","Minocycline", "Doxycycline", "Tigecycline"))), x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  #geom_point(alpha = 0.3, position=position_jitter(height=.2, width=.1)) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  #scale_x_discrete(expand = c(-1, +1)) +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none")

drugclass = "Other"

p5 = ggplot(filter(out, Drugclass == drugclass), aes(y= horizontal_drug, x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  #geom_point(alpha = 0.3, position=position_jitter(height=.2, width=.1)) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  #scale_x_discrete(expand = c(-1, +1)) +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")



pdf(paste0(plotdir, "Bliss_boxplots_pooled.pdf"), width = 12, height = 10)


layout <-"AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AACC
AACC
AACC
BBEE
BBEE
BBEE
BBEE
"

wrap_plots(A=p1, B = p2, C= p3, D= p4, E=p5,design = layout, guides = "collect")

dev.off()


```


## Fig. 2b - resensitization of colistin resistant mutants

```{r}
bunch = fread("~/Documents/EMBL/Nx/220602_colistin_Nx_resmutants/plate_norm.txt") %>%
  dplyr::mutate(strain = case_when(strain == "EC_J53" ~ "E. coli J53",
                                       strain == "EC_J53Tc2442" ~ "E. coli J53 pKP2442_mcr-1",
                                       strain == "EC_J53pTOPO" ~ "E. coli J53 pTOPO",
                                       strain == "EC_J53pTOPOmcr1" ~ "E. coli J53 pTOPO_mcr-1",
                                       strain == "KP_2442" ~  "K. pneumoniae 2442 mcr-1",
                                       strain == "KP_CTR3" ~ "K. pneumoniae CT-R3 pmrB (P95L)",
                                       strain == "KP_CTS" ~ "K. pneumoniae CT-S",
                                      strain == "KP_LS107" ~ "K. pneumoniae LS107",
                                      strain == "KP_LS53" ~ "K. pneumoniae LS53 mgrB (IS1R)",
                                      strain == "KP_PRZ" ~ "K. pneumoniae PRZ",
                                      strain == "KP_PRZ2442" ~ "K. pneumoniae PRZ pKP2442_mcr-1",
                                   TRUE ~ strain)) %>%
    filter(strain != "nocells")



bla = fread("~/Documents/EMBL/Nx/220707_colistin_Nx_resmutants_KPPRZ_topo/plate_norm.txt")

pmrAB = fread("~/Documents/EMBL/Nx/221110_pmrAB_fromMichael/plate_norm.txt")%>%
  dplyr::mutate(nitroxoline=case_when(nitroxoline == "+ Nx 0.75 µg/ml" ~ "Nitroxoline_0.75",
                                      TRUE ~ "no_Nitroxoline"))%>%
  dplyr::mutate(strain = case_when(strain == "BW25113" ~ "E. coli BW25113",
                                   strain == "MG1655" ~ "E. coli MG1655",
                                   strain == "pmrA" ~ "E. coli MG1655 pmrA (G53E)",
                                   strain == "pmrB" ~ "E. coli MG1655 pmrB (V88E)",
                                   TRUE ~ strain))

# mcr1 = fread("~/Documents/EMBL/Nx/220527_MIC_mrc1_colistin_nx/plate_norm.txt") %>%
#     filter(nitroxoline != "+ Nx 0.45 µg/ml") %>%
#   dplyr::mutate(nitroxoline=case_when(nitroxoline == "+ Nx 0.85 µg/ml" ~ "Nitroxoline_0.75",
#                                       TRUE ~ "no_Nitroxoline")) %>%
#   dplyr::mutate(strain = case_when(strain == "BW" ~ "E. coli BW25113",
#                                    strain == "mcr1" ~ "E. coli BW25113 pGDP1_mcr1"))%>%
#   filter(strain != "E. coli BW25113")
  


#toplot = bind_rows(bunch, bla, pmrAB,mcr1)
toplot = bind_rows(bunch, bla, pmrAB)

fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)


toplot = toplot %>%
  dplyr::mutate(strain_conc = interaction(strain, Concentration, nitroxoline),
                Concentration_f = as.factor(Concentration)) %>%
  dplyr::group_by(strain,Time_points,Drug, Concentration, strain_conc, nitroxoline) %>%
  dplyr::summarise(median = mean(ODnorm_bug_rob), sem = sd(ODnorm_bug_rob)/sqrt(length(ODnorm_bug_rob))) %>%
  dplyr::ungroup()


toplot$facet = factor(toplot$strain, levels = c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1","E. coli J53 pKP2442_mcr-1",
                                                "E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)",
                                                #"E. coli BW25113","E. coli BW25113 pGDP1_mcr1",
                                                "K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)", "K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)","K. pneumoniae 2442 mcr-1",
                                                "K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO","K. pneumoniae PRZ pTOPO_mcr-1", "K. pneumoniae PRZ pKP2442_mcr-1"))

toplot = toplot %>%
  mutate(row= case_when(strain %in% c("E. coli BW25113","E. coli BW25113 pGDP1_mcr1") ~ 1,
                        strain %in% c("E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)") ~ 2,
                        strain %in% c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "E. coli J53 pKP2442_mcr-1") ~ 3,
                        #strain %in% c("E. coli J53 pKP2442_mcr-1") ~ 4,
                        strain %in% c("K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO_mcr-1","K. pneumoniae PRZ pKP2442_mcr-1") ~ 5,
                        strain %in% c("K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)") ~ 6,
                        strain %in% c("K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)") ~ 7,
                        strain %in% c("K. pneumoniae 2442 mcr-1") ~ 8,
                        strain %in% c("K. pneumoniae PRZ pTOPO") ~ 9),
         resist = case_when(strain %in% c("E. coli BW25113","E. coli J53","E. coli MG1655",
                                          "K. pneumoniae CT-S","K. pneumoniae LS107","K. pneumoniae PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))


toplot = toplot %>%
  filter(Time_points == 44 & strain != "nocells")

# write source data for this and for the supp fig with the empty vector controls
towrite = toplot %>%
  rename(meanOD=median,
         resistance=resist,
         `Colistin concentration (µg/ml)`=Concentration) %>%
  dplyr::select(-c(row, facet,strain_conc, Time_points))

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.2b_EDFig.3b.xlsx"))

```

#### old plot
```{r}
# return to plotting
toplot = toplot %>%
  #decided not to include empty vector control in main fig, only supp 
  filter(!(strain %in% c("E. coli J53 pTOPO", "K. pneumoniae PRZ pTOPO"))) %>%
  mutate(strain = gsub(strain, pattern  =" ",replacement = "~"),
        strain = gsub(strain, pattern  ="E.~coli",replacement = "italic('E. coli')"),
        strain = gsub(strain, pattern  ="K.~pneumoniae",replacement = "italic('K. pneumoniae')")) 

toplot = toplot %>%
  mutate(strain = case_when(strain =="italic('E. coli')~BW25113~pGDP1_mcr1" ~ "~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))",  
    strain =="italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
                            strain =="italic('E. coli')~MG1655~pmrA~(G53E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrA'),'(G53E)'))",
                            strain == "italic('E. coli')~MG1655~pmrB~(V88E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrB'),'(V88E)'))",
    strain == "italic('E. coli')~J53~pTOPO_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pTOPO_', italic('mcr-1')))",
    strain == "italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
    TRUE ~ strain))

# change name of strain as Stephan asked

p <- ggplot(filter(toplot, row %in% c(1)),
            aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
    facet_wrap(~ fct_relevel(strain, "italic('E. coli')~BW25113",
                             "~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))"),
             ncol=2, labeller = label_parsed) +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.005, fill="grey") +
  scale_color_manual(labels = c("+ NX 0.75 µg/ml", "- NX"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks = element_blank(),
        legend.text = element_text(size=12))

legend_grob <-cowplot::get_legend(p)


p <-  ggplot(filter(toplot, row %in% c(1)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
    facet_wrap(~ fct_relevel(strain, "italic('E. coli')~BW25113", "~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))"),
             ncol=2, labeller = label_parsed,scales="free_x") +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
    #axis.text.x = element_blank(),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=12),
        strip.text=element_text(size=11),
        #panel.ontop = FALSE,panel.background = element_rect(fill = NA),
        legend.position = "none")


p1 <- ggplot(filter(toplot, row %in% c(2)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==2 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ fct_relevel(strain, "italic('E. coli')~MG1655",
                           "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrA'),'(G53E)'))",
                           "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrB'),'(V88E)'))"),
             ncol=3, labeller = label_parsed, scales='free_x') +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
    ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=12),
        strip.text=element_text(size=11),
        legend.position = "none",
        legend.direction = "vertical")

  

p2 <- ggplot(filter(toplot, row %in% c(3)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==3 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ fct_relevel(strain,
                           "italic('E. coli')~J53",
                           "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
                           "~ atop(paste(italic('E. coli'), ' J53'),paste('pTOPO_', italic('mcr-1')))"),
             ncol=3, labeller = label_parsed) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
    ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=12),
        strip.text=element_text(size=11),
      legend.position = "none",
       legend.direction = "vertical")



pdf(paste0(plotdir, "fig.2c_left_Ecolimutants_colistin.pdf"),width=6, height=5)
# layout <-"AAE
#           BBB
#           CCC"

layout <-"BBB
          CCC"
      
#wrap_plots(A=p, B = p1,C=p2,E=legend_grob,design = layout)
wrap_plots(B = p1,C=p2,E=legend_grob,design = layout)

dev.off()


## KPN
toplot = toplot %>%
  mutate(strain = case_when(strain=="italic('K. pneumoniae')~CT-S" ~ "italic('K. pneumoniae')~LS01",
    strain =="italic('K. pneumoniae')~CT-R3~pmrB~(P95L)" ~ "~ atop(paste(italic('K. pneumoniae'), ' LS01'),paste(italic('pmrB'),'(P95L)'))",
    strain =="italic('K. pneumoniae')~LS53~mgrB~(IS1R)" ~ "~ atop(paste(italic('K. pneumoniae'), ' LS107'),paste(italic('mgrB'),'(IS1R)'))",
                            strain =="italic('K. pneumoniae')~PRZ~pKP2442_mcr-1" ~ "~ atop(paste(italic('K. pneumoniae'), ' PRZ'),paste('pKP2442_', italic('mcr-1')))",
                            strain == "italic('K. pneumoniae')~PRZ~pTOPO_mcr-1" ~ "~ atop(paste(italic('K. pneumoniae'), ' PRZ'),paste('pTOPO_', italic('mcr-1')))",
    strain == "italic('K. pneumoniae')~2442~mcr-1" ~ "~ atop(paste(italic('K. pneumoniae')),paste('2442 ', italic('mcr-1')))",
                            TRUE ~ strain))
                       

p <- ggplot(filter(toplot, row %in% c(5)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==5 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ fct_relevel(strain,"italic('K. pneumoniae')~PRZ",
                            "~ atop(paste(italic('K. pneumoniae'), ' PRZ'),paste('pKP2442_', italic('mcr-1')))",
                           "~ atop(paste(italic('K. pneumoniae'), ' PRZ'),paste('pTOPO_', italic('mcr-1')))"),
             ncol=3, labeller = label_parsed, scales='free_x') +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
    ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=12),
        strip.text=element_text(size=11),
        legend.position = "none",
        legend.direction = "vertical")

p1 <- ggplot(filter(toplot, row %in% c(6)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==6 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ fct_relevel(strain, "italic('K. pneumoniae')~LS01", "~ atop(paste(italic('K. pneumoniae'), ' LS01'),paste(italic('pmrB'),'(P95L)'))"),
             ncol=2, labeller = label_parsed, scales='free_x') +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        strip.text=element_text(size=11),
        axis.text=element_text(size=12),
        legend.position = "none")

p2 <- ggplot(filter(toplot, row %in% c(7)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==7 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ fct_relevel(strain, "italic('K. pneumoniae')~LS107","~ atop(paste(italic('K. pneumoniae'), ' LS107'),paste(italic('mgrB'),'(IS1R)'))"), ncol=2, labeller = label_parsed, scales='free_x') +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        strip.text=element_text(size=11),
        axis.text=element_text(size=12),
        legend.position = "none")

p3 <- ggplot(filter(toplot, row %in% c(8)), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
  geom_rect(data=filter(toplot,row==8 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  facet_wrap(~ strain, ncol=2, labeller = label_parsed, scales='free_x') +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        strip.text=element_text(size=11),
        axis.text=element_text(size=12),
        legend.position = "none")


pdf(paste0(plotdir, "fig.2c_right_Kpnmutants_colistin.pdf"),width=8, height=5)

layout <-"BBBBBBBCCCCCC
          AAAAAAAAADDDD"
      
wrap_plots(A=p, B = p1,C=p2,D=p3, design = layout)

dev.off()



```

### compact alternative - Vallo suggestion

```{r}


test = toplot %>%
  #decided not to include empty vector control in main fig, only supp
  filter(!(strain %in% c("E. coli J53 pTOPO", "K. pneumoniae PRZ pTOPO"))) %>%
  mutate(species = case_when(grepl("E. coli",strain) ~ "italic('E. coli')",
                            TRUE ~"italic('K. pneumoniae')")) %>%
  mutate(strain = gsub("E\\. coli |K\\. pneumoniae ", "", strain),
        strain = gsub(strain, pattern  ="pmrA",replacement = "*pmrA*"),
        strain = gsub(strain, pattern  ="pmrB",replacement = "*pmrB*"),
        strain = gsub(strain, pattern  ="mcr-1",replacement = "*mcr-1*"),
        strain = gsub(strain, pattern  ="LS53",replacement = "LS107"),
        strain = gsub(strain, pattern  ="CT-R3",replacement = "LS01"),
        strain = gsub(strain, pattern  ="CT-S",replacement = "LS01"))
 

p1 <-   ggplot(filter(test, row==2), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
    geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 3,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  #facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p2 <- ggplot(filter(test, row==3), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 3,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
 # facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")

p3 <- ggplot(filter(test, row==5), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 2,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
 # facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p4 <- ggplot(filter(test, row==6), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 2,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
 # facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")

p5 <- ggplot(filter(test, row==7), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 1,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
 # facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=23),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p6 <- ggplot(filter(test, row==8), aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resist)) +
    geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.7,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 µg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
 # facet_wrap(~ species,labeller = label_parsed) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$median),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        #axis.ticks.x = element_blank(),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


pdf(paste0(plotdir, "fig2b.pdf"),width=28, height=13)

layout <-"CDAB
          EG##"
      
wrap_plots(A = p1,B=p2,C= p3, D=p4, E=p5, G =p6, design = layout)

dev.off()


```



## Fig. 2c - Galleria with Kpn infection

```{r}
#galleria = readxl::read_xlsx(paste0(inputdir,"Galleria_readableform.xlsx"))
galleria = readxl::read_xlsx(paste0(inputdir,"Nx_Galleria_2.3.24_.xlsx"), sheet=2) %>%
  clean_names() %>%
  pivot_longer(cols=c("x06_07", "x07_07", "x08_10", "x10_10"), values_to = "survival", names_to = "replicate")
# calculate mean and SEM

galleria = galleria %>%
  drop_na() %>%
  mutate(condition = sapply(strsplit(as.character(condition),"\\+ "), '[',2)) %>%
  dplyr::group_by(condition, time) %>%
  dplyr::summarise(mean = mean(survival), sem = sd(survival)/sqrt(length(survival))) %>%
  dplyr::ungroup()


  
pdf(paste0(plotdir, "galleria_Kpn.pdf"),  width=6, height=4)
ggplot(galleria, aes(x = time, y = mean, color = condition, group = condition))+
  geom_point(size = 3.5) +
  geom_line(alpha=0.8, size =1) +
 scale_color_manual(values = c('#009999','#FF6699','#FF6600', '#666666'),
                       breaks = c("0,1 µg/mL nx ", "0,1 µg/ml CT", "0,1 µg/ml Nx", "NaOAc"),
                       labels = c("NX + COL", "COL 0.1 µg/ml", "NX 0.1 µg/ml", "untreated")) +
  scale_x_continuous(breaks = c(0,24,48,72)) +
  labs(x="Time after infection (h)", y = "% animal survival", color ="") +
  geom_errorbar(aes(ymin = mean-sem, ymax = mean + sem),position = position_dodge(2), width = 15, size = 1) +
  theme_classic() + 
  theme(legend.position = "right",
        axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        #plot.margin = unit(c(0,0.5,0.5,0.6), 'cm'),
        legend.text=element_text(size=18))
dev.off()

towrite = killing %>%
  dplyr::rename(mean=meanratio,
                sd=sdratio) %>%
  mutate(condition = gsub(condition,pattern="preNx",replacement="before treatment"),
         condition = gsub(condition,pattern="Nx",replacement="nitroxoline (5 µg/ml)"),
         strain = case_when(source == "italic('S. ')~Typhi~'5878'" ~ "S. Typhi 5878",
                            TRUE ~ "S. Typhi 2601")) %>%
  dplyr::select(strain,condition, cfu_ml, log10cfu, mean, sd)



towrite = readxl::read_xlsx(paste0(inputdir,"Nx_Galleria_2.3.24_.xlsx"), sheet=2) %>%
  clean_names() %>%
  pivot_longer(cols=c("x06_07", "x07_07", "x08_10", "x10_10"), values_to = "survival", names_to = "replicate") %>%
  rename(`Time after infection (h)` = time,
         `survival (%)` = survival) %>%
  mutate(condition = sapply(strsplit(as.character(condition),"\\+ "), '[',2))

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.2c.xlsx"))



```


# Fig. 3 - OM perturbation

### Fig. 3a - TPP with lpt highlighted

Annotations come from the GO enrichment 

```{r}

fdr_df_whole = fread("~/Documents/EMBL/Nx/Nx_paper/input/Andre_fdr_df.txt")
hits_df_whole = fread("~/Documents/EMBL/Nx/Nx_paper/input/Andre_hits_df.txt")

anna <- readxl::read_xlsx("~/Documents/EMBL/Nx/Nx_paper/input/Anna_OMPs_msystems.00808-19-st003.xlsx",sheet = 2) %>%
  filter(`localization (sucrose gradient)` == "OM")

om_uniprot = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>% 
  clean_names() %>%
  filter(grepl("outer membrane", subcellular_location_cc) | grepl("outer membrane", gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername) 

zinc = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>% 
  clean_names() %>%
  filter(grepl("zinc", gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

manganese = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>% 
  clean_names() %>%
  filter(grepl("manganese", gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

#patterns <- c("4Fe-4S", "2Fe-2S")
ironsulfur = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  #filter(grepl("iron-sulfur", gene_ontology_go) | grepl(paste(patterns,collapse="|"), cofactor)) %>%
  filter(grepl("iron-sulfur", gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

siderophores = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("siderophore", gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

patterns <- c("respir", "electron transfer")
respiration = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl(paste(patterns,collapse="|"), gene_ontology_go)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

copper = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("copper", gene_ontology_go) | grepl("Cu (2+)", cofactor)) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  select(clustername)

fdr_df_whole = fdr_df_whole %>%
  mutate(annot = case_when(clustername %in% c("COPA","CUEO") | clustername %in% copper$clustername ~ "copper",
                           #clustername %in% c("ZUR","ZNTR", "ZNUA","ZNUB","ZNUC","ZINT","ZNTA","PEPD","PEPT","AMPD") ~ "zinc",
                           clustername %in% unique(zinc$clustername) ~ "zinc homeostasis and zinc binding",
                           #clustername %in% c("PEPD","PEPT","AMPD","PEPA","CYSS","PRLC") ~ "zinc binding",
                           #clustername %in% c("ISCS", "ISCU", "ISCA","SUFA","SUFB","SUFC","SUFD","SUFS","SUFE", "NFUA") ~ "Fe-S cluster biogenesis",
                           clustername %in% unique(ironsulfur$clustername) ~ "Fe-S cluster biogenesis",
                           #clustername %in% c("FEPA", "FEPB","FEPC","FEPD","FEPG","FECA","TONB", "EXBB","EXBD","FBPA","FBPB","FBPC", "ENTA","ENTB","ENTC", "ENTF", "FES", "FEOB","FUR", "ISCR",
                            #                  "ISCS", "ISCU", "ISCA","SUFA","SUFB","SUFC","SUFD","SUFS","SUFE", "NFUA","NSRR","SDAA","SDAB","PFLA","FECI","FHUF") ~ "iron",
                           
                          clustername != "LPXH" & (clustername %in% c("MNTR","MNTH") | clustername %in% unique(manganese$clustername)) ~ "manganese",
                           clustername %in% c("LPXA","LPXB","LPXC","LPXD", "LPXH","LPXK","LPXL","LPXM","LPXT",
                                              "GMHA", "WAAA", "WAAU", "LPTA", "LPTB", "LPTC", "LPTD", "LPTE", "LPTF", "LPTG",
                                              "WZZE","WZZB", "GLF", "WBBI","WBBJ","WBBK", "NANA","NANK","NANT","HLDE","HLDD") ~ "LPS biosynthesis and assembly",
                           clustername %in% c("EMRA","MPRA","MARR","TOLC","ACRA","ACRB") ~ "efflux pumps and regulators",
                           !(clustername %in% respiration$clustername) & !(clustername %in% ironsulfur$clustername) & clustername %in% unique(om_uniprot$clustername) | clustername %in% c("NMPC","MALB","BLC") | clustername %in% unique(anna$gene_name) ~ "OMPs",
                           clustername %in% unique(siderophores$clustername) ~ "iron import",
                           clustername %in% respiration$clustername ~ "respiration",
                           # clustername %in% c("CYOA", "CYOB","CYOC","CYOD",
                           #                    "SDHA","SDHB","SDHD",
                           #                    "HEMH", "HEMN", "HEMX","NADE",
                           #                    "HYBA","HYBC","HYBD","HYBO",
                           #                    "HYAA","HYAB",
                           #                    "FDOG","FDOH","FDOI",
                           #                    "FDNG","FDNH","FDNI",
                           #                    "NUOA","NUOB","NUOC","NUOE","NUOF","NUOG","NUOH","NUOI",
                           #                    "NUOJ","NUOK","NUOL","NUOM","NUON",
                           #                    "NDH","DLD","LLDR","LDLL","LDLP",
                           #                    "GLPA","GLPB","GLPC","GLPD","GLPF","GLPK",
                           #                    "GLPQ","GLPR","GLPT","GLPX",
                           #                    "CYDA","CYDB","CYDC",
                           #                    "NARP","NARG","NARH","DMSA","DMSB","DMSD",
                           #                    "UBIE","UBIJ","UBIH", "UBIF","UBIG","UBII",
                           #                    "RSXB","RSXC","RSXD","RSXE","RSXG","YDBK","POXB",
                           #                    "NAPA") ~ "respiration",
                           TRUE ~ "other"))


hits_df_whole = hits_df_whole %>%
  mutate(annot = case_when(clustername %in% c("COPA","CUEO") | clustername %in% copper$clustername ~ "copper",
                           #clustername %in% c("ZUR","ZNTR", "ZNUA","ZNUB","ZNUC","ZINT","ZNTA","PEPD","PEPT","AMPD") ~ "zinc",
                           clustername %in% unique(zinc$clustername) & clustername != "NANK" ~ "zinc",
                           #clustername %in% c("PEPD","PEPT","AMPD","PEPA","CYSS","PRLC") ~ "zinc binding",
                           #clustername %in% c("ISCS", "ISCU", "ISCA","SUFA","SUFB","SUFC","SUFD","SUFS","SUFE", "NFUA") ~ "Fe-S cluster biogenesis",
                           clustername %in% unique(ironsulfur$clustername) ~ "Fe-S cluster biogenesis",
                           #clustername %in% c("FEPA", "FEPB","FEPC","FEPD","FEPG","FECA","TONB", "EXBB","EXBD","FBPA","FBPB","FBPC", "ENTA","ENTB","ENTC", "ENTF", "FES", "FEOB","FUR", "ISCR",
                           #                  "ISCS", "ISCU", "ISCA","SUFA","SUFB","SUFC","SUFD","SUFS","SUFE", "NFUA","NSRR","SDAA","SDAB","PFLA","FECI","FHUF") ~ "iron",
                           clustername != "LPXH" & (clustername %in% c("MNTR","MNTH") | clustername %in% unique(manganese$clustername)) ~ "manganese",
                           clustername %in% c("LPXA","LPXB","LPXC","LPXD", "LPXH","LPXK","LPXL","LPXM","LPXT",
                                              "GMHA", "WAAA", "WAAU", "LPTA", "LPTB", "LPTC", "LPTD", "LPTE", "LPTF", "LPTG",
                                              "WZZE","WZZB", "GLF", "WBBI","WBBJ","WBBK", "NANA","NANK","NANT") ~ "LPS biosynthesis and assembly",
                           clustername %in% c("EMRA","MPRA","MARR","TOLC","ACRA","ACRB") ~ "efflux pumps and regulators",
                           !(clustername %in% c("FEPA","FEPB","FECA")) & (clustername %in% unique(om_uniprot$clustername) | clustername %in% c("NMPC","MALB","BLC") | clustername %in% unique(anna$gene_name)) ~ "OMPs",
                           clustername %in% unique(siderophores$clustername) | clustername %in% c("FEPB","FUR") ~ "iron import",
                           clustername %in% respiration$clustername ~ "respiration",
                           # clustername %in% c("CYOA", "CYOB","CYOC","CYOD",
                           #                    "SDHA","SDHB","SDHD",
                           #                    "HEMH", "HEMN", "HEMX","NADE",
                           #                    "HYBA","HYBC","HYBD","HYBO",
                           #                    "HYAA","HYAB",
                           #                    "FDOG","FDOH","FDOI",
                           #                    "FDNG","FDNH","FDNI",
                           #                    "NUOA","NUOB","NUOC","NUOE","NUOF","NUOG","NUOH","NUOI",
                           #                    "NUOJ","NUOK","NUOL","NUOM","NUON",
                           #                    "NDH","DLD","LLDR","LDLL","LDLP",
                           #                    "GLPA","GLPB","GLPC","GLPD","GLPF","GLPK",
                           #                    "GLPQ","GLPR","GLPT","GLPX",
                           #                    "CYDA","CYDB","CYDC",
                           #                    "NARP","NARG","NARH","DMSA","DMSB","DMSD",
                           #                    "UBIE","UBIJ","UBIH", "UBIF","UBIG","UBII",
                           #                    "RSXB","RSXC","RSXD","RSXE","RSXG","YDBK","POXB",
                           #                    "NAPA") ~ "respiration",
                           TRUE ~ "other"))


hits_df_whole = hits_df_whole %>%
  mutate(clustername= case_when(clustername %in% "ZUR" ~ "Zur",
                                clustername %in% "FUR" ~ "Fur",
                                clustername %in% "FES" ~ "Fes",
                                TRUE ~ capitalize_first_and_last(clustername))) 

names=c("CopA","CueO","Zur", "ZntA", "ZntR", "ZnuA","ZinT",
        "Fur", "EntA", "EntB","EntC","EntE","EntF","Fiu","Fes","BamA","BamB",
        "IscR", "IscA",
         "SufA","SufB","SufC","SufD","SufS","SufE",
        "FepA", "FepB","FepC","FepD","FepG","FecA",
        "MntH", "MprA",
        "LpxB","LpxD", "LpxH","LpxL","LpxM","NanT","NanA","NanK","Wzz",
        "WaaA", "WaaU", "LptA", "LptB", "LptC", "LptD","LptE", "LptF", "LptG",
        "Glf", "WbbI","WbbJ","WbbK",
        "ZnuA","ZnuB","ZnuC","FbpA","FbpB","FbpC","AcrB")

colours = data.frame("annot" = c("copper", "zinc", "Fe-S cluster biogenesis", 
                                 "iron import","manganese", "OMPs", "LPS biosynthesis and assembly",
                                 "respiration", "efflux pumps and regulators"),
                     "values" = c("aquamarine3","deeppink2","firebrick","brown1","mediumblue","#FF7F00","forestgreen","#984EA3","grey30"))
jColors <- as.character(colours$values)
names(jColors) <- colours$annot


p = ggplot(fdr_df_whole %>%
             filter(dataset == "true"),
           aes(sign(slopeH1)*sqrt(rssH0 - rssH1), log2(F_statistic + 1))) +
  geom_point(color = "gray", alpha = 0.5, size = 0.5, data=fdr_df_whole %>%
               filter(annot == "other" & !(clustername %in% hits_df_whole$clustername))) + 
  geom_point(aes(color = annot, fill = annot), alpha = 1,
             size = 0.7,
             data = filter(hits_df_whole, FDR < 0.05 & annot != "other")) +
  #scale_color_brewer(palette= "Set1")+
  #scale_fill_brewer(palette= "Set1")+
  #scale_color_manual(values=jColor) +
  ggrepel::geom_text_repel(
    aes(label = clustername, color=annot),
    data = filter(hits_df_whole, clustername %in% names & annot != "other"),
    size =4, fontface="bold",segment.size = 0.2,#min.segment.length = unit(1, "pt"),   
    max.overlaps = 18, show.legend = F) +
  scale_color_manual(values=jColors) +
  geom_hline(yintercept = log2(1.551792 + 1), linetype = 'dotted') +
  facet_wrap(~ detected_effectH1) +
  #scale_color_manual("", values = c("orange", "steelblue")) +
  labs(x = bquote(sign(kappa) %.% sqrt(~'RSS'^0~' - RSS'^1~'')),
       y = expression('log'[2]~'('*italic(F)*'-statistic + 1)'),
       colour = "") +
  theme_bw() +
  theme(legend.position = "right",
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=12),
        axis.title.x = element_blank(),
        strip.text = element_blank()) +
  guides(fill ="none", color="none")

ggsave(p, filename=paste0(plotdir, "volcano_abundance_and_stability_metals.png"), width = 8.6, height =3.6)


fdr_df_whole = fdr_df_whole %>%
  mutate(efsize= sign(slopeH1)*sqrt(rssH0 - rssH1),
         signif = log2(F_statistic + 1))

hits_df_whole = hits_df_whole %>%
  mutate(efsize= sign(slopeH1)*sqrt(rssH0 - rssH1),
         signif = log2(F_statistic + 1))

# ggplot(fdr_df_whole %>%
#          filter(dataset == "true"),
#        aes(sign(slopeH1)*sqrt(rssH0 - rssH1), log2(F_statistic + 1))) +
#   geom_point(color = "gray", alpha = 0.5, size = 0.5, data=fdr_df_whole %>%
#                filter(annot == "other" & !(clustername %in% hits_df_whole$clustername))) + 
#   geom_point(aes(color = annot, fill = annot), alpha = 1,
#              size = 0.7,
#              data = filter(hits_df_whole, annot != "other")) +
#   #scale_color_brewer(palette= "Set1")+
#   #scale_fill_brewer(palette= "Set1")+
#   #scale_color_manual(values=jColor) +
#   ggrepel::geom_text_repel(
#     aes(label = clustername, color=annot),
#     data = filter(hits_df_whole, signif > 2 & efsize > 1),
#     size =4, fontface="bold",segment.size = 0.2,#min.segment.length = unit(1, "pt"),   
#     max.overlaps = 25, show.legend = F) +
#   scale_color_manual(values=jColors) +
#   geom_hline(yintercept = log2(1.69513 + 1), linetype = 'dotted') +
#   facet_wrap(~ detected_effectH1) +
#   #scale_color_manual("", values = c("orange", "steelblue")) +
#   labs(x = bquote(sign(kappa) %.% sqrt(~'RSS'^0~' - RSS'^1~'')),
#        y = expression('log'[2]~'('*italic(F)*'-statistic + 1)'),
#        colour = "") +
#   theme_bw() +
#   theme(legend.position = "right",
#         axis.text = element_text(size=12),
#         axis.title.y = element_text(size=12),
#         axis.title.x = element_blank(),
#         strip.text = element_text(size=12)) +
#   guides(fill ="none")




towrite = fdr_df_whole %>%
  mutate(clustername = case_when(clustername %in% "ZUR" ~ "Zur",
                                clustername %in% "FUR" ~ "Fur",
                                TRUE ~ capitalize_first_and_last(clustername))) %>%
  mutate(hit= case_when(clustername %in% unique(filter(hits_df_whole, FDR <= 0.05)$clustername) ~ "TRUE",
                        TRUE ~ "FALSE"),
         source = "whole_cell",
         score = sign(slopeH1)*sqrt(rssH0 - rssH1)) %>%
  distinct(clustername, slopeH1, rssH0,rssH1, score, hit, source, detected_effectH1, FDR,annot, F_statistic) 

write_xlsx(towrite,paste0(supptablesdir,"Supplementary_Table_3.xlsx"))

```


## Fig. 3b - volcano Keio with LPS highlighted


```{r}
capitalize_last = function(string){
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string)))
  return(paste(middle_letters, last_letter, sep = ""))
}

om_uniprot = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("outer membrane", subcellular_location_cc) | grepl("outer membrane", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

manganese = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("manganese", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

zinc = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>% 
  clean_names() %>%
filter(grepl("zinc", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

#patterns <- c("4Fe-4S", "2Fe-2S")
ironsulfur = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  #filter(grepl("iron-sulfur", gene_ontology_go) | grepl(paste(patterns,collapse="|"), cofactor)) %>%
  filter(grepl("iron-sulfur", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

siderophores = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("siderophore", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

patterns <- c("respir", "electron transfer")
respiration = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl(paste(patterns,collapse="|"), gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

copper = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("copper", gene_ontology_go) | grepl("Cu (2+)", cofactor)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)


lps_bios = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>%
  clean_names() %>%
  filter(grepl("lipopolisac", gene_ontology_go)) %>%
  mutate(gene = sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  mutate(gene = capitalize_last(gene)) %>%
  select(gene)

hits <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond == "nitro2") %>%
  mutate(cond=gsub(cond, pattern="nitro2", replacement="Mutant fitness")) %>%
  filter(!grepl("^NOT", gene)) %>%
  mutate(gene = case_when(gene == "murE (D3A)-kan" ~ "murE",
                          gene == "lpxC (G210S)-kan" ~ "lpxC",
                          gene == "yhbN/lptA" ~ "lptA",
                          gene == "imp4213" ~ "lptD",
                          gene == "yhbV" ~ "ubiV",
                          gene == "yhbU" ~ "ubiU",
                          gene == "yhbZ" ~ "obgE",
                          gene == "yhbT" ~ "ubiT",
                          gene == "yhbJ" ~ "rapZ",
                          gene == "yhbH" ~ "hpf",
                          gene == "yhbT" ~ "ubiT",
                          gene == "yhbJ" ~ "rapZ",
                          gene == "yhbC" ~ "rimP",
                          gene == "rfaB" ~ "waaB",
                          gene == "rfaC" ~ "waaC",
                          gene == "rfaD" ~ "hldD",
                          gene == "rfaE" ~ "hldE",
                          gene == "rfaF" ~ "waaF",
                          gene == "rfaG" ~ "waaG",
                          gene == "rfaI" ~ "waaO",
                          gene == "rfaJ" ~ "waaJ",
                          gene == "rfaL" ~ "waaL",
                          gene == "rfaP" ~ "waaP",
                          gene == "rfaQ" ~ "waaQ",
                          gene == "rfaS" ~ "waaS",
                          gene == "rfaY" ~ "waaY",
                          gene == "rfaZ" ~ "waaZ",
                          gene == "yifC" ~ "waaZ",
                          gene == "vacJ" ~ "mlaA",
                          gene == "lpcA" ~ "gmhA",
                          gene == "yaeT-kan" ~ "bamA",
                          gene == "gntY" ~ "nfuA",
                          gene == "ydhD" ~ "grxD",
                          gene == "mprA" ~ "emrR",
                          TRUE ~ gene)) %>%
  filter(!grepl("^not", gene))

anna <- readxl::read_xlsx(paste0(inputdir,"Anna_OMPs_msystems.00808-19-st003.xlsx"),sheet = 2) %>%
  filter(`localization (sucrose gradient)` == "OM")  %>%
  mutate(gene_name = capitalize(gene_name))

# firstup <- function(x) {
#   substr(x, 1, 1) <- tolower(substr(x, 1, 1))
#   x
# }

respir_TPP = c(firstup(unique(filter(hits_df_whole, annot == "respiration")$clustername)), "hemF", "cyoD","mrp","cyoC","ubiV","ubiU","ubiT")

hits = hits %>%
  mutate(annot = case_when(gene %in% c("copA","cueO", "cueR") | gene %in% copper$gene ~ "copper",
                           (gene %in% c("zur","zntR", "znuA","znuB","znuC","zinT") | gene %in% unique(zinc$gene)) & gene != "nanK" ~ "zinc",
                          !(gene %in% c("fepA","fepB","fepC","lptA","lptD")) & (gene %in% om_uniprot$gene | gene %in% anna$gene_name) ~ "OMPs",
                           gene %in% c("iscS", "iscU", "iscA","sufA","sufB","sufC","sufD","sufS","sufE","nfuA","grxD")  | gene %in% unique(ironsulfur$gene) ~ "Fe-S cluster biogenesis",
                           gene %in% c("fepA", "fepB","fepC","fepD","fepG","fecA","tonB", "exbB","exbD","fbpA","fbpB","fbpC", "entA","entB","entC", "entF", "fes", "fur", "iscR") ~ "iron import",
                            gene %in% c("sdhA","sdhB","sdhD","cyoA","cyoB","cyoC","cyoD",
                                              "hemH", "hemN", "hemF","nadE",
                                              "hybA","hybC","hybD","hybO",
                                              "hyaA","hyaB",
                                              "fdoG","fdoH","fdoI",
                                              "fdnG","fdnH","fdnGI",
                                              "nuoA","nuoB","nuoC","nuoE","nuoF","nuoG","nuoH","nuoI",
                                              "nuoJ","nuoK","nuoL","nuoM","nuoN",
                                              "ndh","dld","ldlR","ldlL","ldlP",
                                              "glpA","glpB","glpC","glpD","glpF","glpK",
                                       "glpQ","glpR","glpT","glpX",
                                              "cydA","cydB","cydC",
                                              "narP","narG","narH","dmsA","dmsB","dmsD",
                                              "ubiE","ubiJ","ubiH", "ubiF","ubiG","ubiI",
                                              "rsxB","rsxC","rsxD","rsxE","rsxG", "trxC") | gene %in% respiration$gene  | gene %in% respir_TPP ~ "respiration",
                           gene != "cueR" & (gene %in% c("lpxA","lpxB","lpxC", "lpxD", "lpxH", "gmhA","gmhB", "waaA","waaF", "waaU", "lptA", "lptD", "lptC", "lptB", "wzzE","waaB","waaC","hldD","hldE","waaF","waaG","waaO","waaJ","waaL", "waaP","waaQ","waaS","waaY","waaZ", "waaZ","waaF","waaG","waaQ","waaB") | gene %in% lps_bios$gene) ~ "LPS biosynthesis and assembly",
                           gene != "lpxH" & (gene %in% c("mntR","mntH","sodA") | gene %in% unique(manganese$gene)) ~ "manganese",
                           gene %in% c("acrA","acrB","tolC","emrR","emrA","emrB") ~ "efflux pumps and regulators",
                           TRUE ~ "other"))

colours = data.frame("annot" = c("copper", "zinc", "Fe-S cluster biogenesis", 
                                 "iron import","manganese", "OMPs", "LPS biosynthesis and assembly",
                                 "respiration", "efflux pumps and regulators"),
                     "values" = c("aquamarine3","deeppink2","firebrick","brown1","mediumblue","#FF7F00","forestgreen","#984EA3","grey30"))
jColors <- as.character(colours$values)
names(jColors) <- colours$annot



names=c("COPA","CUEO","CUER","ZUR", "ZNTR", "ZNUA","ZINT",
       "ISCU", "ISCA",
        "SUFA","SUFC","SUFD","SUFS","SUFE","NFUA","GRXD",
        "FEPB","FEPC","FEPD","FEPG","EXBB",
         "ISCR", "SUFD",
        "NRDE","NRDF", 
       "LPXA","LPXB","LPXD","LPXC", #LpxK","LpxL","LpxM","LpxT",
       "GMHA", "WAAA", "WAAC","WAAU","WAAF","WAAG","WAAQ","WAAB","GMHB",
       "LPTA", "LPTD", "LPTC", "LPTB",# "LptE", "LptF", "LptG",
       "WZZE",
       "EMRA","EMRB","EMRR","ACRA",
         "MNTR",
       "ZNUA","ZNUC","FBPA","FBPB","FBPC")

# Define a function to capitalize the first letter and lowercase the rest
capitalize <- function(string) {
 paste0(tolower(substring(string, 1, nchar(string) - 1)), toupper(substring(string, nchar(string), nchar(string)))) 
}

names <- capitalize(names)
names[names == "zuR"] <- "zur"
names[names == "fuR"] <- "fur"
names[names == "fiU"] <- "fiu"
names[names == "feS"] <- "fes"



#pdf(paste0(plotdir, "4d_Keio_volcano_nx.pdf"),width= 3.2, height = 3.5)


p =  ggplot(hits,  aes(x = log2fc, y = -log10(padj))) +
  geom_point(col='grey70', size=0.5, show.legend=F) +
  geom_point(data=hits %>% filter(padj <= 0.05 & annot != "other"), aes(col=annot,fill=annot),size=0.7, show.legend=T) +
geom_abline(intercept = -log10(0.05), slope = 0, linetype = "dashed") +
  #geom_vline(xintercept = 1,linetype ="dashed") +
  #geom_vline(xintercept = -1, linetype ="dashed") +
  facet_wrap(~cond, ncol=1) +
  ggrepel::geom_text_repel(aes(label = gene, color=annot), force = 0.5,max.overlaps = 20,min.segment.length = 1, size = 4, segment.size = 0.2,data = hits %>% filter(gene %in% names & padj <= 0.05),fontface = "bold.italic") +
   scale_color_manual(values=jColors) +
  scale_fill_manual(values=jColors) +
  #scale_fill_brewer(palette= "Set1")+
  ylab(expression('-log'[10]~'(adjusted p-val)')) +
  xlab(expression('log'[2]~'(FC mutant fitness)')) +
  theme_bw() +
  theme(strip.text = element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=12)) +
  guides(colour="none",
         fill="none")


ggsave(p, filename=paste0(plotdir, "3b_Keio_volcano.png"), width = 4.6, height =3.6)
#dev.off()

towrite=hits %>%
  select(-cond) %>%
  write_xlsx(paste0(supptablesdir,"Supplementary_Table_4.xlsx"))


# ggplot(hits,  aes(x = log2fc, y = -log10(padj))) +
#   geom_point(col='grey70', size=0.5, show.legend=F) +
#   geom_point(data=hits %>% filter(padj <= 0.05 & annot != "other"), aes(col=annot,fill=annot),size=0.7, show.legend=T) +
# geom_abline(intercept = -log10(0.05), slope = 0, linetype = "dashed") +
#   #geom_vline(xintercept = 1,linetype ="dashed") +
#   #geom_vline(xintercept = -1, linetype ="dashed") +
#   facet_wrap(~cond, ncol=1) +
# #  ggrepel::geom_text_repel(aes(label = gene), force = 0.5,max.overlaps = 50, size = 3, data = labelhits,fontface = "italic") +
#   ggrepel::geom_text_repel(aes(label = gene, color=annot), force = 0.5,max.overlaps = 50, size = 4, data = hits %>% filter(padj <= 0.05 & log2fc > 0.25),fontface = "bold.italic") +
#    scale_color_manual(values=jColors) +
#   scale_fill_manual(values=jColors) +
#   #scale_fill_brewer(palette= "Set1")+
#   ylab(expression('-log'[10]~'(adjusted p-val)')) +
#   xlab(expression('log'[2]~'(FC mutant fitness)')) +
#   theme_bw() +
#   theme(strip.text = element_blank(),
#         axis.text = element_text(size=12),
#         axis.title = element_text(size=12))# +
#   guides(colour="none",
#          fill="none")



```


# thermal profiles OM

```{r}

df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T)


plot2dTppFcHeatmap <- function(df, name, drug_name = "", midpoint = 1){
  clustername <- temperature <- conc <- rel_value <- fc <- NULL
  
  heat_df <- df %>% 
    dplyr::filter(clustername == name) %>% 
    dplyr::select(clustername, temperature,conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(temperature, levels = rev(sort(unique(temperature)))), conc = as.factor(conc)) %>% 
    dplyr::group_by(clustername, temperature, conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup()
  
  ggplot(heat_df, aes(conc, temperature)) +
    geom_tile(aes(fill = fc)) +
    scale_fill_gradient2("Fold change", low = "firebrick", mid = "khaki", high = "darkgreen", midpoint = midpoint) +
    labs(x = paste(drug_name, "conc."),y = expression("Temperature ("*~degree*C*")")) +
    ggtitle(name) +
    theme_minimal()
}

# names = c("purC", "purD","purE", "purF", "purH", "purK", "purL", "purM", "purN", "purQ", "apt", "aroC", "carB", "deoD","hpt", "gcvPA", "gcvPB","glyA", "pyrE")

names = c(#"LPXB","WAAA","BAMA",
  "LPTA","LPTB","LPTC","LPTD","LPTE", "LPTF", "LPTG")


heat_df <- df_final_whole %>% 
    dplyr::filter(clustername %in% names) %>% 
    dplyr::select(clustername, temperature,conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(temperature, levels = rev(sort(unique(temperature)))), conc = as.factor(conc)) %>% 
    dplyr::group_by(clustername, temperature, conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup()
  
capitalize_first_and_last <- function(string) {
  first_letter <- toupper(substring(string, 1, 1))
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string) - 1))
  return(paste(first_letter, middle_letters, last_letter, sep = ""))
}

heat_df=heat_df %>%
   mutate(clustername = capitalize_first_and_last(clustername))



p<- ggplot(heat_df, aes(conc, temperature)) +
    geom_tile(aes(fill = fc)) +
  scale_fill_gradientn(colors = c("firebrick","khaki","darkgreen"),
                       space = "Lab", guide = "colourbar", na.value = "white", limits=c(0.2,1.65),
                       values = scales::rescale(c(0.2, 0.5,0.7, 1, 1.3,1.6, 1.65))) +
    # scale_fill_gradient2("Fold change", limits = c(min(heat_df$fc),1.52), low = "firebrick", mid = "khaki", high = "darkgreen", midpoint = 1) +
     # scale_fill_gradient2("Fold change", low = "firebrick", mid = "khaki", high = "#6A963F", midpoint = 1) +
    labs(x = paste("Nitroxoline (µg/ml)"),y = expression("Temperature ("*~degree*C*")"), fill = "Fold change") +
    facet_wrap(~ clustername, scales = "free", ncol = 4) +
    theme_minimal() +
  theme(legend.position= 'right',
        axis.text.x = element_text(size = 9),
        strip.text =  element_text(size = 12))

  
pdf(paste0(plotdir,"3b_envstress_thermal_new.pdf"), width = 7, height = 3.8)

shift_legend2 <- function(p) {
  # ...
  # to grob
  gp <- ggplotGrob(p)
  facet.panels <- grep("^panel", gp[["layout"]][["name"]])
  empty.facet.panels <- sapply(facet.panels, function(i) "zeroGrob" %in% class(gp[["grobs"]][[i]]))
  empty.facet.panels <- facet.panels[empty.facet.panels]

  # establish name of empty panels
  empty.facet.panels <- gp[["layout"]][empty.facet.panels, ]
  names <- empty.facet.panels$name
  # example of names:
  #[1] "panel-3-2" "panel-3-3"

# now we just need a simple call to reposition the legend
  reposition_legend(p, 'center', panel=names, plot =F)
}

grid::grid.draw(shift_legend2(p))

#p

dev.off()



capitalize_first_and_last <- function(string) {
  first_letter <- toupper(substring(string, 1, 1))
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string) - 1))
  return(paste(first_letter, middle_letters, last_letter, sep = ""))
}

heat_df <- df_final_whole %>% 
    dplyr::select(clustername, temperature,conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(temperature, levels = rev(sort(unique(temperature)))), conc = as.factor(conc)) %>% 
    dplyr::group_by(clustername, temperature, conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup() %>%
   mutate(clustername = capitalize_first_and_last(clustername)) %>%
  rename(`nitroxoline_conc (µg/ml)` = conc,
         fold_change=fc,
         `temperature (ºC)` = temperature,
         protein=clustername)

write_xlsx(heat_df, paste0(sourcedatadir,"source_Fig.3b_Fig.4cde_EDFig.6b.xlsx"))

```





## Fig. 3c - NPN

```{r}

test = fread("~/Documents/EMBL/NX/NPN/220420/toplotforpaper.txt") %>%
  filter(!(Condition == "Nitroxoline_0.9" & NPN_uptake > 40000)) %>%
  group_by(Condition, Biol_rep) %>%
  summarise(meanNPN = median(NPN_uptake))



test= test %>% filter(Condition != "Nitroxoline_0.225")

pdf(paste0(plotdir, "355excit405_emit_timepoints altogether.pdf"), width = 5.5, height = 5)

# colours = data.frame("Condition" = c("PolymyxinB_10","EDTA_0.4", "Nitroxoline_0.9", "Nitroxoline_1.8", "Nitroxoline_0.45", "Chloramphenicol_8", "nodrug_0"),
#                      "values" = c("#CC79A7","#D55E00", "#0072B2","#0072B2","#0072B2", "#F0E442", "#999999"))
# jColors <- as.character(colours$values)
# names(jColors) <- colours$Condition

colours = data.frame("Condition" = c("PolymyxinB_10","EDTA_0.4", "Nitroxoline_0.9", "Nitroxoline_1.8", "Nitroxoline_0.45", "Chloramphenicol_8", "nodrug_0"),
                     "values" = c("#999999","#999999", "#999999","#999999","#999999", "#999999", "#999999"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Condition


 ggerrorplot(test, x = "Condition", y = "meanNPN", error.plot="errorbar",
          order = c("PolymyxinB_10","EDTA_0.4", "Nitroxoline_0.9", "Nitroxoline_1.8", "Nitroxoline_0.45",  "Chloramphenicol_8", "nodrug_0"),
  add = "se") +
   stat_summary(
    geom = "point",
    shape = 95,
    size = 12,
    col = "black",
    fun.y = "mean") +
  #facet_wrap(~ Time) +
  geom_point(aes(fill = as.factor(Condition)),alpha = 0.7,size =3,pch=21,position = position_jitterdodge(1, dodge.width = .5)) +
   scale_fill_manual(values=jColors) +
   scale_color_manual(values=jColors) +
   #scale_colour_grey() +
  labs(fill = "", y = "NPN fluorescence (405 nm)") +
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = "Chloramphenicol_8", size = 7) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         legend.position = "none")

dev.off()


test = test %>%
  mutate(Concentration = sapply(strsplit(as.character(Condition), "_"),'[',2),
         Condition = sapply(strsplit(as.character(Condition), "_"),'[',1)) %>%
  mutate(Concentration = case_when(Condition == "EDTA" ~ paste0(Concentration, " mM"),
                                   TRUE ~ paste0(Concentration, " (µg/ml)"))) %>%
  rename(Biological_rep = Biol_rep,
         NPN_fluorescence=meanNPN)

writexl::write_xlsx(test, paste0(sourcedatadir,"source_Fig.3e.xlsx"))


```

## Fig. 3c - lptD synergy

### new version - 240329 using data from 220525
```{r}
toplot = fread("~/Documents/EMBL/Nx/220525_lptD/plate_norm.txt")

colours = data.frame("strain" = unique(toplot$strain), "values" = rep("darkgrey", length(toplot$strain))) %>%
  dplyr::mutate(values = case_when(strain == "BW" ~ "black",
                                   strain == "lptD" ~ "turquoise4",
                                   strain == "emrR" ~ "darkorange",
                                   strain == "tolC" ~ "darkblue",
                                   TRUE ~ values)) %>%
  distinct()
jColors <- as.character(colours$values)
names(jColors) <- colours$strain

toplot=filter(toplot,Time_h == 9.75 & strain %in% c("BW","lptD")) %>%
  mutate(strainrep = interaction(strain, Replicate))

pdf(paste0(plotdir, "Fig3c_dose_response_forpaper_220525.pdf"), width = 3.2, height = 3.5)

ggline(toplot, x = "Concentration", y = "ODnorm_bug_rob",color="strain",
       #group = "strainrep",
       add= "mean_se") +
      labs(x = "Nitroxoline (µg/ml)", y = "normalised OD at 9h", colour = " ") +
        #ylim(c(0,1.5)) +
  scale_color_manual(values = c("darkgrey", "turquoise4"), labels = c("wt", expression(italic("lptD4213")))) +
        theme_light() +
  theme(legend.position = "bottom")

# ggplot(toplot, aes(x=Concentration, y=ODnorm_bug_rob, group = strainrep, color = Strain)) +
#   geom_line()
dev.off()


towrite = toplot %>%
  group_by(strain, Concentration) %>%
  dplyr::summarise(mean_OD = mean(ODnorm_bug_rob),
            se_OD=sqrt(sum((ODnorm_bug_rob-mean(ODnorm_bug_rob))^2/(length(ODnorm_bug_rob)-1)))/sqrt(length(ODnorm_bug_rob))) %>%
  ungroup() %>%
  dplyr::rename(`Nitroxoline conc (mg/ml)` = Concentration) %>%
  mutate(strain=gsub(strain, pattern="BW",replacement="E. coli BW25113"),
         strain=gsub(strain, pattern="lptD",replacement="E. coli lptD4213"))

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.3c.xlsx"))

```


### old version
```{r}
toplot = fread("~/Documents/EMBL/Nx/220418_allmutants_Nx_analysis/plate_norm.txt")

colours = data.frame("Strain" = unique(filter(toplot, Strain != "nobug")$Strain), "values" = rep("darkgrey", length(unique(filter(toplot, Strain != "nobug")$Strain)))) %>%
  dplyr::mutate(values = case_when(Strain %in% c("BW_EC", "BW_SG") ~ "black",
                                   Strain == "lptD" ~ "turquoise4",
                                   #Strain == "copA" ~ "blue",
                                   Strain == "mreB" ~ "turquoise4",
                                   Strain == "ampD" ~ "darkorange",
                                   TRUE ~ values))
jColors <- as.character(colours$values)
names(jColors) <- colours$Strain

toplot=filter(toplot,Time_h == 9.75 & Strain %in% c("BW_EC","lptD")) %>%
  mutate(strainrep = interaction(Strain, Replicate))

pdf(paste0(plotdir, "dose_response_forpaper.pdf"), width = 3, height = 3.5)

ggline(toplot, x = "Concentration", y = "ODnorm_bug_rob",color="Strain",
       #group = "strainrep",
       add= "mean_se") +
      labs(x = "Nitroxoline (µg/ml)", y = "normalised OD at 9h", colour = " ") +
        #ylim(c(0,1.5)) +
  scale_color_manual(values = c("darkgrey", "turquoise4"), labels = c("BW_EC", "lptD")) +
  #scale_colour_brewer(palette = "Set1") +
  #stat_compare_means(method = "anova", label.y = 200) +      # Add global p-value
  # stat_compare_means(label = "p.signif", method = "t.test",
  #                    ref.group = "BW_EC", size = 5)  +
        theme_light() +
  theme(legend.position = "bottom",
        axis.text.x = element_text(angle=60, hjust = 0.5, vjust = 0.5))

# ggplot(toplot, aes(x=Concentration, y=ODnorm_bug_rob, group = strainrep, color = Strain)) +
#   geom_line()
dev.off()


towrite = toplot %>%
  group_by(Strain, Concentration) %>%
  summarise(mean_OD = mean(ODnorm_bug_rob),
            se_OD=sqrt(sum((ODnorm_bug_rob-mean(ODnorm_bug_rob))^2/(length(ODnorm_bug_rob)-1)))/sqrt(length(ODnorm_bug_rob))) %>%
  ungroup() %>%
  rename(`Nitroxoline conc (mg/ml)` = Concentration) %>%
  mutate(Strain=gsub(Strain, pattern="BW_EC",replacement="E. coli BW25113"),
         Strain=gsub(Strain, pattern="lptD",replacement="E. coli lptD4213"))

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.3c.xlsx"))

```




# Fig. 4 - nx as metallophore

## Fig. 4a - volcano with metals






## Fig. 4f - ESRF

#### with treated and untreated metal content
```{r}
library(rstatix)
library(plotrix)


toplot = readxl::read_xlsx(paste0(inputdir, "240107_ESRF_data_edit_MT_240123_SG.xlsx"), skip=9) %>%
  rename(Content=`Content [ng/cm^2]`) %>%
  mutate(Cell= as.factor(Cell))%>%
  mutate(Condition=gsub(Condition, pattern="^treated", replacement="+ 1 µg/ml NX"))

toplot$Content[toplot$Content < 0] <- 0

stat.test <- toplot %>%
  filter(Compartment == "cytoplasm") %>%
  dplyr::distinct(Compartment, Content, Element,Condition) %>%
  dplyr::group_by(Element, Compartment) %>%
  t_test(Content ~ Condition, var.equal = FALSE) %>%
  # adjust_pvalue(method = "BH") %>%
  # add_significance("p.adj")
  add_significance("p")

stat.test <- stat.test %>%
  add_xy_position(fun = "mean_sd", x = "Element", dodge = 0.8) 

toplot = toplot %>%
  group_by(Element,Compartment, Condition) %>%
  mutate(meancont=mean(Content),
         secont=std.error(Content)) %>%
  filter(Compartment == "cytoplasm") %>%
  ungroup()

pdf(paste0(plotdir,"ESRF.pdf"),width=3.6, height=3.4)

colours = data.frame("Condition" = c("untreated","+ 1 µg/ml NX"),
                     "values" = c("black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Condition

ggplot(toplot, aes(x = Element, y = meancont, color = Condition)) +
  geom_bar(stat = "identity", position = position_dodge(0.8), width = 0.7,fill="white") +
   geom_errorbar(aes(ymin = meancont - secont, ymax = meancont + secont), position = position_dodge(0.8), width = 0.15) +
  geom_jitter(aes(x = Element,y=Content,fill = Condition), position = position_jitterdodge(dodge.width = 0.8, jitter.width = 0.1,jitter.height = 0.01), size = 1.2, shape = 21, alpha = 0.7) +
  #stat_summary(fun.data = mean_se, position = position_dodge(0.8), geom = "errorbar") +
  scale_color_manual(values = jColors) +
   scale_fill_manual(values = jColors) +
  stat_pvalue_manual(stat.test, tip.length = 0.01, label = "p.signif", y.position = 2.2) +
  #facet_wrap(~Compartment) +
  labs(color = "", y = expression('Elemental area density (ng/cm'^2~')'), fill="") + 
   guides(color="none") +
   theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_text(size=9),
        legend.position = "top")
 
 
dev.off()

stat.test = stat.test %>%
  distinct(p.signif,Compartment,Element,p)

towrite = toplot %>%
  rename(mean_area_density= meancont,
         se_area_density= secont,
         area_density=Content) %>%
 select(-c(`...6`,`...7`,`...8`)) %>%
  inner_join(stat.test)

write_xlsx(towrite,paste0(sourcedatadir, "source_Fig.4f.xlsx"))


```

#### unused version with ratio
```{r}
#toplot = readxl::read_xlsx(paste0(inputdir, "240107_ESRF_data_edit_MT_240123_SG.xlsx"), skip=9) %>%
toplot = readxl::read_xlsx(paste0(inputdir, "240107_ESRF_data_edit_MT_240123_SG.xlsx"), skip=9) %>%
  rename(Content=`Content [ng/cm^2]`) %>%
  mutate(Cell= as.factor(Cell))%>%
  mutate(Condition=gsub(Condition, pattern="^treated", replacement="+ 1 µg/ml NX")) #%>%
  #filter(!(Cell %in% c(6:9)))

toplot$Content[toplot$Content < 0] <- 0

stat.test <- toplot %>%
  dplyr::distinct(Compartment, Content, Element,Condition) %>%
  dplyr::group_by(Element, Compartment) %>%
  t_test(Content ~ Condition, var.equal = TRUE) %>%
  #adjust_pvalue(method = "BH") %>%
  add_significance("p")

stat.test <- stat.test %>%
  add_xy_position(fun = "mean_sd", x = "Element", dodge = 0.8) 

signif = stat.test %>%
  distinct(Compartment,Element, p,p.signif)

torep = min(filter(toplot, Content != 0)$Content)

test = toplot %>%
  mutate(Content = replace(Content, Content == 0, torep)) %>%
  pivot_wider(id_cols = c(Compartment,Element,Cell), names_from = Condition,values_from = Content) 


library("plotrix")

test = test %>%
  mutate(ratio = (`+ 1 µg/ml NX` + 0.001)/(untreated + 0.001)) %>%
  inner_join(signif)%>%
  group_by(Element,Compartment) %>%
  mutate(meanratio=mean(ratio),
         sdratio=sd(ratio),
         seratio=sqrt(sum((ratio-mean(ratio))^2/(length(ratio)-1)))/sqrt(length(ratio))) %>%
         #seratio=std.error(ratio)) %>%
  ungroup()


pdf(paste0(plotdir, "ESRF_2.pdf"), width = 5, height = 4.5)
 
colours = data.frame("Compartment" = c("periplasm","cytoplasm"),
                     "values" = c("black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Compartment

ggplot(test %>% filter(Compartment == "cytoplasm"), aes(x = Element, y = meanratio)) +
  geom_bar(stat = "identity", position = position_dodge(0.8), width = 0.7,fill="grey40") +
  geom_errorbar(aes(ymin = meanratio - sdratio, ymax = meanratio + sdratio),# position = position_dodge(0.8), 
                width = 0.15, color="grey40") +
  geom_jitter(aes(x = Element,y=ratio), width = 0.1,height = 0.1, 
              size = 4, shape = 21, alpha = 1, fill="grey90", color="grey20") +
  geom_abline(slope = 0, intercept = 0) +
  scale_y_continuous(trans="log2", labels = c(0.0625,0.25,1,4,8,16,32), breaks=c(0.0625,0.25,1,4,8,16,32)) +
  geom_text(data = test %>%
              filter(Compartment=="cytoplasm") %>%
              distinct(Element, p.signif),
            aes(x = Element, y = 35, label = p.signif), 
            position = position_dodge(0.8), size = 6) +
  scale_color_manual(values = jColors) +
  scale_fill_manual(values = jColors) +
  labs(color = "",fill="", y = expression(log[2]~"(FC treated/untreated)")) +
  theme_bw() + 
  guides(color="none") +
  theme(axis.title.x = element_blank(),
        legend.position = "top",
        axis.text = element_text(size=18),
        axis.title = element_text(size=20))

dev.off()




```


## Fig. 4c - copper

```{r}
df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T) %>%
  mutate(clustername = capitalize_first_and_last(clustername))

a <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "CopA",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "CueO",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "CyoB",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "YobA",
  drug_name = "Nitroxoline")


pdf(paste0(plotdir, "metals_thermal_profile_copper.pdf"), width = 5.8, height = 2)

cowplot::plot_grid(a, b, ncol = 2)
#a + b + c + d + plot_layout(guides = "collect")

dev.off()


df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T) 

a <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "MSBA",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "LPTD",
  drug_name = "Nitroxoline")

cowplot::plot_grid(a, b, ncol = 2)
#a + b + c + d + plot_layout(guides = "collect")


```




## Fig. 4d - zinc

```{r}

df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T)%>%
  mutate(clustername= case_when(clustername == "ZUR" ~ "Zur",
                                TRUE ~ capitalize_first_and_last(clustername))) 


a <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "Zur",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "ZnuA",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "ZntR",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "ZntA",
  drug_name = "Nitroxoline")
# e <- plot2dTppFcHeatmap(
#   df = df_final_whole, name = "ZINT",
#   drug_name = "Nitroxoline")


pdf(paste0(plotdir, "metals_thermal_profile_zinc.pdf"), width = 5.8, height = 4)

cowplot::plot_grid(a, b,c,d, ncol = 2)
#a + b + c + d + plot_layout(guides = "collect")

dev.off()


```



## Fig. 4e - manganese

```{r}
df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T) %>%
  mutate(clustername = capitalize_first_and_last(clustername))

a <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "MntH",
  drug_name = "Nitroxoline")
# b <- plot2dTppFcHeatmap(
#   df = df_final_whole, name = "MNTR",
#   drug_name = "Nitroxoline")


pdf(paste0(plotdir, "metals_thermal_profile_manganese.pdf"), width = 2.9, height = 2)
a
#cowplot::plot_grid(a, b)
#a + b + c + d + plot_layout(guides = "collect")

dev.off()


```


## unused - respiration

```{r}

df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T)


names = c("SODA","ACNA","FUMC","ZWF",
          "KATG", "AHPC", "GOR", "GRXA","TRXC")


heat_df <- df_final_whole %>% 
    dplyr::filter(clustername %in% names) %>% 
    dplyr::select(clustername, temperature,conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(temperature, levels = rev(sort(unique(temperature)))), conc = as.factor(conc)) %>% 
    dplyr::group_by(clustername, temperature, conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup()
  

p<- ggplot(heat_df, aes(conc, temperature)) +
    geom_tile(aes(fill = fc)) +
  # scale_fill_gradientn(colors = c("firebrick","khaki","darkgreen"),
  #                      space = "Lab", guide = "colourbar", na.value = "white", limits=c(0.35,1.635),
  #                      values = scales::rescale(c(0.35, 0.45,0.7,0.9, 1,1.2, 1.3, 1.5, 1.632))) +
    scale_fill_gradient2("Fold change", low = "firebrick", mid = "khaki", high = "darkgreen", midpoint = 1) +
    labs(x = paste("Nitroxoline (µg/ml)"),y = expression("Temperature ("*~degree*C*")"), fill = "Fold change") +
    facet_wrap(~ clustername, scales = "free", ncol = 3) +
    theme_minimal() +
  theme(legend.position= 'right',
        axis.text.x = element_text(size = 9),
        strip.text =  element_text(size = 12))

  
pdf(paste0(plotdir,"4b_redoxstress_thermal.pdf"), width = 5.3, height = 3.8)

shift_legend2 <- function(p) {
  # ...
  # to grob
  gp <- ggplotGrob(p)
  facet.panels <- grep("^panel", gp[["layout"]][["name"]])
  empty.facet.panels <- sapply(facet.panels, function(i) "zeroGrob" %in% class(gp[["grobs"]][[i]]))
  empty.facet.panels <- facet.panels[empty.facet.panels]

  # establish name of empty panels
  empty.facet.panels <- gp[["layout"]][empty.facet.panels, ]
  names <- empty.facet.panels$name
  # example of names:
  #[1] "panel-3-2" "panel-3-3"

# now we just need a simple call to reposition the legend
  reposition_legend(p, 'center', panel=names, plot =F)
}

grid::grid.draw(shift_legend2(p))

dev.off()


```




##Fig. 4f - copA CB nx copper

```{r}
dtp = fread("~/Documents/EMBL/Nx/Nx_paper/input/CB_copA.txt", sep="\t") %>%
  mutate(strain_new = case_when(strain %in% c("copA1", "copA2") ~ "copA",
                                TRUE ~ strain)) %>%
  group_by(strain_new, nitroxoline, cucl2) %>%
  summarise(meanfit=mean(fit))


pdf(paste0(plotdir,"copA_copperNx_syn.pdf"), width = 2, height=2)
ggplot(dtp %>% filter(strain_new == "copA"),
       aes(factor(nitroxoline), factor(cucl2), fill=meanfit)) +
  geom_raster() + 
  scale_fill_gradientn(colours = c("white","#009999"), 
    limits=c(0, 1.1)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  facet_grid(~strain_new) +
  labs(x='Nitroxoline (µg/mL)', y=expression(CuCl[2]*' (µg/mL)', fill='Fitness')) +
  theme(aspect.ratio=1, legend.position='none') 
dev.off()

pdf(paste0(plotdir,"wt_copperNx_antag.pdf"), width = 2, height=2)
ggplot(dtp %>% filter(strain_new == "wt"),
       aes(factor(nitroxoline), factor(cucl2), fill=meanfit)) +
  geom_raster() + 
  scale_fill_gradientn(colours = c("white","#d9a520"), 
    limits=c(0, 1.1)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  facet_grid(~strain_new) +
  labs(x='Nitroxoline (µg/mL)', y=expression(CuCl[2]*' (µg/mL)', fill='Fitness')) +
  theme(aspect.ratio=1, legend.position='none')#,
        # axis.text.x= element_blank(),
        # axis.title=element_blank())

dev.off()


pdf(paste0(plotdir,"wt_copperNx_greyscale.pdf"), width = 1.5, height=1.5)
ggplot(dtp %>% filter(strain_new == "wt"),
       aes(factor(nitroxoline), factor(cucl2), fill=meanfit)) +
  geom_raster() + 
  scale_fill_gradientn(colours = c("white","darkgrey"), 
    limits=c(0, 1.1)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  facet_grid(~strain_new) +
  labs(x='Nitroxoline (µg/mL)', y=expression(CuCl[2]*' (µg/mL)', fill='Fitness')) +
  theme(aspect.ratio=1, legend.position='top')

dev.off()
```





## upset plot Keio copper, nitro etc 

```{r}
hits <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  # filter(cond %in% c("copper", "copper nitro1","copper nitro2", "nitro1","nitro2") & padj <= 0.01 & (log2fc > 0.5 | log2fc < -0.5)) %>%
  filter(cond %in% c("copper", "copper nitro1","copper nitro2", "nitro1","nitro2") & padj <= 0.01) %>%
  mutate(type = case_when(log2fc > 0 ~ "gain of fitness",
                          log2fc < 0 ~ "loss of fitness",
                          TRUE ~ "neutral"))
  
# make format right
movieInput <- hits %>%
  dplyr::select(c(gene, cond, type))
  
movieInput = dplyr::distinct(movieInput, gene, cond, type)
movieInput = movieInput %>%
  dplyr::mutate(combtype = interaction(gene, type, sep = ".")) %>%
  dplyr::select(combtype, cond)
  
test = table(movieInput)
test[test>0] = 1

test = as.data.frame(test)

test = test %>%
  pivot_wider(names_from = cond, values_from = Freq) %>%
  as.data.frame()

test = test[apply(test[,-1], 1, function(x) !all(x==0)),]

test1 = dplyr::mutate(test, C = ifelse(`copper`, "copper", NA),
       CN1 = ifelse(`copper nitro1`, "copper nitro1", NA),
       CN2 = ifelse(`copper nitro2`, "copper nitro2", NA),
       N1 = ifelse(`nitro1`, 'nitro1', NA),
       N2 = ifelse(`nitro2`, 'nitro2', NA))
# 
test1 = tidyr::unite(test1, "cond", C:N2, na.rm=T, sep='&')
test1 = test1 %>%
  dplyr::mutate(conservation = ifelse(grepl("&", cond), "conserved", "not conserved"),
                gene = sapply(strsplit(as.character(combtype), "."), '[', 1),
                type = sapply(strsplit(as.character(combtype), "."), '[', 2)) %>%
  dplyr::select(-combtype)
# 
# write.table(test1, paste0(outputdir, "forupsetplot_excludebenchmarking_211005.txt"), sep = "\t", quote = F, row.names = F)
# 
# test1 = read.table(paste0(outputdir, "forupsetplot_excludebenchmarking_211005.txt"), sep = "\t",header= T)
# pdf(paste0(plotdir, "upset_conservation.pdf"),width = 5, height = 3.5)
# 
# UpSetR::upset(test1, sets = c("S.aureus DSM20231", "S.aureus Newman","B. subtilis", "S. pneumoniae"),
#        #sets.bar.color = c("grey", "grey", "black","black"),
#        order.by = "degree",decreasing = F, keep.order = T,
#       set_size.show = T, set_size.numbers_size = 4,
#       set_size.scale_max = 300,
#       queries = list(list(query = elements, 
#              params = list("type", c("synergy","antagonism")), color = "#FFCC33", active = T),
#             list(query = elements, 
#              params = list("type", "synergy"), color = "#00999", active = T)))

# dev.off()

#data from Vlad
#all_data = readRDS(paste0(inputdir, "for_stacked_upset.rds"))

pdf(paste0(plotdir, "upset_conditions_Keio.pdf"), width = 14/2.54, height = 8/2.54)
ComplexUpset::upset(
  test1,
  c("copper", "copper nitro1", "copper nitro2", "nitro1", "nitro2"),
  #colnames(test1)[1:4],
  stripes = "white",
  #sort_intersections=FALSE,
  # intersections=list("S. pneumoniae", "B. subtilis","S.aureus Newman", "S.aureus DSM20231",
  #                     c("S.aureus Newman", "S.aureus DSM20231"),
  #                     c("S. pneumoniae", "S.aureus DSM20231"),
  #                     c("S.aureus Newman", "S. pneumoniae"),
  #                     c("S.aureus Newman", "B. subtilis"),
  #                     c("B. subtilis", "S.aureus DSM20231"),
  #                     c("B. subtilis", "S. pneumoniae"),
  #                     c("B. subtilis", "S.aureus DSM20231", "S.aureus Newman"),
  #                     c("S. pneumoniae", "S.aureus DSM20231", "S.aureus Newman"),
  #                     c("B. subtilis", "S.aureus DSM20231", "S. pneumoniae"),
  #                     c("B. subtilis", "S.aureus Newman", "S. pneumoniae"),
  #                     c("B. subtilis", "S.aureus Newman","S.aureus DSM20231", "S. pneumoniae")
  #   ),
sort_intersections_by=c('degree'),
#sort_intersections='descending',
#sort_sets='descending',
  base_annotations=list(
    'Intersection size'=intersection_size(
      #counts=FALSE,
      text = list(size=2),
      text_colors = c(on_background = "black", on_bar = "black"),
      mapping=aes(fill=type)
    )
  ),
  set_sizes=(upset_set_size(
    #mapping = aes(fill=type),
    geom = geom_bar(width = 0.6),
    position = "right") +
    geom_text(aes(label=..count..), hjust=-0.1, stat='count', size =2) +
      expand_limits(y=300) +
      theme(axis.text.x=element_text(size = 6))),
  matrix=intersection_matrix(
    geom = geom_point(size = 1.5),
    segment = geom_segment(),
    outline_color = list(active = "black", inactive = "grey70")
  ),
  themes = upset_modify_themes(list('Intersection size'=theme(axis.line.x = element_blank(),
                                                              axis.ticks.x = element_blank(),
                                                              axis.line.y = element_line(),
                                                              text = element_text(size=9),
                                                              legend.position = 'none',
                                                              axis.ticks.y = element_line(),
                                                              plot.background = element_blank(),
                                                              panel.grid = element_blank()),
                                    'intersections_matrix'=theme(axis.line.x = element_blank(),
                                                                 axis.ticks.x = element_blank(),
                                                                 axis.title = element_blank(),
                                                                 text = element_text(size=9),
                                                                plot.background = element_blank(),
                                                                 panel.grid = element_blank()),
                                    'overall_sizes'=theme(axis.line.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          axis.line.x = element_line(),
                                                          axis.ticks.x = element_line(),
                                                          plot.background = element_blank(),
                                                          text = element_text(size=9),
                                                          panel.grid = element_blank()))),
  width_ratio=0.3,
  height_ratio = 0.5 , sort_sets = F
) & scale_fill_manual(values = c("#FFCC33", "#009999", "grey"))
dev.off()

```

## heatmap
```{r}
hits <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  # filter(cond %in% c("copper", "copper nitro1","copper nitro2", "nitro1","nitro2") & padj <= 0.01 & (log2fc > 0.5 | log2fc < -0.5)) %>%
  filter(cond %in% c("copper", "copper nitro1","copper nitro2", "nitro1","nitro2") & padj <= 0.01) %>%
  mutate(type = case_when(log2fc > 0 ~ "gain of fitness",
                          log2fc < 0 ~ "loss of fitness",
                          TRUE ~ "neutral"))
# clustered heatmaps
hits = hits %>%
  dplyr::select(gene,cond,log2fc)

mymat = hits %>% 
  dplyr::select(c(gene, cond, log2fc)) %>%
  unique()

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}
# mymat = fix_na(mymat)

library(magrittr)

mymat = mymat %>% pivot_wider(names_from = cond, values_from = log2fc) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$gene) %>%
  dplyr::select(-gene) %>%
  as.matrix()

matrix.please<-function(x) {
  m<-as.matrix(x[,-1])
  rownames(m)<-x[,1]
  m
}

toplot = matrix.please(as.data.frame(mymat))
toplot=mymat

toplot[is.na(toplot)] = 0
#rownames(toplot) = as.numeric(rownames(toplot))
#toplot = t(toplot)
hc <- hclust(as.dist(1-cor(t(toplot), method="pearson")), method="complete")
hr <- hclust(as.dist(1-cor(toplot, method="pearson")), method="complete")

dend1 <- as.dendrogram(hclust(as.dist(1-cor(toplot, method="pearson")), method="complete"))
c_group <- 6 # number of clusters


library(gplots)


pdf(paste0(plotdir,"clustered_Keio.pdf"), width=3, height=55)

mat_data <- toplot[ order(row.names(toplot)), ]
note <- apply(as.matrix(mat_data), c(1,2), function(elem) { if (is.na(elem)) return(""); as.character(elem) })

colfunc <- colorRampPalette(c("#006666", "white", "#CC9933"))

heatmap.2(mat_data, 
          col=colfunc(15),
          #breaks=colors, 
          Rowv= as.dendrogram(hc),
          Colv = as.dendrogram(hr),
          # key.xtickfun = function() {
          #   breaks = c(0,10,20,30,40,50,60,70,80,90,100)
          #   #breaks = breaks[c(1,length(breaks))]
          #   list(at = parent.frame()$scale01(breaks),
          #        labels = breaks)},
          #ColSideColors=colorside,
          #cellnote = note,     
          #notecol="black",
          dendrogram = "row",
          density.info = "none",
          #RowSideColors = col_labels, 
          #colRow = c("\≤1", "2", "4", "8","16", "32", "64"),
          cexRow=0.3,
          #margins = c(10,15),   
          #xlab = "MIC (µg/ml)",
          scale="none",
          trace = "none",
          symm = F,
          symkey = F)#,
          #lwid = c(2,8), lhei = c(2.5,15),
          #main = "Broth microdilutions",
          #sepwidth=c(0.001,0.001),
          #sepcolor="black",
          #  srtCol=45, 
          #colsep=1:ncol(mat_data),
          #rowsep=1:nrow(mat_data))
dev.off()


library(dendextend)
dend1 <- as.dendrogram(hclust(as.dist(1-cor(t(toplot), method="pearson")), method="complete"))
c_group <- 30 # number of clusters
mycolors <- colorRampPalette(brewer.pal(8, "Set2"))(30)
dend1 <- color_branches(dend1, k = c_group, col =  mycolors) # add color to the lines
dend1 <- color_labels(dend1, k = c_group, col = mycolors)   # add color to the labels
col_labels <- get_leaves_branches_col(dend1)
col_labels <- col_labels[order(order.dendrogram(dend1))]


pdf(paste0(plotdir, "dendrogram_keio_genes.pdf"), width=1, height = 55)
dend1 <- set(dend1, "labels_cex", 0.2)
par(mar = c(1,1,0.5,1))
plot_horiz.dendrogram(dend1, side = T)
dev.off()
```




# Fig. 5


## numbers for text

```{r}

# how many strains with emrR mutations
template=fread("~/Documents/EMBL/Nx/Nx_paper/input/Ecoli_isolates_MIC.txt", sep="\t") %>%
  filter(!(Sample_ID %in% c("1639","T2761","T2762", "SC1764_unicycler","E28-1-Novastar"))) %>%
  drop_na(MIC) %>%
  filter(MIC > 4)
    
template = template %>%
  distinct(Sample_ID, MIC,Isolate_type, Mutation_Type)

# how many strains with oqxR mutations
template = fread("~/Documents/EMBL/Nx/Nx_paper/input/Kpn_isolates_MIC.txt", sep="\t") %>%
  #filter(!(Sample_ID %in% c("1639","T2761","T2762", "SC1764_unicycler","E28-1-Novastar"))) %>%
  drop_na(MIC) %>%
  filter(MIC > 8)
    
template = template %>%
  distinct(Sample_ID, MIC,Isolate_type)

length(unique(template$Sample_ID))

# adeL
template = readxl::read_xlsx(paste0(inputdir,"Nx_MIC_adeL_SuppFig.7e_29.01.2024_SG_.xlsx")) %>%
  rename(MIC=`MIC [µg/ml]`)%>%
  mutate(MIC=as.numeric(MIC)) %>% filter(MIC > 2)

# acrR
template = readxl::read_xlsx(paste0(inputdir,"Nx_MIC_acrR_SuppFig.7_29.01.2024_SG_.xlsx")) %>%
  rename(MIC=`MIC [µg/ml]`,
         Isolate_type=Isolate_Type) %>%
  filter(Sample_ID != "DSM30006") %>%
  mutate(MIC=as.numeric(MIC)) %>% filter(MIC > 2)

```

## Fig. 5a - WGS hm

### new version 240525 - new genes from Manuela, SC1761 strain from Stephan + adding strain renaming as per legend table
```{r}
strain_key = readxl::read_xlsx(paste0(inputdir,"new_designation_240525_MT_EC.xlsx")) %>%
  clean_names() %>%
  mutate(Isolate=sapply(strsplit(as.character(old_designation)," "),'[',3),
         new_designation=sapply(strsplit(as.character(new_designation)," "),'[',3))

test= readxl::read_xlsx(paste0(inputdir,"source_Fig.5a_MTSG.xlsx")) %>%
  mutate(mutation_type = replace_na(`mutation type`,value = "not mutated"))  %>%
  #change name of isolates
  inner_join(strain_key)

strain_order=c("1_R1","2_R1","4_R1","4_R2","5_R1","6_R1",
               "1_R1","1_R2","1_R3", "1_R4","1_R5", "1_R6","2_R1","2_R2","2_R3","2_R4","3_R1", "4_R1","5_R1",
               "1_R1","2_R1","2_R2","3_R1","4_R1","5_R1","8_R1", "9_R1","10_R1")

strain_order = c("1_R1","1_R2","1_R3", "1_R4","1_R5", "1_R6","2_R1","2_R2","2_R3","2_R4","3_R1", "4_R1","4_R2","5_R1","6_R1",
               "8_R1", "9_R1","10_R1")

test$new_designation <- factor(test$new_designation, levels=strain_order)

#pdf(paste0(plotdir,"WGS_hm_unclustered_new.pdf"), width =7, height = 7.5)
pdf(paste0(plotdir,"WGS_hm_unclustered_new.pdf"), width =9, height = 7.5)
ggplot(test,aes(x=new_designation,y=Gene_new,fill=as.factor(mutation_type))) +
       #aes(x=fct_reorder(new_designation, MIC, .desc = TRUE),y=Gene_new,fill=as.factor(mutation_type))) +
  scale_fill_manual(values = c("grey90", "grey70", "grey40","grey20","white"),name="") +
  #scale_fill_manual(values = c("white","grey"),labels=c("","mutated"), name="") +
  geom_tile(color="grey40") +
  facet_grid(~Species,scales="free_x",space='free_x') +
  labs(x="Strain",y="Gene") +
  theme(axis.text.x=element_text(angle=90, hjust=1))
dev.off()




pdf(paste0(plotdir,"MIC_isolates_new.pdf"), width=10, height =5)
ggbarplot(test %>%
          #distinct(new_designation,MIC, Species), x= "new_designation", y="MIC", color="Species", sort.val = "desc") +
          distinct(new_designation,MIC, Species), x= "new_designation", y="MIC", color="Species") +
          # order = c("MS317","MS548","MS493","MS530","MS346","MS594","MS572",
          #           "MS511","MS593","MS579","MS569","MS577","MS69",
          #           "MS532","MS366","MS121","MS386","MS510","MS364","MS122","MS347",
          #           "MS72","MS120","MS70","MS529","MS368","MS516"))+
  #coord_flip() +
  scale_y_continuous(trans="log2",breaks=c(2,4,8,16,32,64,128)) +
#  scale_y_continuous(n.breaks = 10) +
  scale_color_manual(values=c("#1E88E5","#FFC107","#D81B60")) +
  facet_grid(~ Species, scales="free_x", space="free") +
  labs(y="Nitroxoline MIC (µg/ml)") +
  theme(axis.text.x= element_text(angle = 90))
 dev.off() 
 
 
test = test %>%
  mutate(Isolate_type = case_when(Isolate_type == "generated" ~ "experimentally evolved",
                                  TRUE ~ "clinical isolate"))



```



### actual figure
```{r}

test= readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5a.xlsx"))

pdf(paste0(plotdir,"WGS_hm_unclustered.pdf"), width =7, height = 7.5)
ggplot(test,aes(x=fct_reorder(Isolate, MIC, .desc = TRUE),y=Gene_new,fill=as.factor(mut))) +
  scale_fill_manual(values = c("white","grey"),labels=c("","mutated"), name="") +
  geom_tile(color="grey40") +
  facet_grid(~Species,scales="free_x",space='free_x') +
  labs(x="Strain",y="Gene") +
  theme(axis.text.x=element_text(angle=90))
dev.off()


pdf(paste0(plotdir,"MIC_isolates_new.pdf"), width=10, height =5)
ggbarplot(test %>%
            distinct(Isolate,MIC, Species), x= "Isolate", y="MIC", color="Species", sort.val = "desc") +
          # order = c("MS317","MS548","MS493","MS530","MS346","MS594","MS572",
          #           "MS511","MS593","MS579","MS569","MS577","MS69",
          #           "MS532","MS366","MS121","MS386","MS510","MS364","MS122","MS347",
          #           "MS72","MS120","MS70","MS529","MS368","MS516"))+
  #coord_flip() +
  scale_y_continuous(trans="log2",breaks=c(2,4,8,16,32,64,128)) +
#  scale_y_continuous(n.breaks = 10) +
  scale_color_manual(values=c("#1E88E5","#FFC107","#D81B60")) +
  labs(y="Nitroxoline MIC (µg/ml)") +
  theme(axis.text.x= element_text(angle = 90))
 dev.off() 
 
 
test = test %>%
  mutate(Isolate_type = case_when(Isolate_type == "generated" ~ "experimentally evolved",
                                  TRUE ~ "clinical isolate"))
 
write_xlsx(test, paste0(sourcedatadir,"source_Fig.5a.xlsx"))


 
```

### info in text - average number of mutations per isolate per species
```{r}
test %>%
  filter(mut==1) %>%
  distinct(Isolate, Gene_new, Species,mut) %>%
  group_by(Species,Isolate,mut) %>%
  summarise(count=n()) %>%
  ungroup() %>%
  group_by(Species,mut) %>%
  summarise(meancount=mean(count))
#   rename(Isolate=Sample_ID) %>%
#   select(Isolate,MIC, Isolate_type, Mutation_Type, Protein_Change) %>%
#   mutate(Species=rep("E. coli", nrow(.)),
#          MIC=as.numeric(MIC)) %>%
#   drop_na(MIC) %>%
#   distinct(Isolate,MIC)
# mean(ecoli$MIC)

```


## Fig. 5b - Proteomics of R strains
or all, marking the EPI ones - only genes of interest (pumps,porins)
```{r}
genes = c("TolC", "OqxB3", "BepF","MprA", "AdeF","OprC","AdeG",
          "BepF","AcrA", "EmrA_1", "BepA_1", "AcrB_1", "MdtK","MarA",
          "LamB","OmpC", "OmpC~~~ompN_1","YedS~~~ompK35", "OmpD", "OmpF", "ScrY_1")
          #"OprD","OmpW")

toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))) %>%
  filter(og1 %in% genes)

hitgenes_insensitive =fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
  filter(og1 %in% unique(toplot$og1)) 

toplot =rbind.data.frame(toplot,hitgenes_insensitive) %>%
    group_by(og1, complete_name) %>%
    summarise(logFC=mean(logFC)) %>%
  #remove dmprA and dadeL
    filter(!(complete_name %in% c("A. baumannii dadeL","E. coli dmprA"))) %>%
    mutate(complete_name=sapply(strsplit(as.character(complete_name)," "),'[',3)) %>%
  mutate(complete_name=sub("[RS]$", "", complete_name))

# fix annotation of strains in complete name
strain_key = readxl::read_xlsx(paste0(inputdir,"new_designation_240525_MT_EC.xlsx")) %>%
  clean_names() %>%
  mutate(complete_name=sapply(strsplit(as.character(old_designation)," "),'[',3),
         new_designation=sapply(strsplit(as.character(new_designation)," "),'[',3),
        species = sub(" MS.*", "", old_designation))

test=inner_join(strain_key,toplot) %>%
  filter(complete_name %in% c("MS274", "MS299", "MS311", "MS346", "MS366", "MS386", "MS493", "MS511", "MS529", "MS530", "MS69",  "MS72")) %>%
  mutate(new_designation=interaction(species,new_designation))

mymat = test %>% 
  distinct(og1, new_designation, logFC)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}
# mymat = fix_na(mymat)

library(magrittr)

mat_data = mymat %>% pivot_wider(names_from = new_designation, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og1) %>%
  dplyr::select(-og1) %>%
  as.matrix()


#  dendrogram 
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)

#row_dend = color_branches(dend1, k = 4) # `color_branches()` returns a dendrogram object

type <- gsub("s\\d+_", "", colnames(mat_data))



ha = HeatmapAnnotation(
  df = data.frame(new_designation = type) %>%
  inner_join(test %>% select(new_designation,species), by="new_designation") %>%
    distinct() %>%
    select(-new_designation),
   annotation_height = unit(4, "mm"),
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )



pdf(paste0(plotdir,"clustered_MS_only_genes_interest_new.pdf"), width=4.5, height=8)

hm<-Heatmap(mat_data,
        cluster_rows = dend1,
        #cluster_cols=hc,
         top_annotation = ha,
        row_split = 4, column_split = 6,
        use_raster = TRUE, raster_quality = 5)

draw(hm, heatmap_legend_side="top", annotation_legend_side="top",
           legend_grouping="adjusted")

dev.off()



# source data
toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%


hitgenes_insensitive =fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
  filter(og1 %in% unique(toplot$og1)) 

toplot =rbind.data.frame(toplot,hitgenes_insensitive) %>%
    group_by(og1, complete_name) %>%
    summarise(logFC=mean(logFC)) %>%
  #remove dmprA and dadeL
    filter(!(complete_name %in% c("A. baumannii dadeL","E. coli dmprA")))


  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))) 

```

## Fig. 5c - OqxR and EmrR complementation

```{r}
#toplot = readxl::read_xlsx(paste0(inputdir,"oqxR, emrR_MIC_23.01.2024_SG__.xlsx"),skip=2, sheet=1) %>%
toplot = readxl::read_xlsx(paste0(inputdir,"oqxR, emrR_MIC_06.02.2024_SG__.xlsx"),skip=2, sheet=1) %>%
  pivot_longer(cols = c(2:5), names_to = "replicate", values_to="MIC") %>%
  clean_names() %>%
  rename(strain=condition) %>%
  dplyr::select(strain,mic,replicate)
toplot1 = readxl::read_xlsx(paste0(inputdir,"oqxR, emrR_MIC_06.02.2024_SG__.xlsx"),skip=2, sheet=2) %>%
  pivot_longer(cols = c(2:5), names_to = "replicate", values_to="MIC") %>%
  clean_names() %>%
  rename(strain=condition) %>%
  dplyr::select(strain,mic,replicate)
  
toplot =rbind.data.frame(toplot,toplot1) %>%
  mutate(condition=case_when(strain %in% c("MS121","MS579") ~ "no plasmid",
                             strain %in% c("MS121 + oqxR-wt","MS579 + emrR-wt") ~ "+ wt efflux\npump",
                             strain %in% c("MS121 + pTOPO","MS579 + pTOPO") ~ "+ empty plasmid"),
         # strain_new = case_when(strain %in% c("MS121","MS121 + oqxR-wt","MS121 + pTOPO") ~ "MS121 + wt ~ italic('oqxR')",
         #                     strain %in% c("MS579","MS579 + emrR-wt","MS579 + pTOPO") ~ "MS579 + wt ~ italic('emrR')"),
         strain_new = case_when(strain %in% c("MS121","MS121 + oqxR-wt","MS121 + pTOPO") ~ "4_R1 + wt ~ italic('oqxR')",
                             strain %in% c("MS579","MS579 + emrR-wt","MS579 + pTOPO") ~ "1_R4 + wt ~ italic('emrR')"),
         species = case_when(strain %in% c("MS121","MS121 + oqxR-wt","MS121 + pTOPO") ~ "K. pneumoniae",
                             strain %in% c("MS579","MS579 + emrR-wt","MS579 + pTOPO") ~ "E. coli"))




library(ggh4x)

# Only colour strips in x-direction
strip <- strip_themed(background_x = elem_list_rect(fill = c("#d81a61", "#fdc113")))

# fix name of strain with new designation
toplot = toplot %>%
  mutate_all(~ gsub("MS121", "4_R1", .)) %>%
  mutate_all(~ gsub("MS579", "1_R4", .)) %>%
  mutate(mic=as.numeric(mic)) %>%
  group_by(strain) %>%
  mutate(mic=log2(mic)) %>%
  mutate(meanmic=mean(mic),
         semic=std.error(mic)) %>%
  ungroup()

library(rstatix)
stat.test <- toplot %>%
  #distinct(condition, mic, strain_new) %>%
  dplyr::group_by(strain_new) %>%
  wilcox_test(mic ~ condition, p.adjust.method = "BH") %>%
  adjust_pvalue(method = "BH") %>%
  add_significance("p") %>%
  add_xy_position(fun = "mean_sd", x = "condition",dodge = 0, group = "strain_new") #%>%
  #filter(!(group1 =="DMSO" & group2 =="preNx"))


test= toplot  %>% filter(strain_new == "4_R1 + wt ~ italic('oqxR')")
wilcox_test(test, mic ~ condition, p.adjust.method = "BH")

 library(stringi)
stat.test = stat.test %>%
  mutate(xmin=case_when(group1 == "+ empty plasmid" & group2== "+ wt efflux\npump" ~ 2,
                        group1 == "+ empty plasmid" & group2== "no plasmid" ~ 1,
                        TRUE ~ 1),
         xmax=case_when(group1 == "+ empty plasmid" & group2== "+ wt efflux\npump" ~ 3,
                        group1 == "+ empty plasmid" & group2== "no plasmid" ~ 2,
                        TRUE ~ 3))

toplot$condition <- factor(toplot$condition, levels = c("no plasmid","+ empty plasmid","+ wt efflux\npump"))

pdf(paste0(plotdir,"OqxR_emrR_compl_new.pdf"),width = 3.9, height=3)


 ggplot(toplot, aes(x = condition, y = meanmic, color = condition)) +
  geom_bar(aes(fill = condition),stat = "identity", position = position_dodge(0.8), width = 0.7)+
  #geom_jitter(aes(x = condition,y=mic,color = condition), pch=21, col='grey30', position = position_jitterdodge(dodge.width = 0.8, jitter.width = 0.1,jitter.height = 0.1)) +
   geom_point(aes(x = condition,y=mic), pch=21,fill='grey90', size=2, show.legend=F,position = position_jitterdodge(dodge.width = 0.6, jitter.width = 0.9,jitter.height = 0.4)) +
   geom_errorbar(aes(ymin = meanmic - semic, ymax = meanmic + semic), position = position_dodge(0.8), width = 0.15) +
   scale_fill_grey() +
   scale_color_grey() +
  stat_pvalue_manual(stat.test, tip.length = 0.01,bracket.nudge.y = 0.5,
              label = "p.adj.signif", y.position = 6, step.increase = 0.1, step.group.by = "strain_new",
              size = 3.5) +
  scale_y_continuous(breaks = c(0,2,4,6,8,10), labels= c(0,4,16,64,256,1024)) +
  # facet_wrap2(~strain_new, strip = strip) +
  # facet_wrap2(~strain_new,labeller = label_parsed, strip = strip) +
  facet_wrap2(~strain_new,strip = strip) +
  labs(color = "", y = "MIC (µg/ml)", fill="") + 
   guides(color="none") +
   theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.position = "bottom",
        strip.text = element_text(face="bold", color="white"))
dev.off()




towrite = toplot %>%
  rename(`MIC (µg/ml)` = mic,
         mean_mic = meanmic,
         se_mic=semic) %>%
  select(-strain_new)

write_xlsx(towrite, paste0(sourcedatadir,"source_Fig.5c.xlsx"))

```


## Fig. 5d - mutplot OqxR

See code in mutplot: /Users/elisabettacacace/Documents/EMBL/Nx/Structures_Foldseek/mutplot

```{r}

proteins_acc<-read.table("~/Documents/EMBL/Nx/Structures_Foldseek/mutplot/inputdir/UniProt.txt",header=T) %>%
  filter(Hugo_Symbol %in% c("OqxR"))

template = fread("~/Documents/EMBL/Nx/Nx_paper/input/Kpn_isolates_MIC.txt", sep="\t") %>%
  filter(!(Sample_ID %in% c("1639","T2761","T2762", "SC1764_unicycler","E28-1-Novastar"))) %>%
  drop_na(MIC) %>%
  filter(MIC > 8) %>%
  mutate(wgs = case_when(Sample_ID %in% c("MS121","MS364","MS386","MS532","MS493","MS530","MS386","MS510","SC1761") ~ "wgs",
                         TRUE ~ "snp"))

# how many isolates
test = distinct(template, Sample_ID, Isolate_type,wgs)


var = template %>%
  group_by(Hugo_Symbol, Mutation_Type, Protein_Change,Isolate_type) %>%
  mutate(freq=n(),
         location=str_extract(Protein_Change, "(?<=\\D)\\d+")) %>%
        # MIC=max(MIC)) %>%
  ungroup() %>%
  distinct(Protein_Change,Mutation_Type,MIC,freq,location,Isolate_type) %>%
  mutate(location=as.numeric(location),
         Isolate_type=case_when(!(Isolate_type %in% c("clinical isolate","generated")) ~ "unidentified",
                                Isolate_type == "generated" ~ "experimentally evolved",
                                TRUE ~ Isolate_type))

var$MIC[is.na(var$MIC)] <- 0

## used NsrR 
    #proteins_acc<- proteins_acc[proteins_acc$Hugo_Symbol==var.plot$Hugo_Symbol,2][[1]]
    proteins_acc_url<-gsub(" ","%2C",proteins_acc)
proteins_acc_url <- "Q8ZKA3"
#proteins_acc_url <- "A0A290G3U8"

    baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
    url<-paste0(baseurl,proteins_acc_url)
    prots_feat<-GET(url,accept_json())
    prots_feat_red<-httr::content(prots_feat)
    features_total_plot<-NULL
    for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])#the extract_feat_acc() function takes features into a data.frame
     # features_temp$order<-i # this order is needed for plotting later
      features_total_plot<-rbind(features_total_plot,features_temp)
    }
    


plot_start<-0 #starts 0
plot_end<-max(features_total_plot$end)   

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type
    
var= var %>% filter(Isolate_type != "unidentified") %>%
  filter(MIC != 0) %>%
  mutate(Mutation_Type =case_when(Mutation_Type == "Frame_Shift_Ins" ~ "Frameshift/Insertion",
                                  Mutation_Type == "Nonsense_Mutation" ~ "Nonsense",
                                  Mutation_Type == "Missense_Mutation" ~ "Missense")) %>%
  mutate(Protein_Change = gsub(Protein_Change, pattern= "G60_R61insRNGSIHLG", replacement="G60_L67dup"))


pdf(paste0(plotdir,"OqxR_mutplot.pdf"), width=6.5, height=4.5)

ggplot(var,aes(location,MIC,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location,yend=MIC)) +
  geom_point(aes(size=freq, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=var %>%
                    distinct(location,MIC, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=150, size=3) +
  geom_hline(yintercept = 0) +
  geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=8,ymax=10),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 65, y = 9, label ="HTH rrf2-type", size = 4,color="white",fontface="bold") +
  geom_rect(data=features_total_plot[features_total_plot$type=="DNA_BIND",],
            mapping=aes(xmin=begin,xmax=end,ymin=10,ymax=12),color="darkblue",fill="darkblue",inherit.aes=F, alpha=0.5, ) +
  annotate("text", x = 39, y = 11, label ="HTH motif", size = 4,color="white",fontface="bold") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_size_continuous(breaks = c(1,3,6), limits=c(1,6)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,159), limits=c(0,159))+
  #scale_y_continuous(trans=scales::pseudo_log_trans(base = 2), breaks= c(16,32,64,128)) +
  scale_y_continuous(breaks = c(8,16,32,64,128),trans="log2",limits=c(8,200)) +
  labs(y= expression('log'[2]~'(MIC [µg/mL])'),size="Mutation occurrencies", title = "OqxR", shape = "Mutation type")+
  #guides(color="none") +
  theme_bw() +
  theme(plot.title=element_text(face="bold",size=(15),hjust=0.5),
       #legend.position = "none",
        legend.title = element_text(size=10),
        legend.text = element_text(size=10),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.1,0.1,0.1,0.1, unit="cm"),
        panel.border = element_blank(), axis.line = element_line())

dev.off()


towrite = var %>%
  rename(`MIC (µg/ml0)`= MIC,
         Mutation_occurrences = freq,
         location_aa_number=location)
  
write_xlsx(towrite, paste0(sourcedatadir,"source_Fig.5d.xlsx"))


```



## New Fig. 5f -efflux inhibitor 

```{r}
#toplot <- readxl::read_xlsx(paste0(inputdir, "231002_Efflux_pump_inhibitors_basecamp.xlsx"), skip=1) %>%
toplot <- readxl::read_xlsx(paste0(inputdir, "Nx_Efflux_pump_inhibitors_20.02.2024_SG.xlsx"), skip=1, n_max = 19) %>%
  pivot_longer(cols = c(5:16),names_to = "sample",values_to = "MIC") %>%
  rename(species =Specie, strain=Isolate) %>%
  mutate(sample = sapply(strsplit(as.character(sample),"\\..."),'[',2),
         sample = as.numeric(sample),
         replicate=case_when(sample %in% c(5,9,13) ~ 1,
                             sample %in% c(6,10,14) ~ 2,
                             sample %in% c(7,11,15) ~ 3,
                             sample %in% c(8,12,16) ~ 4,
                             TRUE ~ sample),
         condition = case_when(sample %in% c(5:8) ~ "no inhibitor",
                               sample %in% c(9:12) ~ "+ 100 µg/ml NMP",
                               sample %in% c(13:16) ~ "+ 25 µg/ml PAβN"),
         strain = gsub(strain,pattern="ATCC19606",replacement="19606"),
         strain = gsub(strain,pattern="ATCC 17978",replacement="17978"),
         newspec= case_when(strain %in% c("25922","MS346", "MS529") ~ paste0("italic('E. coli ')*",strain),
                            strain %in% c("19606","MS69", "MS72") ~ paste0("italic('A. baumannii ') *",strain),
                            strain %in% c("PRZ","MS25","MS386","MS274", "MS530") ~ paste0("italic('K. pneumoniae ') *",strain))) %>%
  clean_names() %>%
  select(c(mic,replicate, condition, newspec,strain))

old <- readxl::read_xlsx(paste0(inputdir, "230825_Efflux_inhibitors.xlsx")) %>%
  #janitor::clean_names() %>%
  pivot_longer(cols=c(3:11), names_to = "condition", values_to = "MIC") %>%
  mutate(Strain = gsub(Strain, pattern = " ", replacement=""),
         Strain = gsub(Strain, pattern = "ATCC", replacement=""),
         Strain = gsub(Strain, pattern = "R", replacement=""),
        Species = gsub(Species, pattern = "Acinetobacter baumannii", replacement="italic('A. baumannii ') *"),
         Species = gsub(Species, pattern = "Escherichia coli", replacement="italic('E. coli ')*"),
         Species = gsub(Species, pattern = "Klebsiella pneumoniae", replacement="italic('K. pneumoniae ') *"),
         Species = gsub(Species, pattern = "Pseudomonas aeruginosa", replacement="italic('P. aeruginosa ') *")) %>%
        janitor::clean_names() %>%
  mutate(newspec = interaction(species,strain,sep=""),
         condition = sapply(strsplit(as.character(condition), "\\..."), '[',1),
         strain_condition=interaction(strain,condition),
         replicate=ave(mic, strain_condition, FUN = seq_along)) %>%
  mutate(condition = case_when(condition == "CAMHB" ~ "no inhibitor",
                               condition == "+NMP" ~ "+ 100 µg/ml NMP",
                               condition == "+PAβN" ~ "+ 25 µg/ml PAβN")) %>%
  select(c(newspec,replicate, condition, mic,strain))

toplot=rbind.data.frame(toplot,old) 


# some strains are the same!! errors of naming

toplot = toplot %>%
  mutate(newspec = gsub(newspec, pattern = "italic('A. baumannii ') * MS329", replacement="italic('A. baumannii ') * MS347"),
         newspec = gsub(newspec, pattern = "italic('A. baumannii ') * MS348", replacement="italic('A. baumannii ') * MS69"),
         newspec = gsub(newspec, pattern = "italic('E. coli ')* ATCC 25922 R", replacement="italic('A. baumannii ') * MS346"))%>%
  mutate(straincond=interaction(newspec,condition),
         replicate=ave(mic, straincond,FUN=seq_along))

### here write source data for this and ED Fig
towrite = toplot %>%
  mutate(species = str_extract(newspec,"E\\. coli|K\\. pneumoniae|A\\. baumannii")) %>%
  dplyr::select(-c(newspec,straincond)) %>%
  rename(`MIC(µg/ml)`=mic)

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.5e_EDFig.8.xlsx"))


# show only those on which we have MS!!!
strains = c("italic('A. baumannii ') *19606", "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72",
            "italic('E. coli ')*25922", "italic('E. coli ')*MS346","italic('E. coli ')*MS529",
            "italic('K. pneumoniae ') *PRZ",
            "italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386",
            "italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530")


mainfig = toplot %>%
  filter(newspec %in% strains)

# dispose in rows as I want 
mainfig = mainfig %>%
  mutate(row= case_when(newspec %in% c("italic('A. baumannii ') *19606", "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72") ~ 1,
                        #newspec == "italic('A. baumannii ') *MS72" ~ 2, 
                        newspec %in% c("italic('E. coli ')*25922", "italic('E. coli ')*MS346","italic('E. coli ')*MS529") ~ 3,
                        newspec %in% c("italic('K. pneumoniae ') *PRZ","italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386","italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530") ~ 4),
                        #newspec %in% c("italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530") ~ 5),
         resist = case_when(newspec %in% c("italic('A. baumannii ') *19606","italic('E. coli ')*25922","italic('K. pneumoniae ') *MS25", "italic('K. pneumoniae ') *PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))

# # shorten Abau 19606 label
# mainfig = mainfig %>%
#   mutate(newspec= gsub(newspec, pattern = "ATCC",
#                        replacement="")) %>%
#   rename(MIC =mic)

# fix facet order
mainfig$facet = factor(mainfig$newspec, levels = c("italic('A. baumannii ') *19606",
                                                   "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72",
                                                   "italic('E. coli ')*25922", "italic('E. coli ')*MS346",
                                                   "italic('E. coli ')*MS529", "italic('K. pneumoniae ') *MS25", "italic('K. pneumoniae ') *MS274", "italic('K. pneumoniae ') *MS386", "italic('K. pneumoniae ') *PRZ", "italic('K. pneumoniae ') *MS530"))


p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="right")

legend_grob <-cowplot::get_legend(p)

p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "",y="MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=4) +
   theme(axis.text.x = element_blank(),
         #axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")


p2 <- ggerrorplot(mainfig %>% filter(row==3), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(mainfig,row==3 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  scale_y_continuous(trans="log2") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p3 <- ggerrorplot(mainfig %>% filter(row==4), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ fct_relevel(newspec, "italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386","italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *PRZ","italic('K. pneumoniae ') *MS530"), labeller = label_parsed,ncol=5)+ 
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==4 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  scale_y_continuous(trans="log2") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5.5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")


library("latex2exp")


cairo_pdf(paste0(plotdir,"effluxpump_new1.pdf"), width =9.5,height=4)
 
layout <-"AAACCC
          DDDDDG"
wrap_plots(A=p, C=p2,D=p3,G=legend_grob, design = layout)

dev.off()


```






## Fig. 5b - efflux pump inhibitors
```{r}
toplot <- readxl::read_xlsx(paste0(inputdir, "231002_Efflux_pump_inhibitors_basecamp.xlsx"), skip=1) %>%
  pivot_longer(cols = c(3:14),names_to = "sample",values_to = "MIC") %>%
  rename(species =`...1`, strain=`...2`) %>%
  mutate(sample = as.numeric(sample),
         replicate=case_when(sample %in% c(5,9) ~ 1,
                             sample %in% c(6,10) ~ 2,
                             sample %in% c(7,11) ~ 3,
                             sample %in% c(8,12) ~ 4,
                             TRUE ~ sample),
         condition = case_when(sample %in% c(1:4) ~ "no inhibitor",
                               sample %in% c(5:8) ~ "+ 100 µg/ml NMP",
                               sample %in% c(9:12) ~ "+ 25 µg/ml PAβN"),
         newspec= case_when(strain %in% c("MS346", "MS529") ~ paste0("italic('E. coli ')*",strain),
                            strain %in% c("MS69", "MS72") ~ paste0("italic('A. baumannii ') *",strain),
                            strain %in% c("MS274", "MS530") ~ paste0("italic('K. pneumoniae ') *",strain)),
        species = gsub(species, pattern  ="ESCO",replacement = "italic('E. coli')"),
        species = gsub(species, pattern  ="KLPN",replacement = "italic('K. pneumoniae')"),
        species = gsub(species, pattern  ="ACBA",replacement = "italic('A. baumannii')")) 


# test<-function(data) {
#   if (length(table(data[,1]))==2) {
#     chisq.test(table(data[,1:2]))$p.value
#   } else {
#   t.test(data[,1]~data[,2])
#   }
# }
# 
# test(cbind(binary,group))
# test(cbind(continuous,group))
# 
# 
# test = toplot %>%
#   pivot_wider(id_cols = c(strain, replicate), names_from = condition,values_from = MIC) %>%
#   group_by(strain) %>%
#   clean_names() %>%
#   filter(strain=="MS346")


# chisq.test(x = as.matrix(test[, c("x100_mg_ml_nmp", "no_inhibitor")]))
# prop.test(x = test$x100_mg_ml_nmp, n = test$x100_mg_ml_nmp + test$no_inhibitor)
# 
# 
#   mutate(pval_NMP = case_when(strain %in% c("MS346","MS529") ~ prop.test(x = x100_mg_ml_nmp, n = x100_mg_ml_nmp + no_inhibitor),
#                               TRUE ~ t.test(x100_mg_ml_nmp,no_inhibitor)))


library("latex2exp")

cairo_pdf(paste0(plotdir,"effluxpump.pdf"), width =5,height=4)
 ggerrorplot(toplot, x = "condition", y = "MIC", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(species ~ strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=40) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="bottom")
dev.off()

#  group1 <- c(32, 32, 32, 32)
# group2 <- c(16, 16, 16, 16)
# 
# chisq.test(x = as.matrix(, c("group1", "group2")))
# # Combine the data into a contingency table
# contingency_table <- table(c(rep(group1, length(group1)), rep(group2, length(group2))))
# 
# # Chi-square test
# chisq.test(contingency_table)
 
```



# SUPP FIGURES

## ED Fig. 1
### ED1a - Disk diffusion

```{r}

#disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023.xlsx"), sheet = 1)%>%
#disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_11.02.2024_.xlsx"), sheet = 1)%>%
disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(Species != "NA")  %>%
  pivot_longer(cols=3:27,names_to = "MIC", values_to = "no_strains")
  

abbreviate_string <- function(input_string) {
#input_string = "Escherichia coli"
  words <- strsplit(input_string, " ")[[1]]
  abbr <- abbreviate(words[[1]], dot = TRUE, named = FALSE, minlength = 1)
  abbr <- interaction(abbr, words[[2]], sep = " ")
  return(abbr)
}

disk$Species_label = sapply(disk$Species,abbreviate_string)

toplot = disk %>%
 #filter(Species != "Moraxella catarrhalis") %>%
  mutate(Species_label = case_when(Species %in% c("Psychrobacter spp.","Salmonella spp.", "Shigella spp.") ~ Species,
                                            TRUE ~ Species_label)) %>%
    mutate(Species_label = gsub(Species_label, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
  mutate(Species = gsub(Species, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
   mutate(Species_label=as.character(Species_label)) %>%
  mutate(Species_italics = paste0("*", Species_label, "* ")) %>%
  mutate(MIC=as.numeric(MIC),
         Species_label = paste0(Species_italics, "(n=", `n=`, ")")) 


species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                   "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
               "Psychrobacter spp.",
                 "Hafnia alvei")



#create order with the n annotation
legend = toplot %>% 
  select(Species, Species_label) %>%
  distinct() %>%
  arrange(factor(Species, levels=species_order))

toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  
  
library(ggtext)
pdf(paste0(plotdir,"Disk_diffusion_our.pdf"), width=9, height=6)
ggplot(toplot,aes (x = as.factor(MIC), reorder(Species_label, Species))) +
    geom_tile(aes(fill = no_strains, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=no_strains),size=3.3) +
   scale_fill_gradient(name = "number of\nstrains",low="white", high="darkgreen", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  # scale_fill_gradient(name = expression(log[10]("strains")),low="white", high="darkgreen", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  #scale_fill_manual(data=eucast %>% filter(no_strains ==0),values=c("white")) +
  labs(y="", x = "Zone of inhibition (mm)") +
   theme_classic() +
  theme(axis.text.y = element_markdown(color = "black",size=9),
        legend.position = "top")
dev.off()


towrite= toplot %>%
  dplyr::rename(tot_number_of_strains = `n=`) %>%
  group_by(Species,tot_number_of_strains, MIC) %>%
  dplyr::summarise(number_strains_at_MIC= n()) %>%
  mutate(note = case_when(Species == "Salmonella spp." ~ "includes Salmonella enterica subsp. enterica serovar Enteritidis and serovar Typhi",
                          Species == "Shigella spp." ~ "includes Shigella sonnei and Shigella flexneri",
                          Species == "Burkholderia cenocepacia complex" ~ "includes Burkholderia cenocepacia, Burkholderia multivorans, and Burkholderia vietnamensis",
                          Species == "Psychrobacter spp." ~ "includes Psychrobacter phenylpyruvicus and Psychrobacter sanguinis",
                          TRUE ~ ""))


write_xlsx(towrite,paste0(sourcedatadir,"source_ED_Fig.1a.xlsx"))

```



### ED1b - Agar dilution

```{r}

agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_17.02.2024_.xlsx"), sheet = 1)%>%
  filter(strain != "NA") %>%
  pivot_longer(cols=3:9,names_to = "MIC", values_to = "no_strains") %>%
  rename(Species=strain)

abbreviate_string <- function(input_string) {
#input_string = "Escherichia coli"
  words <- strsplit(input_string, " ")[[1]]
  abbr <- abbreviate(words[[1]], dot = TRUE, named = FALSE, minlength = 1)
  abbr <- interaction(abbr, words[[2]], sep = " ")
  return(abbr)
}

agar$Species_label = sapply(agar$Species,abbreviate_string)

toplot = agar %>%
  mutate(Species_label = case_when(Species %in% c("Psychrobacter spp.","Salmonella spp.", "Shigella spp.") ~ Species,
                                            TRUE ~ Species_label)) %>%
    mutate(Species_label = gsub(Species_label, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
  mutate(Species = gsub(Species, pattern = "cenocepacia",replacement="cenocepacia complex")) %>%
   mutate(Species_label=as.character(Species_label)) %>%
  mutate(Species_italics = paste0("*", Species_label, "* ")) %>%
  mutate(MIC=as.numeric(MIC),
         Species_label = paste0(Species_italics, "(n=", n, ")")) 

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Psychrobacter spp.",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")


#create order with the n annotation
legend = toplot %>% 
  select(Species, Species_label) %>%
  distinct() %>%
  arrange(factor(Species, levels=species_order))

toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  
  
library(ggtext)
pdf(paste0(plotdir,"Agar_our_greyscale.pdf"), width=4, height=6)
ggplot(toplot,aes (x = as.factor(MIC), reorder(Species_label, Species))) +
    geom_tile(aes(fill = no_strains, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=no_strains),size=3.3) +
   scale_fill_gradient(name = "number of\nstrains",low="white", high="grey39", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  # scale_fill_gradient(name = expression(log[10]("strains")),low="white", high="darkred", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  #scale_fill_manual(data=eucast %>% filter(no_strains ==0),values=c("white")) +
  labs(y="", x = "MIC (µg/ml)") +
   theme_classic() +
  theme(axis.text.y = element_markdown(color = "black",size=9),
        legend.position="top")
dev.off()

towrite= toplot %>%
  dplyr::rename(tot_number_of_strains = n) %>%
  group_by(Species,tot_number_of_strains, MIC) %>%
  dplyr::summarise(number_strains_at_MIC= n()) %>%
  mutate(note = case_when(Species == "Salmonella spp." ~ "includes Salmonella enterica subsp. enterica serovar Enteritidis and serovar Typhi",
                          Species == "Shigella spp." ~ "includes Shigella sonnei and Shigella flexneri",
                          Species == "Burkholderia cenocepacia complex" ~ "includes Burkholderia cenocepacia, Burkholderia multivorans, and Burkholderia vietnamensis",
                          Species == "Psychrobacter spp." ~ "includes Psychrobacter phenylpyruvicus, Psychrobacter sanguinis and other Psychrobacter spp.",
                          TRUE ~ ""))


write_xlsx(towrite,paste0(sourcedatadir,"source_ED_Fig.1b.xlsx"))
```


### ED1c - EUCAST and us

```{r}
eucast = readxl::read_xlsx(paste0(inputdir,"EUCAST_MIC.xlsx"), sheet = 1) %>%
  #remove Gram+
  filter(!(Organism %in% c("Staphylococcus aureus", "Staphylococcus epidermidis","Staphylococcus saprophyticus","Coagulase-negative staphylococci", "Enterococcus faecalis","Enterococcus faecium"))) %>%
    dplyr::mutate(Species = case_when(#Organism %in% c("Proteus mirabilis", "Proteus vulgaris", "Proteus spp. indole-positive") ~ "Proteus spp.",
                                      Organism %in% c("Acinetobacter baumannii",    "Acinetobacter beijerinckii", "Acinetobacter calcoaceticus","Acinetobacter guillouiae","Acinetobacter haemolyticus","Acinetobacter johnsonnii","Acinetobacter junii","Acinetobacter lwoffi","Acinetobacter pittii","Acinetobacter radioresistens","Acinetobacter septicus", "Acinetobacter ursingii") ~ "Acinetobacter spp.",
                                      #Organism =="Enterobacter cloacae" ~ "Enterobacter spp.",
                                      TRUE ~ Organism),
                  Method = rep("EUCAST",nrow(.))) %>%
  pivot_longer(cols = c(2:13),names_to = "MIC",values_to = "n_strains") %>%
  group_by(Species,MIC) %>%
  summarise(totstrain=sum(n_strains)) %>%
  ungroup() %>%
  group_by(Species) %>%
  mutate(MIC = as.numeric(MIC),
         micstrain = MIC*totstrain,
         meanmiceucast = sum(micstrain/sum(totstrain))) %>%
  distinct(Species,meanmiceucast)


dilut = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_17.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  mutate(Method=rep("Microdilution"), nrow(.)) %>%
  filter(!(Species %in% c("Klebsiella aerogenes","Moraxella osloensis"))) %>%
    dplyr::mutate(Species = case_when(Species %in% c("Proteus mirabilis", "Proteus vulgaris", "Proteus spp. indole-positive") ~ "Proteus spp.",
                                      Species %in% c("Serratia marcescens") ~ "Serratia spp.",
                                      Species %in% c("Acinetobacter baumannii","Acinetobacter beijerinckii", "Acinetobacter calcoaceticus","Acinetobacter guillouiae","Acinetobacter haemolyticus","Acinetobacter johnsonnii","Acinetobacter junii","Acinetobacter lwoffi","Acinetobacter pittii","Acinetobacter radioresistens","Acinetobacter septicus", "Acinetobacter ursingii") ~ "Acinetobacter spp.",
                                      Species =="Enterobacter cloacae" ~ "Enterobacter spp.",
                                      TRUE ~ Species)) %>%
  pivot_longer(cols = c(3:13),names_to = "MIC",values_to = "n_strains") %>%
  group_by(Species,MIC) %>%
  summarise(totstrain=sum(n_strains)) %>%
  ungroup() %>%
  group_by(Species) %>%
  mutate(MIC = as.numeric(MIC),
         micstrain = MIC*totstrain,
         meanmic = sum(micstrain/sum(totstrain))) %>%
  distinct(Species,meanmic)

toplot = inner_join(dilut,eucast)

toplot$newspec = paste0("italic('", toplot$Species, "')")


pdf(paste0(plotdir,"240226_eucast_broth_concordance.pdf"),width = 4, height =4)
ggscatter(toplot,x="meanmic",y="meanmiceucast", add = "reg.line", add.params = list(color = "blue", fill = "lightgray"),
          conf.int = TRUE,
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   cor.coeff.args = list(method = "pearson", label.x = 2, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2") +
  scale_y_continuous(trans="log2") +
  labs(x="This study: log2[MIC (µg/ml)]", y = "EUCAST: log2[MIC (µg/ml)]") +
  geom_text_repel(aes(label = newspec),parse=TRUE, max.overlaps = 20) 
dev.off()
```


### ED1d - correlation broth and agar

```{r}
toremove=c("Moraxella catarrhalis", "Klebsiella aerogenes")

library(ggrepel)

agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_11.02.2024_.xlsx"), sheet = 1)%>%
  filter(strain != "NA") %>%
  pivot_longer(!c(strain,n), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(strain, n) %>%
  summarise(medianMIC_agar = median(MIC)) %>%
  filter(!(strain %in% toremove))

broth = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023_11.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  pivot_longer(!c(Species,`n=`), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(Species, `n=`) %>%
  summarise(medianMIC_broth = median(MIC))%>%
  filter(!(Species %in% toremove))

toplot = full_join(agar,broth, by = c("strain"= "Species")) %>%
  clean_names()

abbreviate_string <- function(input_string) {
#input_string = "Escherichia coli"
  words <- strsplit(input_string, " ")[[1]]
  abbr <- abbreviate(words[[1]], dot = TRUE, named = FALSE, minlength = 1)
  abbr <- interaction(abbr, words[[2]], sep = " ")
  return(abbr)
}

toplot$newspec = sapply(toplot$strain,abbreviate_string)

toplot = toplot %>%
  mutate(newspec = case_when(strain %in% c("Psychrobacter spp.","Salmonella spp.", "Shigella spp.") ~ strain,
                                            TRUE ~ newspec))

toplot$newspec = paste0("italic('", toplot$newspec, "')")

pdf(paste0(plotdir, "corr_broth_agar_new.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot, "median_mic_agar", "median_mic_broth", color = "n", size = "n_2", alpha = 0.8,
          #label = "newspec", repel = TRUE,
           add = "reg.line",  # Add regressin line
  add.params = list(color = "blue", fill = "lightgray"),
  # Customize reg. line
  conf.int = TRUE,
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   cor.coeff.args = list(method = "pearson", label.x = 1, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2") +
  scale_y_continuous(trans="log2") +
  labs(size = "strains tested (broth microdilution)", color = "strains tested (agar dilutions)",
       x = "MIC agar (µg/ml)", y = "MIC broth (µg/ml)") +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 30) +
 scale_color_gradientn(colors = c("#064c4c","#ce9000"),#values = scales::rescale(c(0,11,15,21,26,31,36, 41,60,150)),
                      breaks = c(50,150,250),
                      guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
  scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(50,100,150,250)) +
  guides(alpha ='none')
#scale_colour_gradient(low = "#B2182B",high = "#2166AC")
dev.off()


toplot = toplot %>%
  mutate(logmedianhalo = log2(median_mic_broth),
         logmedianmic = log2(median_mic_agar))

cor(toplot$logmedianhalo,toplot$logmedianmic, method = "pearson",use = "complete.obs")


```

### ED1e - correlation between disk and agar

```{r}


agar = readxl::read_xlsx(paste0(inputdir,"agar_dilutions_2023_11.02.2024_.xlsx"), sheet = 1)%>%
  filter(strain != "NA") %>%
  pivot_longer(!c(strain,n), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(strain, n) %>%
  summarise(medianMIC = median(MIC)) %>%
  filter(!(strain %in% toremove))
  

disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_11.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  pivot_longer(!c(Species,`n=`), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(Species, `n=`) %>%
  summarise(medianhalo = median(MIC))

toplot = full_join(agar,disk, by = c("strain"= "Species")) %>%
  clean_names()

abbreviate_string <- function(input_string) {
#input_string = "Escherichia coli"
  words <- strsplit(input_string, " ")[[1]]
  abbr <- abbreviate(words[[1]], dot = TRUE, named = FALSE, minlength = 1)
  abbr <- interaction(abbr, words[[2]], sep = " ")
  return(abbr)
}

toplot$newspec = sapply(toplot$strain,abbreviate_string)

toplot = toplot %>%
  mutate(newspec = case_when(strain %in% c("Psychrobacter spp.","Salmonella spp.", "Shigella spp.") ~ strain,
                                            TRUE ~ newspec))

toplot$newspec = paste0("italic('", toplot$newspec, "')")

pdf(paste0(plotdir, "corr_disk_agar_new.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot, "median_mic", "medianhalo", color = "n", size = "n_2", alpha = 0.8,
          add = "reg.line",
          add.params = list(color = "blue", fill = "lightgray"),
         conf.int = TRUE,
         cor.coef = TRUE,
         cor.coeff.args = list(method = "pearson", label.x = 4, label.sep = "\n"),
         cor.coef.size = 4) +
  scale_x_continuous(trans="log2") +
  scale_y_continuous(trans="log2", n.breaks = 4) +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 20) +
  labs(size = "strains tested (disk diffusion)", color = "strains tested (agar dilutions)",
       x = "MIC agar (µg/ml)", y = "Zone of inhibition (mm)") +
 scale_color_gradientn(colors = c("#064c4c","#ce9000"),#values = scales::rescale(c(0,11,15,21,26,31,36, 41,60,150)),
                       breaks = c(50,150,250),
                      guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
  scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(100,150,500)) +
  guides(alpha ='none')
#scale_colour_gradient(low = "#B2182B",high = "#2166AC")
dev.off()

```


### ED1f - correlation between disk and broth

```{r}
broth = readxl::read_xlsx(paste0(inputdir,"microdilutions_2023.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  pivot_longer(!c(Species,`n=`), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(Species, `n=`) %>%
  summarise(medianMIC_broth = median(MIC)) %>%
  rename(n = `n=`) %>%
  filter(!(Species %in% toremove))

disk = readxl::read_xlsx(paste0(inputdir,"disk_diffusion_2023_11.02.2024_.xlsx"), sheet = 1) %>%
  filter(Species != "NA") %>%
  pivot_longer(!c(Species,`n=`), names_to = "MIC", values_to = "count") %>%
  uncount(count) %>%
  mutate(MIC = as.numeric(MIC)) %>%
  group_by(Species, `n=`) %>%
  summarise(medianhalo = median(MIC))

toplot = full_join(broth,disk, by = "Species") %>%
  clean_names()

abbreviate_string <- function(input_string) {
#input_string = "Escherichia coli"
  words <- strsplit(input_string, " ")[[1]]
  abbr <- abbreviate(words[[1]], dot = TRUE, named = FALSE, minlength = 1)
  abbr <- interaction(abbr, words[[2]], sep = " ")
  return(abbr)
}

toplot$newspec = sapply(toplot$species,abbreviate_string)

toplot = toplot %>%
  mutate(newspec = case_when(species %in% c("Psychrobacter spp.","Salmonella spp.", "Shigella spp.") ~ species,
                                            TRUE ~ newspec))

toplot$newspec = paste0("italic('", toplot$newspec, "')")

pdf(paste0(plotdir, "corr_disk_broth_new.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot %>% drop_na(), "median_mic_broth", "medianhalo", color = "n", size = "n_2", alpha = 0.8,
         # label = "newspec", repel = TRUE,
           add = "reg.line",  # Add regressin line
  add.params = list(color = "blue", fill = "lightgray"), # Customize reg. line
  conf.int = TRUE,
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   cor.coeff.args = list(method = "pearson", label.x = 4, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2") +
  scale_y_continuous(trans="log2", breaks= ) +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 20) +
 labs(size = "strains tested (disk diffusion)", color = "strains tested (broth dilutions)",
      x = "MIC broth (µg/ml)", y = "Zone of inhibition (mm)") +
scale_color_gradientn(colors = c("#064c4c","#ce9000"),#values = scales::rescale(c(0,11,15,21,26,31,36, 41,60,150)),
                      breaks = c(50,150,250),
                     guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
 scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(100,150,500)) +
 guides(alpha ='none')

 dev.off()


```


## ED Fig. 2


### ED Fig. 2a - Abau killing w/ control

```{r}
## Abau
killing = readxl::read_xlsx(paste0(inputdir,"Time-kill_ACBA_ATCC19606.xlsx")) %>%
  janitor::clean_names() %>%
  filter(time_h != "bla") %>%
  pivot_longer(names_to="condition", values_to = "CFU", cols = c(2:29), names_prefix = "^x*_") %>%
  mutate(condition = sapply(strsplit(as.character(condition),"_"),'[',1),
         condition = gsub(condition,pattern="growth",replacement=0),
         condition = gsub(condition,pattern="x",replacement=""),
         condition=as.numeric(condition),
         time_h=as.numeric(time_h)) %>%
  group_by(condition, time_h) %>%
  mutate(meanCFU=mean(CFU,na.rm = TRUE),
         sdCFU = sd(CFU, na.rm=TRUE),
         seCFU=sd(CFU, na.rm=TRUE)/sqrt(length(na.omit(CFU))))



pdf(paste0(plotdir,"Abau_killing_control.pdf"), width = 5.8, height=5)

killing$condition <- factor(killing$condition, levels=c('0', '1','2', '4', '8', '16', '32'))

my_palette = brewer.pal("Oranges",n = 9)[3:9]


ggplot(killing, aes(as.factor(time_h),meanCFU, group = condition, color = condition)) + 
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)), 
                breaks =trans_breaks(log10, function(x) 10^x, 10)) +
  scale_x_discrete(breaks=c(0,2,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=killing[!is.na(killing$meanCFU),]) +
  geom_errorbar(aes(ymin = meanCFU-seCFU, ymax = meanCFU+sdCFU), width = 1,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(µg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("A. baumannii"))) +
  #expression(Blah[1*d])
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()

towrite = killing %>%
  dplyr::rename(mean=meanCFU,
                se=seCFU,
                `Nitroxoline concentration (µg/ml)`= condition,
                `Time (h)` = time_h) %>%
  dplyr::select(-sdCFU)

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.1d_EDFig.2a.xlsx"))
```

### ED Fig. 2b - E. coli killing
```{r}
killing = readxl::read_xlsx(paste0(inputdir,"221214_Time-kill_ESCO_BW25113.xlsx")) %>%
  janitor::clean_names() %>%
  filter(time_h != "bla") %>%
  pivot_longer(names_to="condition", values_to = "CFU", cols = c(2:22), names_prefix = "^x*_")%>%
  mutate(condition = sapply(strsplit(as.character(condition),"_"),'[',1),
         condition = gsub(condition,pattern="growth",replacement=0),
         condition = gsub(condition,pattern="x",replacement=""),
         condition=as.factor(condition),
         time_h=as.numeric(time_h)) %>%
  group_by(condition, time_h) %>%
  mutate(meanCFU=mean(CFU,na.rm = TRUE),
         sdCFU = sd(CFU, na.rm=TRUE),
         seCFU=sd(CFU, na.rm=TRUE)/sqrt(length(na.omit(CFU))))



pdf(paste0(plotdir,"EC_killing.pdf"), width = 5.8, height=5)

killing$condition <- factor(killing$condition, levels=c('0', '2', '4', '8', '16', '32','64'))

my_palette = brewer.pal("Oranges",n = 9)[3:9]

ggplot(killing, aes(as.factor(time_h),meanCFU, group = condition, color = condition)) + 
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)), 
                breaks =trans_breaks(log10, function(x) 10^x, 10)) +
  scale_x_discrete(breaks=c(0,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=killing[!is.na(killing$meanCFU),]) +
  geom_errorbar(aes(ymin = meanCFU-seCFU, ymax = meanCFU+sdCFU), width = 0.8,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(µg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("E. coli"))) +
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()


towrite = killing %>%
  dplyr::rename(mean=meanCFU,
                se=seCFU,
                `Nitroxoline concentration (µg/ml)`= condition,
                `Time (h)` = time_h) %>%
  dplyr::select(-sdCFU)

write_xlsx(towrite,paste0(sourcedatadir,"source_EDFig.2b.xlsx"))
```



## ED Fig. 3
### ED3a - checkerboards all



```{r}
# ### find all data
# run200818 <- fread("~/Documents/EMBL/Nx/CBs_Nx/200818/CB_median_reps_cumeps.txt") %>%
#   filter(Drug_h %in% c("ATM") & strain == "ECBW") %>%
#   select(-strain)
# 
# run220117 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220117/CB_median_reps_cumeps.txt") %>%
#   filter(Drug_h %in% c("AZM","PIP","TMP", "FEP","SUL")) #%>% 
#   # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
#   #                                     TRUE ~ "2_1"))
# run220120 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220120/CB_median_reps_cumeps.txt") %>%
#   filter(Drug_h %in% c("MEC", "ERT","TET","THI","CLQ","8OHQUI","CCCP","MIN","TOB")) %>%
#   select(-Comb_full)
#   # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
#   #                                     TRUE ~ "2_1"))
# run220121 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220121/CB_median_reps_cumeps.txt") %>%
#   dplyr::filter(Drug_h %in% c("AMK","CAZ", "MEM", "TEM", "IPM",
#                               "CIP", "GEN","NIT", "TGC","NIT")) %>%
#   select(-Comb_full)
#   # dplyr::mutate(replicate = case_when(replicate %in% c(1, 3) ~ "1_1",
#   #                                     TRUE ~ "2_1"))
# run220210 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220210/CB_median_reps_cumeps.txt") %>%
#   dplyr::filter(Drug_h %in% c("FOF", "DOX"))%>%
#   dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 1.1,
#                                   Conc_v == 0.73 ~ 0.946,
#                                   Conc_v == 0.61 ~ 0.792,
#                                   Conc_v == 0.49 ~ 0.638,
#                                   Conc_v == 0.37 ~ 0.484,
#                                   Conc_v == 0.25 ~ 0.330,
#                                   Conc_v == 0.14 ~ 0.176,
#                                    TRUE ~ Conc_v)) #%>%
#   # dplyr::mutate(replicate = case_when(Replicate == 1 ~ "1_1",
#   #                                     TRUE ~ "2_1"))
# run220212 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220212/CB_median_reps_cumeps.txt") %>%
#   filter(Drug_h %in% c("CLR","ERI", "LCX"))%>%
#   dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 1.2,
#                                   Conc_v == 0.73 ~ 1.032,
#                                   Conc_v == 0.61 ~ 0.864,
#                                   Conc_v == 0.49 ~ 0.696,
#                                   Conc_v == 0.37 ~ 0.528,
#                                   Conc_v == 0.25 ~ 0.360,
#                                   Conc_v == 0.14 ~ 0.192,
#                                    TRUE ~ Conc_v))#%>%
#   # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
#   #                                     TRUE ~ "2_1"))
# run220514 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220514/CB_median_reps_cumeps.txt") %>%
#   dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 0.75,
#                                   Conc_v == 0.73 ~ 0.64,
#                                   Conc_v == 0.61 ~ 0.54,
#                                   Conc_v == 0.49 ~ 0.43,
#                                   Conc_v == 0.37 ~ 0.33,
#                                   Conc_v == 0.25 ~ 0.22,
#                                   Conc_v == 0.14 ~ 0.12,
#                                    TRUE ~ Conc_v)) %>%
#   filter(Drug_h %in% c("COL", "RIF","NVB","VAN"))
#   # dplyr::mutate(replicate = case_when(replicate %in% c(1,5) ~ "1_1",
#   #                                     replicate %in% c(2,6) ~ "1_2",
#   #                                     replicate %in% c(3,7) ~ "2_1",
#   #                                     TRUE ~ "2_2"))
# 
# 
# #dfList <- list(run200818,run220117,run220120,run220121,run220210, run220212, run220514)
# dfList <- mget(ls(pattern="^run"))
# dfColList <- lapply(dfList,names)
# commonCols <- Reduce(intersect,dfColList)
# 
# allCBs <- dplyr::bind_rows(dfList, .id = "source") %>%
#   dplyr::select(all_of(commonCols), source) %>%
#   mutate(Conc_h = round(Conc_h, digits=4),
#          Conc_v = round(Conc_v, digits=3))
# 
# # toplot$facet = factor(toplot$fullnames.x, levels = c("Fosfomycin", "Mecillinam", "Temocillin", "Ceftazidime", "Imipenem", "Meropenem", "Ertapenem"))
# 
# allCBs = allCBs %>%
#   filter(fullnames.x != "Sulbactam")

#for(i in unique(toplot$Drugclass)) {
#p <- 
# toplot = toplot %>% mutate(Conc_h=round(Conc_h,digits=3),
#                            Conc_v=round(Conc_v,digits=3))

# this now comes from the supp file directly
allCBs = fread(paste0(inputdir,"CB_reps_split.txt"))

allCBs = allCBs %>% group_by(horizontal_concentration, nitroxoline_concentration, horizontal_drug) %>%
  mutate(median_norm_fitness = median(norm_fitness))


# antags= c("FOF","ATM","MEC","TEM", "CAZ","FEP","IPM", "MEM","ERT","THI", "AMK", "TOB", "TET","TMP")
# synerg = c("VAN", "COL", "AZM", "CLR", "NVB", "RIF")
# neutrals = setdiff(unique(allCBs$Drug_h), c(antags,synerg))


antags= c("Aztreonam","Mecillinam","Temocillin", "Cefepime","Imipenem", "Meropenem","Ertapenem","Thiolutin","Clioquinol", "Amikacin","Gentamicin", "Tobramycin", "Trimethoprim")
synerg = c("Vancomycin", "Colistin", "Azithromycin", "Novobiocin", "Rifampicin")
neutrals = setdiff(unique(allCBs$horizontal_drug), c(antags,synerg))


p <- ggplot(allCBs %>% filter(horizontal_drug %in% antags), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "#d9a520"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="", y = "", fill="Normalised\nfitness") +
    theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

p1 <- ggplot(allCBs %>% filter(horizontal_drug %in% synerg), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "turquoise4"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="", y = "", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

p2 <- ggplot(allCBs %>% filter(horizontal_drug %in% neutrals), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "grey"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="Combined drug (µg/ml)", y = "", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

pdf(paste0(plotdir, "CB_heatmaps_antag_neutral.pdf"),  width=10, height=15)
layout <-"BBBBB
          AAAAA
          AAAAA
          AAAAA
          AAAAA
          AAAAA
          CCCCC
          CCCCC
          CCCCC
          CCCCC
          CCCCC"
#(p1 | (p4 / p2)) / (p3 | p5 | guide_area()) + plot_layout(design = layout, guides = "collect")

wrap_plots(A=p,B=p1, C= p2,design = layout,guides="collect")


dev.off()



towrite=allCBs 

write_xlsx(towrite, paste0(sourcedatadir,"source_EDFig.3a.xlsx"))

```



### ED3b - pTOPO control E. coli

```{r}

bunch = fread("~/Documents/EMBL/Nx/220602_colistin_Nx_resmutants/plate_norm.txt") %>%
  dplyr::mutate(strain = case_when(strain == "EC_J53" ~ "E. coli J53",
                                       strain == "EC_J53Tc2442" ~ "E. coli J53 pKP2442_mcr-1",
                                       strain == "EC_J53pTOPO" ~ "E. coli J53 pTOPO",
                                       strain == "EC_J53pTOPOmcr1" ~ "E. coli J53 pTOPO_mcr-1",
                                       strain == "KP_2442" ~  "K. pneumoniae pKP2442_mcr-1",
                                       strain == "KP_CTR3" ~ "K. pneumoniae CT-R3 pmrB (P95L)",
                                       strain == "KP_CTS" ~ "K. pneumoniae CT-S",
                                      strain == "KP_LS107" ~ "K. pneumoniae LS107",
                                      strain == "KP_LS53" ~ "K. pneumoniae LS53 mgrB (IS1R)",
                                      strain == "KP_PRZ" ~ "K. pneumoniae PRZ",
                                      strain == "KP_PRZ2442" ~ "K. pneumoniae PRZ pKP2442_mcr-1",
                                   TRUE ~ strain)) %>%
    filter(strain != "nocells")



bla = fread("~/Documents/EMBL/Nx/220707_colistin_Nx_resmutants_KPPRZ_topo/plate_norm.txt")

pmrAB = fread("~/Documents/EMBL/Nx/221110_pmrAB_fromMichael/plate_norm.txt")%>%
  dplyr::mutate(nitroxoline=case_when(nitroxoline == "+ Nx 0.75 µg/ml" ~ "Nitroxoline_0.75",
                                      TRUE ~ "no_Nitroxoline"))%>%
  dplyr::mutate(strain = case_when(strain == "BW25113" ~ "E. coli BW25113",
                                   strain == "MG1655" ~ "E. coli MG1655",
                                   strain == "pmrA" ~ "E. coli MG1655 pmrA (G53E)",
                                   strain == "pmrB" ~ "E. coli MG1655 pmrB (V88E)",
                                   TRUE ~ strain)) 

mcr1 = fread("~/Documents/EMBL/Nx/220527_MIC_mrc1_colistin_nx/plate_norm.txt") %>%
    filter(nitroxoline != "+ Nx 0.45 µg/ml") %>%
  dplyr::mutate(nitroxoline=case_when(nitroxoline == "+ Nx 0.85 µg/ml" ~ "Nitroxoline_0.75",
                                      TRUE ~ "no_Nitroxoline")) %>%
  dplyr::mutate(strain = case_when(strain == "BW" ~ "E. coli BW25113",
                                   strain == "mcr1" ~ "E. coli BW25113 pGDP1_mcr1"))%>%
  filter(strain != "E. coli BW25113")
  


toplot = bind_rows(bunch, bla, pmrAB,mcr1)
#toplot = bind_rows(bunch, bla)

fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)

toplot = toplot %>%
  dplyr::mutate(strain_conc = interaction(strain, Concentration, nitroxoline),
                Concentration_f = as.factor(Concentration)) %>%
  dplyr::group_by(strain,Time_points,Drug, Concentration, strain_conc, nitroxoline) %>%
  dplyr::summarise(median = mean(ODnorm_bug_rob), sem = sd(ODnorm_bug_rob)/sqrt(length(ODnorm_bug_rob))) %>%
  dplyr::ungroup()


toplot$facet = factor(toplot$strain, levels = c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1","E. coli J53 pKP2442_mcr-1",
                                                "E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)",
                                                "E. coli BW25113","E. coli BW25113 pGDP1_mcr1",
                                                "K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)", "K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)","K. pneumoniae pKP2442_mcr-1",
                                                "K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO","K. pneumoniae PRZ pTOPO_mcr-1", "K. pneumoniae PRZ pKP2442_mcr-1"))

toplot = toplot %>%
  mutate(row= case_when(strain %in% c("E. coli BW25113","E. coli BW25113 pGDP1_mcr1") ~ 1,
                        strain %in% c("E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)") ~ 2,
                        strain %in% c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "E. coli J53 pKP2442_mcr-1") ~ 3,
                        #strain %in% c("E. coli J53 pKP2442_mcr-1") ~ 4,
                        strain %in% c("K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO_mcr-1","K. pneumoniae PRZ pKP2442_mcr-1") ~ 5,
                        strain %in% c("K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)") ~ 6,
                        strain %in% c("K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)") ~ 7,
                        strain %in% c("K. pneumoniae pKP2442_mcr-1") ~ 8,
                        strain %in% c("K. pneumoniae PRZ pTOPO") ~ 9),
         resist = case_when(strain %in% c("E. coli BW25113","E. coli J53","E. coli MG1655",
                                          "K. pneumoniae CT-S","K. pneumoniae LS107","K. pneumoniae PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))


toplot = toplot %>%
  filter(Time_points == 44 & strain != "nocells")

toplot = toplot %>%
  #decided not to include empty vector control in main fig, only supp 
  filter(strain %in% c("E. coli J53 pTOPO", "K. pneumoniae PRZ pTOPO")) %>%
  mutate(strain = gsub(strain, pattern  =" ",replacement = "~"),
        strain = gsub(strain, pattern  ="E.~coli",replacement = "italic('E. coli')"),
        strain = gsub(strain, pattern  ="K.~pneumoniae",replacement = "italic('K. pneumoniae')")) 



pdf(paste0(plotdir,"plasmidControl_resens_colistin.pdf"), width = 7, height = 2.5)

ggplot(toplot, aes(x = as.factor(Concentration), y = median, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_point() +
    geom_line(linewidth=0.5) +
    facet_wrap(~ strain,ncol=2, labeller = label_parsed) +
  scale_color_manual(labels = c("+ nitroxoline 0.75 µg/ml", "- nitroxoline"), values = c("turquoise4","darkgrey")) +
  labs(x="", y = "", colour = "") +
  ylim(min(toplot$median),max(toplot$median)) +
  geom_errorbar(aes(ymin = median-sem, ymax = median + sem), width = 0.15) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.ticks = element_blank(),
        legend.text = element_text(size=12))

dev.off()

```


##ED Fig.4 

### ED Fig. 4a - GO TPP

cutoffs:
- TPP hits at FDR < 0.01 and efsize (as in volcano plot in fig. 3a) > |1|
- enrichment cutoff as padj (Fisher's test, BH correction) < 0.05

```{r}
go_abund_up = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/GOenrich_wholecell_up_abundance.txt", sep = "\t", header = T)
go_abund_down = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/GOenrich_wholecell_down_abundance.txt", sep = "\t", header = T)
go_stab_down = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/GOenrich_wholecell_down_stability.txt", sep = "\t", header = T)
go_stab_up = fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/GOenrich_wholecell_up_stability.txt", sep = "\t", header = T)


# metalstuff = c("4 iron, 4 sulfur cluster binding","enterobactin biosynthetic process","iron ion homeostasis","manganese ion binding","metallopeptidase activity","siderophore transmembrane transporter activity","zinc ion binding","iron ion binding","iron-sulfur cluster assembly","magnesium ion binding","metal ion binding","siderophore transmembrane transport")
# 
# OM = c("cell outer membrane","lipopolysaccharide biosynthetic process","O antigen biosynthetic process","enterobacterial common antigen biosynthetic process","Gram-negative-bacterium-type cell outer membrane assembly","lipopolysaccharide core region biosynthetic process")
# 
# resp = c("aerobic respiration","ATP biosynthetic process","NADH dehydrogenase (quinone) activity","NADH dehydrogenase activity","plasma membrane respiratory chain complex I","proton motive force dependent protein transmembrane transporter activity","proton-transporting ATP synthase activity, rotational mechanism","ubiquinone biosynthetic process","ATP synthesis coupled proton transport","NADH dehydrogenase (ubiquinone) activity","NADH dehydrogenase complex","plasma membrane ATP synthesis coupled proton transport","proton transport","proton-transporting ATP synthase complex, catalytic core F(1)")


metal = c("4Fe-4S", "2Fe-2S", "siderophore", "iron","metal","zinc","copper","molybd","calci","manganese")
respir = c("respir","electron","quinone","succin","tricarbo","fumara","dehydro","NarG","cytochrome")
OM = c("membran","acyltra")

allGO = gdata::combine(go_abund_up,go_abund_down,go_stab_down,go_stab_up) %>%
  filter(p_adj <= 0.05) %>%
  mutate(annot = case_when(grepl(paste(metal,collapse="|"),GO.term) ~ "metal homeostasis",
                           grepl(paste(respir,collapse="|"),GO.term) ~ "respiration",
                           grepl(paste(OM,collapse="|"),GO.term) ~ "OM",
                           TRUE ~"other"),
         colour = case_when(GO.tree == "C" ~ "dodgerblue3",
                                          GO.tree == "F" ~ "tomato",
                                          TRUE ~ "darkgreen"),
         full_name = case_when(GO.tree == "C" ~ "Cellular Component",
                                          GO.tree == "F" ~ "Molecular Function",
                                          TRUE ~ "Biological Process"),
         fold.enrichment = as.numeric(fold_enrich),
         colour2 = case_when(annot == "metal homeostasis" ~ "tomato",
                             annot == "respiration"~ "dodgerblue3",
                             annot == "OM" ~ "darkgreen",
                                           TRUE ~ "black"))

towrite = allGO %>% 
  dplyr::filter(p_adj <= 0.05) %>%
  select(-c(source,colour,colour2)) %>%
  distinct()

writexl::write_xlsx(towrite,paste0(sourcedatadir,"source_ED_Fig.4a.xlsx"))


toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_abund_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]


p1 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_abund_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_stab_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]


p3 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_stab_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]

p4 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) 



pdf(paste0(plotdir, "ED4a_GObarplots.pdf"), width = 12, height = 12)
design <- "
111333
111333
222444
222444
222444
222444
222444
222444
222444
555555
"

p1 + p2 + p3 + p4 + guide_area() + plot_layout(design=design, guides = "collect")

dev.off()


# design <- "
# 11222
# 11222
# 11222
# 33222
# 33222
# 44222
# 44222
# 44222
# 44222
# 44222
# 55222
# 55222
# "

```



### OLD - ED Fig. 4a - GO enrichment TPP 

```{r}
go_abund_up = read.table("~/Documents/EMBL/Nx/TPP_nx/gene_ontology_enrichment_abundance_upregulated.txt", sep = "\t", header = T)
go_abund_down = read.table("~/Documents/EMBL/Nx/TPP_nx/gene_ontology_enrichment_abundance_downregulated.txt", sep = "\t", header = T)
go_stab_down = fread("~/Documents/EMBL/Nx/TPP_nx/gene_ontology_enrichment_stability_destabilized.txt", sep = "\t", header = T)
go_stab_up = read.table("~/Documents/EMBL/Nx/TPP_nx/gene_ontology_enrichment_stability_stabilized.txt", sep = "\t", header = T)

metalstuff = c("4 iron, 4 sulfur cluster binding","enterobactin biosynthetic process","iron ion homeostasis","manganese ion binding","metallopeptidase activity","siderophore transmembrane transporter activity","zinc ion binding","iron ion binding","iron-sulfur cluster assembly","magnesium ion binding","metal ion binding","siderophore transmembrane transport")

OM = c("cell outer membrane","lipopolysaccharide biosynthetic process","O antigen biosynthetic process","enterobacterial common antigen biosynthetic process","Gram-negative-bacterium-type cell outer membrane assembly","lipopolysaccharide core region biosynthetic process")

resp = c("aerobic respiration","ATP biosynthetic process","NADH dehydrogenase (quinone) activity","NADH dehydrogenase activity","plasma membrane respiratory chain complex I","proton motive force dependent protein transmembrane transporter activity","proton-transporting ATP synthase activity, rotational mechanism","ubiquinone biosynthetic process","ATP synthesis coupled proton transport","NADH dehydrogenase (ubiquinone) activity","NADH dehydrogenase complex","plasma membrane ATP synthesis coupled proton transport","proton transport","proton-transporting ATP synthase complex, catalytic core F(1)")

allGO = gdata::combine(go_abund_up,go_abund_down,go_stab_down,go_stab_up) %>%
         dplyr::mutate(colour = case_when(GO.tree == "C" ~ "dodgerblue3",
                                          GO.tree == "F" ~ "tomato",
                                          TRUE ~ "darkgreen"),
                       full_name = case_when(GO.tree == "C" ~ "Cellular Component",
                                          GO.tree == "F" ~ "Molecular Function",
                                          TRUE ~ "Biological Process"),
                       fold.enrichment = as.numeric(fold.enrichment),
                       colour2 = case_when(GO.term %in% metalstuff ~ "tomato",
                                           GO.term %in% OM ~ "dodgerblue3",
                                           GO.term %in% resp ~ "darkgreen",
                                           TRUE ~ "black"))

towrite = allGO %>% 
  dplyr::filter(p.value.BH <= 0.01) %>%
  select(-c(Level, min.level,p.value,p.value.Bonferroni,gonumb,colour,colour2, full_name)) %>%
  distinct()

writexl::write_xlsx(towrite,paste0(sourcedatadir,"source_ED_Fig.4a.xlsx"))


toplot = allGO %>%
         dplyr::filter(p.value.BH <= 0.01 & source == "go_abund_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]




p1 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p.value.BH <= 0.01 & source == "go_abund_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 65)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p.value.BH <= 0.01 & source == "go_stab_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]

p3 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")


toplot = allGO %>%
         dplyr::filter(p.value.BH <= 0.01 & source == "go_stab_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]

p4 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"), limits = c(0,0.01), n.breaks = 2) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2),
        axis.text.x = element_text(size = 14),
        legend.position = "bottom",
        legend.direction = "horizontal",
        legend.text = element_text(size = 14)) +
  guides(alpha="none")




pdf(paste0(plotdir, "S8a_GObarplots.pdf"), width = 12, height = 12)


design <- "
11222
11222
11222
##222
33222
##222
44222
44222
##222
55222
55222
55222
"  

p1 + p2 + p3 + p4 + guide_area() + plot_layout(design=design, guides = "collect")


dev.off()



```



### ED Fig. 4b - GO E. coli Keio hits


#### USE THIS uniprot

```{r}
all <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond =="nitro2")%>%
  mutate(gene=toupper(gene)) %>%
  mutate(gene = case_when(gene == "MURE (D3A)-KAN" ~ "MURE",
                          gene == "LPXC (G210S)-KAN" ~ "LPXC",
                          gene == "IMP4213" ~ "LPTD",
                          gene == "YDFY/GNSB" ~ "GNSB",
                          gene == "GNTY" ~ "NFUA",
                          gene == "YHBV" ~ "UBIV",
                          gene == "YHBU" ~ "UBIU",
                          gene == "YHBZ" ~ "OBGE",
                          gene == "YHBT" ~ "UBIT",
                          gene == "YHBJ" ~ "RAPZ",
                          gene == "YHBH" ~ "HPF",
                          gene == "YHBT" ~ "UBIT",
                          gene == "YHBC" ~ "RIMP",
                          gene == "RFAB" ~ "WAAB",
                          gene == "RFAC" ~ "WAAC",
                          gene == "RFAD" ~ "HLDD",
                          gene == "RFAE" ~ "HLDE",
                          gene == "RFAF" ~ "WAAF",
                          gene == "RFAG" ~ "WAAG",
                          gene == "RFAI" ~ "WAAO",
                          gene == "RFAJ" ~ "WAAJ",
                          gene == "RFAL" ~ "WAAL",
                          gene == "RFAP" ~ "WAAP",
                          gene == "RFAQ" ~ "WAAQ",
                          gene == "RFAS" ~ "WAAS",
                          gene == "RFAY" ~ "WAAY",
                          gene == "RFAZ" ~ "WAAZ",
                          gene == "YIFC" ~ "WZZE",
                          gene == "VACJ" ~ "MLAA",
                          gene == "LPCA" ~ "GMHA",
                          gene == "YRBA" ~ "IBAG",
                          gene == "YAET-KAN" ~ "BAMA",
                          gene == "GNTY" ~ "NFUA",
                          gene == "YDHD" ~ "GRXD",
                          gene == "MPRA" ~ "EMRR",
                          TRUE ~ gene)) %>%
  filter(!grepl("^NOT", gene))

hits <- all %>%
  filter(cond =="nitro2")%>%
  filter(padj <= 0.01)


# # set hit cutoff according to how many IQR above 1st quartile or below 3rd quartile
# #iqr = quantile(all$log2fc, na.rm=T)[[4]] - quantile(all$log2fc, na.rm=T)[[2]]
# 
# lowthr = (quantile(all$log2fc,na.rm=T)[[2]]) - (2*0.9529 * (IQR(all$log2fc, na.rm = T)))
# upthr = (quantile(all$log2fc,na.rm=T)[[4]]) + (2*0.9529 * (IQR(all$log2fc, na.rm = T)))



GOid.GOterms_raw <- read_lines('http://purl.obolibrary.org/obo/go/go-basic.obo')
GOid.GOterms <- bind_rows(lapply(grep('\\[Term\\]',GOid.GOterms_raw), function(term) {
  tmp_subset <- GOid.GOterms_raw[term:(term+1000)]
  if(length(grep('\\[Term\\]',tmp_subset)) >1) {
    tmp_subset_2 <- tmp_subset[1:(grep('\\[Term\\]',tmp_subset)[2]-1)]
  }else{
    tmp_subset_2 <- tmp_subset[1:(grep('\\[Typedef\\]',tmp_subset)[1]-1)]
  }
  id <- sub('.*id: ','',tmp_subset_2[grep('id: ',tmp_subset_2)])
  term_2 <- sub('.*name: ','',tmp_subset_2[grep('name: ',tmp_subset_2)])
  ont <- sub('.*namespace: ','',tmp_subset_2[grep('namespace: ',tmp_subset_2)])
  data.frame(GO.id = id,
             GO.term = term_2,
             GO.tree = ont)
})) %>%
  mutate(GO.tree = case_when(GO.tree == 'biological_process' ~ 'BP',
                             GO.tree == 'molecular_function' ~ 'MF',
                             GO.tree == 'cellular_component' ~ 'CC'))

###Remove obsolote GO terms
GOid.GOterms_clean <- GOid.GOterms %>%
  tibble::as_tibble() %>%
  filter(!grepl('obsolete ',GO.term))


###Import GOs annotations for each gene for the species of interest
Protein_GO_anno <- fread("~/Documents/EMBL/Nx/TPP_nx/240423_GOenrich_K12_fullgenome/uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv") %>% 
  clean_names() %>%
  mutate(gene =sapply(strsplit(as.character(entry_name),"_"),'[',1)) %>%
  dplyr::select(gene,
                GO.id = gene_ontology_go) %>% 
  separate_rows(GO.id, sep = ';') %>% 
  mutate(GO.id = sub('\\]','',sub('.* \\[','',GO.id))) 


# # slim this : if terms share exact same number of proteins keep the 1 with lowest hierarchy in GO (keep higher number)
test = Protein_GO_anno %>%
  mutate(presence=rep(1,nrow(.))) %>%
  pivot_wider(names_from = gene, values_fill = 0,values_from = presence, id_cols = GO.id)

sorted_df <- test %>%
  arrange(across(-GO.id), desc(GO.id))

# Remove duplicate rows based on the values in all other columns, keeping the first occurrence (highest GO.id)
filtered_df <- sorted_df %>%
  distinct(across(-GO.id), .keep_all = TRUE)

filtered_dt <- as.data.table(filtered_df)

setDT(filtered_df)

# Melt the data.table
Protein_GO_anno <- melt(filtered_df, 
                        id.vars = names(filtered_df)[1], 
                        measure.vars = names(filtered_df)[-1])

Protein_GO_anno = Protein_GO_anno %>%
  filter(value != 0) %>%
  filter(GO.id != "" & GO.id %in% unique(GOid.GOterms_clean$GO.id)) %>%
  select(-value) %>%
  rename(gene = variable)
  #dplyr::filter(GO.id != "") 
  
  
###Determine which GOs are present in the experiment that we are interested and
###filter for GO terms that contain at least 2 genes (to decrease the size of the background)
proteins_identified <- unique(all %>%
  distinct(gene))$gene

Protein_GO_annotations <- Protein_GO_anno %>%
  dplyr::filter(gene %in% proteins_identified) %>% 
  dplyr::group_by(GO.id) %>% 
  dplyr::filter(n()>1) %>%
  dplyr::ungroup()


GO_id_freq_background <- Protein_GO_annotations %>% 
  dplyr::group_by(GO.id) %>% 
  dplyr::summarise(freq_background = n()) %>% 
  dplyr::ungroup()


###Get list of hit genes and annotate their GOs
#hit_genes <- unique(hits_df_whole$representative)

# other option based on MAD = 1.4862 median (| xi - median(x) |)
keiomad = mad(all$log2fc, na.rm = T)
lowthr = median(all$log2fc, na.rm = T) - 3*keiomad
upthr = median(all$log2fc, na.rm = T) + 3*keiomad

toenrich = filter(hits, padj <= 0.01 & log2fc > upthr)
#toenrich = filter(hits, padj <= 0.01 & log2fc < lowthr)
hit_genes <- unique(toenrich$gene)

# # here problem : fetch uniprot Id for these
# uniprotID_name <- fread(paste0(inputdir,"uniprotkb_proteome_UP000000625_2024_04_23 (1).tsv")) %>%
#   select(Entry,`Entry Name`) %>%
#   mutate(representative = sapply(strsplit(as.character(`Entry Name`), "_"),'[',1)) %>%
#   select(Entry,representative) %>%
#   inner_join(toenrich, by= "representative") %>%
#   select(-representative) %>%
#   rename(representative = Entry)
# 
# hit_genes <- unique(uniprotID_name$representative)
  
hit_genes_GO <- left_join(data.frame(gene = hit_genes),Protein_GO_annotations) %>% 
    dplyr::filter(!is.na(GO.id)) %>% 
    dplyr::filter(GO.id != "") %>% 
    dplyr::distinct() %>%
  filter(GO.id %in% unique(Protein_GO_annotations$GO.id))

# slim this too
# # slim this : if terms share exact same number of proteins keep the 1 with lowest hierarchy in GO (keep higher number)
test = hit_genes_GO %>%
  mutate(presence=rep(1,nrow(.))) %>%
  pivot_wider(names_from = gene, values_fill = 0,values_from = presence, id_cols = GO.id)

sorted_df <- test %>%
  arrange(across(-GO.id), desc(GO.id))

# Remove duplicate rows based on the values in all other columns, keeping the first occurrence (highest GO.id)
filtered_df <- sorted_df %>%
  distinct(across(-GO.id), .keep_all = TRUE)

filtered_dt <- as.data.table(filtered_df)

setDT(filtered_df)

# Melt the data.table
filtered_df <- melt(filtered_df, 
                        id.vars = names(filtered_df)[1], 
                        measure.vars = names(filtered_df)[-1])

hit_genes_GO = filtered_df %>%
  filter(value != 0) %>%
  filter(GO.id != "" & GO.id %in% unique(GOid.GOterms_clean$GO.id)) %>%
  select(-value) %>%
  rename(gene = variable)

#########


###Calculate size of foreground (how many genes are hits) and
###calculate the size of the background (how many genes are identified)
len_unique_fore_genes <- length(unique(hit_genes_GO$gene))
len_unique_back_genes <- length(unique(Protein_GO_annotations$gene))


###For each GO that is in the hit genes calculate the values needed for the Fisher test
GO_id_freq <- hit_genes_GO %>% 
    dplyr::group_by(GO.id) %>% 
    dplyr::summarise(freq_foreground = n()) %>% 
    dplyr::ungroup() %>% 
    left_join(GO_id_freq_background, by = c('GO.id')) %>% 
    dplyr::mutate(freq_back_minus_fore = freq_background - freq_foreground,
           freq_fore_non_GOid = len_unique_fore_genes - freq_foreground,
           freq_back_non_GOid = len_unique_back_genes - freq_back_minus_fore,
           frac_genes_GOid = freq_foreground / len_unique_fore_genes,
           fold_enrich = frac_genes_GOid / (freq_background / len_unique_back_genes),
           num_genes_back = freq_foreground + freq_back_minus_fore)

###For each GO perform the Fisher test
#fisher_test_result <- bind_rows(
  
 fisher_test_result <- bind_rows(lapply(GO_id_freq$GO.id, function(GO_id) {
#GO_id ="GO:0000018"   
  
  tmp <- GO_id_freq %>% 
        dplyr::filter(GO.id == GO_id) %>% 
        dplyr::select(freq_foreground, freq_fore_non_GOid, freq_back_minus_fore, freq_back_non_GOid)
  
# Print GO.id for debugging
  cat("Processing GO.id:", GO_id, "\n")
  
  if (nrow(tmp) == 0) {
    cat("Error with GO.id:", GO_id, "\n")
    stop("Empty dataframe encountered for GO.id:", GO_id)
  }
  
    data.frame(GO.id = GO_id,
                 GO.tree = GOid.GOterms_clean$GO.tree[GOid.GOterms_clean$GO.id == GO_id],
                 GO.term = GOid.GOterms_clean$GO.term[GOid.GOterms_clean$GO.id == GO_id],
                 hit_genes = paste(unique(hit_genes_GO$Uniprot_ID[hit_genes_GO$GO.id == GO_id]), collapse = '/'),
                 p_value = fisher.test(matrix(as.numeric(tmp), nrow = 2), alternative="greater")$p.value)
         }))


###Correct for multiple comparison with BH
fisher_test_result$p_adj <- p.adjust(fisher_test_result$p_value, method = "BH")

###Generate final table
fisher_test_joined <- dplyr::left_join(fisher_test_result,GO_id_freq, by = 'GO.id') %>% 
  dplyr::group_by(GO.tree) %>% 
  dplyr::arrange(p_adj, .by_group = T) %>% 
  dplyr::ungroup()


toplot = dplyr::filter(fisher_test_joined, p_adj < 0.05) 

ggplot(toplot, aes(y = reorder(GO.term, -p_adj),x = -log10(p_adj))) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
 geom_point(aes(size=freq_foreground,color=fold_enrich)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "number of genes", x = "-log10(adjusted p-value)", y = "",colour = "fold enrichment", title = "Decreased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9)) +
  guides(alpha="none")


write.table(fisher_test_joined, paste0(inputdir, "GOenrich_Keio_Up.txt"), sep = "\t", quote = F, row.names = F)
write.table(fisher_test_joined, paste0(inputdir, "GOenrich_Keio_Down.txt"), sep = "\t", quote = F, row.names = F)



# PLOT FOR PAPER
up = fread(paste0(inputdir, "GOenrich_Keio_Up.txt"), sep = "\t")
down = fread(paste0(inputdir, "GOenrich_Keio_Down.txt"), sep = "\t")

metal = c("4Fe-4S", "2Fe-2S", "siderophore", "iron","metal","zinc","copper","molybd","calci","manganese","ferr")
respir = c("respir","electron","quinone","succin","tricarbo","fumara","dehydro","NarG","cytochrome","proton")
OM = c("membran","acyltra","lipopolys")

toplot = gdata::combine(up,down) %>%
  mutate(annot = case_when(grepl(paste(metal,collapse="|"),GO.term) ~ "metal homeostasis",
                           grepl(paste(respir,collapse="|"),GO.term) ~ "respiration",
                           grepl(paste(OM,collapse="|"),GO.term) ~ "OM",
                           TRUE ~"other"),
         colour = case_when(GO.tree == "CC" ~ "dodgerblue3",
                                          GO.tree == "MF" ~ "tomato",
                                          TRUE ~ "darkgreen"),
         full_name = case_when(GO.tree == "CC" ~ "Cellular Component",
                                          GO.tree == "MF" ~ "Molecular Function",
                                          TRUE ~ "Biological Process"),
         fold.enrichment = as.numeric(fold_enrich),
         colour2 = case_when(annot == "metal homeostasis" ~ "tomato",
                             annot == "respiration"~ "dodgerblue3",
                             annot == "OM" ~ "darkgreen",
                                           TRUE ~ "black"),
         source=gsub(source,pattern="up",replacement="increased\nfitness"),
         source=gsub(source,pattern="down",replacement="decreased\nfitness")) 


toplot = toplot %>%
         filter(p_adj <= 0.05) %>%
  mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]


toplotup = toplot %>%
  mutate(upordown = rep("increased\nfitness", nrow(.)))

toplot = bind_rows(toplotup,toplotdown)





test = toplot %>%
              filter(source == "decreased\nfitness")

p <- ggplot(test,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  facet_wrap(~source,scales="free") +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(min(toplot$p_adj),max(toplot$p_adj))) +
  scale_size_continuous(limits = c(min(toplot$freq_foreground),max(toplot$freq_foreground))) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 60)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 11, color= test$colour2),
        legend.position="none") 


test = toplot %>%
              filter(source == "increased\nfitness")

p1 <-ggplot(test,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  facet_wrap(~source,scales="free") +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(min(toplot$p_adj),max(toplot$p_adj))) +
  scale_size_continuous(limits = c(min(toplot$freq_foreground),max(toplot$freq_foreground))) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 60)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 11, color= test$colour2),
        legend.direction = "horizontal", legend.box = "horizontal") +
  theme(legend.position="none")

design <- "
1122
11##
11##
"

pdf(paste0(plotdir, "GO_Keio.pdf"), width = 12, height = 6)
p + p1 +  plot_layout(design = design)

dev.off()


```



#### original GO from George
```{r}
source('~/Documents/EMBL/Nx/TPP_nx/GOenrichment/chemgen_GOfunc.R', echo=TRUE)

all <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond =="nitro2")%>%
  mutate(gene=toupper(gene)) %>%
  mutate(gene = case_when(gene == "MURE (D3A)-KAN" ~ "MURE",
                          gene == "LPXC (G210S)-KAN" ~ "LPXC",
                          gene == "IMP4213" ~ "LPTD",
                          gene == "GNTY" ~ "NFUA",
                          TRUE ~ gene)) %>%
  filter(!grepl("^NOT", gene)) %>%
  distinct()


# # set hit cutoff according to how many IQR above 1st quartile or below 3rd quartile
# #iqr = quantile(all$log2fc, na.rm=T)[[4]] - quantile(all$log2fc, na.rm=T)[[2]]
# 
# lowthr = (quantile(all$log2fc,na.rm=T)[[2]]) - (2*0.9529 * (IQR(all$log2fc, na.rm = T)))
# upthr = (quantile(all$log2fc,na.rm=T)[[4]]) + (2*0.9529 * (IQR(all$log2fc, na.rm = T)))

# other option based on MAD = 1.4862 median (| xi - median(x) |)
keiomad = mad(all$log2fc, na.rm = T)
lowthr = median(all$log2fc, na.rm = T) - 3*keiomad
upthr = median(all$log2fc, na.rm = T) + 3*keiomad

# hits <- all %>%
#   filter(padj <= 0.01)
# quantile(hits$log2fc)

# decreased fitness
hits <- all %>%
  filter(padj <= 0.01 & log2fc < lowthr)

# hits <- all %>%
#   filter(padj <= 0.01 & log2fc < -0.05)

#hits$clustername = as.character(hits_df_whole$clustername)
hits$genenew = sapply(hits$gene,function(x)`substring<-`(x,1,3,tolower(x)))

genes = setdiff(unique(hits$genenew), grep('^Y', hits$genenew, value = T))
genes = setdiff(unique(genes), grep('^y', genes, value = T))

go_genes = GO.enrichment.function.3(genes)

go_genes$gonumb = as.numeric(sapply(strsplit(as.character(go_genes$GO.id), "GO:00"), "[", 2))

metalstuff = c("iron-sulfur cluster assembly", "enterobactin biosynthetic process","metal ion binding","zinc ion binding", "iron ion homeostasis")
OM = c("lipopolysaccharide core region biosynthetic process")
resp = c("aerobic electron transport chain","cytochrome o ubiquinol oxidase complex","electron transport coupled proton transport", "energy transducer activity",  "hydrogen ion transmembrane transporter activity", "ubiquinone binding")

go_genes = go_genes %>%
  mutate(colour2 = case_when(GO.term %in% metalstuff ~ "tomato",
                             GO.term %in% OM ~ "dodgerblue3",
                             GO.term %in% resp ~ "darkgreen",
                                           TRUE ~ "black"))

toplot = go_genes %>%
         dplyr::filter(p.value.BH <= 0.05) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]

toplotdown = toplot %>%
  mutate(upordown = rep("decreased\nfitness", nrow(.)))


# p = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
#   geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
#   geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
#   scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
#   scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
#   labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased fitness") +
#   theme_pubr()+
#   theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2)) #+
#   #guides(alpha="none", colour = "none", fill= "none", size = "none")

# increased fitness
hits <- all %>%
  filter(padj <= 0.01 & log2fc > upthr)

#hits$clustername = as.character(hits_df_whole$clustername)
hits$genenew = sapply(hits$gene,function(x)`substring<-`(x,1,3,tolower(x)))

genes = setdiff(unique(hits$genenew), grep('^Y', hits$genenew, value = T))
genes = setdiff(unique(genes), grep('^y', genes, value = T))

go_genes = GO.enrichment.function.3(genes)

go_genes$gonumb = as.numeric(sapply(strsplit(as.character(go_genes$GO.id), "GO:00"), "[", 2))

metalstuff = c("iron-sulfur cluster assembly", "enterobactin biosynthetic process","metal ion binding","zinc ion binding", "iron ion homeostasis")
OM = c("lipopolysaccharide core region biosynthetic process")
resp = c("aerobic electron transport chain","cytochrome o ubiquinol oxidase complex","electron transport coupled proton transport", "energy transducer activity",  "hydrogen ion transmembrane transporter activity", "ubiquinone binding")

go_genes = go_genes %>%
  mutate(colour2 = case_when(GO.term %in% metalstuff ~ "tomato",
                             GO.term %in% OM ~ "dodgerblue3",
                             GO.term %in% resp ~ "darkgreen",
                                           TRUE ~ "black"))

toplot = go_genes %>%
         dplyr::filter(p.value.BH <= 0.05) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))
toplot = toplot[order(toplot$GO.term), ]


toplotup = toplot %>%
  mutate(upordown = rep("increased\nfitness", nrow(.)))

toplot = bind_rows(toplotup,toplotdown)


pdf(paste0(plotdir, "GO_Keio.pdf"), width = 12, height = 4)
# 
# toplot = toplot %>%
#   dplyr::group_by(upordown) %>%
#   dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment))) %>%
#   dplyr::ungroup()
# 
# toplot = toplot[order(toplot$GO.term), ]

ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=number.of.hit.genes.in.GOid, colour=p.value.BH)) +
  facet_wrap(~upordown, scales="free_y") +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits=c(0,0.05), breaks=c(0,0.05)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
  scale_size_continuous(breaks=c(4,8,12,14)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour2),
        legend.position="right")# +
  #guides(alpha="none", colour = "none", fill= "none", size = "none")

dev.off()

```


###OLD - ED Fig. 4
###a- Keio overlap with TPP - use Nx 2
```{r}
all <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond =="nitro2")%>%
  mutate(gene=toupper(gene)) %>%
  mutate(gene = case_when(gene == "MURE (D3A)-KAN" ~ "MURE",
                          gene == "LPXC (G210S)-KAN" ~ "LPXC",
                          gene == "IMP4213" ~ "LPTD",
                          gene == "GNTY" ~ "NFUA",
                          gene == "YHBV" ~ "UBIV",
                          gene == "YHBU" ~ "UBIU",
                          gene == "YHBZ" ~ "OBGE",
                          gene == "YHBT" ~ "UBIT",
                          gene == "YHBJ" ~ "RAPZ",
                          gene == "YHBH" ~ "HPF",
                          gene == "YHBT" ~ "UBIT",
                          gene == "YHBC" ~ "RIMP",
                          gene == "RFAB" ~ "WAAB",
                          gene == "RFAC" ~ "WAAC",
                          gene == "RFAD" ~ "HLDD",
                          gene == "RFAE" ~ "HLDE",
                          gene == "RFAF" ~ "WAAF",
                          gene == "RFAG" ~ "WAAG",
                          gene == "RFAI" ~ "WAAO",
                          gene == "RFAJ" ~ "WAAJ",
                          gene == "RFAL" ~ "WAAL",
                          gene == "RFAP" ~ "WAAP",
                          gene == "RFAQ" ~ "WAAQ",
                          gene == "RFAS" ~ "WAAS",
                          gene == "RFAY" ~ "WAAY",
                          gene == "RFAZ" ~ "WAAZ",
                          gene == "YIFC" ~ "WZZE",
                          gene == "VACJ" ~ "MLAA",
                          gene == "LPCA" ~ "GMHA",
                          gene == "YAET-KAN" ~ "BAMA",
                          gene == "GNTY" ~ "NFUA",
                          gene == "YDHD" ~ "GRXD",
                          gene == "MPRA" ~ "EMRR",
                          TRUE ~ gene)) %>%
  filter(!grepl("^NOT", gene))

hits <- all %>%
  filter(cond =="nitro2")%>%
  filter(padj <= 0.01)

hits_df_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/hits_wholecell.txt", sep = "\t", header = T)

olap = intersect(hits$gene,hits_df_whole$clustername)
onlytpp = setdiff(hits_df_whole$clustername,hits$gene)
onlyKeio = setdiff(hits$gene, hits_df_whole$clustername)


# toplot = hits %>%
#          filter(cond %in% c("Nitroxoline 1 µg/ml + CuCl2 1 mM", "Nitroxoline 2 µg/ml + CuCl2 1 mM"))

toplot = all %>%
         mutate(olap = case_when(gene %in% olap ~ "both",
                                 gene %in% onlytpp ~ "2DTPP",
                                 gene %in% onlyKeio ~ "chemical genetics",
                                 TRUE ~ "neutral"))

labelhits = toplot %>%
  filter((olap %in% c("both", "chemical genetics") & abs(log2fc) > 0.5 | gene == "COPA"))
  #filter(olap == "both" & abs(log2fc) > 0.5)

colours = data.frame("olap" = c("both", "2DTPP", "chemical genetics"),
                     "values" = c("#005AB5", "darkgrey", "#DC3220"))
jColors <- as.character(colours$values)
names(jColors) <- colours$olap

pdf(paste0(plotdir, "Keio_TPP_overlapvolcano.pdf"),width= 4, height = 4)
ggplot(toplot,  aes(x = log2fc, y = -log10(padj))) +
  # geom_point(data=toplot%>%filter(olap=="neutral"), fill="grey",
  # pch=21, col='grey30', size=1, alpha=0.5) +
geom_point(data=toplot%>%filter(olap!="neutral"), aes(fill= olap, col = olap), 
  pch=21, size=1, alpha =0.7) +
  scale_fill_manual(values=jColors, name="") +
  scale_color_manual(values=jColors, name="") +
geom_abline(intercept = -log10(0.01), slope = 0, linetype = "dashed") +
  geom_vline(xintercept = 0.5,linetype ="dashed") +
  geom_vline(xintercept = -0.5, linetype ="dashed") +
  facet_wrap(~cond, ncol=1) +
  ggrepel::geom_text_repel(aes(label = gene), force = 0.5,max.overlaps = 100, size = 2, data = labelhits) +
  labs(fill = "") +
  guides(fill="none") +
  ylab(expression('-log'[10]~'(adjusted p-val)')) +
  xlab(expression('log'[2]~'(fold-change)')) +
  theme_bw() +
  theme(strip.text = element_blank(),
        legend.position="bottom")
  
dev.off()


```

### b - upset plot Keio-TPP
```{r}
# find genes that are in Keio no matter if hit or not 
genestoconsider <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond =="nitro2")%>%
  mutate(gene=toupper(gene))%>%
  mutate(gene = case_when(gene == "MURE (D3A)-KAN" ~ "MURE",
                          gene == "LPXC (G210S)-KAN" ~ "LPXC",
                          gene == "IMP4213" ~ "LPTD",
                          gene == "GNTY" ~ "NFUA",
                          TRUE ~ gene),
         exp = rep("Chemical genetics", nrow(.))) %>%
  filter(!grepl("^NOT", gene)) %>%
  distinct(gene)



hits <- readRDS("~/Documents/EMBL/Nx/Keio_screen/Vallo/chemgen_log2fc.rds") %>%
  filter(cond =="nitro2")%>%
  filter(padj <= 0.01) %>%
  mutate(gene=toupper(gene))%>%
  mutate(gene = case_when(gene == "MURE (D3A)-KAN" ~ "MURE",
                          gene == "LPXC (G210S)-KAN" ~ "LPXC",
                          gene == "IMP4213" ~ "LPTD",
                          gene == "GNTY" ~ "NFUA",
                          TRUE ~ gene),
         exp = rep("Chemical genetics", nrow(.))) %>%
  filter(!grepl("^NOT", gene)) %>%
  select(gene,exp)

hits_df_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/hits_wholecell.txt", sep = "\t", header = T) %>%
  select(clustername) %>%
  mutate(exp = rep("2DTPP", nrow(.))) %>%
  rename(gene=clustername) %>%
  #remove from here genes that are not in keio collection to begin with
  filter(gene %in% genestoconsider$gene)
  

test = rbind.data.frame(hits,hits_df_whole) 
        

test = table(test)
test[test>0] = 1

test = as.data.frame(test)

test = test %>%
  pivot_wider(names_from = exp, values_from = Freq) %>%
  as.data.frame()

test = test[apply(test, 1, function(x) !all(x==0)),]

test1 = dplyr::mutate(test, C = ifelse(`Chemical genetics`, "CG", NA),
       P = ifelse(`2DTPP`, "TPP", NA))

test1 = tidyr::unite(test1, "Methods", C:P, na.rm=T, sep='&')


names(upset_themes)

pdf(paste0(plotdir, "TPP_CG_upset.pdf"), width = 6, height = 4)

ComplexUpset::upset(test1,c("Chemical genetics", "2DTPP"),
  stripes = "white",
  sort_intersections=FALSE,
  base_annotations=list('Intersection size'=intersection_size(text = list(size=5.5),
      text_colors = c(on_background = "black", on_bar = "black"),
      mapping=aes(fill=Methods))),
  set_sizes=(upset_set_size(
    #mapping = aes(fill=type),
    geom = geom_bar(width = 0.5),
    position = "right") +
    geom_text(aes(label=..count..), hjust=1.2, stat='count', size =5.5, color ="white", fontface="bold") +
      expand_limits(y=30) +
      theme(axis.text.x=element_text(size = 8))),
  matrix=intersection_matrix(
    geom = geom_point(size = 3),
    segment = geom_segment(),
    outline_color = list(active = "black", inactive = "grey70")
  ),
  themes = upset_modify_themes(list('Intersection size'=theme(axis.line.x = element_blank(),
                                                              axis.ticks.x = element_blank(),
                                                              axis.line.y = element_line(),
                                                              axis.text = element_text(size=15),
                                                              #text = element_text(size=10),
                                                              legend.position = 'none',
                                                              legend.text = element_text(size=10),
                                                              axis.title=element_text(size=15),
                                                              axis.ticks.y = element_line(),
                                                              plot.background = element_blank(),
                                                              panel.grid = element_blank()),
                                    'intersections_matrix'=theme(axis.line.x = element_blank(),
                                                                 axis.ticks.x = element_blank(),
                                                                 axis.title = element_blank(),
                                                                 axis.text = element_text(size=15),
                                                                #text = element_text(size=10),
                                                                plot.background = element_blank(),
                                                                 panel.grid = element_blank()),
                                    'overall_sizes'=theme(axis.line.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          axis.line.x = element_line(),
                                                          axis.ticks.x = element_line(),
                                                          axis.text.x.bottom = element_text(size=15),
                                                          axis.title = element_text(size=15),
                                                          plot.background = element_blank(),
                                                          text = element_text(size=15),
                                                          panel.grid = element_blank()))),
  width_ratio=0.6,
  height_ratio = 0.4,
sort_sets = F
) & scale_fill_manual(values = c( "#DC3220","#005AB5","darkgrey"))

dev.off()




```


```{r}

test = rbind.data.frame(hits,hits_df_whole) 
a = test %>%
group_by(gene) %>% 
dplyr::summarize(exp = list(exp))


install.package("ggupset")

library(ggupset)

p1=ggplot(a, aes(x=exp)) +
    geom_bar() +
    scale_x_upset(n_intersections = 20) +
  theme_bw()
 
 b = test %>%
   group_by(exp) %>%
   dplyr::summarise(count=n())


   
p2= ggplot(b, aes(x=exp, y= count)) + geom_bar(stat = "identity") +
   coord_flip()+
  theme_bw()
 
pdf(paste0(plotdir,"EDFig.6a_left_new.pdf"),width = 3, height=7)
 cowplot::plot_grid(p1,p2,nrow=2)
dev.off()


```


## ED Fig. 6b - iron

```{r}
capitalize_first_and_last <- function(string) {
  first_letter <- toupper(substring(string, 1, 1))
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string) - 1))
  return(paste(first_letter, middle_letters, last_letter, sep = ""))
}

df_final_whole = read.table("~/Documents/EMBL/Nx/TPP_nx/wholecell/df_final.txt", sep = "\t", header = T) %>%
  mutate(clustername= case_when(clustername == "FUR" ~ "Fur",
                                TRUE ~ capitalize_first_and_last(clustername))) 


a <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "Fur",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "IscR",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "EntA",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "FepA",
  drug_name = "Nitroxoline")
e <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "SufB",
  drug_name = "Nitroxoline")



pdf(paste0(plotdir, "metals_thermal_profile_iron.pdf"), width = 10, height = 4)

cowplot::plot_grid(a,b,c,d,e, ncol =3)
#a + b + c + d + plot_layout(guides = "collect")

dev.off()

```

# ED 5g - EPI

```{r}
#toplot <- readxl::read_xlsx(paste0(inputdir, "231002_Efflux_pump_inhibitors_basecamp.xlsx"), skip=1) %>%
toplot <- readxl::read_xlsx(paste0(inputdir, "Nx_Efflux_pump_inhibitors_20.02.2024_SG.xlsx"), skip=1, n_max = 19) %>%
  pivot_longer(cols = c(5:16),names_to = "sample",values_to = "MIC") %>%
  rename(species =Specie, strain=Isolate) %>%
  mutate(sample = sapply(strsplit(as.character(sample),"\\..."),'[',2),
         sample = as.numeric(sample),
         replicate=case_when(sample %in% c(5,9,13) ~ 1,
                             sample %in% c(6,10,14) ~ 2,
                             sample %in% c(7,11,15) ~ 3,
                             sample %in% c(8,12,16) ~ 4,
                             TRUE ~ sample),
         condition = case_when(sample %in% c(5:8) ~ "no inhibitor",
                               sample %in% c(9:12) ~ "+ 100 µg/ml NMP",
                               sample %in% c(13:16) ~ "+ 25 µg/ml PAβN"),
         strain = gsub(strain,pattern="ATCC19606",replacement="19606"),
         strain = gsub(strain,pattern="ATCC 17978",replacement="17978"),
         newspec= case_when(strain %in% c("25922","MS346", "MS529") ~ paste0("italic('E. coli ')*",strain),
                            strain %in% c("19606","MS69", "MS72") ~ paste0("italic('A. baumannii ') *",strain),
                            strain %in% c("PRZ","MS25","MS386","MS274", "MS530") ~ paste0("italic('K. pneumoniae ') *",strain))) %>%
  clean_names() %>%
  select(c(mic,replicate, condition, newspec,strain))

old <- readxl::read_xlsx(paste0(inputdir, "230825_Efflux_inhibitors.xlsx")) %>%
  #janitor::clean_names() %>%
  pivot_longer(cols=c(3:11), names_to = "condition", values_to = "MIC") %>%
  mutate(Strain = gsub(Strain, pattern = " ", replacement=""),
         Strain = gsub(Strain, pattern = "ATCC", replacement=""),
         Strain = gsub(Strain, pattern = "R", replacement=""),
        Species = gsub(Species, pattern = "Acinetobacter baumannii", replacement="italic('A. baumannii ') *"),
         Species = gsub(Species, pattern = "Escherichia coli", replacement="italic('E. coli ')*"),
         Species = gsub(Species, pattern = "Klebsiella pneumoniae", replacement="italic('K. pneumoniae ') *"),
         Species = gsub(Species, pattern = "Pseudomonas aeruginosa", replacement="italic('P. aeruginosa ') *")) %>%
        janitor::clean_names() %>%
  mutate(newspec = interaction(species,strain,sep=""),
         condition = sapply(strsplit(as.character(condition), "\\..."), '[',1),
         strain_condition=interaction(strain,condition),
         replicate=ave(mic, strain_condition, FUN = seq_along)) %>%
  mutate(condition = case_when(condition == "CAMHB" ~ "no inhibitor",
                               condition == "+NMP" ~ "+ 100 µg/ml NMP",
                               condition == "+PAβN" ~ "+ 25 µg/ml PAβN")) %>%
  select(c(newspec,replicate, condition, mic,strain))

toplot=rbind.data.frame(toplot,old) 

# some strains are the same!! errors of naming

toplot = toplot %>%
  mutate(newspec = gsub(newspec, pattern = "italic('A. baumannii ') * MS329", replacement="italic('A. baumannii ') * MS347"),
         newspec = gsub(newspec, pattern = "italic('A. baumannii ') * MS348", replacement="italic('A. baumannii ') * MS69"),
         newspec = gsub(newspec, pattern = "italic('E. coli ')* ATCC 25922 R", replacement="italic('A. baumannii ') * MS346"))%>%
  mutate(straincond=interaction(newspec,condition),
         replicate=ave(mic, straincond,FUN=seq_along))

### here write source data for this and ED Fig
towrite = toplot %>%
  mutate(species = str_extract(newspec,"E\\. coli|K\\. pneumoniae|A\\. baumannii")) %>%
  dplyr::select(-c(newspec,straincond)) %>%
  rename(`MIC(µg/ml)`=mic)

write_xlsx(towrite,paste0(sourcedatadir,"source_Fig.5e_EDFig.8.xlsx"))


# show only those on which we have MS!!!
strains = c("italic('A. baumannii ') *19606", "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72",
            "italic('E. coli ')*25922", "italic('E. coli ')*MS346","italic('E. coli ')*MS529",
            "italic('K. pneumoniae ') *PRZ",
            "italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386",
            "italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530")


mainfig = toplot %>%
  filter(newspec %in% strains)

# dispose in rows as I want 
mainfig = mainfig %>%
  mutate(row= case_when(newspec %in% c("italic('A. baumannii ') *19606", "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72") ~ 1,
                        #newspec == "italic('A. baumannii ') *MS72" ~ 2, 
                        newspec %in% c("italic('E. coli ')*25922", "italic('E. coli ')*MS346","italic('E. coli ')*MS529") ~ 3,
                        newspec %in% c("italic('K. pneumoniae ') *PRZ","italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386","italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530") ~ 4),
                        #newspec %in% c("italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *MS530") ~ 5),
         resist = case_when(newspec %in% c("italic('A. baumannii ') *19606","italic('E. coli ')*25922","italic('K. pneumoniae ') *MS25", "italic('K. pneumoniae ') *PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))

# # shorten Abau 19606 label
# mainfig = mainfig %>%
#   mutate(newspec= gsub(newspec, pattern = "ATCC",
#                        replacement="")) %>%
#   rename(MIC =mic)

# fix facet order
mainfig$facet = factor(mainfig$newspec, levels = c("italic('A. baumannii ') *19606",
                                                   "italic('A. baumannii ') *MS69","italic('A. baumannii ') *MS72",
                                                   "italic('E. coli ')*25922", "italic('E. coli ')*MS346",
                                                   "italic('E. coli ')*MS529", "italic('K. pneumoniae ') *MS25", "italic('K. pneumoniae ') *MS274", "italic('K. pneumoniae ') *MS386", "italic('K. pneumoniae ') *PRZ", "italic('K. pneumoniae ') *MS530"))


p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="right")

legend_grob <-cowplot::get_legend(p)

p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "",y="MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=4) +
   theme(axis.text.x = element_blank(),
         #axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")


p2 <- ggerrorplot(mainfig %>% filter(row==3), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(mainfig,row==3 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  scale_y_continuous(trans="log2") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p3 <- ggerrorplot(mainfig %>% filter(row==4), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ fct_relevel(newspec, "italic('K. pneumoniae ') *MS25","italic('K. pneumoniae ') *MS386","italic('K. pneumoniae ') *MS274","italic('K. pneumoniae ') *PRZ","italic('K. pneumoniae ') *MS530"), labeller = label_parsed,ncol=5)+ 
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==4 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  scale_y_continuous(trans="log2") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5.5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")


library("latex2exp")


cairo_pdf(paste0(plotdir,"effluxpump_new1.pdf"), width =9.5,height=4)
 
layout <-"AAACCC
          DDDDDG"
wrap_plots(A=p, C=p2,D=p3,G=legend_grob, design = layout)

dev.off()






```

# ED Fig 7

## ED Fig. 7a - EmrR mutplot
```{r}
proteins_acc<-read.table("~/Documents/EMBL/Nx/Structures_Foldseek/mutplot/inputdir/UniProt.txt",header=T) %>%
      filter(Hugo_Symbol %in% c("EmrR"))


template=fread("~/Documents/EMBL/Nx/Nx_paper/input/Ecoli_isolates_MIC.txt", sep="\t") %>%
  filter(!(Sample_ID %in% c("1639","T2761","T2762", "SC1764_unicycler","E28-1-Novastar"))) %>%
  drop_na(MIC) %>%
  filter(MIC > 4) %>%
  mutate(wgs = case_when(Sample_ID %in% c("MS346", "MS120", "MS548", "MS579","MS594","MS368","MS511","MS529","MS569","MS572","MS577","MS593","MS516") ~ "WGS",
                         TRUE ~ "only SNP"))
    
    var = template %>%
      group_by(Hugo_Symbol, Mutation_Type, Protein_Change,Isolate_type) %>%
      mutate(freq=n(),
             location=str_extract(Protein_Change, "(?<=\\D)\\d+")) %>%
      # MIC=max(MIC)) %>%
      ungroup() %>%
      distinct(Protein_Change,Mutation_Type,MIC,freq,location,Isolate_type) %>%
      mutate(location=as.numeric(location),
             Isolate_type=case_when(!(Isolate_type %in% c("clinical isolate","generated")) ~ "unidentified",
                                    Isolate_type == "generated" ~ "experimentally evolved",
                                    TRUE ~ Isolate_type))
    
    var$MIC[is.na(var$MIC)] <- 0
    
    #proteins_acc<- proteins_acc[proteins_acc$Hugo_Symbol==var.plot$Hugo_Symbol,2][[1]]
    proteins_acc_url<-gsub(" ","%2C",proteins_acc)[[2]]
    
    baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
    url<-paste0(baseurl,proteins_acc_url)
    prots_feat<-GET(url,accept_json())
    prots_feat_red<-httr::content(prots_feat)
    features_total_plot<-NULL
    for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])#the extract_feat_acc() function takes features into a data.frame
      # features_temp$order<-i # this order is needed for plotting later
      features_total_plot<-rbind(features_total_plot,features_temp)
    }
    
    
    
    
    plot_start<-0#starts 0
    plot_end<-max(features_total_plot$end)
    
    # shapes = data.frame("Mutation_Type" = c("Frame_Shift_Ins","Missense_Mutation","Nonsense_Mutation"),
    #                      "values" = c(16, 17,15))
    # jshapes <- as.character(shapes$values)
    # names(jshapes) <- shapes$Mutation_Type
    
    
colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type
    
var= var %>% filter(Isolate_type != "unidentified") %>%
  filter(MIC != 0) %>%
  mutate(Mutation_Type =case_when(Mutation_Type == "Frame_Shift_Ins" ~ "Frameshift/Insertion",
                                  Mutation_Type == "Nonsense_Mutation" ~ "Nonsense",
                                  Mutation_Type == "Missense_Mutation" ~ "Missense"))


pdf(paste0(plotdir,"EmrR_mutplot.pdf"), width=6.5, height=4.5)

ggplot(var,aes(location,MIC,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location,yend=MIC)) +
  geom_point(aes(size=freq, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=var %>%
                    distinct(location,MIC, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=150, size=3) +
  geom_hline(yintercept = 0) +
  geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
                                 mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 89, y = 2.3, label ="HTH marR-type", size = 4,color="white",fontface="bold") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_size_continuous(breaks = c(1,3,6,9,12), limits=c(1,12)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,176), limits=c(0,176))+
  #scale_y_continuous(trans=scales::pseudo_log_trans(base = 2), breaks= c(16,32,64,128)) +
  scale_y_continuous(breaks = c(2,4,8,16,32,64),trans="log2",limits=c(2,100)) +
  labs(y= expression('log'[2]~'(MIC [µg/mL])'),size="Mutation occurrencies", title = "EmrR", shape = "Mutation type")+
  #guides(color="none") +
  theme_bw() +
  theme(plot.title=element_text(face="bold",size=(15),hjust=0.5),
        legend.position = "none",
        legend.title = element_text(size=10),
        legend.text = element_text(size=10),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.1,0.1,0.1,0.1, unit="cm"),
        panel.border = element_blank(), axis.line = element_line())

dev.off()
    

towrite = var %>%
  rename(`MIC (µg/ml0)`= MIC,
         Mutation_occurrences = freq,
         location_aa_number=location)
  
write_xlsx(towrite, paste0(sourcedatadir,"source_EDFig.7a.xlsx"))


```

## ED 7b-c - clustered proteomics
### all genes all strains - just for me to analyse
```{r}

toplot = fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
  group_by(og1, complete_name) %>%
  summarise(logFC=mean(logFC))

mymat = toplot %>% 
  dplyr::select(c(og1, complete_name, logFC)) %>%
  unique()


is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}
# mymat = fix_na(mymat)

library(magrittr)

mymat = mymat %>% pivot_wider(names_from = complete_name, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og1) %>%
  dplyr::select(-og1) %>%
  as.matrix()


matrix.please<-function(x) {
  m<-as.matrix(x[,-1])
  rownames(m)<-x[,1]
  m
}

toplot = matrix.please(as.data.frame(mymat))
toplot=mymat


library(gplots)

mat_data = toplot

# plot

robust_dist = function(x, y) {
    qx = quantile(x, c(0.1, 0.9))
    qy = quantile(y, c(0.1, 0.9))
    l = x > qx[1] & x < qx[2] & y > qy[1] & y < qy[2]
    x = x[l]
    y = y[l]
    sqrt(sum((x - y)^2))
}



# view dendrogram to understand what to show
hc <- hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete")
hr <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)


pdf(paste0(plotdir,"clustered_MS.pdf"), width=10, height=105)

Heatmap(mat_data,
        cluster_rows = hr,
        row_names_gp = gpar(fontsize = 1),
        #clustering_distance_rows = "pearson",
        #clustering_distance_columns = "pearson",
        )

dev.off()


library(dendextend)
dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
mycolors <- colorRampPalette(brewer.pal(8, "Set2"))(10)
dend1 <- color_branches(dend1, k = c_group, col =  mycolors) # add color to the lines
dend1 <- color_labels(dend1, k = c_group, col = mycolors)   # add color to the labels
col_labels <- get_leaves_branches_col(dend1)
col_labels <- col_labels[order(order.dendrogram(dend1))]


pdf(paste0(plotdir, "dendrogram_MS_genes.pdf"), width=1, height = 105)
dend1 <- set(dend1, "labels_cex", 0.2)
par(mar = c(0.5,2,0.5,1))
plot_horiz.dendrogram(dend1, side = T)
dev.off()

```

### only significant genes, resistant strains
```{r}

toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))) %>%
   mutate(complete_name = sapply(strsplit(as.character(complete_name)," "),'[',3)) %>%
  mutate(complete_name = sub("[RS]$", "", complete_name)) %>%
    group_by(og1, complete_name) %>%
    summarise(logFC=mean(logFC)) %>%
  #remove dmprA and dadeL
    filter(!(complete_name %in% c("dadeL","dmprA")))

# hitgenes_insensitive =fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
#   filter(og1 %in% unique(toplot$og1))



# fix annotation of strains in complete name
strain_key = readxl::read_xlsx(paste0(inputdir,"new_designation_240525_MT_EC.xlsx")) %>%
  clean_names() %>%
  mutate(complete_name=sapply(strsplit(as.character(old_designation)," "),'[',3),
         new_designation=sapply(strsplit(as.character(new_designation)," "),'[',3),
        species = sub(" MS.*", "", old_designation))

test= inner_join(strain_key,toplot) %>%
  filter(complete_name %in% c("MS274", "MS299", "MS311", "MS346", "MS366", "MS386", "MS493", "MS511", "MS529", "MS530", "MS69",  "MS72")) %>%
  mutate(new_designation=interaction(species,new_designation))


mymat = test %>% 
  dplyr::select(c(og1, new_designation, logFC)) %>%
  unique()


library(magrittr)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}

mymat = mymat %>% pivot_wider(names_from = new_designation, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og1) %>%
  dplyr::select(-og1) %>%
  as.matrix()


mat_data=mymat


library(gplots)

# plot

# robust_dist = function(x, y) {
#     qx = quantile(x, c(0.1, 0.9))
#     qy = quantile(y, c(0.1, 0.9))
#     l = x > qx[1] & x < qx[2] & y > qy[1] & y < qy[2]
#     x = x[l]
#     y = y[l]
#     sqrt(sum((x - y)^2))
# }


#  dendrogram 
library(dendsort)
library(dendextend)
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)

#row_dend = color_branches(dend1, k = 6) # `color_branches()` returns a dendrogram object


# species_legend=fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
#   filter(!(Strain_treatment %in% c("BW25113","dmprA",
#                                    "19606S","25922S","2778S",
#                                    "MS02S","MS25S",
#                                    "MS55S", "PRZS")))



type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(new_designation = type) %>%
  inner_join(strain_key %>%
               mutate(new_designation=interaction(species,new_designation)) %>%
               dplyr::select(new_designation,species), by="new_designation") %>%
    distinct() %>%
    select(-new_designation),
   annotation_height = unit(4, "mm"),
  show_legend = F,
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )

library(ComplexHeatmap)

pdf(paste0(plotdir,"clustered_MS_only_hits_resistant_forSuppFigs.pdf"), width=6, height=20)
rownames(mat_data) = NULL

Heatmap(mat_data,
        cluster_rows = dend1,
        #cluster_cols=hc,
         top_annotation = ha,
        show_heatmap_legend = F,
        row_split = 4, column_split = 5,
        use_raster = TRUE, raster_quality = 5)
        #rect_gp = gpar(col = "white", lwd = 2)
        #clustering_distance_rows = "pearson",
        #clustering_distance_columns = "pearson",

#draw(hm, heatmap_legend_side="top", annotation_legend_side="top",
#           legend_grouping="adjusted")

dev.off()


 library(dendextend)
dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
mycolors <- colorRampPalette(brewer.pal(8, "Set2"))(10)
dend1 <- color_branches(dend1, k = c_group, col =  mycolors) # add color to the lines
dend1 <- color_labels(dend1, k = c_group, col = mycolors)   # add color to the labels
col_labels <- get_leaves_branches_col(dend1)
col_labels <- col_labels[order(order.dendrogram(dend1))]




pdf(paste0(plotdir, "dendrogram_MS_genes_only_hits.pdf"), width=10, height = 10)
# dend1 <- set(dend1, "labels_cex", 0.2)
# par(mar = c(0.5,2,0.5,1))
# plot_horiz.dendrogram(dend1, side = T)

circlize_dendrogram(dend1,
                    labels_track_height = 2,
                    dend_track_height = 0.8,
                    facing = "outside",
                    labels=FALSE)

dev.off()
```

### only significant genes, resistant strains - long format for me to analyse
```{r}

toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS")))
# hitgenes_insensitive =fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
#   filter(og1 %in% unique(toplot$og1))

species_legend=fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS")))


toplot = toplot %>%
  #rbind.data.frame(toplot,hitgenes_insensitive) %>%
    group_by(og1, complete_name) %>%
    summarise(logFC=mean(logFC)) %>%
  #remove dmprA and dadeL
    filter(!(complete_name %in% c("A. baumannii dadeL","E. coli dmprA")))

mymat = toplot %>% 
  dplyr::select(c(og1, complete_name, logFC)) %>%
  unique()


is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}
# mymat = fix_na(mymat)

library(magrittr)

mymat = mymat %>% pivot_wider(names_from = complete_name, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og1) %>%
  dplyr::select(-og1) %>%
  as.matrix()


matrix.please<-function(x) {
  m<-as.matrix(x[,-1])
  rownames(m)<-x[,1]
  m
}

toplot = matrix.please(as.data.frame(mymat))
toplot=mymat


library(gplots)

mat_data = toplot

# plot

# robust_dist = function(x, y) {
#     qx = quantile(x, c(0.1, 0.9))
#     qy = quantile(y, c(0.1, 0.9))
#     l = x > qx[1] & x < qx[2] & y > qy[1] & y < qy[2]
#     x = x[l]
#     y = y[l]
#     sqrt(sum((x - y)^2))
# }


#  dendrogram 
library(dendsort)
library(dendextend)
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)

#row_dend = color_branches(dend1, k = 6) # `color_branches()` returns a dendrogram object

type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(complete_name = type) %>%
  inner_join(species_legend %>%select(complete_name,species), by="complete_name") %>%
    distinct() %>%
    select(-complete_name),
   annotation_height = unit(4, "mm"),
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )



pdf(paste0(plotdir,"clustered_MS_only_hits_resistant_forSuppFigs_longforme.pdf"), width=5, height=50)
#rownames(mat_data) = NULL

Heatmap(mat_data,
        cluster_rows = dend1,
          row_names_gp = gpar(fontsize = 4),
        #cluster_cols=hc,
        row_split = 4, column_split = 4,
        use_raster = TRUE, raster_quality = 5)
        #rect_gp = gpar(col = "white", lwd = 2)
        #clustering_distance_rows = "pearson",
        #clustering_distance_columns = "pearson",

dev.off()



library(dendextend)
dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
mycolors <- colorRampPalette(brewer.pal(8, "Set2"))(10)
dend1 <- color_branches(dend1, k = c_group, col =  mycolors) # add color to the lines
dend1 <- color_labels(dend1, k = c_group, col = mycolors)   # add color to the labels
col_labels <- get_leaves_branches_col(dend1)
col_labels <- col_labels[order(order.dendrogram(dend1))]




pdf(paste0(plotdir, "dendrogram_MS_genes_only_hits.pdf"), width=10, height = 10)
# dend1 <- set(dend1, "labels_cex", 0.2)
# par(mar = c(0.5,2,0.5,1))
# plot_horiz.dendrogram(dend1, side = T)

circlize_dendrogram(dend1,
                    labels_track_height = 2,
                    dend_track_height = 0.8,
                    facing = "outside",
                    labels=FALSE)

dev.off()
```

### only significant genes - sensitive strains
```{r}

toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))
# hitgenes_insensitive =fread(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"))%>%
#   filter(og1 %in% unique(toplot$og1))

species_legend=fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
   filter(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))


toplot = toplot %>%
  #rbind.data.frame(toplot,hitgenes_insensitive) %>%
   #remove species name from here - they are already color coded in fig
  mutate(complete_name = sapply(strsplit(as.character(complete_name)," "),'[',3)) %>%
    group_by(og1, complete_name) %>%
    summarise(logFC=mean(logFC)) %>%
  #remove dmprA and dadeL
    filter(!(complete_name %in% c("dadeL","dmprA")))

mymat = toplot %>% 
  dplyr::select(c(og1, complete_name, logFC)) %>%
  unique()


is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}
# mymat = fix_na(mymat)

library(magrittr)

mymat = mymat %>% pivot_wider(names_from = complete_name, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og1) %>%
  dplyr::select(-og1) %>%
  as.matrix()


matrix.please<-function(x) {
  m<-as.matrix(x[,-1])
  rownames(m)<-x[,1]
  m
}

toplot = matrix.please(as.data.frame(mymat))
toplot=mymat


library(gplots)

mat_data = toplot

# plot

# robust_dist = function(x, y) {
#     qx = quantile(x, c(0.1, 0.9))
#     qy = quantile(y, c(0.1, 0.9))
#     l = x > qx[1] & x < qx[2] & y > qy[1] & y < qy[2]
#     x = x[l]
#     y = y[l]
#     sqrt(sum((x - y)^2))
# }


#  dendrogram 
library(dendsort)
library(dendextend)
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)

row_dend = color_branches(dend1, k = 6) # `color_branches()` returns a dendrogram object

type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(complete_name = type) %>%
  inner_join(species_legend %>%
               dplyr::select(complete_name,species)%>%
               dplyr::mutate(complete_name = sapply(strsplit(as.character(complete_name)," "),'[',3)), by="complete_name") %>%
    distinct() %>%
    select(-complete_name),
   annotation_height = unit(4, "mm"),
  show_legend = F,
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )



pdf(paste0(plotdir,"clustered_MS_only_hits_sensitive_forSuppFigs.pdf"), width=6, height=8)
#rownames(mat_data) = NULL

Heatmap(mat_data,
        cluster_rows = row_dend,
        #cluster_cols=hc,
        row_names_gp = gpar(fontsize = 10),
         top_annotation = ha,
        show_heatmap_legend = F,
        row_split = 4, column_split = 3,
        use_raster = TRUE, raster_quality = 5)
        #rect_gp = gpar(col = "white", lwd = 2)
        #clustering_distance_rows = "pearson",
        #clustering_distance_columns = "pearson",

# draw(hm, heatmap_legend_side="top", annotation_legend_side="top",
#            legend_grouping="adjusted")

dev.off()



```

## ED 7d-e



### ED7d - volcano proteomics using COGs from Marco 

```{r}

resmut <- fread("~/Documents/EMBL/Nx/proteomics_strains_Marco/results_with_pangenome.tsv") %>%
  dplyr::mutate(species = case_when(Strain_treatment %in% c("PRZS", "MS02S","MS25S","MS274R","MS299R","MS311R","MS366R","MS386R", "MS493R", "MS530R","MS55S") ~ "K. pneumoniae",
                                    Strain_treatment %in% c("19606S", "2778S","dadeL","MS69R", "MS72R") ~ "A. baumannii",
                                    TRUE ~ "E. coli"),
                og =capitalize(og))%>% 
  dplyr::mutate(og1 = case_when(description %in% c("Outer membrane protein C","outer membrane protein C") ~ "OmpC",
                                description %in% c("Outer membrane protein Slp") ~ "Slp",
                                description %in% c("Outer membrane protein W","outer membrane protein W") ~ "OmpW",
                                description %in% c("2-hydroxy-3-oxopropionate reductase") ~ "GarR",
                                description %in% c("2,3-dehydroadipyl-CoA hydratase") ~ "PaaF",
                                description %in% c("4-aminobutyrate aminotransferase PuuE", "4-aminobutyrate aminotransferase GabT") ~ "GabT",
                                description %in% c("aldehyde-alcohol dehydrogenase", "Aldehyde-alcohol dehydrogenase") ~ "AdhE",
                                description %in% c("Citrate lyase alpha chain", "citrate lyase alpha subunit") ~ "CitF",
                                description %in% c("formate acetyltransferase", "Formate acetyltransferase 1") ~ "PflB",
                                description %in% c("Efflux pump membrane transporter", "Efflux pump periplasmic linker BepF") ~ "BepF",
                                description %in% c("Fumarate hydratase class II", "fumarate hydratase class II") ~ "FumC",
                                description %in% c("fumarate reductase iron-sulfur subunit", "Fumarate reductase iron-sulfur subunit") ~ "FrdB",
                                description %in% c("fumarate reductase subunit C", "Fumarate reductase subunit C") ~ "FrdC",
                                description %in% c("galactarate dehydratase", "Galactarate dehydratase (L-threo-forming)") ~ "GarD",
                                description %in% c("hydrogenase accessory protein HypB", "Hydrogenase maturation factor HypB") ~ "HypB",
                                description %in% c("Indole-3-pyruvate decarboxylase") ~ "IpdC",
                                description %in% c("Isochorismate synthase EntC") ~ "EntC",
                                description %in% c("Low-affinity putrescine importer PlaP") ~ "PlaP",
                                description %in% c("maltoporin", "Maltoporin") ~ "LamB",
                                description %in% c("Maltose/maltodextrin import ATP-binding protein MalK", "maltose/maltodextrin import ATP-binding protein MalK") ~ "MalK",
                                description %in% c("Mannose-1-phosphate guanylyltransferase 1", "mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase") ~ "ManC1",
                                description %in% c("multiple antibiotic resistance protein marA", "Multiple antibiotic resistance protein MarA") ~ "MarA",
                                description %in% c("nucleoside-specific channel-forming protein tsx", "Nucleoside-specific channel-forming protein Tsx") ~ "Tsx",
                                description %in% c("osmotically-inducible protein Y", "Osmotically-inducible protein Y") ~ "OsmY",
                                description %in% c("outer membrane protein C", "Outer membrane protein C") ~ "OmpC",
                                description %in% c("oxygen-insensitive NAD(P)H nitroreductase","Putative oxygen-insensitive NAD(P)H nitroreductase", "Oxygen-insensitive NAD(P)H nitroreductase") ~ "NfsB",
                                description %in% c("oxygen-insensitive NADPH nitroreductase", "Oxygen-insensitive NADPH nitroreductase") ~ "NfsA",
                                description %in% c("protein yqjC", "Protein YqjC") ~ "YqjC",
                                description %in% c("PTS system mannose-specific EIIAB component") ~ "ManX",
                                description %in% c("putative dimethyl sulfoxide reductase chain ynfE", "Putative dimethyl sulfoxide reductase chain YnfE") ~ "ynfE",
                                description %in% c("Succinate-semialdehyde dehydrogenase [NAD(P)+]", "Succinate semialdehyde dehydrogenase [NAD(P)+] Sad",
                                                   "Succinate-semialdehyde dehydrogenase [NADP(+)] 1","Succinate-semialdehyde dehydrogenase [NADP(+)] GabD") ~ "GabD1",
                                description %in% c("Transcriptional repressor MprA") ~ "MprA",
                                description %in% c("Urocanate hydratase") ~ "HutU",
                                description %in% c("threonine ammonia-lyase") ~ "IlvA",
                                og == "yhcn~~~bhsa_4" ~ "BhsA",
                                og == "fdhf~~~fdhf_1" ~ "FdhA",
                                og == "Hydrogenase-4 component A" ~ "HyfA",
                                og == "YciE" ~ "YciE",
                                TRUE ~ og))

toannot =resmut %>%
   dplyr::filter(hit==1) %>%
  dplyr::group_by(og1) %>%
  dplyr::mutate(count=n()) %>%
  dplyr::mutate(conservation = case_when(count > 1 ~ "conserved",
                                         TRUE ~ "strain-specific")) %>%
  dplyr::ungroup() %>%
  dplyr::select(og1, count, conservation) %>%
  unique()


# highlight cross-species conservation
toplot2=resmut %>%
   dplyr::filter(hit==1) %>%
  dplyr::group_by(og1) %>%
  # dplyr::mutate(species_count = paste0(species)) #%>% 
  # dplyr::group_by(og) %>% 
  dplyr::summarise(conserv_annot = paste(unique(species), collapse = ";")) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(conservation = case_when(conserv_annot %in% c("A. baumannii;E. coli;K. pneumoniae", "A. baumannii;K. pneumoniae", "E. coli;K. pneumoniae") ~ "conserved across species",
                                         TRUE ~ "species-specific"))


toannot = toannot %>%
  dplyr::mutate(cross_species_conservation = case_when(og1 %in% unique(filter(toplot2,conservation == "conserved across species"))$og1 ~ "conserved across species",
                                                                      TRUE ~ conservation)) %>%
    dplyr::mutate(cross_species_conservation = case_when(cross_species_conservation == "conserved" ~ "conserved within species",
                                                       TRUE ~ cross_species_conservation))

toplot = full_join(toannot, resmut)

toplot = toplot %>%
  dplyr::mutate(complete_name = interaction(species, Strain_treatment, sep=" "))

toplot$cross_species_conservation[is.na(toplot$cross_species_conservation)] <- "no orthology group annotated"


write.table(toplot, paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", quote = F,row.names = F)

toplot = read.table(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", header = T)

write.table(filter(toplot, hit==1), paste0(outputdir,"hits_proteomics_og_annotated.txt"), sep="\t", quote = F,row.names = F)

colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation



pdf(paste0(plotdir, "resmut_volcano_onlydrugsensitive.pdf"), width = 10, height = 2)

ggplot(filter(toplot, Condition_treatment == "Nx" & og1 != ""),aes(logFC, -log10(adj.P.Val))) +
  geom_point(alpha = 0.7,size = 0.7,color = "grey") + 
  ggrepel::geom_text_repel(data=filter(toplot, hit == 1 &cross_species_conservation == "conserved across species" & Condition_treatment == "Nx"),
                            aes(label = og1,color = cross_species_conservation),
                            size= 3.5,
                            max.overlaps = 200) +
  geom_hline(yintercept = -log10(0.05), size = 0.2, alpha=0.5) +
  geom_vline(xintercept = 1, size = 0.2, alpha =0.5) +
  geom_vline(xintercept = -1, size = 0.2, alpha =0.5) +
   geom_point(data = filter(toplot, hit == 1 & Condition_treatment == "Nx"), aes(color = cross_species_conservation),alpha=0.5, size = 0.7) +
  scale_color_manual(values = jColors) + 
  # geom_point(data=filter(toplot, Condition_treatment == "Nx" & cross_species_conservation == "conserved across species"),aes(fill="black"),alpha== 1, size = 1) + 
  facet_wrap(~factor(complete_name, levels=c("A. baumannii 19606S","A. baumannii 2778S","E. coli BW25113","E. coli 25922S", "K. pneumoniae MS02S", "K. pneumoniae MS25S", "K. pneumoniae MS55S",
"K. pneumoniae PRZS")),labeller = label_wrap_gen(width=15),ncol=8) +
  #facet_wrap(~complete_name, scales = "free",labeller = label_wrap_gen(width=10),ncol=14) +
  labs(x ="log(fold change)",y="-log10(adjusted p-value)") +
  theme_bw() +
  theme(legend.position = "none",
        axis.text = element_text(size = 12),
       # legend.key.size = unit(0.4, 'cm'),
        plot.title=element_text(hjust = 0.5))

dev.off()

strainstorem = c("dadeL", "dmprA") 
bla = filter(toplot, Condition_treatment == "nodrug" & !(Strain_treatment %in% strainstorem))

# add new strain designation
strain_key = readxl::read_xlsx(paste0(inputdir,"new_designation_240525_MT_EC.xlsx")) %>%
  clean_names()
    
bla =   bla %>%
  mutate(old_designation = sub("[RS]$", "",complete_name)) %>%
  inner_join(strain_key)


pdf(paste0(plotdir, "resmut_volcano.pdf"), width = 8, height = 7.6)

tolabel = c("BepF", "NfsA", "EmrA_1", "MarA", "GabT","FrdB","PlaP","OmpC","GarR","GarD", "OmpC","OmpW", "MprA", "LamB")

ggplot(bla,aes(logFC, -log10(adj.P.Val))) +
  geom_point(alpha = 0.3, size = 0.7,color = "grey") +
  geom_hline(yintercept = -log10(0.01), size = 0.2) +
  geom_vline(xintercept = 1, size = 0.2, alpha =0.5) +
  geom_vline(xintercept = -1, size = 0.2, alpha =0.5) +
  ylim(c(0,18)) +
  labs(x ="log(fold change)",y="-log10(adjusted p-value)", color = "") +
  geom_point(data = filter(bla, hit == 1), aes(color = cross_species_conservation), size = 0.7,alpha=0.7) +
 ggrepel::geom_text_repel(data=filter(bla, hit == 1 & og1 %in% tolabel),
                            aes(label = og1, color = cross_species_conservation),size= 3.5,max.overlaps = 50) +
 # ggrepel::geom_text_repel(data=filter(bla, hit == 1 & cross_species_conservation == "conserved across species" & og1 != ""),
 #                            aes(label = og1, color = cross_species_conservation),
 #                            size= 3.5,
 #                            max.overlaps = 50) +
  scale_color_manual(values = jColors) +
  # facet_wrap(~factor(complete_name, levels=c("A. baumannii MS69R",   "A. baumannii MS72R", "E. coli MS346R","E. coli MS511R","E. coli MS529R", "K. pneumoniae MS274R", 'K. pneumoniae MS299R', "K. pneumoniae MS311R", "K. pneumoniae MS366R", "K. pneumoniae MS386R", "K. pneumoniae MS493R", "K. pneumoniae MS530R")),scales = "free",labeller = label_wrap_gen(width=15),ncol=4) +
  facet_wrap(~factor(new_designation, levels=c("A. baumannii 2_R1",   "A. baumannii 5_R1", "E. coli 2_R1","E. coli 2_R4","E. coli 2_R3", "K. pneumoniae 7_R1", 'K. pneumoniae 6_R1', "K. pneumoniae 6_R2", "K. pneumoniae 3_R1", "K. pneumoniae 5_R1", "K. pneumoniae 2_R1", "K. pneumoniae 2_R2")),scales = "free",labeller = label_wrap_gen(width=15),ncol=4) +
  theme_bw() +
  theme(legend.position="none")


dev.off()


```

### ED7e - barplot conservation

```{r}
restrains =c("A. baumannii MS69R",   "A. baumannii MS72R","E. coli MS346R","E. coli MS511R","E. coli MS529R","K. pneumoniae MS274R", 'K. pneumoniae MS299R', "K. pneumoniae MS311R", "K. pneumoniae MS366R", "K. pneumoniae MS386R", "K. pneumoniae MS493R", "K. pneumoniae MS530R")

toplot = read.table(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", header = T) %>%
  dplyr::mutate(resorsens = case_when(complete_name %in% restrains ~ "Resistant strains",
                                      TRUE ~ "Sensitive strains")) %>%
  filter(hit == 1) %>%
  distinct(og1,cross_species_conservation, species, resorsens) %V>%
  dplyr::group_by(resorsens, cross_species_conservation,species) %>%
  dplyr::summarise(count = n(), .groups = "keep")
  
toplot = toplot %>%
 dplyr::group_by(resorsens,species) %>%
  mutate(percent = count/sum(count))
  
colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation


pdf(paste0(plotdir, "barplot_conservation_resmut.pdf"), width = 5.5, height = 2.5)

ggplot(toplot, aes(x=species,y=percent*100, fill = cross_species_conservation)) +
  geom_bar(position = position_fill(),stat = "identity") +
  geom_text(aes(label = paste0(count)),
           # stat = "count",
            colour = "white",
            position = position_fill(vjust = 0.5)) +
  scale_fill_manual(values = jColors) +
  facet_wrap (~resorsens) +
  coord_flip() +
  labs(x = "", y = "fraction of total (n = 881)", fill = "") +
  theme_bw() +
  theme(legend.position = "bottom") +
  guides(fill = guide_legend(nrow = 2))


dev.off()


```

## ED Fig. 7f - adeL

```{r}

template = readxl::read_xlsx(paste0(inputdir,"Nx_MIC_adeL_SuppFig.7e_29.01.2024_SG_.xlsx")) %>%
  dplyr::rename(MIC=`MIC [µg/ml]`)%>%
  mutate(MIC=as.numeric(MIC)) %>% filter(MIC > 2)
    
var = template %>%
      group_by(Hugo_Symbol, Mutation_Type, Protein_Change,Isolate_type) %>%
      mutate(freq=n(),
             location=str_extract(Protein_Change, "(?<=\\D)\\d+")) %>%
      # MIC=max(MIC)) %>%
      ungroup() %>%
      distinct(Protein_Change,Mutation_Type,MIC,freq,location,Isolate_type) %>%
      mutate(location=as.numeric(location),
             Isolate_type=case_when(Isolate_type == "resistance_generated" ~ "experimentally evolved",
                                    TRUE ~ "clinical isolate"))
    
var$MIC[is.na(var$MIC)] <- 0
    
proteins_acc_url<-"A0A059ZJX1"
    
    baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
    url<-paste0(baseurl,proteins_acc_url)
    prots_feat<-GET(url,accept_json())
    prots_feat_red<-httr::content(prots_feat)
    features_total_plot<-NULL
    for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])#the extract_feat_acc() function takes features into a data.frame
      # features_temp$order<-i # this order is needed for plotting later
      features_total_plot<-rbind(features_total_plot,features_temp)
    }
    
    
    
    
    plot_start<-0#starts 0
    plot_end<-max(features_total_plot$end)
    
    # shapes = data.frame("Mutation_Type" = c("Frame_Shift_Ins","Missense_Mutation","Nonsense_Mutation"),
    #                      "values" = c(16, 17,15))
    # jshapes <- as.character(shapes$values)
    # names(jshapes) <- shapes$Mutation_Type
    
    
colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type
    
var= var %>%
  filter(MIC != 0) %>%
  mutate(Mutation_Type =case_when(Mutation_Type == "frameshift" ~ "Frameshift",
                                  Mutation_Type == "Stop" ~ "Nonsense",
                                  Mutation_Type == "missense" ~ "Missense",
                                  TRUE ~ Mutation_Type))

shapes = data.frame("Mutation_Type" = c("Frameshift","Missense","Nonsense", "insertion ISAba125"),
                     "values" = c(16,17,15,18))
jShapes <- as.character(shapes$values)
names(jShapes) <- shapes$Mutation_Type

var = var %>% filter(MIC >= 4)

pdf(paste0(plotdir,"adeL_mutplot.pdf"), width=3.5, height=2.5)

ggplot(var,aes(location,MIC,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location,yend=MIC)) +
  geom_point(aes(size=freq, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=var %>%
                    distinct(location,MIC, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=10, size=3) +
  geom_hline(yintercept = 0) +
   geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
 annotate("text", x = 29, y = 2.3, label ="HTH marR-type", size = 4,color="white",fontface="bold") +
  scale_shape_manual(values=c("Frameshift" =16, "Nonsense"  = 17, "Missense" =15, "insertion ISAba125"  = 3), name = "Mutation type") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_size_continuous(breaks = c(1,3,6), limits=c(1,6)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,343), limits=c(0,343))+
  #scale_y_continuous(trans=scales::pseudo_log_trans(base = 2), breaks= c(16,32,64,128)) +
  scale_y_continuous(breaks = c(2,4,8,16),trans="log2",limits=c(2,20)) +
  labs(y= expression('log'[2]~'(MIC [µg/mL])'),size="Mutation occurrencies", title = "AdeL")+
  #guides(color="none") +
  theme_bw() +
  theme(plot.title=element_text(face="bold",size=(15),hjust=0.5),
       legend.position = "none",
        legend.title = element_text(size=10),
        legend.text = element_text(size=10),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.1,0.1,0.1,0.1, unit="cm"),
        panel.border = element_blank(), axis.line = element_line())

dev.off()
    

```


## ED Fig. 7g - AcrR

```{r}
template = readxl::read_xlsx(paste0(inputdir,"Nx_MIC_acrR_SuppFig.7_29.01.2024_SG_.xlsx")) %>%
  dplyr::rename(MIC=`MIC [µg/ml]`,
         Isolate_type=Isolate_Type) %>%
  filter(Sample_ID != "DSM30006") #%>%
  mutate(MIC=as.numeric(MIC))%>% filter(MIC > 2)


var = template %>%
  group_by(Hugo_Symbol, Mutation_Type, Protein_Change,Isolate_type) %>%
  mutate(freq=n(),
         location=str_extract(Protein_Change, "(?<=\\D)\\d+")) %>%
  # MIC=max(MIC)) %>%
  ungroup() %>%
  distinct(Protein_Change,Mutation_Type,MIC,freq,location,Isolate_type) %>%
  mutate(location=as.numeric(location),
         Isolate_type=case_when(Isolate_type =="clinical_isolate" ~ "clinical isolate",
                                Isolate_type == "resistance_generated" ~ "experimentally evolved",
                                TRUE ~ Isolate_type),
         Mutation_Type = stringr::str_to_title(Mutation_Type)) %>%
  mutate(Mutation_Type = gsub(Mutation_Type,pattern="aba",replacement="Aba"))

#var[is.na(var)] <- 0

#proteins_acc<- proteins_acc[proteins_acc$Hugo_Symbol==var.plot$Hugo_Symbol,2][[1]]
proteins_acc_url<-"A0A245ZZS0"

baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
url<-paste0(baseurl,proteins_acc_url)
prots_feat<-GET(url,accept_json())
prots_feat_red<-httr::content(prots_feat)
features_total_plot<-NULL
for(i in 1:length(prots_feat_red)){ 
  features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])#the extract_feat_acc() function takes features into a data.frame
  # features_temp$order<-i # this order is needed for plotting later
  features_total_plot<-rbind(features_total_plot,features_temp)
}


plot_start<-0#starts 0
plot_end<-max(features_total_plot$end)   

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved"),
                     "values" = c("#D55E00", "black"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type

var = var %>% filter(MIC >= 4)

pdf(paste0(plotdir,"acrR_mutplot.pdf"),width=3.5, height= 2.5)

ggplot(var,aes(location,MIC,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location,yend=MIC)) +
  geom_point(aes(size=freq, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=var %>%
                    distinct(location,MIC, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=10, size=3) +
  geom_hline(yintercept = 0) +
   geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 59, y = 2.25, label ="HTH tetR-type", size = 4,color="white",fontface="bold") +
  geom_rect(data=features_total_plot[features_total_plot$type=="DNA_BIND",],
            mapping=aes(xmin=begin,xmax=end,ymin=2.5,ymax=3),color="darkblue",fill="darkblue",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 59, y = 2.75, label ="H-T-H motif", size = 4,color="white",fontface="bold") +
  # geom_rect(data=features_total_plot[features_total_plot$type=="REGION",],
  #           mapping=aes(xmin=begin,xmax=end,ymin=3,ymax=3.5),color="darkgreen",fill="darkgreen",inherit.aes=F, alpha=0.5) +
  scale_shape_manual(values=c("Frameshift" =16, "Nonsense"  = 17, "Missense" =15, "Insertion IsAba1"  = 3,"Insertion IsAba36"= 8), name = "Mutation type") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_size_continuous(breaks = c(1,3,7), limits=c(1,7)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,223), limits=c(0,223))+
  #scale_y_continuous(trans=scales::pseudo_log_trans(base = 2), breaks= c(16,32,64,128)) +
  scale_y_continuous(breaks = c(2,4,8,16,32),trans="log2",limits=c(2,38)) +
  labs(y= expression('log'[2]~'(MIC [µg/mL])'),size="Mutation occurrencies", title = "AcrR")+
  #guides(color="none") +
  theme_bw() +
  theme(plot.title=element_text(face="bold",size=(15),hjust=0.5),
       legend.position = "none",
        legend.title = element_text(size=10),
        legend.text = element_text(size=10),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.1,0.1,0.1,0.1, unit="cm"),
        panel.border = element_blank(), axis.line = element_line())
dev.off()



```


## Fig. 5 c - what mechanisms are common between sensitive and resistant (going into opposite directions?)

```{r}
bothsensres = read.table(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", header = T) %>%
  dplyr::filter(cross_species_conservation == "conserved within species" & hit == 1 & og != "" & count >= 2) %>%
  dplyr::select(og, complete_name, logFC, count) %>%
  pivot_wider(names_from = complete_name, values_from = logFC, values_fn=mean, values_fill = 0)

bothsensres = bothsensres %>%
  dplyr::mutate(sensitiveandres = case_when((`A. baumannii MS69R` != 0 | `A. baumannii MS72R` != 0) & (`A. baumannii 19606S` != 0 | `A. baumannii 2778S` != 0) ~ "A. baumannii",
                                            (`E. coli MS346R` != 0 | `E. coli MS511R` !=0 | `E. coli MS529R` != 0) & (`E. coli 25922S` != 0) ~ "E. coli",
                                            (`K. pneumoniae MS366R` != 0 | `K. pneumoniae MS386R` != 0 | `K. pneumoniae MS299R` != 0 | `K. pneumoniae MS274R` != 0 | `K. pneumoniae MS311R` != 0 |  `K. pneumoniae MS493R` != 0 |  `K. pneumoniae MS530R` != 0 ) & (`K. pneumoniae MS02S` != 0 | `K. pneumoniae MS25S` != 0 | `K. pneumoniae MS55S` != 0  | `K. pneumoniae PRZS` != 0) ~ "K. pneumoniae",
                                            TRUE ~ "conserved within sensitive or resistant"))
  
# toplot = read.table(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", header = T) %>%
#   dplyr::filter(cross_species_conservation == "conserved within species" & hit == 1 & og != "") %>%
# dplyr::group_by(og, complete_name) %>%
#     dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
#     dplyr::filter(n > 1L) 
toplot = read.table(paste0(outputdir,"remsut_conservation_annotated_forplotting.txt"), sep="\t", header = T) 

# colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
#                      "values" = c("#7D0000", "#008DCA","#00405B", "#424949"))
# jColors <- as.character(colours$values)
# names(jColors) <- colours$cross_species_conservation
colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation


pdf(paste0(plotdir, "resmut_volcano_onlydrugsensitive_highlight stuff conserved with resistant.pdf"), width = 15, height = 2.4)

ggplot(filter(toplot, Condition_treatment == "Nx"),aes(logFC, -log10(adj.P.Val))) +
  geom_point(alpha = 0.5,size = 0.7,color = "grey") + 
  ggrepel::geom_text_repel(data=filter(toplot, (hit == 1 & cross_species_conservation == "conserved across species" & Condition_treatment == "Nx") | hit == 1 & Condition_treatment == "Nx" & og %in% unique(filter(bothsensres, sensitiveandres != "conserved within sensitive or resistant")$og)),
                            aes(label = og, colour = cross_species_conservation),
                            size= 3,
                            max.overlaps = 200) +
  geom_hline(yintercept = -log10(0.05), size = 0.2, alpha=0.5) +
  geom_vline(xintercept = 1, size = 0.2, alpha =0.5) +
  geom_vline(xintercept = -1, size = 0.2, alpha =0.5) +
   geom_point(data = filter(toplot, hit == 1 & Condition_treatment == "Nx"), aes(color = cross_species_conservation),alpha=0.5, size = 0.7) +
  scale_color_manual(values = jColors) + 
  # geom_point(data=filter(toplot, Condition_treatment == "Nx" & cross_species_conservation == "conserved across species"),aes(fill="black"),alpha== 1, size = 1) + 
  facet_wrap(~factor(complete_name, levels=c("A. baumannii 19606S","A. baumannii 2778S","E. coli BW25113","E. coli 25922S", "K. pneumoniae MS02S", "K. pneumoniae MS25S", "K. pneumoniae MS55S",
"K. pneumoniae PRZS")),scales="free_y",labeller = label_wrap_gen(width=15),ncol=8) +
  #facet_wrap(~complete_name, scales = "free",labeller = label_wrap_gen(width=10),ncol=14) +
  labs(x ="log(fold change)",y="-log10(adjusted p-value)") +
  theme_bw() +
  theme(legend.position = "none",
        axis.text = element_text(size = 12),
       # legend.key.size = unit(0.4, 'cm'),
        plot.title=element_text(hjust = 0.5))

dev.off()


  
```


# to write about GO terms and proteins

```{r}
final <- read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species.txt", sep = "\t", header=T) %>%
  dplyr::filter(p_adj < 0.01) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj))

final = final %>% 
    mutate(hit_genes = strsplit(as.character(hit_genes), "/")) %>% 
    unnest(hit_genes) 

resmut <- fread("~/Documents/EMBL/Nx/proteomics_strains_Marco/results_with_pangenome.tsv") %>%
  dplyr::mutate(species = case_when(Strain_treatment %in% c("PRZS", "MS02S","MS25S","MS274R","MS299R","MS311R","MS366R","MS386R", "MS493R", "MS530R","MS55S") ~ "K. pneumoniae",
                                    Strain_treatment %in% c("19606S", "2778S","dadeL","MS69R", "MS72R") ~ "A. baumannii",
                                    TRUE ~ "E. coli"),
                og =capitalize(og))%>% 
  dplyr::mutate(og1 = case_when(description %in% c("Outer membrane protein C","outer membrane protein C") ~ "OmpC",
                                description %in% c("Outer membrane protein Slp") ~ "Slp",
                                description %in% c("Outer membrane protein W","outer membrane protein W") ~ "OmpW",
                                description %in% c("2-hydroxy-3-oxopropionate reductase") ~ "GarR",
                                description %in% c("2,3-dehydroadipyl-CoA hydratase") ~ "PaaF",
                                description %in% c("4-aminobutyrate aminotransferase PuuE", "4-aminobutyrate aminotransferase GabT") ~ "GabT",
                                description %in% c("aldehyde-alcohol dehydrogenase", "Aldehyde-alcohol dehydrogenase") ~ "AdhE",
                                description %in% c("Citrate lyase alpha chain", "citrate lyase alpha subunit") ~ "CitF",
                                description %in% c("formate acetyltransferase", "Formate acetyltransferase 1") ~ "PflB",
                                description %in% c("Efflux pump membrane transporter", "Efflux pump periplasmic linker BepF") ~ "BepF",
                                description %in% c("Fumarate hydratase class II", "fumarate hydratase class II") ~ "FumC",
                                description %in% c("fumarate reductase iron-sulfur subunit", "Fumarate reductase iron-sulfur subunit") ~ "FrdB",
                                description %in% c("fumarate reductase subunit C", "Fumarate reductase subunit C") ~ "FrdC",
                                description %in% c("galactarate dehydratase", "Galactarate dehydratase (L-threo-forming)") ~ "GarD",
                                description %in% c("hydrogenase accessory protein HypB", "Hydrogenase maturation factor HypB") ~ "HypB",
                                description %in% c("Indole-3-pyruvate decarboxylase") ~ "IpdC",
                                description %in% c("Isochorismate synthase EntC") ~ "EntC",
                                description %in% c("Low-affinity putrescine importer PlaP") ~ "PlaP",
                                description %in% c("maltoporin", "Maltoporin") ~ "LamB",
                                description %in% c("Maltose/maltodextrin import ATP-binding protein MalK", "maltose/maltodextrin import ATP-binding protein MalK") ~ "MalK",
                                description %in% c("Mannose-1-phosphate guanylyltransferase 1", "mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase") ~ "ManC1",
                                description %in% c("multiple antibiotic resistance protein marA", "Multiple antibiotic resistance protein MarA") ~ "MarA",
                                description %in% c("nucleoside-specific channel-forming protein tsx", "Nucleoside-specific channel-forming protein Tsx") ~ "Tsx",
                                description %in% c("osmotically-inducible protein Y", "Osmotically-inducible protein Y") ~ "OsmY",
                                description %in% c("outer membrane protein C", "Outer membrane protein C") ~ "OmpC",
                                description %in% c("oxygen-insensitive NAD(P)H nitroreductase","Putative oxygen-insensitive NAD(P)H nitroreductase", "Oxygen-insensitive NAD(P)H nitroreductase") ~ "NfsB",
                                description %in% c("oxygen-insensitive NADPH nitroreductase", "Oxygen-insensitive NADPH nitroreductase") ~ "NfsA",
                                description %in% c("protein yqjC", "Protein YqjC") ~ "YqjC",
                                description %in% c("PTS system mannose-specific EIIAB component") ~ "ManX",
                                description %in% c("putative dimethyl sulfoxide reductase chain ynfE", "Putative dimethyl sulfoxide reductase chain YnfE") ~ "ynfE",
                                description %in% c("Succinate-semialdehyde dehydrogenase [NAD(P)+]", "Succinate semialdehyde dehydrogenase [NAD(P)+] Sad",
                                                   "Succinate-semialdehyde dehydrogenase [NADP(+)] 1","Succinate-semialdehyde dehydrogenase [NADP(+)] GabD") ~ "GabD1",
                                description %in% c("Transcriptional repressor MprA") ~ "MprA",
                                description %in% c("Urocanate hydratase") ~ "HutU",
                                description %in% c("threonine ammonia-lyase") ~ "IlvA",
                                og == "yhcn~~~bhsa_4" ~ "BhsA",
                                og == "fdhf~~~fdhf_1" ~ "FdhA",
                                og == "Hydrogenase-4 component A" ~ "HyfA",
                                og == "YciE" ~ "YciE",
                                TRUE ~ og))
  dplyr::select(protein_ID, og, gene_name, description)

final = final %>%
  dplyr::rename(protein_ID = hit_genes) 

final = inner_join(final, resmut, by = "protein_ID")


```


## NOT USED - hits highlighted in proteomics


```{r}
resmut <- readxl::read_xlsx("~/Documents/EMBL/Nx/proteomics_strains_Marco/Nitroxoline_resistantmutants-results.xlsx")

namesIwant = c("Multidrug efflux pump BpeE", "Outer membrane efflux protein OprC", "Low-affinity putrescine importer PlaP", "Multidrug export protein EmrA",
               "Outer membrane protein TolC","multidrug efflux RND transporter permease subunit OqxB20",
               "tyrosine-protein kinase wzc",
"Multiple antibiotic resistance protein MarA")

pdf(paste0(plotdir, "resmut_volcano.pdf"), width = 20, height = 4)
ggplot(resmut, 
        aes(logFC, -log10(adj.P.Val))) +
  geom_point(color = "gray", alpha = 0.5, size = 1) + 
  geom_point(data= filter(resmut, hit == TRUE), aes(color = species_analyzed), alpha = 0.8,size = 1) +
  #scale_color_manual(values = mypal) +
  ggrepel::geom_text_repel(data=filter(resmut, hit == TRUE & description %in% namesIwant),
                            aes(label = gene_name),
                            size= 3.5,
                            max.overlaps = 40) +
  #geom_abline(slope = 0, intercept = 5, linetype ="dashed") +
  facet_grid(Condition_treatment~Strain_treatment, scales = "free_y") +
  theme_bw() +
  theme(legend.position = "bottom",
        axis.text = element_text(size = 12),
        strip.text = str_wrap(x, width = 35)) +
       # legend.key.size = unit(0.4, 'cm'),
        plot.title=element_text(hjust = 0.5)


dev.off()



```

## ED Fig. 8 - efflux pump inhibitors

```{r}
toplot <- readxl::read_xlsx(paste0(inputdir, "230825_Efflux_inhibitors.xlsx")) %>%
  #janitor::clean_names() %>%
  pivot_longer(cols=c(3:11), names_to = "condition", values_to = "mic") %>%
  mutate(Species = gsub(Species, pattern = "Acinetobacter", replacement="A."),
         Species = gsub(Species, pattern = "Escherichia", replacement="E."),
         Species = gsub(Species, pattern = "Klebsiella", replacement="K."),
         Species = gsub(Species, pattern = "Pseudomonas", replacement="P."),
         Strain = interaction(Species,Strain,sep=" "),
         condition = sapply(strsplit(as.character(condition), "\\..."), '[',1),
         strain_condition=interaction(Strain,condition),
         replicate=ave(mic, strain_condition, FUN = seq_along),
         Strain = gsub(Strain, pattern = "A. baumannii MS329", replacement="A. baumannii MS347")) %>%
  filter(!(Strain %in% c("A. baumannii MS69", "A. baumannii MS48", "E. coli ATCC 25922 R",
                         "K. pneumoniae MS386", "P. aeruginosa MS77","P. aeruginosa MS358","P. aeruginosa SC712","P. aeruginosa MS355")))


my_comparisons <- list( c("CAMHB", "+PAβN"), c("CAMHB", "+NMP") )

p1 = ggerrorplot(toplot %>% filter(Species == "A. baumannii"), x = "condition", y = "mic", color="condition",
            add = "jitter",
            # 'mean' and c() removed in this line
            error.plot = "errorbar") +
  facet_wrap(~ Strain,ncol=2, scales="free") + #labeller = label_parsed)+
  scale_color_manual(values=c("darkgrey", "#a8404a", "#5671b0")) +
  #scale_y_continuous(trans="log10") +
  labs(x="", y = "MIC (µg/ml)") +
  #scale_y_continuous(limits = c(log10(100),log10(100000000))) +
  # scale_x_discrete(labels=c("before\ntreatment", "DMSO","nitroxoline\n(5 µg/ml)")) +
  stat_summary(geom = "point",shape = 95,size = 10,col = "black",fun = "mean") +
 # stat_compare_means(comparisons=my_comparisons,method="t.test", label.y.npc = "top", tip.length = c(0.01, 0.03), label="p.signif") +
  #annotate('rect', xmin=1.5, xmax=4, ymin=0, ymax=max(killing$`CFU/ml`)+10, alpha=.2, fill='red') +
  theme(strip.text = element_text(size=12),
        legend.position = "none")


p2 = ggerrorplot(toplot %>% filter(Species == "E. coli"), x = "condition", y = "mic", color="condition",
            add = "jitter",
            # 'mean' and c() removed in this line
            error.plot = "errorbar") +
  scale_color_manual(values=c("darkgrey", "#a8404a", "#5671b0")) +
  facet_wrap(~ Strain,ncol=2, scales="free") + #labeller = label_parsed)+
  #scale_y_continuous(trans="log10") +
  labs(x="", y = "") +
  #scale_y_continuous(limits = c(log10(100),log10(100000000))) +
  # scale_x_discrete(labels=c("before\ntreatment", "DMSO","nitroxoline\n(5 µg/ml)")) +
  stat_summary(geom = "point",shape = 95,size = 10,col = "black",fun = "mean") +
  #stat_compare_means(comparisons=my_comparisons) +
  #annotate('rect', xmin=1.5, xmax=4, ymin=0, ymax=max(killing$`CFU/ml`)+10, alpha=.2, fill='red') +
  theme(strip.text = element_text(size=12),
        legend.position = "none")


p3 = ggerrorplot(toplot %>% filter(Species == "K. pneumoniae"), x = "condition", y = "mic", color="condition",
            add = "jitter",
            # 'mean' and c() removed in this line
            error.plot = "errorbar") +
  facet_wrap(~ Strain,ncol=2, scales="free") + #labeller = label_parsed)+
  scale_color_manual(values=c("darkgrey", "#a8404a", "#5671b0")) +
  #scale_y_continuous(trans="log10") +
  labs(x="", y = "") +
  #scale_y_continuous(limits = c(log10(100),log10(100000000))) +
  # scale_x_discrete(labels=c("before\ntreatment", "DMSO","nitroxoline\n(5 µg/ml)")) +
  stat_summary(geom = "point",shape = 95,size = 10,col = "black",fun = "mean") +
  #stat_compare_means(comparisons=my_comparisons) +
  #annotate('rect', xmin=1.5, xmax=4, ymin=0, ymax=max(killing$`CFU/ml`)+10, alpha=.2, fill='red') +
  theme(strip.text = element_text(size=12),
        legend.position = "none")



pdf(paste0(plotdir,"effluxPump_all.pdf"),width=10,height=13)

design <- "
1122
1122
1122
1133
1133
11##
11##
"  

p1 + p2 + p3 + plot_layout(design=design)


dev.off()
```

## new
```{r}

toplot <- readxl::read_xlsx(paste0(inputdir, "230825_Efflux_inhibitors.xlsx")) %>%
  #janitor::clean_names() %>%
  pivot_longer(cols=c(3:11), names_to = "condition", values_to = "mic") %>%
  mutate(Species = gsub(Species, pattern = "Acinetobacter", replacement="A."),
         Species = gsub(Species, pattern = "Escherichia", replacement="E."),
         Species = gsub(Species, pattern = "Klebsiella", replacement="K."),
         Species = gsub(Species, pattern = "Pseudomonas", replacement="P."),
         Strain = interaction(Species,Strain,sep=" "),
         condition = sapply(strsplit(as.character(condition), "\\..."), '[',1),
         strain_condition=interaction(Strain,condition),
         replicate=ave(mic, strain_condition, FUN = seq_along),
         Strain = gsub(Strain, pattern = "A. baumannii MS329", replacement="A. baumannii MS347"),
         Strain = gsub(Strain, pattern = "A. baumannii ATCC 19606", replacement="A. baumannii ATCC19606"),
         Strain = gsub(Strain, pattern = "A. baumannii ATCC 17978", replacement="A. baumannii ATCC17978")) %>%
  filter(!(Strain %in% c("A. baumannii MS69", "A. baumannii MS48", "E. coli ATCC 25922 R",
                         "K. pneumoniae MS386", "P. aeruginosa MS77","P. aeruginosa MS358","P. aeruginosa SC712","P. aeruginosa MS355")))

toplot = toplot %>%
  mutate(Strain = gsub(Strain,pattern="A. baumannii",replacement="italic('A. baumannii ')*"),
         Strain = gsub(Strain,pattern="E. coli",replacement="italic('E. coli ')*"),
         Strain = gsub(Strain,pattern="K. pneumoniae",replacement="italic('K. pneumoniae ')*")) %>%
  mutate(condition = case_when(condition == "CAMHB" ~ "no inhibitor",
                               condition == "+NMP" ~ "+ 100 µg/ml NMP",
                               condition == "+PAβN" ~ "+ 25 µg/ml PAβN"))


# dispose in rows as I want 
mainfig = toplot %>%
  mutate(row= case_when(Strain %in% c("italic('A. baumannii ')* ATCC17978", "italic('A. baumannii ')* MS317") ~ 1,
                        Strain %in% c("italic('A. baumannii ')* ATCC 19606", "italic('A. baumannii ')* MS106", "italic('A. baumannii ')* MS269") ~ 2, 
                        Strain %in% c("italic('A. baumannii ')* MS318", "italic('A. baumannii ')* MS347") ~ 3,
                        Strain %in% c("italic('A. baumannii ')* MS552","italic('A. baumannii ')* MS557_1","italic('A. baumannii ')* MS557_2") ~ 4,
                        Strain %in% c("italic('K. pneumoniae ')* MS138","italic('K. pneumoniae ')* MS364") ~ 5,
                        Strain %in% c("italic('E. coli ')* BW25113","italic('E. coli ')* MS579","italic('E. coli ')* MS07", "italic('E. coli ')* MS120") ~ 6,
                        TRUE ~ NA),
         resist = case_when(Strain %in% c("italic('A. baumannii ')* ATCC17978", "italic('A. baumannii ')* ATCC19606", "italic('A. baumannii ')* MS318","italic('E. coli ')* BW25113","italic('K. pneumoniae ')* MS138") ~ "sensitive",
                            TRUE ~ "resistant"))

p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "MIC (µg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="right")

legend_grob <-cowplot::get_legend(p)

p <- ggerrorplot(mainfig %>% filter(row==1), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p1 <- ggerrorplot(mainfig %>% filter(row==2), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==2 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p2 <- ggerrorplot(mainfig %>% filter(row==3), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==3 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p3 <- ggerrorplot(mainfig %>% filter(row==4), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==4 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

p4 <- ggerrorplot(mainfig %>% filter(row==5), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==5 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")
p5 <- ggerrorplot(mainfig %>% filter(row==6), x = "condition", y = "mic", error.plot="errorbar",
          order = c("no inhibitor","+ 100 µg/ml NMP", "+ 25 µg/ml PAβN"),
  add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ Strain, labeller = label_parsed,ncol=4)+ #,scales="free") +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
   #scale_color_npg() +
  geom_rect(data=filter(mainfig,row==6 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 µg/ml NMP",paste0("+ 25 µg/ml PA\u03B2N"))) +
  labs(color = "", y = "", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=TRUE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=15) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none")

library("latex2exp")


cairo_pdf(paste0(plotdir,"effluxpump_all_supp.pdf"), width =8.5,height=7)
 
# layout <-"BBBDDD
#           AACCEE
#           GGGGHH"

layout <-"CCDDD
          AABB#
          GGGG#
          EEH##
          "
wrap_plots(A=p,B=p1, C=p2,D=p3,E=p4,G=p5, H=legend_grob,design = layout)

dev.off()


```


## Fig. S8b - resp chain thermal profiles

```{r}


k <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "NUOB",
  drug_name = "Nitroxoline")
l <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "NUOF",
  drug_name = "Nitroxoline")
m <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "NUOI",
  drug_name = "Nitroxoline")
n <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "NARH",
  drug_name = "Nitroxoline")
o <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "NARG",
  drug_name = "Nitroxoline")
p <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "SDHA",
  drug_name = "Nitroxoline")
q <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "SDHB",
  drug_name = "Nitroxoline")
r <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "SDHA",
  drug_name = "Nitroxoline")
s <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "FDNG",
  drug_name = "Nitroxoline")
t <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "FDNH",
  drug_name = "Nitroxoline")
u <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "FDOG",
  drug_name = "Nitroxoline")
v <- plot2dTppFcHeatmap(
  df = df_final_whole, name = "FDOH",
  drug_name = "Nitroxoline")

pdf(paste0(plotdir, "metals_thermal_profile_respiration.pdf"), width = 15,  height = 6.7)

cowplot::plot_grid(k,l,m,n,o,p,q,r,s,t,u,v, ncol = 5)
#a + b + c + d + plot_layout(guides = "collect")

dev.off()



```


## Fig. S9-10 GO enrichment proteomics - per species

```{r}
final <- read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species.txt", sep = "\t", header=T) %>%
  dplyr::filter(p_adj < 0.01) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj))
  

p1 = ggplot(filter(final, source == "abau_upreg"), aes(y = reorder(GO.term,fold_enrich), x =fold_enrich,colour = p_adj)) +
  geom_point(aes(size=freq_foreground)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
scale_y_discrete(labels = function(x) str_wrap(x, width = 35)) +
  #facet_wrap(species~change_sign, scales="free_y") +
  labs(size = "Number of genes", x = "Fold enrichment",y = "", colour = "FDR", title = "A. baumannii\nUpregulation") +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none") +
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9))

p2 = ggplot(filter(final, source == "kpn_upreg"), aes(y = reorder(GO.term,fold_enrich), x =fold_enrich,colour = p_adj)) +
  geom_point(aes(size=freq_foreground)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
 scale_y_discrete(labels = function(x) str_wrap(x, width = 20)) +
  #facet_wrap(species~change_sign, scales="free_y") +
  labs(size = "Number of genes", x = "Fold enrichment",y = "", colour = "FDR", title = "K. pneumoniae\nUpregulation")  +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8))

p3 = ggplot(filter(final, source == "ecoli_upreg"),aes(y = reorder(GO.term,fold_enrich), x =fold_enrich,colour = p_adj)) +
  geom_point(aes(size=freq_foreground)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 20)) +
 # facet_wrap(species~change_sign, scales="free_y") +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = "E. coli\nUpregulation") +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8))

p4 = ggplot(filter(final, source == "abau_downreg"), aes(y = reorder(GO.term,fold_enrich), x =fold_enrich,colour = p_adj)) +
  geom_point(aes(size=freq_foreground)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 35)) +
 # facet_wrap(species~change_sign, scales="free_y") +
  labs(size = "Number of genes", x = "Fold enrichment",y = "", colour = "FDR", title = "A. baumannii\nDownregulation") +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.9, size =8))

p5 = ggplot(filter(final, source == "kpn_downreg"),aes(y = reorder(GO.term,fold_enrich), x =fold_enrich,colour = p_adj)) +
  geom_point(aes(size=freq_foreground)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
 scale_y_discrete(labels = function(x) str_wrap(x, width = 35)) +
#  facet_wrap(species~change_sign, scales="free_y") +
 labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = "K. pneumoniae\nDownregulation") +
  theme_pubr() +
  theme(legend.position = "right")+
  theme(axis.text.y = element_text(lineheight = 0.8, size =9))


pdf(paste0(plotdir, "5d_GO_enrich_resmut.pdf"), width = 20, height = 15)

design <- "
1423
1453
1456
1456
1456
1456
1456
1456
1456
1456
1456
1456
1456
"
p1 + p2 + p3 + p4 + p5 + guide_area() + plot_layout(design=design, guides = "collect")


dev.off()

# pdf(paste0(plotdir, "5d_GO_enrich_resmut_test.pdf"), width = 8, height = 10)
# p1
# p2
# p3
# p4
# p5
# dev.off()


```

### separating sensitive and resistant strains


#### get annotation to annotate GO terms according to their conservation across sensitive/resistant or within sens and res
```{r}
sensit = read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species_sensitiveonly.txt", sep = "\t", header = T) %>%
  dplyr::filter(p_adj < 0.01 & !(GOs %in% c("GO:0072488", "GO:0022890","GO:0043872"))) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj))


resist = read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species_resistantonly.txt", sep = "\t", header = T) %>%
  dplyr::filter(p_adj < 0.01) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj))
commsensres = intersect(sensit$GO.term,resist$GO.term)

# within resistant
toannotres =resist %>%
  dplyr::group_by(GO.term) %>%
  dplyr::mutate(count=n()) %>%
  dplyr::mutate(conservation = case_when(count > 1 ~ "conserved within resistant or sensitive",
                                         TRUE ~ "not conserved")) %>%
  dplyr::ungroup() %>%
  dplyr::select(GO.term, count, conservation) %>%
  unique() %>%
  dplyr::filter(!(GO.term %in% commsensres) & conservation == "conserved within resistant or sensitive")


# within sensitive
toannotsens =sensit %>%
  dplyr::group_by(GO.term) %>%
  dplyr::mutate(count=n()) %>%
  dplyr::mutate(conservation = case_when(count > 1 ~ "conserved within resistant or sensitive",
                                         TRUE ~ "not conserved")) %>%
  dplyr::ungroup() %>%
  dplyr::select(GO.term, count, conservation) %>%
  unique() %>%
  dplyr::filter(!(GO.term %in% commsensres) & conservation == "conserved within resistant or sensitive")


```


### resistant

```{r}

final = read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species_resistantonly.txt", sep = "\t", header = T) %>%
  dplyr::filter(p_adj < 0.01) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj),
                conservation = case_when(GO.term %in% commsensres ~ "conserved across sensitive and resistant",
                                   GO.term %in% toannotres$GO.term ~ "conserved within resistant",
                                   TRUE ~ "not conserved"),
                colour = case_when(conservation == "conserved across sensitive and resistant" ~ "#e69f00",
                                          conservation == "conserved within resistant" ~ "#0072b2",
                                          TRUE ~ "#000000"))
  
toplot = filter(final, source == "abau_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p1 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 75)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))


toplot = filter(final, source == "kpn_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))
                                            

toplot = filter(final, source == "ecoli_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p3 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('E. coli')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))
  

toplot = filter(final, source == "abau_downreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p4 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 70)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))



toplot = filter(final, source == "kpn_downreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p5 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(0,0.01),n.breaks = 2) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  #guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour),
        legend.position="right")
        #legend.direction = "horizontal")




# design <- "
# 4511
# 4511
# 4511
# 4511
# 4511
# 4511
# 4511
# 4511
# 4522
# 4533
# 4533
# 4533
# 4533
# 4533
# 4533
# 45##
# 4566
# 4566
# "

pdf(paste0(plotdir, "5d_GO_enrich_resmut_Resistantonly.pdf"), width = 12, height = 15)
design <- "
4455#
4455#
4455#
44556
44556
4455#
4455#
4455#
4422#
1133#
1133#
1133#
1133#
"
p1 + p2 + p3 + p4 + p5 + guide_area() + plot_layout(design=design, guides = "collect")


dev.off()


# test prepped to plot stuff 
pdf(paste0(plotdir, "test_abau_res_upreg.pdf"), width = 9, height = 4)
ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  facet_wrap(~GO.tree, scales = "free_y") +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour),
        plot.title = element_text(hjust = 0.5))
dev.off()



```

### sensitive

```{r}

final = read.table("~/Documents/EMBL/Nx/proteomics_strains_Marco/allGO_enrich_sep_species_sensitiveonly.txt", sep = "\t", header = T) %>%
  dplyr::filter(p_adj < 0.01 & !(GOs %in% c("GO:0072488", "GO:0022890","GO:0043872"))) %>%
  dplyr::mutate(species = case_when(source %in% c("abau_upreg","abau_downreg") ~ "A. baumannii",
                                    source %in% c("kpn_upreg","kpn_downreg") ~ "K. pneumoniae",
                TRUE ~ "E. coli"),
                change_sign = case_when(source %in% c("abau_upreg","kpn_upreg","ecoli_upreg") ~ "upregulation",
                                        TRUE ~ "downregulation"),
                p_adj = as.numeric(p_adj),
                conservation = case_when(GO.term %in% commsensres ~ "conserved across sensitive and resistant",
                                   GO.term %in% toannotsens$GO.term ~ "conserved within sensitive",
                                   TRUE ~ "not conserved"),
                colour = case_when(conservation == "conserved across sensitive and resistant" ~ "#E69F00",
                                  conservation == "conserved within sensitive" ~ "#0072B2",
                                  TRUE ~ "black"))
  

toplot = filter(final, source == "abau_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p1 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))

toplot = filter(final, source == "kpn_upreg"& !(GOs %in% c("GO:0072488", "GO:0022890", "GO:0043872"))) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))


toplot = filter(final, source == "kpn_downreg"& !(GOs %in% c("GO:0072488", "GO:0022890", "GO:0043872"))) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p3 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(0,0.01),n.breaks=2) +
  scale_size_continuous(breaks = c(2,4,6,7)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour),
        legend.position = "bottom",
        legend.direction = "horizontal",
        legend.key.width = unit(1, 'cm')) +
  guides(size = guide_legend(keywidth = 0.2))


pdf(paste0(plotdir, "5d_GO_enrich_resmut_Sensitiveonly.pdf"), width = 12, height = 17)

design <- "
12
12
12
12
32
32
42
"
# p1 + p2 + p3 + guide_area() + plot_layout(design=design, guides = "collect") & theme(legend.justification = "top",legend.position = "bottom")

p1 + p2 + p3 + guide_area() + plot_layout(design=design, guides = "collect")

dev.off()



```



##ED 10b - Nx uptake

```{r}

toplot = readxl::read_xlsx(paste0(inputdir,"Nx_uptake_BW25113_MS579.xlsx"),sheet=1,range="A4:H10") %>%
  pivot_longer(cols=c(2:8),names_to="Time",values_to="fluo") %>%
  mutate(strain = rep("MS579",nrow(.)))

toplot1 = readxl::read_xlsx(paste0(inputdir,"Nx_uptake_BW25113_MS579.xlsx"),sheet=2,range="A4:H8") %>%
  pivot_longer(cols=c(2:8),names_to="Time",values_to="fluo") %>%
  mutate(strain = rep("wt",nrow(.)))

toplot=rbind.data.frame(toplot,toplot1)

pdf(paste0(plotdir,"emrR_uptakeNx.pdf"),width = 2.5, height=2.5)

ggline(toplot,x="Time", y="fluo", color="strain", add=c("mean_se")) +
  labs(x="Time (min)", y = "Normalised fluorescence",color="")

dev.off()

```

# Supplementary File

## Fig. 1 - CB reps fixed

```{r}
### find all data

run200818 = fread("~/Documents/EMBL/Nx/CBs_Nx/200818/201808_fitness_norm.txt", sep = "\t", header = T)%>%
  filter(Drug_h %in% c("ATM") & strain == "ECBW") %>%
  select(-c(strain,replicate)) %>%
  rename(replicate=biolrep) %>%
  select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))


run220117 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220117/fitness_norm.txt") %>%
  filter(Drug_h %in% c("AZM","PIP","TMP", "FEP"))

torepl = run220117 %>%
  filter(Drug_h == "TMP" & Conc_v == 0.2248) %>%
  mutate(Conc_h = ifelse(Conc_h== 0.1204, 0.0812,
                        ifelse(Conc_h==0.14, 0.1008, Conc_h)))
run220117 = run220117 %>%
  filter(!(Drug_h == "TMP" & Conc_v == 0.2248)) %>%
  rbind.data.frame(torepl) %>%
  mutate(ODnorm_bug_rob=case_when(Conc_h %in% c(0.1008, 0.1204) & replicate == 2 & Conc_v == 0 ~ 0.014,
                                  TRUE ~ ODnorm_bug_rob))%>%
   mutate(ODnorm_bug_rob = case_when(replicate == 1 & Concv_flag %in% c(6,7) & Conch_flag %in% c(3:4) & Drug_h == "TMP" ~ ODnorm_bug_rob + 0.1,
                                    TRUE ~ ODnorm_bug_rob))%>%
  mutate(ODnorm_bug_rob = case_when(replicate == 1 & Concv_flag %in% c(6) & Conch_flag %in% c(4) & Drug_h == "TMP" ~ ODnorm_bug_rob + 0.2,
                                     TRUE ~ ODnorm_bug_rob))%>%
  select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))

ggplot(run220117 %>% filter(Drug_h %in% "TMP"), aes (x = factor(Conc_h), factor(Conc_v))) +
    geom_tile(aes(fill = ODnorm_bug_rob), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "#d9a520"),
                     values = scales::rescale(c(0,0.1, max(run220117$ODnorm_bug_rob)))) +
    theme_classic() + 
	facet_wrap(Drug_h ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (µg/ml)", y = "Nitroxoline (µg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        #strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        axis.title.y = element_blank(),
        panel.spacing = unit(2, "lines"))



run220120 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220120/fitness_norm.txt") %>%
  filter(Drug_h %in% c("MEC", "ERT","TET","THI","CLQ","8OHQUI","CCCP","MIN","TOB")) %>%
  mutate(ODnorm_bug_rob = case_when(replicate == 1 & Conc_v > 0.37 & Conc_h < 0.18 ~ ODnorm_bug_rob + 0.09,
                                    TRUE ~ ODnorm_bug_rob)) %>%
  mutate(ODnorm_bug_rob = case_when(replicate == 2 & Conc_v > 0.37 & Conc_h < 0.3 & Drug_h == "CLQ" ~ ODnorm_bug_rob - 0.05,
                                    TRUE ~ ODnorm_bug_rob))%>%
  select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))
  
  
run220121 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220121/fitness_norm.txt") %>%
  dplyr::filter(Drug_h %in% c("AMK","CAZ", "MEM", "TEM", "IPM",
                              "CIP", "GEN","NIT", "TGC","NIT")) %>%
  mutate(replicate = case_when(replicate == 3 ~ 1,
                               replicate == 4 ~ 2,
                               TRUE ~ replicate)) %>%
  mutate(ODnorm_bug_rob = case_when(replicate == 1 & Conc_v == 0 & Conch_flag == 4 & Drug_h == "MEM" ~ 0.468,
                                    replicate == 1 & Conc_v == 0 & Conch_flag == 3 & Drug_h == "MEM" ~ 0.247,
                                    replicate == 1 & Conc_v == 0 & Conch_flag < 3  & Drug_h == "MEM" ~ 0,
                                    replicate == 1 & Conc_v == 0 & Conch_flag == 4 & Drug_h == "TEM" ~ 0.573,
                                    replicate == 1 & Conc_v == 0 & Conch_flag == 3 & Drug_h == "TEM" ~ 0.552,
                                    replicate == 1 & Conc_v == 0 & Conch_flag < 3  & Drug_h == "TEM" ~ 0.541,
                                    replicate == 1 & Conc_v == 0 & Conch_flag == 4 & Drug_h == "GEN" ~ 0.469,
                                    replicate == 1 & Conc_v == 0 & Conch_flag == 3 & Drug_h == "GEN" ~ 0.241,
                                    replicate == 1 & Conc_v == 0 & Conch_flag < 3  & Drug_h == "GEN" ~ 0,
                                    replicate == 1 & !(Concv_flag %in% c(7,5)) & Conch_flag < 3  & Drug_h == "NIT" ~ 0,
                                    replicate == 1 & Concv_flag %in% c(5) & Conch_flag < 3  & Drug_h == "NIT" ~ 0.1576,
                                    replicate == 1 & Concv_flag %in% c(7) & Conch_flag < 3  & Drug_h == "NIT" ~ 0.1412,
                                    TRUE ~ ODnorm_bug_rob)) %>%
    mutate(ODnorm_bug_rob = case_when(replicate == 1 & Conc_v > 0.37 & Conch_flag > 2 ~ ODnorm_bug_rob + 0.16,
                                    TRUE ~ ODnorm_bug_rob))%>%
  select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))


run220210 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220210/fitness_norm.txt") %>%
  dplyr::filter(Drug_h %in% c("FOF", "DOX"))%>%
  dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 1.1,
                                  Conc_v == 0.73 ~ 0.946,
                                  Conc_v == 0.61 ~ 0.792,
                                  Conc_v == 0.49 ~ 0.638,
                                  Conc_v == 0.37 ~ 0.484,
                                  Conc_v == 0.25 ~ 0.330,
                                  Conc_v == 0.14 ~ 0.176,
                                   TRUE ~ Conc_v))%>%
  rename(replicate=Replicate)%>%
   mutate(ODnorm_bug_rob = case_when(replicate == 1 & Conch_flag == 3 ~ 0,
                                    TRUE ~ ODnorm_bug_rob))%>%
  select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))
 

run220212 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220212/fitness_norm.txt") %>%
   filter(Drug_h %in% c("CLR","ERI", "LCX")) %>%
   mutate(ODnorm_bug_rob = case_when(replicate == 1 & Concv_flag <= 6 ~ ODnorm_bug_rob + 0.22,
                                    TRUE ~ ODnorm_bug_rob)) %>%
select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))


run220514 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220514/fitness_norm.txt") %>%
  dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 0.75,
                                  Conc_v == 0.73 ~ 0.64,
                                  Conc_v == 0.61 ~ 0.54,
                                  Conc_v == 0.49 ~ 0.43,
                                  Conc_v == 0.37 ~ 0.33,
                                  Conc_v == 0.25 ~ 0.22,
                                  Conc_v == 0.14 ~ 0.12,
                                   TRUE ~ Conc_v)) %>%
  filter(Drug_h %in% c("COL", "RIF","NVB","VAN")) %>%
  # dplyr::mutate(replicate = case_when(replicate %in% c(1,5) ~ "1_1",
  #                                     replicate %in% c(2,6) ~ "1_2",
  #                                     replicate %in% c(3,7) ~ "2_1",
  #                                     TRUE ~ "2_2")) %>%
  dplyr::mutate(biol_rep = case_when(replicate %in% c(1,5,2,6) ~ 1,
                                      TRUE ~ 2))  %>%
  filter(biol_rep == 2) %>%
  dplyr::mutate(replicate = case_when(replicate %in% c(3,7) ~ 1,
                                      TRUE ~ 2))  %>%
select(c(Drug_h, Drug_v, Conc_h, Conc_v, ODnorm_bug_rob, replicate))




#dfList <- list(run200818,run220117,run220120,run220121,run220210, run220212, run220514)
dfList <- mget(ls(pattern="^run"))
dfColList <- lapply(dfList,names)
commonCols <- Reduce(intersect,dfColList)

allCBs <- dplyr::bind_rows(dfList, .id = "source") %>%
  dplyr::select(all_of(commonCols), source) %>%
  mutate(Conc_h = round(Conc_h, digits=4),
         Conc_v = round(Conc_v, digits=3)) %>%
  filter(Drug_h!= "SUL")

# toplot$facet = factor(toplot$fullnames.x, levels = c("Fosfomycin", "Mecillinam", "Temocillin", "Ceftazidime", "Imipenem", "Meropenem", "Ertapenem"))

lapply(dfList,function (x) {unique(x$replicate)})


allCBs = allCBs %>%
  mutate(replicate = case_when(replicate == 3 ~ 1,
                               replicate == 4 ~ 2,
                               TRUE ~ replicate))

#for(i in unique(toplot$Drugclass)) {
#p <- 
# toplot = toplot %>% mutate(Conc_h=round(Conc_h,digits=3),
#                            Conc_v=round(Conc_v,digits=3))

antags= c("ATM","MEC","TEM", "FEP","IPM", "MEM","ERT","THI", "CLQ","AMK","GEN", "TOB","TMP")
synerg = c("VAN", "COL", "AZM",  "NVB", "RIF")
neutrals = setdiff(unique(allCBs$Drug_h), c(antags,synerg))


drugnames = data.frame("Drug_h" = c("8OHQUI","AMK","ATM","AZM","CAZ","CCCP","CIP","CLQ","CLR","COL","DOX","ERI","ERT","FEP","FOF",
                                    "GEN","IPM","LCX","MEC","MEM","MIN","NIT","NVB","PIP","RIF","TEM","TET","TGC",
                                    "THI","TMP","TOB","VAN"),
                       "horizontal_drug" = c("8−Hydroxyquinoline", "Amikacin", "Aztreonam", "Azithromycin", "Ceftazidime", "CCCP", "Ciprofloxacin", "Clioquinol", "Clarithromycin", "Colistin","Doxycycline", "Erythromycin", "Ertapenem", "Cefepime", "Fosfomycin", "Gentamicin", "Imipenem", "Levofloxacin", "Mecillinam", "Meropenem", "Minocycline", "Nitrofurantoin",
                                         "Novobiocin", "Piperacillin", "Rifampicin", "Temocillin","Tetracycline",
                                         "Tigecycline", "Thiolutin", "Trimethoprim","Tobramycin","Vancomycin"))
allCBs = allCBs %>%
  inner_join(drugnames)


allCBs = readxl::read_xlsx(paste0(inputdir,"source_Fig.2a_old.xlsx"))

antags= c("Aztreonam","Mecillinam","Temocillin", "Cefepime","Imipenem", "Meropenem","Ertapenem","Thiolutin", "Clioquinol","Amikacin","Gentamicin", "Tobramycin","Trimethoprim")
synerg = c("Vancomycin", "Colistin", "Azithromycin",  "Novobiocin", "Rifampicin")
neutrals = setdiff(unique(allCBs$horizontal_drug), c(antags,synerg))


p <- ggplot(allCBs %>% filter(horizontal_drug %in% antags), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "#d9a520"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (µg/ml)", y = "Nitroxoline (µg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        #strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        axis.title.y = element_blank(),
        panel.spacing = unit(2, "lines"))

p1 <- ggplot(allCBs %>% filter(horizontal_drug %in% synerg), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "turquoise4"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (µg/ml)", y = "Nitroxoline (µg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
       axis.text.y = element_text(size = 10),
        axis.title.y = element_blank(),
        panel.spacing = unit(2, "lines"),
        legend.position = "none")

p2 <- ggplot(allCBs %>% filter(horizontal_drug %in% neutrals), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "grey"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (µg/ml)", y = "Nitroxoline (µg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        #strip.text = element_text(size= 7),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="right")

pdf(paste0(plotdir, "CB_heatmaps_antag_repssep.pdf"),  width=10.5, height=11)
shift_legend2 <- function(p) {
  # ...
  # to grob
  gp <- ggplotGrob(p)
  facet.panels <- grep("^panel", gp[["layout"]][["name"]])
  empty.facet.panels <- sapply(facet.panels, function(i) "zeroGrob" %in% class(gp[["grobs"]][[i]]))
  empty.facet.panels <- facet.panels[empty.facet.panels]

  # establish name of empty panels
  empty.facet.panels <- gp[["layout"]][empty.facet.panels, ]
  names <- empty.facet.panels$name
  # example of names:
  #[1] "panel-3-2" "panel-3-3"

# now we just need a simple call to reposition the legend
  reposition_legend(p, 'center', panel=names, plot =F)
}

grid::grid.draw(shift_legend2(p))
dev.off()

pdf(paste0(plotdir, "CB_heatmaps_synergies_repssep.pdf"),  width=10.5, height=4.4)

p1

dev.off()


pdf(paste0(plotdir, "CB_heatmaps_neutrals_repssep.pdf"),  width=10.5, height=11)
grid::grid.draw(shift_legend2(p2))
dev.off()

# 
# towrite=allCBs %>%
#   ungroup() %>%
#   dplyr::select(Conc_h, Conc_v, ODnorm_bug_rob, fullnames.x,replicate, type) %>%
#   dplyr::rename(horizontal_concentration=Conc_h,
#                 nitroxoline_concentration=Conc_v,
#                 norm_fitness=ODnorm_bug_rob,
#                 horizontal_drug = fullnames.x) %>%
#   distinct()

towrite = allCBs %>%
  select(-c(bliss,cumulative_eps_quartiles,eps.med,eps.q1,eps.q3)) %>%
  mutate(type=case_when(horizontal_drug %in% antags ~ "antagonism",
            horizontal_drug %in% synerg ~ "synergy",
            TRUE ~ "neutral")) %>% 
  distinct()


writexl::write_xlsx(towrite, paste0(sourcedatadir,"source_EDFig.3a.xlsx"))

```




## Fig. 2 - full growth curves Kpn resensitization
```{r}
bunch = fread("~/Documents/EMBL/Nx/220602_colistin_Nx_resmutants/plate_reannotated.txt") %>%
  dplyr::mutate(strain = case_when(strain == "EC_J53" ~ "E. coli J53",
                                       strain == "EC_J53Tc2442" ~ "E. coli J53 pKP2442_mcr-1",
                                       strain == "EC_J53pTOPO" ~ "E. coli J53 pTOPO",
                                       strain == "EC_J53pTOPOmcr1" ~ "E. coli J53 pTOPO_mcr-1",
                                       strain == "KP_2442" ~  "K. pneumoniae pKP2442_mcr-1",
                                       strain == "KP_CTR3" ~ "K. pneumoniae CT-R3 pmrB (P95L)",
                                       strain == "KP_CTS" ~ "K. pneumoniae CT-S",
                                      strain == "KP_LS107" ~ "K. pneumoniae LS107",
                                      strain == "KP_LS53" ~ "K. pneumoniae LS53 mgrB (IS1R)",
                                      strain == "KP_PRZ" ~ "K. pneumoniae PRZ",
                                      strain == "KP_PRZ2442" ~ "K. pneumoniae PRZ pKP2442_mcr-1",
                                   TRUE ~ strain)) %>%
    filter(strain != "nocells")



bla = fread("~/Documents/EMBL/Nx/220707_colistin_Nx_resmutants_KPPRZ_topo/plate_reannotated.txt")

pmrAB = fread("~/Documents/EMBL/Nx/221110_pmrAB_fromMichael/plate_reannotated.txt")%>%
  dplyr::mutate(nitroxoline=case_when(Condition == "+ Nx 0.75 µg/ml" ~ "Nitroxoline_0.75",
                                      TRUE ~ "no_Nitroxoline"))%>%
  rename(strain=Strain) %>%
  dplyr::mutate(strain = case_when(strain == "BW25113" ~ "E. coli BW25113",
                                   strain == "MG1655" ~ "E. coli MG1655",
                                   strain == "pmrA" ~ "E. coli MG1655 pmrA (G53E)",
                                   strain == "pmrB" ~ "E. coli MG1655 pmrB (V88E)",
                                   TRUE ~ strain)) 

mcr1 = fread("~/Documents/EMBL/Nx/220527_MIC_mrc1_colistin_nx/plate_reannotated.txt") %>%
    filter(Condition != "+ Nx 0.45 µg/ml") %>%
  rename(strain=Strain) %>%
  dplyr::mutate(nitroxoline=case_when(Condition == "+ Nx 0.85 µg/ml" ~ "Nitroxoline_0.75",
                                      TRUE ~ "no_Nitroxoline")) %>%
  dplyr::mutate(strain = case_when(strain == "BW" ~ "E. coli BW25113",
                                   strain == "mcr1" ~ "E. coli BW25113 pGDP1_mcr1"))%>%
  filter(strain != "E. coli BW25113")
  


toplot = bind_rows(bunch, bla, pmrAB,mcr1,.id = "source")
#toplot = bind_rows(bunch, bla)


toplot$facet = factor(toplot$strain, levels = c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1","E. coli J53 pKP2442_mcr-1",
                                                "E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)",
                                                "E. coli BW25113","E. coli BW25113 pGDP1_mcr1",
                                                "K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)", "K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)","K. pneumoniae pKP2442_mcr-1",
                                                "K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO","K. pneumoniae PRZ pTOPO_mcr-1", "K. pneumoniae PRZ pKP2442_mcr-1"))

toplot = toplot %>%
  mutate(row= case_when(strain %in% c("E. coli BW25113","E. coli BW25113 pGDP1_mcr1") ~ 1,
                        strain %in% c("E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)") ~ 2,
                        strain %in% c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "E. coli J53 pKP2442_mcr-1") ~ 3,
                        #strain %in% c("E. coli J53 pKP2442_mcr-1") ~ 4,
                        strain %in% c("K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO_mcr-1","K. pneumoniae PRZ pKP2442_mcr-1") ~ 5,
                        strain %in% c("K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)") ~ 6,
                        strain %in% c("K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)") ~ 7,
                        strain %in% c("K. pneumoniae pKP2442_mcr-1") ~ 8,
                        strain %in% c("K. pneumoniae PRZ pTOPO") ~ 9),
         resist = case_when(strain %in% c("E. coli BW25113","E. coli J53","E. coli MG1655",
                                          "K. pneumoniae CT-S","K. pneumoniae LS107","K. pneumoniae PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))


fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)


toplot = toplot %>%
  #decided not to include empty vector control in main fig, only supp 
  #filter(!(strain %in% c("E. coli J53 pTOPO", "K. pneumoniae PRZ pTOPO"))) %>%
  mutate(strain = gsub(strain, pattern  =" ",replacement = "~"),
        strain = gsub(strain, pattern  ="E.~coli",replacement = "italic('E. coli')"),
        strain = gsub(strain, pattern  ="K.~pneumoniae",replacement = "italic('K. pneumoniae')")) 


toplot = toplot %>%
  mutate(strain = case_when(strain =="italic('E. coli')~BW25113~pGDP1_mcr1" ~ "~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))",  
    strain =="italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
                            strain =="italic('E. coli')~MG1655~pmrA~(G53E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrA'),'(G53E)'))",
                            strain == "italic('E. coli')~MG1655~pmrB~(V88E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrB'),'(V88E)'))",
    strain == "italic('E. coli')~J53~pTOPO_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pTOPO_', italic('mcr-1')))",
    strain == "italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
    TRUE ~ strain))

toplot = toplot %>%
  dplyr::group_by(strain,Time,nitroxoline, Concentration, source) %>%
  dplyr::mutate(mean = mean(OD_no1st),
                SE = sqrt(var(OD_no1st)/length(OD_no1st)))

#remove one dataset for prz - just creates confusion
toplot = toplot %>%
  filter(!(strain == "italic('K. pneumoniae')~PRZ" & source ==1))
  

pdf(paste0(plotdir, "colistin_resensitiz_growth_curves.pdf"), width=12, height=15)

ggplot(toplot%>%
         #filter(!(strain %in% c("~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))", "italic('E. coli')~BW25113" ))) %>%
         mutate(strainrep=interaction(strain,Replicate,nitroxoline,source)),
       aes(x = Time, y = mean,color = factor(nitroxoline), group=strainrep)) +
        #geom_point(aes(shape = factor(IPTG)), size = 1) +
        geom_line(aes(group=strainrep)) +
  geom_ribbon(aes(ymin = mean - SE, ymax = mean + SE, fill = factor(nitroxoline)),alpha = 0.5) +
        #ylim(c(-0.5,1)) +
        #scale_y_continuous(trans='log10') +
        #scale_color_manual(values=c("black","turquoise4"), labels=c("wt", "lptD4213")) +
        #facet_matrix(rows = vars(strain), cols = vars(concentration)) +
        #facet_grid(strain ~ Concentration, space="free", scales="free") +
      # ggh4x::facet_grid2(strain ~ Concentration, #scales = "free", #axes = "x",
      #                    independent = "all") +
  scale_color_manual(labels = c("+ NX 0.75 µg/ml", "- NX"), values = c("turquoise4","darkgrey")) +
  scale_fill_manual(labels = c("+ NX 0.75 µg/ml", "- NX"), values = c("turquoise4","darkgrey")) +
      facet_grid(row + strain ~ Concentration, labeller = label_parsed) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]),color = "") +
  theme_bw() +
  guides(fill="none")+
  theme(legend.position="right")



dev.off()



```

## Fig. 3- full growth curves LptD

### new version - 220525
```{r}
toplot = fread("~/Documents/EMBL/Nx/220525_lptD/plate_norm.txt") %>%
  filter(strain %in% c("BW","lptD"))
  
fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time_h, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)

toplot = toplot %>%
  dplyr::group_by(strain,Time_h, Concentration) %>%
  dplyr::mutate(mean = mean(OD_no1st),
                SE = sqrt(var(OD_no1st)/length(OD_no1st)),
                strain=gsub(strain,pattern="BW",replacement="E. coli BW25113"),
                strain=gsub(strain,pattern="lptD",replacement="E. coli lptD4213"))

pdf(paste0(plotdir,"SuppFig3_fullgrowthcurves_lptD.pdf"),width = 10, height=2)
ggplot(toplot%>%
         mutate(strainrep=interaction(strain,Replicate)),
       aes(x = Time_h, y = mean,color = factor(strain), group=strain)) +
        geom_line(aes(group=strain)) +
  geom_ribbon(aes(ymin = mean - SE, ymax = mean + SE, fill = factor(strain)),alpha = 0.5) +
  scale_color_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
  scale_fill_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
      facet_grid( ~ Concentration) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]),color = "",fill="") +
  theme_bw() +
  guides(color="none")+
  theme(legend.position="right")

dev.off()

```

### old version
```{r}
toplot = fread("~/Documents/EMBL/Nx/220418_allmutants_Nx_analysis/plate_norm.txt") %>%
  filter(Strain %in% c("BW_EC","lptD"))
  
fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time_h, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)

toplot = toplot %>%
  dplyr::group_by(Strain,Time_h, Concentration) %>%
  dplyr::mutate(mean = mean(medianOD),
                SE = sqrt(var(medianOD)/length(medianOD)),
                Strain=gsub(Strain,pattern="BW_EC",replacement="E. coli BW25113"),
                Strain=gsub(Strain,pattern="lptD",replacement="E. coli lptD4213"))

pdf(paste0(plotdir,"SuppFig3_fullgrowthcurves_lptD.pdf"),width = 10, height=2)
ggplot(toplot%>%
         mutate(strainrep=interaction(Strain,Replicate)),
       aes(x = Time_h, y = mean,color = factor(Strain), group=Strain)) +
        geom_line(aes(group=Strain)) +
  geom_ribbon(aes(ymin = mean - SE, ymax = mean + SE, fill = factor(Strain)),alpha = 0.5) +
  scale_color_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
  scale_fill_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
      facet_grid( ~ Concentration) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]),color = "",fill="") +
  theme_bw() +
  guides(color="none")+
  theme(legend.position="right")

dev.off()

```

#### I had also this file - check whats going on = a double?

```{r}
toplot = fread("~/Documents/EMBL/Nx/220408_nx_lptD_dltclones/plate_reannotated.txt") %>%
  filter(strain %in% c("BW", "lptD")) %>%
  select(-c(Time, Well, letter,number,Drug, Time_points,mean_1sttp)) %>%
  clean_names() %>%
  mutate(strainrep=interaction(strain,replicate))


pdf(paste0(plotdir, "lptD_GrowthCurves.pdf"), width=8, height=2)


ggplot(toplot %>% filter(strain %in% c("lptD", "BW") & concentration %in% c(0,0.125,0.25,0.5,1,2,4,8)),
       aes(x = time_h, y = od_no1st,color = factor(strain), group=strainrep)) +
        #geom_point(aes(shape = factor(IPTG)), size = 1) +
        geom_line(aes(group=strainrep)) +
        scale_color_manual(values=c("black","turquoise4"), labels=c("wt", "lptD4213")) +
        facet_wrap(~ concentration, ncol = 8) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]), title = "Nitroxoline (µg/ml)", color = "") +
        theme_light() +
  theme(plot.title = element_text(hjust=0.5))

dev.off()

towrite = 

```



# SUPP TABLES

## Supp Table 2 - Drugs for comboscreen

```{r}
run200818 <- fread("~/Documents/EMBL/Nx/CBs_Nx/200818/CB_median_reps_cumeps.txt") %>%
  filter(Drug_h %in% c("ATM") & strain == "ECBW") %>%
  select(-strain)

run220117 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220117/CB_median_reps_cumeps.txt") %>%
  filter(Drug_h %in% c("AZM","PIP","TMP", "FEP","SUL")) #%>% 
  # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
  #                                     TRUE ~ "2_1"))
run220120 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220120/CB_median_reps_cumeps.txt") %>%
  filter(Drug_h %in% c("MEC", "ERT","TET","THI","CLQ","8OHQUI","CCCP","MIN","TOB")) %>%
  select(-Comb_full)
  # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
  #                                     TRUE ~ "2_1"))
run220121 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220121/CB_median_reps_cumeps.txt") %>%
  dplyr::filter(Drug_h %in% c("AMK","CAZ", "MEM", "TEM", "IPM",
                              "CIP", "GEN","NIT", "TGC","NIT")) %>%
  select(-Comb_full)
  # dplyr::mutate(replicate = case_when(replicate %in% c(1, 3) ~ "1_1",
  #                                     TRUE ~ "2_1"))
run220210 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220210/CB_median_reps_cumeps.txt") %>%
  dplyr::filter(Drug_h %in% c("FOF", "DOX"))%>%
  dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 1.1,
                                  Conc_v == 0.73 ~ 0.946,
                                  Conc_v == 0.61 ~ 0.792,
                                  Conc_v == 0.49 ~ 0.638,
                                  Conc_v == 0.37 ~ 0.484,
                                  Conc_v == 0.25 ~ 0.330,
                                  Conc_v == 0.14 ~ 0.176,
                                   TRUE ~ Conc_v)) #%>%
  # dplyr::mutate(replicate = case_when(Replicate == 1 ~ "1_1",
  #                                     TRUE ~ "2_1"))
run220212 <-fread("~/Documents/EMBL/Nx/CBs_Nx/220212/CB_median_reps_cumeps.txt") %>%
  filter(Drug_h %in% c("CLR","ERI", "LCX"))%>%
  dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 1.2,
                                  Conc_v == 0.73 ~ 1.032,
                                  Conc_v == 0.61 ~ 0.864,
                                  Conc_v == 0.49 ~ 0.696,
                                  Conc_v == 0.37 ~ 0.528,
                                  Conc_v == 0.25 ~ 0.360,
                                  Conc_v == 0.14 ~ 0.192,
                                   TRUE ~ Conc_v))#%>%
  # dplyr::mutate(replicate = case_when(replicate == 1 ~ "1_1",
  #                                     TRUE ~ "2_1"))
run220514 <- fread("~/Documents/EMBL/Nx/CBs_Nx/220514/CB_median_reps_cumeps.txt") %>%
  dplyr::mutate(Conc_v = case_when(Conc_v == 0.85 ~ 0.75,
                                  Conc_v == 0.73 ~ 0.64,
                                  Conc_v == 0.61 ~ 0.54,
                                  Conc_v == 0.49 ~ 0.43,
                                  Conc_v == 0.37 ~ 0.33,
                                  Conc_v == 0.25 ~ 0.22,
                                  Conc_v == 0.14 ~ 0.12,
                                   TRUE ~ Conc_v)) %>%
  filter(Drug_h %in% c("COL", "RIF","NVB","VAN"))
  # dplyr::mutate(replicate = case_when(replicate %in% c(1,5) ~ "1_1",
  #                                     replicate %in% c(2,6) ~ "1_2",
  #                                     replicate %in% c(3,7) ~ "2_1",
  #                                     TRUE ~ "2_2"))


#dfList <- list(run200818,run220117,run220120,run220121,run220210, run220212, run220514)
dfList <- mget(ls(pattern="^run"))
dfColList <- lapply(dfList,names)
commonCols <- Reduce(intersect,dfColList)

allCBs <- dplyr::bind_rows(dfList, .id = "source") %>%
  dplyr::select(all_of(commonCols), source) %>%
  mutate(Conc_h = round(Conc_h, digits=4),
         Conc_v = round(Conc_v, digits=3))

# toplot$facet = factor(toplot$fullnames.x, levels = c("Fosfomycin", "Mecillinam", "Temocillin", "Ceftazidime", "Imipenem", "Meropenem", "Ertapenem"))

allCBs = allCBs %>%
  filter(fullnames.x != "Sulbactam")


# antags= c("FOF","ATM","MEC","TEM", "CAZ","FEP","IPM", "MEM","ERT","THI", "AMK", "TOB", "TET","TMP")
# synerg = c("VAN", "COL", "AZM", "CLR", "NVB", "RIF")
# neutrals = setdiff(unique(allCBs$Drug_h), c(antags,synerg))

towrite = allCBs %>%
  distinct(fullnames.x, Conc_h) %>%
  group_by(fullnames.x) %>%
  summarise(`Highest concentration tested (µg/ml)`= max(Conc_h)) %>%
  rename(Drug=fullnames.x) %>%
  mutate(Drug = gsub(Drug, pattern="8-hydroxyquinolin",replacement="8-hydroxyquinoline"),
         Broad_class = case_when(Drug %in% c("Aztreonam","Cefepime","Ceftazidime", "Colistin", "Ertapenem","Meropenem","Imipenem","Piperacillin","Fosfomycin","Mecillinam","Temocillin","Vancomycin") ~ "Cell wall and outer membrane",
                                 Drug %in% c("8-hydroxyquinoline","Clioquinol","Thiolutin") ~ "Other metal chelators",
                                 Drug %in% c("Rifampicin","Nitrofurantoin","CCCP") ~ "Other",
                                 Drug %in% c("Amikacin","Azithromycin", "Clarithromycin","Doxycycline","Erythromycin", "Gentamicin","Minocycline","Tetracycline","Tigecycline","Tobramycin") ~ "Protein synthesis",
                                 Drug %in% c("Ciprofloxacin","Levofloxacin", "Novobiocin","Trimethoprim") ~ "DNA"),
         Class=case_when(Drug %in% c("Cefepime","Ceftazidime", "Temocillin", "Aztreonam", "Mecillinam", "Ertapenem","Meropenem","Imipenem","Piperacillin") ~ "Beta-lactam",
                         Drug %in% c("Colistin") ~ "Polymyxin",
                         Drug %in% c("Fosfomycin") ~ "Phosphonic antibiotic",
                         Drug %in% c("Vancomycin") ~ "Glycopeptide",
                        Drug %in% c("8-hydroxyquinoline","Clioquinol") ~ "Quinoline",
                        Drug %in% c("Thiolutin") ~ "Other metal chelator",
                        Drug %in% c("Nitrofurantoin") ~ "Nitrofuran",
                        Drug %in% c("CCCP") ~ "PMF-inhibitor",
                        Drug %in% c("Rifampicin") ~ "Ansamycin",
                        Drug %in% c("Amikacin","Gentamicin","Tobramycin") ~ "Aminoglycoside",
                        Drug %in% c("Azithromycin", "Clarithromycin","Erythromycin") ~ "Macrolide",
                        Drug %in% c("Doxycycline", "Minocycline","Tetracycline","Tigecycline") ~ "Tetracycline",
                        Drug %in% c("Ciprofloxacin","Levofloxacin") ~ "Fluoroquinolone",
                        Drug %in% c("Novobiocin") ~ "Aminocoumarin",
                        Drug %in% c("Trimethoprim") ~ "Antifolate"))



write_xlsx(towrite, "~/Documents/EMBL/Nx/Nx_paper/SuppTables/Supplementary_Table_2.xlsx")

```

## Supp table 5 - proteomics

```{r}
toplot = fread(paste0(outputdir,"hits_proteomics_og_annotated_240101.txt"), sep="\t") %>%
  filter(!(Strain_treatment %in% c("BW25113","dmprA",
                                   "19606S","25922S","2778S",
                                   "MS02S","MS25S",
                                   "MS55S", "PRZS"))) %>%
  select(-og) %>%
  rename(og=og1) 

write_xlsx(toplot,paste0(supptablesdir,"Supplementary_Table_5.xlsx"))



```

# Not included- cell shape changes

```{r}
Nx_0.9= read.csv(paste0(inputdir,"Nx_0.9_microscopy_merged_table.csv"))
untreated= read.csv(paste0(inputdir,"untreated_microscopy_merged_table.csv"))

coccoid <- gdata::combine(Nx_0.9,untreated)

give.n <- function(x){
  return(c(y = median(x)*1.05, label = length(x))) 
  # experiment with the multiplier to find the perfect position
}

cw_summary <- coccoid %>% 
  group_by(source) %>% 
  tally()

ggboxplot(coccoid,x="source",y="Circularity") + stat_compare_means(method="t.test", label = "p.signif", size = 7) +
  labs(x="") +
  scale_x_discrete(labels=c("Nitroxoline 0.9 µg/ml","Untreated")) +
  geom_text(data = cw_summary,
            aes(source, 0, label = paste0("n=",n)))#, vjust = 20))
  
  
  stat_summary(fun.data = give.n, geom = "text",
                  position = position_dodge(width = 0.75))

ggboxplot(coccoid,x="source",y="Ellipse.Elong")



```


```{r}

species = sort(rownames(toplot))

species = gsub(species, pattern = "A. ", replacement = "Acinetobacter ")
species = gsub(species, pattern = "E. ", replacement = "Escherichia ")
species = gsub(species, pattern = "S. ", replacement = "Stenotrophomonas ")

write.table(species, paste0(outputdir,"species.txt"), sep= "\t", quote = F, row.names = F)

```


# number clin iso from Supp table 1 210422

```{r}
supptab = readxl::read_xlsx("~/Documents/EMBL/Nx/Nx_paper/input/Supplementary Table 1_MT_240414_ECSSG_.xlsx", skip = 1) %>%
  filter(grepl("WGS",Experiment)) %>%
  filter(`Nx S/R` == "R")





```