---
title: "code_nx_source_code_figs"
output: html_document
date: "2024-05-26"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r}
#load packages
install.packages("pacman")

if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install("ComplexHeatmap", force = TRUE)

library(pacman)
pacman::p_load(data.table,tidyverse,stringr, tools, smoothmest, ggforce,directlabels,RColorBrewer, Rmisc, tools, lemon, gridExtra, ggbeeswarm, raster, writexl,ggpubr, RColorBrewer, UpSetR, grid, circlize, ComplexUpset, Hmisc,corrplot, heatmaply,devtools, sjmisc, patchwork,ggalt,janitor,gplots, fuzzyjoin, R.utils, TreeTools, TreeDist, word2vec, ComplexHeatmap,scales,httr,drawProteins,ggrepel, plotrix, rstatix,ggtext, dendextend,dendsort,ggh4x, rstatix,gdata,latex2exp,plotrix, stringi,ggh4x, lemon,castor,ggtree,forcats)

```

# data dirs
```{r}
home_dir = "~/Documents/EMBL/Nx/Nx_paper/REVISION/nitroxoline_2024/figures/"

inputdir = paste0(home_dir, "input/")
plotdir =  paste0(home_dir, "plots/")

sourcedatadir = paste0(home_dir, "source_data/")
supptablesdir = paste0(home_dir, "SuppTables/")
```


# required functions

```{r}
plot2dTppFcHeatmap <- function(df, name, 
                               drug_name = "",
                               midpoint = 1){
  protein <- temperature <- nitroxoline_conc <- 
    rel_value <- fc <- NULL
  
  heat_df <- df %>% 
    dplyr::filter(protein == name) %>% 
    dplyr::select(protein, temperature, 
                  nitroxoline_conc, rel_value) %>% 
    dplyr::mutate(temperature = factor(
      temperature, levels = rev(sort(unique(temperature)))),
      nitroxoline_conc = as.factor(nitroxoline_conc)) %>% 
    dplyr::group_by(protein, temperature, nitroxoline_conc) %>% 
    dplyr::summarise(fc = mean(rel_value, na.rm = TRUE)) %>% 
    dplyr::ungroup()
  
  ggplot(heat_df, aes(nitroxoline_conc, temperature)) +
    geom_tile(aes(fill = fc)) +
    scale_fill_gradient2(
      "Fold\nchange", 
      low = "firebrick", mid = "khaki", 
      high = "darkgreen", midpoint = midpoint) +
    labs(x = paste(drug_name, "(Âµg/mL)"),
         y = expression("Temperature ("*~degree*C*")")) +
    ggtitle(name) +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5),
          axis.text = element_text(size = 10),
          axis.title = element_text(size = 12),
          legend.key.size = unit(1,"line"),
          legend.title = element_text(size=10),
          legend.text = element_text(size=10))
}


capitalize_first_and_last <- function(string) {
  first_letter <- toupper(substring(string, 1, 1))
  last_letter <- toupper(substring(string, nchar(string), nchar(string)))
  middle_letters <- tolower(substring(string, 2, nchar(string) - 1))
  return(paste(first_letter, middle_letters, last_letter, sep = ""))
}

```

#MAIN FIGURES

# Fig. 1

## Fig. 1b - how many species in our three methods and in EUCAST

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.1b.xlsx"))

toplot = toplot %>%
  distinct(Species, Method)
toplot = table(toplot)
toplot[toplot>0] = 1

toplot = as.data.frame(toplot)

toplot = toplot %>%
  pivot_wider(names_from = Method, values_from = Freq) %>%
  as.data.frame()

toplot = toplot[apply(toplot[,-1], 1, function(x) !all(x==0)),]

toplot = dplyr::mutate(toplot, A = ifelse(`Agar dilution`, "Agar dilution", NA),
       D = ifelse(`Disk diffusion`, "Disk diffusion", NA),
       M = ifelse(`Microdilution`, "Microdilution", NA),
       R = ifelse(`Microdilution`, "EUCAST", NA))

toplot = tidyr::unite(toplot, "Methods", A:M, na.rm=T, sep='&')


pdf(paste0(plotdir,"Fig.1b_upset_EUCAST_this_study.pdf"), width = 6, height = 4.5)
ComplexUpset::upset(toplot,c("Agar dilution", "Disk diffusion", "Microdilution", "EUCAST"),
  stripes = "white",
  sort_intersections=FALSE,
  intersections=list("EUCAST",
                     c("Microdilution", "Disk diffusion","Agar dilution","EUCAST"), 
                     c("Microdilution", "Disk diffusion","Agar dilution"),
                    "Disk diffusion"),
  base_annotations=list('Intersection size'=intersection_size(
      text = list(size=5.5),
      text_colors = c(on_background = "black", on_bar = "white"))),
  set_sizes=(upset_set_size(
    geom = geom_bar(width = 0.5),
    position = "right") +
    geom_text(aes(label=..count..), hjust=1.2, stat='count', size =5.5, color ="white", fontface="bold") +
      expand_limits(y=30) +
      theme(axis.text.x=element_text(size = 8))),
  matrix=intersection_matrix(
    geom = geom_point(size = 3),
    segment = geom_segment(),
    outline_color = list(active = "black", inactive = "grey70")
  ),
  themes = upset_modify_themes(list('Intersection size'=theme(axis.line.x = element_blank(),
                                                              axis.ticks.x = element_blank(),
                                                              axis.line.y = element_line(),
                                                              axis.text = element_text(size=15),
                                                              legend.position = 'none',
                                                              legend.text = element_text(size=10),
                                                              axis.title=element_text(size=15),
                                                              axis.ticks.y = element_line(),
                                                              plot.background = element_blank(),
                                                              panel.grid = element_blank()),
                                    'intersections_matrix'=theme(axis.line.x = element_blank(),
                                                                 axis.ticks.x = element_blank(),
                                                                 axis.title = element_blank(),
                                                                 axis.text = element_text(size=15),
                                                                plot.background = element_blank(),
                                                                 panel.grid = element_blank()),
                                    'overall_sizes'=theme(axis.line.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          axis.line.x = element_line(),
                                                          axis.ticks.x = element_line(),
                                                          axis.text.x.bottom = element_text(size=15),
                                                          axis.title = element_text(size=15),
                                                          plot.background = element_blank(),
                                                          text = element_text(size=15),
                                                          panel.grid = element_blank()))),
  width_ratio=0.7,
  height_ratio = 0.6,
sort_sets = F)

dev.off()

```

## Fig. 1c - MIC in broth and phylogeny

#### Left: tree from gdtb

```{r}
pruned.tree = ape::read.tree(paste0(sourcedatadir,"Fig1c.tree"))

pdf(paste0(plotdir,"Fig.1c_left_tree.pdf"), width=10, height=10)
ggtree(pruned.tree) + 
  geom_tiplab(size = 5,offset = 0.001,align = TRUE) + 
  theme_tree2()+
  xlim(0, 1.6)+
  theme(
    axis.title.x = element_text(size = 15),
    legend.title = element_text(face = "bold", size = 12),
    legend.text = element_text(face = "bold", size = 10),
    plot.title = element_text(size = 12, face = "bold"))
dev.off()

```

### Right: species total MIC heatmap with GTDB dendro

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.1c.xlsx")) %>%
  mutate(Species_italics = case_when(Species == "Salmonella spp." ~ "*Salmonella* spp.",
                                     Species == "Shigella spp." ~ "*Shigella* spp.",
                                    TRUE ~ paste0("*", Species, "* ")),
         MIC=as.numeric(MIC),
         Species_label = as.character(paste0(Species_italics, "(n=", tot_number_of_strains, ")"))) 

species_order= c("Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")

#create order with the n annotation
legend = toplot %>% 
  distinct(Species, Species_label) %>%
  arrange(factor(Species, levels=species_order))


toplot = toplot %>%
  arrange(factor(Species, levels=species_order)) %>%
  distinct(Species_label,MIC, MIC50,number_strains_at_MIC) %>%
  complete(Species_label, MIC, fill = list(number_of_strains = 0))


toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  


toplot$number_strains_at_MIC[is.na(toplot$number_strains_at_MIC)] <- 0

toplot$highlight_MIC50 <- ifelse(toplot$MIC == toplot$MIC50, TRUE, FALSE)

toplot$highlight_MIC50[is.na(toplot$highlight_MIC50)] <- FALSE

pdf(paste0(plotdir,"Fig.1c_right_MICs_broth.pdf"), width =8.181818, height=10)
ggplot(toplot, aes (x = as.factor(MIC), Species_label)) +
    geom_tile(aes(fill = number_strains_at_MIC, color="black"), colour = "black", size = 0.25) +   
     geom_tile(data = toplot[toplot$highlight_MIC50, ],
            aes(x = as.factor(MIC), y = Species_label),
            fill = NA, colour = "black", size = 1.5) +
    geom_text(aes(label=number_strains_at_MIC), size =5) +
  scale_fill_gradient(name = "Number of strains\nwith MIC value", low="white", high="grey40") +
  labs(y="", x =  "MIC (Âµg/ml)") +
 scale_fill_gradientn(colors = c("white","grey70", "grey60", "grey40"),
                      values = scales::rescale(c(0,5,20, max(toplot$number_strains_at_MIC)))) +
   theme_classic() +
  theme(axis.text.y = element_markdown(color = "black", size = 15),
        axis.text.x=element_text(size=13, angle =60, hjust = 1),
        legend.position="right",
        axis.title=element_text(size=18))

 dev.off()
 

```


## Fig. 1d - Salmonella intracellular

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.1d.xlsx")) %>%
  mutate(condition = factor(condition, levels=c("before treatment", "DMSO", "nitroxoline (5 Âµg/ml)")),
         source = plyr::mapvalues(strain, from = c("S. Typhi 5878","S. Typhi 2601"),
                                         to = c("italic('S. ')~Typhi~'5878'","italic('S. ')~Typhi~'2601'")))

stat.test <- toplot %>%
  distinct(source, mean,condition) %>%
  dplyr::group_by(source) %>%
  t_test(cfu_ml ~ condition, var.equal = TRUE) %>%
  add_significance("p") %>%
  add_xy_position(fun = "mean_sd", x = "condition", dodge = 0.8) %>%
  filter(!(group1 =="DMSO" & group2 =="preNx"))

                 
library(stringi)      
stat.test = stat.test %>%
  mutate(xmin=case_when(xmin == 1 ~ 2,
                        xmin == 2 ~ 3,
                        TRUE ~ 1),
         xmax=case_when(xmin == 1 ~ 2,
                        xmin == 2 ~ 3,
                        TRUE ~ 1))
                                                


pdf(paste0(plotdir, "Fig.1d_Salmonella_intracell_killing.pdf"), width=7.5, height=5.5)


ggplot(toplot, aes(x = condition, y = mean)) +
  geom_bar(stat = "identity", position = position_dodge(0.8), width = 0.5,fill="grey50") +
  geom_errorbar(aes(ymin = mean - sd, ymax = mean + sd),width = 0.15, color="grey50") +
  geom_jitter(aes(x = condition,y=cfu_ml), width = 0.1,height = 0.1, 
              size = 3, shape = 21, alpha = 1, fill="grey90", color="grey20") +
   stat_pvalue_manual(stat.test, tip.length = 0.01,
                     label = "p.signif",  y.position = 5, step.increase = 0.1, step.group.by = "source",size = 5) +
  facet_wrap(~ source, labeller = label_parsed)+
  scale_y_continuous(breaks = c(0,2,4), labels= math_format(10^.x)) +
  scale_x_discrete(labels=c("before\ntreatment","solvent\ncontrol","nitroxoline\n(5 Âµg/ml)")) +
  labs(x="", y= expression(CFU/ml)) +
  theme_bw() +
  guides(color="none") +
  theme(strip.text = element_text(size=16),
        axis.text=element_text(size=16),
        axis.title=element_text(size=16))
dev.off()

```


## Fig. 1e - killing A. baumannii

See source data ED Fig. 2b
```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.1e_EDFig.2b.xlsx")) %>%
  filter(`Nitroxoline concentration (Âµg/ml)` != 0)


pdf(paste0(plotdir,"Fig.1e_Abau_killing.pdf"), width = 5.8, height=5)

my_palette = brewer.pal("Oranges",n = 9)[3:9]

toplot = toplot %>% clean_names()

ggplot(toplot, aes(x=as.factor(time_h),y=mean, group =nitroxoline_concentration_mg_ml, color = nitroxoline_concentration_mg_ml)) + 
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)), 
                breaks =trans_breaks(log10, function(x) 10^x, 10)) +
  scale_x_discrete(breaks=c(0,2,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=toplot[!is.na(toplot$mean),]) +
  geom_errorbar(aes(ymin = mean-se, ymax = mean+se), width = 1,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(Âµg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("A. baumannii"))) +
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()


```



# Fig. 2


## Fig. 2a - Bliss scores


```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.2a_EDFig.3a.xlsx"))

  hzt_alone = toplot %>% filter(nitroxoline_concentration == 0.00) %>%
    dplyr::select(c("horizontal_drug","horizontal_concentration", "norm_fitness")) %>%
    dplyr::rename("fit_hzt"= "norm_fitness")
  
  vert_alone = toplot %>% filter(horizontal_concentration == 0.00) %>%
    dplyr::select(c("horizontal_drug", "nitroxoline_concentration", "norm_fitness")) %>%
    dplyr::rename("fit_vert"= "norm_fitness")
  
  out = inner_join(hzt_alone, toplot)
  out = inner_join(vert_alone, out)
  
  out$fit_comb = out$norm_fitness
  
  out = mutate(out, exp_fit = fit_vert * fit_hzt,
               bliss = fit_comb - exp_fit)

# remove non-growing space, useless for Bliss calculation
out = out %>% filter(!(horizontal_drug == "Colistin" & horizontal_concentration > 0.2))

#assign interaction sign: synergy if 1st quartile < -0.1, antag if 3rd > 0.1
out = out %>% dplyr::group_by(horizontal_drug) %>%
  dplyr::mutate(eps.med = median(bliss),
         eps.q1 = quantile(bliss, 0.25),
         eps.q3 = quantile(bliss, 0.75)) %>%
     dplyr::mutate(type = ifelse(eps.q1 < -0.1 & eps.med < -0.039, "synergy",
                        ifelse(eps.q3 > 0.1 & eps.med >= 0.03, "antagonism","neutral")),
                  cumulative_eps_quartiles = ifelse(type == "synergy", eps.q1,
                                  ifelse(type == "antagonism", eps.q3, eps.med))) %>%
  select(horizontal_drug,bliss,horizontal_concentration, nitroxoline_concentration,replicate,median_norm_fitness, cumulative_eps_quartiles, eps.med,eps.q1,eps.q3, type) %>%
  distinct()

rm(vert_alone)
rm(hzt_alone)
rm(toplot)


out = out %>%
  mutate(Drugclass= case_when(horizontal_drug %in% c("Fosfomycin", "Aztreonam", "Mecillinam","Temocillin","Piperacillin","Ceftazidime", "Cefepime", "Imipenem", "Meropenem", "Ertapenem", "Vancomycin", "Colistin") ~ "Cell wall and outer membrane",
                              horizontal_drug %in% c("Ciprofloxacin", "Levofloxacin", "Novobiocin","Trimethoprim") ~ "DNA",
                              horizontal_drug %in% c("8âHydroxyquinoline", "Clioquinol", "Thiolutin") ~ "Other metal chelators",
                              horizontal_drug %in% c("Amikacin", "Gentamicin", "Tobramycin","Azithromycin", "Clarithromycin", "Erythromycin","Tetracycline","Minocycline", "Doxycycline", "Tigecycline") ~ "Protein synthesis",
                              TRUE ~ "Other"))
out = out %>%
  mutate(cumulative_eps_quartiles=round(cumulative_eps_quartiles, digits =2))

out = distinct(out)

# plot
colours = data.frame("type" = c("synergy", "antagonism", "neutral"), "values" = c("turquoise4", "#d9a520", "grey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$type

drugclass = "Cell wall and outer membrane"

p1 = ggplot (filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Fosfomycin", "Aztreonam", "Mecillinam","Temocillin","Piperacillin","Ceftazidime", "Cefepime", "Imipenem", "Meropenem", "Ertapenem", "Vancomycin", "Colistin"))),
                                                       x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  labs (x = "Interaction score", y="") +
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y", space = "free") +
  xlim(-1, 1)+
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3,check_overlap = T) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "DNA"

p2 = ggplot(filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Ciprofloxacin", "Levofloxacin", "Novobiocin","Trimethoprim"))), x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  labs (x = "Interaction score", y="") +
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  xlim(-1, 1)+
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3,check_overlap = T) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "Other metal chelators"

p3 = ggplot(filter(out, Drugclass == drugclass), aes(y= horizontal_drug, x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3,check_overlap = T) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")

drugclass = "Protein synthesis"

p4 = ggplot(filter(out, Drugclass == drugclass), aes(y= factor(horizontal_drug, level=rev(c("Amikacin", "Gentamicin", "Tobramycin","Azithromycin", "Clarithromycin", "Erythromycin","Tetracycline","Minocycline", "Doxycycline", "Tigecycline"))), x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3,check_overlap = T) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none")

drugclass = "Other"

p5 = ggplot(filter(out, Drugclass == drugclass), aes(y= horizontal_drug, x = bliss, colour = type)) +
  geom_boxplot(lwd = 0.5) +
  labs (x = "Interaction score", y="") +
  xlim(-1, 1)+
  geom_vline(data = filter(out, Drugclass == drugclass),
              aes(xintercept = 0), linetype = "dashed") +
  facet_grid(~ Drugclass, scales =  "free_y") +
  geom_text(data=filter(out, Drugclass == drugclass),
            aes(y=horizontal_drug, x=0.95, label=cumulative_eps_quartiles), col='#58595b', size=3,check_overlap = T) +
  scale_color_manual(name = "",values = jColors) +
  theme_light() +
  theme(strip.text = element_text(colour = "white",size=10)) +
  guides(alpha="none", colour = "none")



pdf(paste0(plotdir, "Fig2a_Bliss_boxplots_pooled.pdf"), width = 12, height = 10)


layout <-"AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AADD
AACC
AACC
AACC
BBEE
BBEE
BBEE
BBEE
"

wrap_plots(A=p1, B = p2, C= p3, D= p4, E=p5,design = layout, guides = "collect")

dev.off()


```


## Fig. 2b - resensitization of colistin resistant mutants

```{r}
toplot= readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.2b.xlsx")) %>%
  mutate(row= case_when(strain %in% c("E. coli BW25113","E. coli BW25113 pGDP1_mcr1") ~ 1,
                        strain %in% c("E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)") ~ 2,
                        strain %in% c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "E. coli J53 pKP2442_mcr-1") ~ 3,
                        strain %in% c("K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO_mcr-1","K. pneumoniae PRZ pKP2442_mcr-1") ~ 5,
                        strain %in% c("K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)") ~ 6,
                        strain %in% c("K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)") ~ 7,
                        strain %in% c("K. pneumoniae 2442 mcr-1") ~ 8,
                        strain %in% c("K. pneumoniae PRZ pTOPO") ~ 9),
         resist = case_when(strain %in% c("E. coli BW25113","E. coli J53","E. coli MG1655",
                                          "K. pneumoniae CT-S","K. pneumoniae LS107","K. pneumoniae PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))

test = toplot %>%
  #don't include empty vector control in main fig, but in ED fig.
  filter(!(strain %in% c("E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "K. pneumoniae PRZ pTOPO"))) %>%
  mutate(species = case_when(grepl("E. coli",strain) ~ "italic('E. coli')",
                            TRUE ~"italic('K. pneumoniae')")) %>%
  mutate(strain = gsub("E\\. coli |K\\. pneumoniae ", "", strain),
        strain = gsub(strain, pattern  ="pmrA",replacement = "*pmrA*"),
        strain = gsub(strain, pattern  ="pmrB",replacement = "*pmrB*"),
        strain = gsub(strain, pattern  ="mcr-1",replacement = "*mcr-1*"),
        strain = gsub(strain, pattern  ="LS53",replacement = "LS107"),
        strain = gsub(strain, pattern  ="CT-R3",replacement = "LS01"),
        strain = gsub(strain, pattern  ="CT-S",replacement = "LS01")) %>%



p1 <-   ggplot(filter(test, row==2), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
    geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 3,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p2 <- ggplot(filter(test, row==3), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 3,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")

p3 <- ggplot(filter(test, row==5), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 2,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p4 <- ggplot(filter(test, row==6), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 2,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")

p5 <- ggplot(filter(test, row==7), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 1,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(21,22,23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=23),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


p6 <- ggplot(filter(test, row==8), aes(x = as.factor(`Colistin concentration (Âµg/ml)`), y = meanOD, group = interaction(strain, nitroxoline), colour = nitroxoline)) +
   geom_line(linewidth=1,aes(linetype=resistance)) +
    geom_errorbar(aes(ymin = meanOD-sem, ymax = meanOD + sem), width = 0.7,size=1.3, position=position_dodge(0.2)) +
   geom_point(aes(shape=strain, color = nitroxoline),fill="white",stroke=1.5,size = 5.5,position=position_dodge(0.1)) +
  scale_color_manual(labels = c("+ Nitroxoline 0.75 Âµg/ml", "- Nitroxoline"), values = c("turquoise4","darkgrey")) +
  scale_shape_manual(values=c(23), guide =
                         guide_legend(label.theme = element_markdown(), keyheight = 2)) +
  scale_linetype_manual(values=c("solid","longdash")) +
  labs(x="", y = "", colour = "") +
    ylim(min(test$meanOD),1.8) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=60, hjust = 1),
        axis.text=element_text(size=23),
        strip.text=element_text(size=22),
        legend.direction = "vertical",
        legend.position = c(0.36,0.85),
        legend.text=element_text(size=25),
        legend.title=element_blank(),
        legend.spacing = unit(0.4, "cm")) +
  guides(color="none", linetype ="none", fill="none")


pdf(paste0(plotdir, "Fig.2b_colistinR_strains.pdf"),width=28, height=13)

layout <-"CDAB
          EG##"
      
wrap_plots(A = p1,B=p2,C= p3, D=p4, E=p5, G =p6, design = layout)

dev.off()


```



## Fig. 2c - Galleria with Kpn infection

```{r}


toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.2c.xlsx")) %>%
  clean_names() %>%
  dplyr::group_by(condition, time_after_infection_h) %>%
  dplyr::summarise(mean = mean(survival_percent), sem = sd(survival_percent)/sqrt(length(survival_percent))) %>%
  dplyr::ungroup()


  
pdf(paste0(plotdir, "Fig.2c_galleria_Kpn.pdf"),  width=6, height=4)
ggplot(toplot, aes(x = time_after_infection_h, y = mean, color = condition, group = condition))+
  geom_point(size = 3.5) +
  geom_line(alpha=0.8, size =1) +
 scale_color_manual(values = c('#009999','#FF6699','#FF6600', '#666666'),
                       breaks = c("0,1 Âµg/mL nx + 0,1 Âµg/ml CT", "0,1 Âµg/ml CT", "0,1 Âµg/ml Nx", "NaOAc"),
                       labels = c("NX + COL", "COL 0.1 Âµg/ml", "NX 0.1 Âµg/ml", "untreated")) +
  scale_x_continuous(breaks = c(0,24,48,72)) +
  labs(x="Time after infection (h)", y = "% animal survival", color ="") +
  geom_errorbar(aes(ymin = mean-sem, ymax = mean + sem),position = position_dodge(2), width = 15, size = 1) +
  theme_classic() + 
  theme(legend.position = "right",
        axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        legend.text=element_text(size=18))
dev.off()


```


# Fig. 3

### Fig. 3a - TPP - SUPP TABLE 3

See "~/Documents/EMBL/Nx/Nx_paper/REVISION/nitroxoline_2024/2DTPP/" for 2DTPP analysis and FDR calculation.
Protein functional annotations were obtained from Gene Ontology annotations of E. coli proteome (UP000000625), downloaded from Uniprot on the 23.04.2024 and manually curated to integrate missing annotations
To annotate proteins as localised in the OM, experimental data were also used (PMID: 32127419, Methods).


```{r}
fdr_df_whole = fread("~/Documents/EMBL/Nx/Nx_paper/REVISION/nitroxoline_2024/2DTPP/output/fdr_df.txt")
hits_df_whole = fread("~/Documents/EMBL/Nx/Nx_paper/REVISION/nitroxoline_2024/2DTPP/output/hits_df.txt")

# Helper function to extract cluster names
extract_clustername <- function(df, pattern_col, pattern) {
  df %>%
    filter(str_detect(!!sym(pattern_col), pattern)) %>%
    mutate(clustername = sapply(strsplit(as.character(entry_name), "_"), '[', 1)) %>%
    select(clustername)
}

# Read data
sueki_OMPs_msystems <- readxl::read_xlsx(file.path(inputdir, "Sueki_OMPs_msystems.00808-19-st003.xlsx"), sheet = 2) %>%
  filter(`localization (sucrose gradient)` == "OM")

GO_annotation <- fread(paste0(inputdir,"uniprotkb_proteome_UP000000625_2024_04_23.tsv")) %>%
  clean_names()

# Create filtered datasets
om_uniprot <- extract_clustername(GO_annotation, "subcellular_location_cc", "outer membrane") %>%
  bind_rows(extract_clustername(GO_annotation, "gene_ontology_go", "outer membrane"))

zinc <- extract_clustername(GO_annotation, "gene_ontology_go", "zinc")
manganese <- extract_clustername(GO_annotation, "gene_ontology_go", "manganese")
ironsulfur <- extract_clustername(GO_annotation, "gene_ontology_go", "iron-sulfur")
siderophores <- extract_clustername(GO_annotation, "gene_ontology_go", "siderophore")

respiration_patterns <- c("respir", "electron transfer")
respiration <- extract_clustername(GO_annotation, "gene_ontology_go", paste(respiration_patterns, collapse = "|"))

copper <- GO_annotation %>%
  filter(str_detect(gene_ontology_go, "copper") | str_detect(cofactor, "Cu (2+)")) %>%
  mutate(clustername = sapply(strsplit(as.character(entry_name), "_"), '[', 1)) %>%
  select(clustername)


LPS_clusters <- c("LPXA", "LPXB", "LPXC", "LPXD", "LPXH", "LPXK", "LPXL", "LPXM", "LPXT",
                  "GMHA", "WAAA", "WAAU", "LPTA", "LPTB", "LPTC", "LPTD", "LPTE", "LPTF", "LPTG",
                  "WZZE", "WZZB", "GLF", "WBBI", "WBBJ", "WBBK", "NANA", "NANK", "NANT")
efflux_clusters <- c("EMRA", "MPRA", "MARR", "TOLC", "ACRA", "ACRB")

# Annotate fdr_df_whole and hits_df_whole
annotate_data <- function(df) {
  df %>%
    mutate(annot = case_when(
      clustername %in% c("COPA", "CUEO") | clustername %in% copper$clustername ~ "copper",
      clustername %in% zinc$clustername & clustername != "NANK" ~ "zinc",
      clustername %in% ironsulfur$clustername ~ "Fe-S cluster biogenesis",
      clustername != "LPXH" & clustername %in% c("MNTR", "MNTH") | clustername %in% manganese$clustername ~ "manganese",
      clustername %in% LPS_clusters ~ "LPS biosynthesis and assembly",
      clustername %in% efflux_clusters ~ "efflux pumps and regulators",
      clustername %in% om_uniprot$clustername | clustername %in% c("NMPC", "MALB", "BLC") | clustername %in% sueki_OMPs_msystems$gene_name ~ "OMPs",
      clustername %in% siderophores$clustername | clustername %in% c("FEPB","FUR", "ENTA","ENTB","ENTC") ~ "iron import",
      clustername %in% respiration$clustername ~ "respiration",
      TRUE ~ "other"
    )) %>%
    mutate(clustername = case_when(
      clustername == "ZUR" ~ "Zur",
      clustername == "FUR" ~ "Fur",
      clustername == "FES" ~ "Fes",
      TRUE ~ capitalize_first_and_last(clustername)
    ))
}

fdr_df_whole <- annotate_data(fdr_df_whole)
hits_df_whole <- annotate_data(hits_df_whole)


names=c("CopA","CueO","Zur", "ZntA", "ZntR", "ZnuA","ZinT",
        "Fur", "EntA", "EntB","EntC","EntE","EntF","Fiu","Fes","BamA","BamB",
        "IscR", "IscA",
         "SufA","SufB","SufC","SufD","SufS","SufE",
        "FepA", "FepB","FepC","FepD","FepG","FecA",
        "MntH", "MprA",
        "LpxB","LpxD", "LpxH","LpxL","LpxM","NanT","NanA","NanK","Wzz",
        "WaaA", "WaaU", "LptA", "LptB", "LptC", "LptD","LptE", "LptF", "LptG",
        "Glf", "WbbI","WbbJ","WbbK",
        "ZnuA","ZnuB","ZnuC","FbpA","FbpB","FbpC","AcrB")

colours = data.frame("annot" = c("copper", "zinc", "Fe-S cluster biogenesis", 
                                 "iron import","manganese", "OMPs", "LPS biosynthesis and assembly",
                                 "respiration", "efflux pumps and regulators"),
                     "values" = c("aquamarine3","deeppink2","firebrick","brown1","mediumblue","#FF7F00","forestgreen","#984EA3","grey30"))
jColors <- as.character(colours$values)
names(jColors) <- colours$annot


fdr_df_whole = fdr_df_whole %>%
  mutate(efsize = sign(slopeH1)*sqrt(rssH0 - rssH1),
         significance = log2(F_statistic + 1))

hits_df_whole = hits_df_whole %>%
  mutate(efsize = sign(slopeH1)*sqrt(rssH0 - rssH1),
         significance = log2(F_statistic + 1))


p = ggplot(fdr_df_whole %>%
             filter(dataset == "true"),
           aes(sign(slopeH1)*sqrt(rssH0 - rssH1), log2(F_statistic + 1))) +
  geom_point(color = "gray", alpha = 0.5, size = 0.5,
             data=fdr_df_whole %>%
               filter(annot == "other" & !(clustername %in% hits_df_whole$clustername))) + 
  geom_point(aes(color = annot, fill = annot), alpha = 1,
             size = 0.7,
             data = filter(hits_df_whole, annot != "other")) +
  ggrepel::geom_text_repel(
    aes(label = clustername, color=annot),
    data = filter(hits_df_whole, clustername %in% names & annot != "other"),
    size =4, fontface="bold",segment.size = 0.2,
    max.overlaps = 18, show.legend = F) +
  scale_color_manual(values=jColors) +
  geom_hline(yintercept = log2(1.551792 + 1), linetype = 'dotted') +
  facet_wrap(~ detected_effectH1, scales="free_x") +
  labs(x = bquote(sign(kappa) %.% sqrt(~'RSS'^0~' - RSS'^1~'')),
       y = expression('log'[2]~'('*italic(F)*'-statistic + 1)'),
       colour = "") +
  theme_bw() +
  theme(legend.position = "right",
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=12),
        strip.text = element_blank()) +
  guides(fill ="none", color="none")


ggsave(p, filename=paste0(plotdir, "Fig.3a_2DTPP.png"), width = 8.6, height =3.6)


fdr_df_whole = fdr_df_whole %>%
  mutate(efsize= sign(slopeH1)*sqrt(rssH0 - rssH1),
         signif = log2(F_statistic + 1))

hits_df_whole = hits_df_whole %>%
  mutate(efsize= sign(slopeH1)*sqrt(rssH0 - rssH1),
         signif = log2(F_statistic + 1))



towrite = fdr_df_whole %>%
  mutate(clustername = case_when(clustername %in% "ZUR" ~ "Zur",
                                clustername %in% "FUR" ~ "Fur",
                                TRUE ~ capitalize_first_and_last(clustername))) %>%
  mutate(hit= case_when(clustername %in% unique(filter(hits_df_whole, FDR <= 0.05)$clustername) ~ "TRUE",
                        TRUE ~ "FALSE"),
         source = "whole_cell",
         score = sign(slopeH1)*sqrt(rssH0 - rssH1)) %>%
  distinct(clustername, slopeH1, rssH0,rssH1, score, hit, source, detected_effectH1, FDR,annot, F_statistic) 

write_xlsx(towrite,paste0(supptablesdir,"Supplementary_Table_3.xlsx"))

```



## Fig. 3b - chemgen - SUPP TABLE 4

```{r}

hits = readxl::read_xlsx(paste0(sourcedatadir, "source_Fig.3b.xlsx"))


colours = data.frame("annot" = c("copper", "zinc", "Fe-S cluster biogenesis", 
                                 "iron import","manganese", "OMPs", "LPS biosynthesis and assembly",
                                 "respiration", "efflux pumps and regulators"),
                     "values" = c("aquamarine3","deeppink2","firebrick","brown1","mediumblue","#FF7F00","forestgreen","#984EA3","grey30"))
jColors <- as.character(colours$values)
names(jColors) <- colours$annot



names=c("COPA","CUEO","CUER","ZUR", "ZNTR", "ZNUA","ZINT","ZNUB",
       "ISCU", "ISCA",
        "SUFA","SUFC","SUFD","SUFS","SUFE","NFUA","GRXD",
        "FEPB","FEPC","FEPD","FEPG","EXBB",
         "ISCR", "SUFD",
        "NRDE","NRDF", 
       "LPXA","LPXB","LPXD","LPXC",
       "GMHA", "WAAA", "WAAC","WAAU","WAAF","WAAG","WAAQ","WAAB","GMHB",
       "LPTA", "LPTD", "LPTC", "LPTB",
       "WZZE",
       "EMRA","EMRB","EMRR","ACRA",
         "MNTR",
       "ZNUA","ZNUC","FBPA","FBPB","FBPC")

# Capitalize the first letter and lowercase the rest for gene names
capitalize <- function(string) {
 paste0(tolower(substring(string, 1, nchar(string) - 1)), toupper(substring(string, nchar(string), nchar(string)))) 
}

names <- capitalize(names)
names[names == "zuR"] <- "zur"
names[names == "fuR"] <- "fur"
names[names == "fiU"] <- "fiu"
names[names == "feS"] <- "fes"


p =  ggplot(hits,  aes(x = log2fc, y = -log10(padj))) +
  geom_point(col='grey70', size=0.5, show.legend=F) +
  geom_point(data=hits %>% filter(padj <= 0.05 & annot != "other"), aes(col=annot,fill=annot),size=0.7, show.legend=T) +
geom_abline(intercept = -log10(0.05), slope = 0, linetype = "dashed") +
  facet_wrap(~cond, ncol=1) +
  ggrepel::geom_text_repel(aes(label = gene, color=annot), force = 0.5,max.overlaps = 20,min.segment.length = 0.2, size = 4, segment.size = 0.2,data = hits %>% filter(gene %in% names & padj <= 0.05),fontface = "bold.italic") +
   scale_color_manual(values=jColors) +
  scale_fill_manual(values=jColors) +
  ylab(expression('-log'[10]~'(adjusted p-val)')) +
  xlab(expression('log'[2]~'(FC mutant fitness)')) +
  theme_bw() +
  theme(strip.text = element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=12)) +
  guides(colour="none",
         fill="none")


ggsave(p, filename=paste0(plotdir, "Fig.3b_Keio_volcano.png"), width = 5, height =4.2)


towrite=hits %>%
  select(-cond) %>%
  write_xlsx(paste0(supptablesdir,"Supplementary_Table_4.xlsx"))

```


## Fig. 3c - NPN

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.3c.xlsx")) %>%
  mutate(Condition = interaction(Condition, Concentration, sep = "_")) %>%
  mutate(Condition = sapply(strsplit(as.character(Condition),"\\ "), '[',1))


pdf(paste0(plotdir, "Fig.3c_NPN.pdf"), width = 5.5, height = 5)

colours = data.frame("Condition" = c("PolymyxinB_10","EDTA_0.4", "Nitroxoline_0.9", "Nitroxoline_1.8", "Nitroxoline_0.45", "Chloramphenicol_8", "nodrug_0"),
                     "values" = c("#999999","#999999", "#999999","#999999","#999999", "#999999", "#999999"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Condition


 ggerrorplot(toplot, x = "Condition", y = "NPN_fluorescence", error.plot="errorbar",
          order = c("PolymyxinB_10","EDTA_0.4", "Nitroxoline_0.9", "Nitroxoline_1.8", "Nitroxoline_0.45",  "Chloramphenicol_8", "nodrug_0"),
  add = "se") +
   stat_summary(
    geom = "point",
    shape = 95,
    size = 12,
    col = "black",
    fun.y = "mean") +
  geom_point(aes(fill = Condition),alpha = 0.7,size =3,pch=21,position = position_jitterdodge(1, dodge.width = .5)) +
  scale_fill_manual(values=jColors) +
  labs(fill = "", y = "NPN fluorescence (405 nm)") +
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = "Chloramphenicol_8", size = 7) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         legend.position = "none")

dev.off()


```



# Fig. 4

## Fig. 4a - copper TPP

```{r}
toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx"))

a <- plot2dTppFcHeatmap(
  df = toplot, name = "CopA",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = toplot, name = "CueO",
  drug_name = "Nitroxoline")


pdf(paste0(plotdir, "Fig.4a_thermal_profiles_copper.pdf"), width = 5.8, height = 2)

cowplot::plot_grid(a, b, ncol = 2)

dev.off()

```

## Fig. 4b - zinc TPP

```{r}

toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx"))

a <- plot2dTppFcHeatmap(
  df = toplot, name = "Zur",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = toplot, name = "ZnuA",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = toplot, name = "ZntR",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = toplot, name = "ZntA",
  drug_name = "Nitroxoline")



pdf(paste0(plotdir, "Fig.4b_thermal_profile_zinc.pdf"), width = 5.8, height = 4)

cowplot::plot_grid(a, b,c,d, ncol = 2)

dev.off()

```


## Fig. 4c - manganese TPP

```{r}
toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx"))

a <- plot2dTppFcHeatmap(
  df = toplot, name = "MntH",
  drug_name = "Nitroxoline")


pdf(paste0(plotdir, "Fig.4c_thermal_profile_manganese.pdf"), width = 2.9, height = 2)
a
dev.off()

```


## Fig. 4d - ESRF

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4d.xlsx"))

pdf(paste0(plotdir,"Fig.4d_ESRF.pdf"),width=3.6, height=3.4)

colours = data.frame("Condition" = c("untreated","+ 1 Âµg/ml NX"),
                     "values" = c("black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Condition

ggplot(toplot, aes(x = Element, y = mean_area_density, color = Condition)) +
  geom_bar(stat = "identity", position = position_dodge(0.8), width = 0.7,fill="white") +
   geom_errorbar(aes(ymin = mean_area_density - se_area_density, 
                     ymax = mean_area_density + se_area_density),
                 position = position_dodge(0.8), width = 0.15) +
  geom_jitter(aes(x = Element,y=area_density,fill = Condition), position = position_jitterdodge(dodge.width = 0.8, jitter.width = 0.1,jitter.height = 0.01), size = 1.2, shape = 21, alpha = 0.7) +
  scale_color_manual(values = jColors) +
   scale_fill_manual(values = jColors) +
  stat_pvalue_manual(toplot %>% 
                       distinct(Element,y.position,n1,n2,group1, group2,groups,x,xmin,xmax,
                                statistic, p, p.signif),
                     tip.length = 0.01, label = "p.signif", y.position = 2.2) +
  labs(color = "", y = expression('Elemental area density (ng/cm'^2~')'), fill="") + 
   guides(color="none") +
   theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_text(size=9),
        legend.position = "top")
 
 
dev.off()

```


# Fig. 5

## Fig. 5a - WGS heatmap

```{r}
toplot= readxl::read_xlsx(paste0(sourcedatadir, "source_Fig.5a.xlsx"))

strain_order = c("1_R1","1_R2","1_R3", "1_R4","1_R5", "1_R6","2_R1","2_R2","2_R3","2_R4","3_R1","3_R2", "4_R1","5_R1","6_R1","7_R1","8_R1")

toplot$isolate <- factor(toplot$isolate, levels=strain_order)

pdf(paste0(plotdir,"Fig.5a_WGS_hm.pdf"),width =7, height = 7)

toplot$effect <- factor(toplot$effect, levels=c("conservative inframe deletion", "disruptive inframe insertion", "frameshift variant", "loss of locus", "missense variant","partial loss", "stop gained", "stop lost&splice region variant","TN insertion", "not mutated"))

colours = data.frame("effect" = c("conservative inframe deletion", "disruptive inframe insertion", "frameshift variant", "loss of locus", "missense variant","not mutated",
                                "partial loss", "stop gained", "TN insertion","stop lost&splice region variant"),
                     "values" = c("#66C2A5", "#DA9870", "#BE979C", "#AB98C8","#B3B3B3","white", "#E1D83B", "#F3CF5B", "#D9C09A", "#DF92B6"))
mycolors <- as.character(colours$values)
names(mycolors) <- colours$effect

ggplot(toplot,aes(x=isolate,y=gene,fill=as.factor(effect))) +
  scale_fill_manual(values=mycolors, name ="") +
  geom_tile(color="grey40") +
  facet_grid(~ species,space="free_x",scales="free_x") +
  labs(x="Strain",y="Gene") +
  theme(axis.text.x=element_text(angle=60, hjust=1),
        axis.text.y=element_text(face="italic"),
        strip.text = element_text(face="italic"),
        legend.text = element_text(size=7))

dev.off()


pdf(paste0(plotdir,"Fig.5a_MICs.pdf"), width=10, height =5)
ggbarplot(toplot %>%
          distinct(isolate,nx_mic, species), x= "isolate", y="nx_mic", color="species",
          order = c("1_R1","1_R2","1_R3", "1_R4","1_R5","1_R6",
                    "2_R1","2_R2","2_R3","2_R4",
                    "3_R1","3_R2", "4_R1","5_R1",
                    "6_R1","7_R1","8_R1")) + 
  scale_y_continuous(trans="log2",breaks=c(2,4,8,16,32,64,128)) +
  scale_color_manual(values=c("#1E88E5","#FFC107","#D81B60")) +
  facet_grid(~ species, scales="free_x", space="free") +
  labs(y="Nitroxoline MIC (Âµg/ml)") +
  theme(axis.text.x= element_text(angle = 90))
 dev.off() 


```


## Fig. 5b - Proteomics of R strains

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5b_EDFig.5b.xlsx"))

genes = c("TolC", "OqxB3", "OprC","EmrR",
          "AcrA", "AcrB_1", "MdtK","MarA",
          "LamB","OmpC", "OmpD", "OmpF","BepA",
          "BpeF/AdeG/OqxB" ,"BpeE/AdeF/OqxA/BepFF","OmpR",
          "OprC","OprD","OmpW","EmrR","EmrA", "AcrB",
          "EntC","OmpD","LamB","OmpC","OmpN","OmpF","OmpW",
          "OprD","OprM","MdtK","MarA","YedS", "ScrY")

strains = c("A. baumannii 2_R1","A. baumannii 4_R1",
            "E. coli 2_R1","E. coli 2_R3","E. coli 2_R4",
            "K. pneumoniae 10_R1","K. pneumoniae 2_R1","K. pneumoniae 2_R2",
            "K. pneumoniae 3_R1", "K. pneumoniae 5_R1", "K. pneumoniae 9_R1", "K. pneumoniae 9_R2")


toplot = toplot %>% filter(og %in% genes & Strain_treatment %in% strains)


mymat = toplot %>% 
  distinct(og, Strain_treatment, logFC)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}


library(magrittr)

mat_data = mymat %>% 
  pivot_wider(names_from = Strain_treatment, values_from = logFC, values_fn = median) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og) %>%
  dplyr::select(-og) %>%
  as.matrix()

mymat |>
  dplyr::summarise(n = dplyr::n(), .by = c(og, Strain_treatment)) |>
  dplyr::filter(n > 1L) 


#  dendrogram 
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)


type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(Strain_treatment = type) %>%
  inner_join(toplot %>% select(Strain_treatment,species), by="Strain_treatment") %>%
    distinct() %>%
    select(-Strain_treatment),
   annotation_height = unit(4, "mm"),
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )


mymat2 = toplot %>% 
  distinct(og, Strain_treatment, hit)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}


library(magrittr)

significance_data = mymat2 %>% pivot_wider(names_from = Strain_treatment, values_from = hit, values_fn = unique) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og) %>%
  dplyr::select(-og) %>%
  as.matrix()



# add annotation according to significance
star_annotation = function(j, i, x, y, width, height, fill) {
    if (significance_data[i, j]) {
        grid.text("*", x = x, y = y, gp = gpar(fontsize = 10, col = "black"))
    }
}

pdf(paste0(plotdir,"Fig.5b_hm_proteomics.pdf"), width=7, height=7)

hm<-Heatmap(mat_data,
        cluster_rows = dend1,
         top_annotation = ha,
        cell_fun = star_annotation,
         row_names_side = "left",
        row_split = 4, column_split = 6,
        use_raster = TRUE, raster_quality = 5)

draw(hm, heatmap_legend_side="right", annotation_legend_side="right",row_dend_side = "right",
           legend_grouping="adjusted")

dev.off()


```

## Fig. 5c - OqxR and EmrR complementation

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5c.xlsx"))

stat.test <- toplot %>%
  dplyr::group_by(strain_condition) %>%
  wilcox_test(`MIC (Âµg/ml)` ~ condition, p.adjust.method = "BH") %>%
  adjust_pvalue(method = "BH") %>%
  add_significance("p") %>%
  add_xy_position(fun = "mean_sd", x = "condition",dodge = 0, group = "strain_condition") 



pdf(paste0(plotdir,"Fig.5c_OqxR_emrR_complementation.pdf"),width = 3.9, height=3)

# color according to species
strip <- strip_themed(background_x = elem_list_rect(fill = c("#fdc113","#d81a61")))

#reorder bars
toplot$condition <- factor(toplot$condition, levels = c("no plasmid","+ empty plasmid","+ wt efflux\npump"))

# code position of significance stars in plot
stat.test = stat.test %>%
  mutate(xmin=case_when(group1 == "+ empty plasmid" & group2== "+ wt efflux\npump" ~ 2,
                        group1 == "+ empty plasmid" & group2== "no plasmid" ~ 1,
                        TRUE ~ 1),
         xmax=case_when(group1 == "+ empty plasmid" & group2== "+ wt efflux\npump" ~ 3,
                        group1 == "+ empty plasmid" & group2== "no plasmid" ~ 2,
                        TRUE ~ 3))

 ggplot(towrite, aes(x = condition, y = mean_mic, color = condition)) +
  geom_bar(aes(fill = condition),stat = "identity", position = position_dodge(0.8), width = 0.7)+
   geom_point(aes(x = condition,y=`MIC (Âµg/ml)`), pch=21,fill='grey90', size=2, show.legend=F,position = position_jitterdodge(dodge.width = 0.6, jitter.width = 0.9,jitter.height = 0.4)) +
   geom_errorbar(aes(ymin = mean_mic - se_mic, ymax = mean_mic + se_mic), position = position_dodge(0.8), width = 0.15) +
   scale_fill_grey() +
   scale_color_grey() +
  stat_pvalue_manual(stat.test, tip.length = 0.01,bracket.nudge.y = 0.5,
              label = "p.adj.signif", y.position = 6, step.increase = 0.1, step.group.by = "strain_condition",
              size = 3.5) +
  scale_y_continuous(breaks = c(0,2,4,6,8,10), labels= c(0,4,16,64,256,1024)) +
  facet_wrap2(~strain_condition,strip = strip) +
  labs(color = "", y = "MIC (Âµg/ml)", fill="") + 
   guides(color="none") +
   theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.position = "bottom",
        strip.text = element_text(face="bold", color="white"))
dev.off()


```


## Fig. 5d - OqxR mutations, MIC and isolate type

```{r}

# get domain annotations

## used NsrR as closest structural homolog of OqxR which doesnt have a uniprot ID (see Methods)
proteins_acc_url <- "Q8ZKA3"
baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
url<-paste0(baseurl,proteins_acc_url)
prots_feat<-GET(url,accept_json())
prots_feat_red<-httr::content(prots_feat)
features_total_plot<-NULL
for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])
      features_total_plot<-rbind(features_total_plot,features_temp)
      }
  

plot_start<-0 #starts 0
plot_end<-max(features_total_plot$end)   


# read data with MIC and isolate type info
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5d.xlsx"))

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type
    

pdf(paste0(plotdir,"Fig.5d_OqxR_mutplot.pdf"), width=4.5, height=4)


ggplot(toplot,aes(location_aa_number,`MIC (Âµg/ml)`,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location_aa_number,yend=`MIC (Âµg/ml)`)) +
  geom_point(aes(size=Mutation_occurrences, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=toplot %>%
                    distinct(location_aa_number,`MIC (Âµg/ml)`, Protein_Change,Isolate_type,Mutation_occurrences),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=150) +
  geom_hline(yintercept = 0) +
  geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=8,ymax=10),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 65, y = 9, label ="HTH rrf2-type", size = 4,color="white",fontface="bold") +
  geom_rect(data=features_total_plot[features_total_plot$type=="DNA_BIND",],
            mapping=aes(xmin=begin,xmax=end,ymin=10,ymax=12),color="darkblue",fill="darkblue",inherit.aes=F, alpha=0.5, ) +
  annotate("text", x = 39, y = 11, label ="HTH motif", size = 4,color="white",fontface="bold") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_shape_manual(values=c("missense" =16, "nonsense"  = 15, "frameshift/insertion" =17, "insertion IS1S"  = 8), name = "Mutation type") +
  scale_size_continuous(breaks = c(1,2,3), limits=c(1,3)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,159), limits=c(0,159))+
  scale_y_continuous(breaks = c(8,16,32,64,128),trans="log2",limits=c(8,150)) +
  labs(y= "MIC (Âµg/mL)",size="Mutation occurrencies", title = "OqxR", shape = "Mutation type")+
  theme_bw() +
  theme(plot.title=element_blank(),
       legend.position = "top",
       legend.direction="vertical",
        axis.text = element_text(size=10),
        legend.title = element_text(size=9, face="bold"),
        legend.text = element_text(size=8),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.05,0.05,0.05,0.05, unit="cm"),
        panel.border = element_blank(), axis.line = element_line()) +
  guides(size=guide_legend(direction='horizontal', title.position="top", title.hjust = 0.5))


dev.off()

```



## Fig. 5e - efflux pump inhibitors

All strains are in the new file, except for 2390, which is important bc wt of MS274.

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5e_EDFig.5g.xlsx")) %>%
  mutate(newspec = paste0("italic('", species, " ')~'", isolate, "'"),
         facet = factor(newspec, levels = c("italic('A. baumannii ')~'2'","italic('A. baumannii ')~'2_R1'",
                                            "italic('A. baumannii ')~'4'","italic('A. baumannii ')~'4_R1'",
                                            "italic('E. coli ')~'2'", "italic('E. coli ')~'2_R1'","italic('E. coli ')~'2_R3'",
                                            "italic('K. pneumoniae ')~'2'", "italic('K. pneumoniae ')~'2_R2'",
                                            "italic('K. pneumoniae ')~'5'", "italic('K. pneumoniae ')~'5_R1'")),
         row= case_when(species == "A. baumannii" ~ 1,
                        species == "K. pneumoniae" ~ 2,
                        species == "E. coli" ~ 3),
         resist = case_when(isolate %in% c("1","2","4","5") ~ "sensitive",
                             TRUE ~ "resistant")) %>%
  # only plot selected strains - rest in ED Fig
  drop_na(facet)



p <- ggerrorplot(toplot %>% filter(row==1),
                 x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
                 order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"), add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun = "mean") +
  facet_wrap(~ facet,nrow=1,labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),
            aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2", limits = c(0.25,32), breaks = c(0.25,1,4,16)) +
  labs(color = "", y = "MIC (Âµg/ml)", x= "") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="right",
         strip.background =element_rect(fill="#3e83c4"),
         strip.text=element_text(face="bold",colour="white"))

legend_grob <-cowplot::get_legend(p)

p <- ggerrorplot(toplot %>% filter(row==1),
                 x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
                 order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"), add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun = "mean") +
  facet_wrap(~ facet,nrow=1,labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),
            aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2", limits = c(0.25,32), breaks = c(0.25,1,4,16)) +
  labs(color = "", y = "MIC (Âµg/ml)", x= "") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#3e83c4"),
         strip.text=element_text(face="bold",colour="white"))



p2 <- ggerrorplot(toplot %>% filter(row==2),
                 x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
                 order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"), add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun = "mean") +
  facet_wrap(~ facet,nrow=1,labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==2 & resist=="resistant"),
            aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2", limits = c(0.25,32), breaks = c(0.25,1,4,16)) +
  labs(color = "", y = "MIC (Âµg/ml)", x= "") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#d81a61"),
         strip.text=element_text(face="bold",colour="white"))

p3 <- ggerrorplot(toplot %>% filter(row==3),
                 x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
                 order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"), add = "se") +
   stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun = "mean") +
  facet_wrap(~ facet,nrow=1,labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==3 & resist=="resistant"),
            aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2", limits = c(0.25,32), breaks = c(0.25,1,4,16)) +
  labs(color = "", y = "MIC (Âµg/ml)", x= "") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#fdc113"),
         strip.text=element_text(face="bold",colour="white"))


library("latex2exp")


cairo_pdf(paste0(plotdir,"Fig.5g_EPI.pdf"), width =7,height=6)
 
layout <-"AAAA
          BBBB
          CCCD
          "
wrap_plots(A=p, B=p2,C=p3,D=legend_grob, design = layout)

dev.off()


```


# EXTENDED DATA FIGURES

## ED Fig. 1
### ED1a - Disk diffusion

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1a.xlsx")) %>%
  mutate(Species_italics = case_when(Species == "Salmonella spp." ~ "*Salmonella* spp.",
                                     Species == "Shigella spp." ~ "*Shigella* spp.",
                                    TRUE ~ paste0("*", Species, "* ")),
         MIC=as.numeric(MIC),
         Species_label = as.character(paste0(Species_italics, "(n=", tot_number_of_strains, ")"))) 

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                   "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
               "Psychrobacter spp.",
                 "Hafnia alvei")

#create order with the n annotation
legend = toplot %>% 
  distinct(Species, Species_label) %>%
  arrange(factor(Species, levels=species_order))


toplot = toplot %>%
  arrange(factor(Species, levels=species_order)) %>%
  distinct(Species_label,MIC,number_strains_at_MIC) %>%
  complete(Species_label, MIC, fill = list(number_of_strains = 0))


toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label))  


toplot$number_strains_at_MIC[is.na(toplot$number_strains_at_MIC)] <- 0

pdf(paste0(plotdir,"EDFig.1a_Disk_diffusion.pdf"), width=9, height=6)
ggplot(toplot,aes (x = as.factor(MIC), Species_label)) +
    geom_tile(aes(fill = number_strains_at_MIC, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=number_strains_at_MIC),size=3.3) +
   scale_fill_gradient(name = "number of\nstrains",low="white", high="darkgreen", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  labs(y="", x = "Zone of inhibition (mm)") +
   theme_classic() +
  theme(axis.text.y = element_markdown(color = "black",size=9),
        legend.position = "top")

dev.off()

```


### ED1b - Agar dilution

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1b.xlsx")) %>%
  mutate(Species_italics = case_when(Species == "Salmonella spp." ~ "*Salmonella* spp.",
                                     Species == "Shigella spp." ~ "*Shigella* spp.",
                                    TRUE ~ paste0("*", Species, "* ")),
         MIC=as.numeric(MIC),
         Species_label = as.character(paste0(Species_italics, "(n=", tot_number_of_strains, ")"))) 

species_order= c(  "Acinetobacter pittii",
                   "Acinetobacter baumannii",
                   "Acinetobacter junii",
                   "Acinetobacter ursingii",
                   "Acinetobacter lwoffii",
                   "Acinetobacter johnsonii",
                  "Moraxella catarrhalis",
                  "Moraxella osloensis",
                  "Psychrobacter spp.",
                  "Pseudomonas aeruginosa",
                "Pseudomonas putida",
                "Stenotrophomonas maltophilia",
                 "Achromobacter xylosoxidans",
               "Burkholderia cenocepacia complex",
                "Neisseria gonorrhoeae",
                 "Escherichia coli",
                "Shigella spp.",
                 "Salmonella spp.",
                "Citrobacter koseri",
                "Citrobacter freundii",
                "Klebsiella pneumoniae",
                "Klebsiella aerogenes",
                "Klebsiella oxytoca",
                "Enterobacter cloacae",
                "Proteus vulgaris",
                 "Proteus mirabilis",
                "Morganella morganii",
                  "Serratia marcescens",
                 "Hafnia alvei")


#create order with the n annotation
legend = toplot %>% 
  distinct(Species, Species_label) %>%
  arrange(factor(Species, levels=species_order))


toplot = toplot %>%
  arrange(factor(Species, levels=species_order)) %>%
  distinct(Species_label,MIC,number_strains_at_MIC) %>%
  complete(Species_label, MIC, fill = list(number_of_strains = 0))


toplot$Species_label <- factor(toplot$Species_label, levels = rev(legend$Species_label)) 

toplot$number_strains_at_MIC[is.na(toplot$number_strains_at_MIC)] <- 0


pdf(paste0(plotdir,"EDFig.1b_agar_microdilutions.pdf"), width=4, height=6)
ggplot(toplot,aes (x = as.factor(MIC), Species_label)) +
    geom_tile(aes(fill = number_strains_at_MIC, color="black"), colour = "black", size = 0.25) +   
    geom_text(aes(label=number_strains_at_MIC),size=3.3) +
   scale_fill_gradient(name = "number of\nstrains",low="white", high="grey39", trans=scales::pseudo_log_trans(base = 10), breaks=c(0,75)) +
  labs(y="", x = "MIC (Âµg/ml)") +
   theme_classic() +
  theme(axis.text.y = element_markdown(color = "black",size=9),
        legend.position="top")
dev.off()
```



### ED1c - EUCAST and us

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1c.xlsx")) %>%
  mutate(newspec = paste0("italic('", toplot$Species, "')"))


pdf(paste0(plotdir,"EDFig.1c_eucast_broth_concordance.pdf"),width = 4, height =4)
ggscatter(toplot,x="MIC_this_study",y="MIC_EUCAST", add = "reg.line", add.params = list(color = "blue", fill = "lightgray"),
          conf.int = TRUE,
   cor.coef = TRUE,
   cor.coeff.args = list(method = "pearson", label.x = 2, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2") +
  scale_y_continuous(trans="log2") +
  labs(x="This study: log2[MIC (Âµg/ml)]", y = "EUCAST: log2[MIC (Âµg/ml)]") +
  geom_text_repel(aes(label = newspec),parse=TRUE, max.overlaps = 20) 
dev.off()
```



### ED1d - correlation between disk and agar

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1d.xlsx")) %>%
  mutate(newspec = paste0("italic('", Species, "')"))


pdf(paste0(plotdir, "EDFig.1d_corr_disk_agar.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot, "MIC_agar", "MIC_disk", color = "n_strains_agar", size = "n_strains_disk", alpha = 0.8,
          add = "reg.line",
          add.params = list(color = "blue", fill = "lightgray"),
         conf.int = TRUE,
         cor.coef = TRUE,
         cor.coeff.args = list(method = "pearson", label.x = 4, label.sep = "\n"),
         cor.coef.size = 4) +
  scale_x_continuous(trans="log2",breaks = c(2,4,8,16,32)) +
  scale_y_continuous(trans="log2", n.breaks=4) +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 20) +
  labs(size = "strains tested (disk diffusion)", color = "strains tested (agar dilutions)",
       x = "MIC agar (Âµg/ml)", y = "Zone of inhibition (mm)") +
 scale_color_gradientn(colors = c("#064c4c","#ce9000"),
                       breaks = c(50,150,250),
                      guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
  scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(100,150,500)) +
  guides(alpha ='none')

dev.off()

```


### ED1e - correlation broth and agar

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1e.xlsx")) %>%
  mutate(newspec = paste0("italic('", Species, "')"))


pdf(paste0(plotdir, "EDFig.1e_corr_broth_agar.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot, "MIC_agar", "MIC_broth", color = "n_strains_agar", size = "n_strains_broth", alpha = 0.8,
           add = "reg.line",
  add.params = list(color = "blue", fill = "lightgray"),
  conf.int = TRUE,
   cor.coef = TRUE,
   cor.coeff.args = list(method = "pearson", label.x = 0.5, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2", breaks = c(2,4,8,16,32)) +
  scale_y_continuous(trans="log2", breaks = c(2,4,8,16,32)) +
  labs(size = "strains tested (broth microdilution)", color = "strains tested (agar dilutions)",
       x = "MIC agar (Âµg/ml)", y = "MIC broth (Âµg/ml)") +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 30) +
 scale_color_gradientn(colors = c("#064c4c","#ce9000"),
                      breaks = c(50,150,250),
                      guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
  scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(50,100,150,250)) +
  guides(alpha ='none')

dev.off()


```


### ED1f - correlation between disk and broth

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.1f.xlsx")) %>%
  mutate(newspec = paste0("italic('", Species, "')"))


pdf(paste0(plotdir, "EDFig,1f_corr_disk_broth.pdf"), width = 5.5, height = 5.5)
ggscatter(toplot %>% drop_na(), "MIC_broth", "MIC_disk", color = "n_strains_broth", size = "n_strains_disk", alpha = 0.8,
           add = "reg.line",  
  add.params = list(color = "blue", fill = "lightgray"), 
  conf.int = TRUE,
   cor.coef = TRUE, 
   cor.coeff.args = list(method = "pearson", label.x = 4,label.y=5, label.sep = "\n"),
   cor.coef.size = 4) +
  scale_x_continuous(trans="log2",breaks = c(2,4,8,16,32)) +
  scale_y_continuous(trans="log2",n.breaks=4) +
  geom_text_repel(aes(label = newspec), parse = TRUE, max.overlaps = 20) +
 labs(size = "strains tested (disk diffusion)", color = "strains tested (broth dilutions)",
      x = "MIC broth (Âµg/ml)", y = "Zone of inhibition (mm)") +
scale_color_gradientn(colors = c("#064c4c","#ce9000"),
                      breaks = c(50,150,250),
                     guide = guide_legend(direction = "horizontal", title.position = "top", title.hjust = 0.5,alpha = "none")) +
 scale_size(guide = guide_legend(direction = "horizontal",title.position = "top",title.hjust = 0.1),breaks=c(100,150,500)) +
 guides(alpha ='none')

dev.off()


```


## ED Fig. 2


### ED Fig. 2a - Abau killing w/ control

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.1e_EDFig.2b.xlsx"))


pdf(paste0(plotdir,"EDFig.2b_Abau_killing_control.pdf"), width = 5.8, height=5)

my_palette = brewer.pal("Oranges",n = 9)[3:9]

toplot = toplot %>% clean_names()

ggplot(toplot, aes(x=as.factor(time_h),y=mean, group =nitroxoline_concentration_mg_ml, color = nitroxoline_concentration_mg_ml)) + 
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)), 
                breaks =trans_breaks(log10, function(x) 10^x, 10)) +
  scale_x_discrete(breaks=c(0,2,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=toplot[!is.na(toplot$mean),]) +
  geom_errorbar(aes(ymin = mean-se, ymax = mean+se), width = 1,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(Âµg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("A. baumannii"))) +
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()

```

### ED Fig. 2b - E. coli killing
```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.2a.xlsx"))


pdf(paste0(plotdir,"EDFig.2b_Ecoli_killing.pdf"), width = 5.8, height=5)

toplot$`Nitroxoline concentration (Âµg/ml)` <- factor(toplot$`Nitroxoline concentration (Âµg/ml)`, levels=c('0', '2', '4', '8', '16', '32','64'))

my_palette = brewer.pal("Oranges",n = 9)[3:9]

ggplot(toplot, aes(as.factor(`Time (h)`),mean, group = `Nitroxoline concentration (Âµg/ml)`, color = `Nitroxoline concentration (Âµg/ml)`)) +
  geom_point(size=0.8) +
  scale_y_log10(labels= trans_format(log10, math_format(10^.x)),
                breaks =trans_breaks(log10, function(x) 10^x, 10)) +
  scale_x_discrete(breaks=c(0,4,8,24,30)) +
  scale_colour_manual(values=my_palette) +
  geom_line(data=toplot[!is.na(toplot$mean),]) +
  geom_errorbar(aes(ymin = mean-se, ymax = mean+se), width = 0.8,position=position_dodge(0.1)) +
  labs(color = "Nitroxoline\n(Âµg/ml)", x="Time (h)", y= "CFU/ml", title=expression(italic("E. coli"))) +
  theme_bw() +
  theme(axis.text=element_text(size = 15),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13),
        plot.title =element_text(size = 15, hjust=0.5))

dev.off()

```


## ED Fig. 3
### ED3a - all checkerboards

```{r}
allCBs = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.2a_EDFig.3a.xlsx")) %>%
  group_by(horizontal_concentration, nitroxoline_concentration, horizontal_drug) %>%
  mutate(median_norm_fitness = median(norm_fitness))

antags= c("Aztreonam","Mecillinam","Temocillin", "Cefepime","Imipenem", "Meropenem","Ertapenem","Thiolutin", "Amikacin","Gentamicin", "Tobramycin", "Trimethoprim", "Tetracycline")
synerg = c("Vancomycin", "Colistin", "Azithromycin", "Novobiocin", "Rifampicin")
neutrals = setdiff(unique(allCBs$horizontal_drug), c(antags,synerg))


p <- ggplot(allCBs %>% filter(horizontal_drug %in% antags), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "#d9a520"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="", y = "", fill="Normalised\nfitness") +
    theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

p1 <- ggplot(allCBs %>% filter(horizontal_drug %in% synerg), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "turquoise4"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="", y = "", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

p2 <- ggplot(allCBs %>% filter(horizontal_drug %in% neutrals), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = median_norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "grey"),
                     values = scales::rescale(c(0,0.1, max(allCBs$median_norm_fitness)))) +
    theme_classic() + 
	facet_wrap(~ horizontal_drug, scales = "free", ncol = 5) +
  labs(x ="Combined drug (Âµg/ml)", y = "", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="none")

pdf(paste0(plotdir, "EDFig.3a_CB_heatmaps_hits.pdf"),  width=10, height=15)
layout <-"BBBBB
          AAAAA
          AAAAA
          AAAAA
          AAAAA
          AAAAA
          CCCCC
          CCCCC
          CCCCC
          CCCCC
          CCCCC"

wrap_plots(A=p,B=p1, C= p2,design = layout,guides="collect")


dev.off()


```


##ED Fig.4 

### ED Fig. 4a - GO TPP

cutoffs to define hits (foreground for enrichment):
- TPP hits at FDR < 0.01 and efsize (as in volcano plot in fig. 3a) > |1|
- enrichment cutoff as padj (Fisher's test, BH correction) < 0.05

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_ED_Fig.4a.xlsx")) %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_abund_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]


p1 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_abund_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased abundance") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_stab_up") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]


p3 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Increased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) +
  guides(alpha="none", colour = "none", fill= "none", size = "none")

toplot = allGO %>%
         dplyr::filter(p_adj <= 0.05 & source == "go_stab_down") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold.enrichment)))

toplot = toplot[order(toplot$GO.term), ]

p4 = ggplot(toplot,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = "Decreased stability") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, color= toplot$colour2)) 



pdf(paste0(plotdir, "EDFig.4a_GObarplots.pdf"), width = 12, height = 12)
design <- "
111333
111333
222444
222444
222444
222444
222444
222444
222444
555555
"

p1 + p2 + p3 + p4 + guide_area() + plot_layout(design=design, guides = "collect")

dev.off()


```



### ED Fig. 4b - Lpt thermal profiles

```{r}
toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx")) %>%
  filter(protein %in% c("LptA","LptB","LptC","LptD","LptE", "LptF", "LptG")) %>%
  dplyr::mutate(temperature = factor(temperature, levels = rev(sort(unique(temperature)))),
                nitroxoline_conc = as.factor(nitroxoline_conc))


p<- ggplot(toplot, aes(as.factor(nitroxoline_conc), temperature)) +
    geom_tile(aes(fill = rel_value)) +
  scale_fill_gradientn(colors = c("firebrick","khaki","darkgreen"),
                       space = "Lab", guide = "colourbar", na.value = "white", limits=c(0.2,1.65),
                       values = scales::rescale(c(0.2, 0.5,0.7, 1, 1.3,1.6, 1.65))) +
    labs(x = paste("Nitroxoline (Âµg/ml)"),y = expression("Temperature ("*~degree*C*")"), fill = "Fold change") +
    facet_wrap(~ protein, scales = "free", ncol = 4) +
    theme_minimal() +
  theme(legend.position= 'right',
        axis.text.x = element_text(size = 9),
        strip.text =  element_text(size = 12))


pdf(paste0(plotdir,"EDFig.4b_thermal_profile_envstress.pdf"), width = 7, height = 3.8)

shift_legend2 <- function(p) {
  # ...
  # to grob
  gp <- ggplotGrob(p)
  facet.panels <- grep("^panel", gp[["layout"]][["name"]])
  empty.facet.panels <- sapply(facet.panels, function(i) "zeroGrob" %in% class(gp[["grobs"]][[i]]))
  empty.facet.panels <- facet.panels[empty.facet.panels]

  # establish name of empty panels
  empty.facet.panels <- gp[["layout"]][empty.facet.panels, ]
  names <- empty.facet.panels$name

  reposition_legend(p, 'center', panel=names, plot =F)
}

grid::grid.draw(shift_legend2(p))


dev.off()


```


### ED Fig. 4c - GO enrichment hits from chemical genetics


```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.4c.xlsx")) %>%
 mutate(colour = case_when(GO.tree == "CC" ~ "dodgerblue3",
                                          GO.tree == "MF" ~ "tomato",
                                          TRUE ~ "darkgreen"),
         full_name = case_when(GO.tree == "CC" ~ "Cellular Component",
                                          GO.tree == "MF" ~ "Molecular Function",
                                          TRUE ~ "Biological Process"),
         fold.enrichment = as.numeric(fold_enrich),
         colour2 = case_when(annotation == "metal homeostasis" ~ "tomato",
                             annotation == "respiration"~ "dodgerblue3",
                             annotation == "OM" ~ "darkgreen",
                                           TRUE ~ "black"))

test = toplot %>%
              filter(effect == "decreased fitness")

p <- ggplot(test,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  facet_wrap(~effect,scales="free") +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(min(toplot$p_adj),max(toplot$p_adj))) +
  scale_size_continuous(limits = c(min(toplot$freq_foreground),max(toplot$freq_foreground))) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 60)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 11, color= test$colour2),
        legend.position="none") 


test = toplot %>%
              filter(effect == "increased fitness")

p1 <-ggplot(test,aes(y = GO.term, x = fold.enrichment)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  facet_wrap(~effect,scales="free") +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(min(toplot$p_adj),max(toplot$p_adj))) +
  scale_size_continuous(limits = c(min(toplot$freq_foreground),max(toplot$freq_foreground))) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 60)) +
  labs(size = "Number of genes", x = "Fold enrichment", y = "",colour = "FDR") +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 11, color= test$colour2),
        legend.direction = "horizontal", legend.box = "horizontal") +
  theme(legend.position="none")

design <- "
1122
11##
11##
"

pdf(paste0(plotdir, "EDFig.4c_GO_Keio.pdf"), width = 12, height = 6)
p + p1 +  plot_layout(design = design)

dev.off()

```



### ED Fig. 4d - LptD synergy

New version - 240329 using data from 220525
```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.4d.xlsx")) %>%
  mutate(concentration= as.factor(`Nitroxoline conc (Âµg/ml)`))

colours = data.frame("strain" = unique(toplot$strain), "values" = rep("darkgrey", length(toplot$strain))) %>%
  dplyr::mutate(values = case_when(strain == "E. coli BW25113" ~ "black",
                                   strain == "E. coli lptD4213" ~ "turquoise4",
                                   TRUE ~ values)) %>%
  distinct()
jColors <- as.character(colours$values)
names(jColors) <- colours$strain


pdf(paste0(plotdir, "EDFig.4d_dose_response_lptD.pdf"), width = 3.2, height = 3.5)

ggplot(toplot, aes(x = concentration, y = mean_OD,color=strain, group = strain)) +
  geom_line() +
  geom_point() +
  geom_errorbar(aes(ymin = mean_OD-se_OD, ymax=mean_OD+se_OD),width = 0.15) +
  labs(x = "Nitroxoline (Âµg/ml)", y = "normalised OD at 9h", colour = " ") +
  scale_color_manual(values = c("darkgrey", "turquoise4"), labels = c("wt", expression(italic("lptD4213")))) +
        theme_light() +
  theme(legend.position = "bottom")


dev.off()

```



### ED Fig. 4f - iron

```{r}
toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx"))

a <- plot2dTppFcHeatmap(
  df = toplot, name = "Fur",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = toplot, name = "IscR",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = toplot, name = "EntA",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = toplot, name = "FepA",
  drug_name = "Nitroxoline")
e <- plot2dTppFcHeatmap(
  df = toplot, name = "SufB",
  drug_name = "Nitroxoline")


pdf(paste0(plotdir, "EDFig.4f_thermal_profile_iron.pdf"), width = 5.8, height = 4)

cowplot::plot_grid(a, b,c,d,e, nrow = 2)

dev.off()

```

## ED Fig. 5

### ED Fig. 5a - volcano R strains
```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.5a.xlsx"))

colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))

jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation

toplot = toplot %>% filter(Condition_treatment=="nodrug")

pdf(paste0(plotdir, "EDFig.5a_resistant_strains_volcano.pdf"), width = 8, height = 7.6)


tolabel = c("BepF", "NfsA", "MarA", "GabT","FrdB","PlaP","OmpC","GarR","GarD", "OmpW", "MprA", "LamB", "TolC", "OprC","EmrR",
          "AcrA", "AcrB_1", "MdtK",
          "LamB","OmpC", "OmpD", "OmpF","BepA",
          "AdeG/OqxB","AdeF/OqxA","OmpR",
          "OprC","OprD","OmpW","EmrR","EmrA", "AcrB", "EntC","OmpD","LamB","OmpC","OmpN","OmpF","OmpW","OprD","OprM","MdtK","MarA","YedS", "ScrY")

ggplot(toplot,aes(logFC, -log10(adj.P.Val))) +
  geom_point(alpha = 0.3, size = 0.7,color = "grey") +
  geom_hline(yintercept = -log10(0.01), size = 0.2) +
  geom_vline(xintercept = 1, size = 0.2, alpha =0.5) +
  geom_vline(xintercept = -1, size = 0.2, alpha =0.5) +
  ylim(c(0,30)) +
  labs(x =expression('log'[2]~'(fold change)'),y= expression('-log'[10]~'(adjusted p-value)'), color = "") +
  geom_point(data = filter(toplot, hit == 1), aes(color = cross_species_conservation), size = 0.7,alpha=0.7) +
 ggrepel::geom_text_repel(data=filter(toplot, hit == 1 & og %in% tolabel),
                            aes(label = og, color = cross_species_conservation),size= 3,max.overlaps = 50) +
  scale_color_manual(values = jColors) +
  facet_wrap(~factor(Strain_treatment, levels=c("A. baumannii 2_R1",   "A. baumannii 4_R1", "E. coli 2_R1","E. coli 2_R3","E. coli 2_R4", "K. pneumoniae 2_R1", 'K. pneumoniae 2_R2', "K. pneumoniae 3_R1", "K. pneumoniae 5_R1", "K. pneumoniae 9_R1", "K. pneumoniae 9_R2", "K. pneumoniae 10_R1")),scales = "free",labeller = label_wrap_gen(width=15),ncol=4) +
  theme_bw() +
  theme(legend.position="none")


dev.off()


```


### ED Fig. 5b - clustered MS - only significant genes, resistant strains

Source data is same as Fig. 5b (that is a small subset)
```{r}

strains = c("A. baumannii 2_R1","A. baumannii 4_R1",
            "E. coli 2_R1","E. coli 2_R3","E. coli 2_R4",
            "K. pneumoniae 10_R1","K. pneumoniae 2_R1","K. pneumoniae 2_R2","K. pneumoniae 3_R1", "K. pneumoniae 5_R1", "K. pneumoniae 9_R1", "K. pneumoniae 9_R2")


toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5b_EDFig.5b.xlsx")) %>%
  filter(hit==1 & og != "" & Strain_treatment %in% strains)

mymat = toplot %>% 
  distinct(og, Strain_treatment, logFC)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}


library(magrittr)

mat_data = mymat %>% 
  # only 9 doubles with same orthologous group annotation
  pivot_wider(names_from = Strain_treatment, values_from = logFC, values_fn = median) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og) %>%
  dplyr::select(-og) %>%
  as.matrix()


hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)


type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(Strain_treatment = type) %>%
  inner_join(toplot %>% select(Strain_treatment,species), by="Strain_treatment") %>%
    distinct() %>%
    select(-Strain_treatment),
   annotation_height = unit(4, "mm"),
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )



pdf(paste0(plotdir,"EDFig.5b_clustered_proteomics.pdf"), width=6, height=30)
rownames(mat_data) = NULL

Heatmap(mat_data,
        cluster_rows = dend1,
        top_annotation = ha,
        show_heatmap_legend = F,
        row_split = 4, column_split = 5,
        use_raster = TRUE, raster_quality = 5)
       
dev.off()


```


### ED Fig. 5c - EmrR mutplot
```{r}

# get domain annotations
proteins_acc_url <- "P0ACR9"
baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
url<-paste0(baseurl,proteins_acc_url)
prots_feat<-GET(url,accept_json())
prots_feat_red<-httr::content(prots_feat)
features_total_plot<-NULL
for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])
      features_total_plot<-rbind(features_total_plot,features_temp)
      }
  

plot_start<-0 #starts 0
plot_end<-max(features_total_plot$end)   

# read data with MIC and isolate type info
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.5c.xlsx"))

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type
    

pdf(paste0(plotdir,"EDFig.5c_EmrR_mutplot.pdf"), width=4.5, height=3.5)

ggplot(toplot,aes(location_aa_number,`MIC (Âµg/ml)`,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location_aa_number,yend=`MIC (Âµg/ml)`)) +
  geom_point(aes(size=Mutation_occurrences, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=toplot %>%
                    distinct(location_aa_number,`MIC (Âµg/ml)`, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=150, size=3) +
  geom_hline(yintercept = 0) +
  geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 89, y = 2.3, label ="HTH marR-type", size = 4,color="white",fontface="bold") +
  scale_color_manual(values=jColors, name = "Isolate type") +
   scale_shape_manual(values=c("missense" =16, "nonsense"  = 15, "frameshift/insertion" =17, "insertion IS1S"  = 8), name = "Mutation type") +
  scale_size_continuous(breaks = c(1,2,3,4), transform = "log10",range = c(1,4)) +
  coord_cartesian(clip = 'off') +
  scale_y_continuous(breaks = c(2,4,8,16,32,64),trans="log2",limits=c(2,100)) +
  labs(y= "MIC (Âµg/ml)",size="Total mutation\noccurrencies", shape = "Mutation type")+
  theme_bw() +
  theme(
        plot.title=element_blank(),
       legend.position = "top",
       legend.direction="vertical",
        axis.text = element_text(size=10),
        legend.title = element_text(size=9, face="bold"),
        legend.text = element_text(size=8),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.05,0.05,0.05,0.05, unit="cm"),
        panel.border = element_blank(), axis.line = element_line()) +
  guides(size=guide_legend(direction='horizontal', title.position="top", title.hjust = 0.5))

 dev.off()
    

```


### ED Fig. 5e - AdeL mutplot

```{r}
# get domain annotations
proteins_acc_url <- "A0A059ZJX1"
baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
url<-paste0(baseurl,proteins_acc_url)
prots_feat<-GET(url,accept_json())
prots_feat_red<-httr::content(prots_feat)
features_total_plot<-NULL
for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])
      features_total_plot<-rbind(features_total_plot,features_temp)
      }
  

plot_start<-0 #starts 0
plot_end<-max(features_total_plot$end)   

# read data with MIC and isolate type info
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.5e.xlsx"))

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type


pdf(paste0(plotdir,"adeL_mutplot.pdf"), width=4.5, height=4)

ggplot(toplot,aes(location_aa_number,`MIC (Âµg/ml)`,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location_aa_number,yend=`MIC (Âµg/ml)`)) +
  geom_point(aes(size=Mutation_occurrences, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=toplot %>%
                    distinct(location_aa_number,`MIC (Âµg/ml)`, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=10, size=3) +
  geom_hline(yintercept = 0) +
   geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
 annotate("text", x = 29, y = 2.3, label ="HTH marR-type", size = 4,color="white",fontface="bold") +
  scale_shape_manual(values=c("frameshift" =17, "nonsense"  = 15, "missense" =16, "insertion ISAba125"  = 8), name = "Mutation type") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,343), limits=c(0,343))+
  scale_y_continuous(breaks = c(2,4,8,16,32),trans="log2",limits=c(2,40)) +
  labs(y= "MIC (Âµg/mL)",size="Mutation occurrencies", title = "AdeL")+
  theme_bw() +
  theme(plot.title=element_blank(),
       legend.position = "top",
       legend.direction="vertical",
        axis.text = element_text(size=10),
        legend.title = element_text(size=9, face="bold"),
        legend.text = element_text(size=8),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.05,0.05,0.05,0.05, unit="cm"),
        panel.border = element_blank(), axis.line = element_line()) +
  guides(size=guide_legend(direction='horizontal', title.position="top", title.hjust = 0.5))

dev.off()
    


```


### ED Fig. 5f - AcrR mutplot

```{r}

# get domain annotations
proteins_acc_url <- "A0A245ZZS0"
baseurl<-"https://www.ebi.ac.uk/proteins/api/features?offset=0&size=100&accession="
url<-paste0(baseurl,proteins_acc_url)
prots_feat<-GET(url,accept_json())
prots_feat_red<-httr::content(prots_feat)
features_total_plot<-NULL
for(i in 1:length(prots_feat_red)){ 
      features_temp<-drawProteins::extract_feat_acc(prots_feat_red[[i]])
      features_total_plot<-rbind(features_total_plot,features_temp)
      }
  

plot_start<-0 #starts 0
plot_end<-max(features_total_plot$end)   

# read data with MIC and isolate type info
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig.5f.xlsx"))

colours = data.frame("Isolate_Type" = c("clinical isolate","experimentally evolved","unidentified"),
                     "values" = c("#D55E00", "black","darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$Isolate_Type


pdf(paste0(plotdir,"EDFig.5f_acrR_mutplot.pdf"),width=4.5, height=4)

ggplot(toplot,aes(location_aa_number,`MIC (Âµg/ml)`,color=Isolate_type,label=Protein_Change))+
  geom_segment(aes(y=0,xend=location_aa_number,yend=`MIC (Âµg/ml)`)) +
  geom_point(aes(size=Mutation_occurrences, shape=Mutation_Type, color=Isolate_type)) +
  geom_text_repel(data=toplot %>%
                    distinct(location_aa_number,`MIC (Âµg/ml)`, Protein_Change,Isolate_type),
                  aes(label=Protein_Change,color=Isolate_type),fontface="bold",nudge_y=0.2, max.overlaps=10, size=3) +
  geom_hline(yintercept = 0) +
   geom_rect(data=features_total_plot[features_total_plot$type=="DOMAIN",],
            mapping=aes(xmin=begin,xmax=end,ymin=2,ymax=2.5),color="black",fill="black",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 59, y = 2.25, label ="HTH tetR-type", size = 4,color="white",fontface="bold") +
  geom_rect(data=features_total_plot[features_total_plot$type=="DNA_BIND",],
            mapping=aes(xmin=begin,xmax=end,ymin=2.5,ymax=3),color="darkblue",fill="darkblue",inherit.aes=F, alpha=0.5) +
  annotate("text", x = 59, y = 2.75, label ="H-T-H motif", size = 4,color="white",fontface="bold") +
  scale_shape_manual(values=c("frameshift" =17, "nonsense"  = 15, "missense" =16, "Insertion IsAba1"  = 23,"Insertion IsAba36"= 3), name = "Mutation type") +
  scale_color_manual(values=jColors, name = "Isolate type") +
  scale_size_continuous(breaks = c(1,2), limits=c(1,2)) +
  coord_cartesian(clip = 'off') +
  scale_x_continuous(name="aa number", breaks=c(0,223), limits=c(0,223))+
  scale_y_continuous(breaks = c(2,4,8,16,32),trans="log2",limits=c(2,38)) +
  labs(y= "MIC (Âµg/ml)",size="Mutation occurrencies", title = "AcrR")+
  theme_bw() +
  theme(plot.title=element_blank(),
       legend.position = "top",
       legend.direction="vertical",
        axis.text = element_text(size=10),
        legend.title = element_text(size=9, face="bold"),
        legend.text = element_text(size=8),
        legend.box.margin=margin(1,1,1,1),
        legend.margin = margin(0.05,0.05,0.05,0.05, unit="cm"),
        panel.border = element_blank(), axis.line = element_line()) +
  guides(size=guide_legend(direction='horizontal', title.position="top", title.hjust = 0.5))
dev.off()


```



### ED 5g - EPI


```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5e_EDFig.5g.xlsx")) %>%
  mutate(newspec = paste0("italic('", species, " ')~'", isolate, "'"), 
         facet = factor(newspec, levels = c("italic('A. baumannii ')~'1'","italic('A. baumannii ')~'1_R1'",
                                            "italic('A. baumannii ')~'5'","italic('A. baumannii ')~'5_R1'",
                                            "italic('E. coli ')~'1'", "italic('E. coli ')~'1_R4'")),
         row= case_when(species == "A. baumannii" ~ 1,
                        species == "E. coli" ~ 2),
         resist = case_when(isolate %in% c("1","2","4","5") ~ "sensitive",
                             TRUE ~ "resistant")) %>%
  # only plot selected strains - rest in main Fig
  drop_na(facet)


p <- ggerrorplot(toplot %>% filter(row==1), x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
          order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"),
  add = "se") +
 #  stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, nrow=1, labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "",y="MIC (Âµg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=FALSE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=4) +
   theme(axis.text.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#3e83c4"),
         strip.text=element_text(face="bold",colour="white"))

legend_grob <-cowplot::get_legend(p)

p <- ggerrorplot(toplot %>% filter(row==1), x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
          order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"),
  add = "se") +
  # stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, nrow=1, labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==1 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  scale_y_continuous(trans="log2") +
  labs(color = "",y="MIC (Âµg/ml)", x= "") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=FALSE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=4) +
   theme(axis.text.x = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#3e83c4"),
         strip.text=element_text(face="bold",colour="white"))


p2 <- ggerrorplot(toplot %>% filter(row==2), x = "condition", y = "`MIC(Âµg/ml)`", error.plot="errorbar",
          order = c("no inhibitor","+ 100 Âµg/ml NMP", "+ 25 Âµg/ml PAÎ²N"),
  add = "se") +
 #  stat_summary(geom = "point",shape = 95,size = 12,col = "black",fun.y = "mean") +
  facet_wrap(~ newspec, nrow=1, labeller = label_parsed) +
   geom_point(aes(colour = as.factor(condition)),alpha = 0.7,size =1.5,position = position_jitterdodge(1.5, dodge.width = .5)) +
  geom_rect(data=filter(toplot,row==2 & resist=="resistant"),aes(fill = resist),xmin = -Inf,xmax = Inf,ymin = -Inf,ymax = Inf,alpha = 0.01, fill="grey") +
  scale_color_manual(values=c("darkgrey", "#F39B7FB2", "#DC0000B2"),labels = c("untreated", "+ 100 Âµg/ml NMP",paste0("+ 25 Âµg/ml PA\u03B2N"))) +
  labs(color = "", y = "MIC (Âµg/ml)", x= "") +
  scale_y_continuous(trans="log2") +
  stat_compare_means(label = "p.signif", method = "t.test",paired=FALSE,
                     ref.group = "no inhibitor", size = 4, label.y.npc = c("top","top"),label.y=5) +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks.x = element_blank(),
         legend.position="none",
         strip.background =element_rect(fill="#fdc113"),
         strip.text=element_text(face="bold",colour="white"))


library("latex2exp")


cairo_pdf(paste0(plotdir,"EDFig.5g_effluxpump.pdf"), width =7,height=4)
 
layout <-"AAAA
          BBDD
          "
wrap_plots(A=p, B=p2,D=legend_grob, design = layout)

dev.off()


```




# Supplementary File

### Fig. 1 - MTT toxicity

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.S1.xlsx"))


toplot$condition = factor(toplot$condition, levels = c("0.5", "1", "2","5","10","20","50","0.1%\nTriton", "RPMI"), ordered = TRUE)


pdf(paste0(plotdir,"Fig.S1_MTT_assay.pdf"),width =4,height=2)
ggplot(toplot, aes(x=condition,y=mean_viability, fill=drug_ctrl)) +
  geom_bar(stat="identity") +
  geom_hline(yintercept=100, linetype="dashed", linewidth=0.3) +
  scale_y_continuous(breaks=c(0,20,40,60,80,100)) +
  scale_fill_manual(values=c("#DD4444", "#555599", "#66BBBB"), labels=c("nitroxoline","rpmi","triton")) +
  labs(x="",y="Cell Viability (%)") +
  geom_errorbar(aes(ymin=lowersd_viab,ymax=uppersd_viab), width=0.1) +
  theme_bw() +
  theme(legend.position="none")
dev.off()


```


## Fig. 2 - checkerboard replicates

```{r}

allCBs = readxl::read_xlsx(paste0(sourcedatadir,"source_2a_EDFig.3a.xlsx"))


antags= c("Aztreonam","Mecillinam","Temocillin", "Cefepime","Imipenem", "Meropenem","Ertapenem","Thiolutin", "Amikacin","Gentamicin", "Tobramycin","Trimethoprim","Tetracycline")
synerg = c("Vancomycin", "Colistin", "Azithromycin",  "Novobiocin", "Rifampicin")
neutrals = setdiff(unique(allCBs$horizontal_drug), c(antags,synerg))


p <- ggplot(allCBs %>% filter(horizontal_drug %in% antags), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "#d9a520"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (Âµg/ml)", y = "Nitroxoline (Âµg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        axis.title.y = element_blank(),
        panel.spacing = unit(2, "lines"))

p1 <- ggplot(allCBs %>% filter(horizontal_drug %in% synerg), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "turquoise4"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (Âµg/ml)", y = "Nitroxoline (Âµg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.text = element_text(size= 10),
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
       axis.text.y = element_text(size = 10),
        axis.title.y = element_blank(),
        panel.spacing = unit(2, "lines"),
        legend.position = "none")

p2 <- ggplot(allCBs %>% filter(horizontal_drug %in% neutrals), aes (x = factor(horizontal_concentration), factor(nitroxoline_concentration))) +
    geom_tile(aes(fill = norm_fitness), colour = "white", size = 0.25) +   
scale_fill_gradientn(colors = c("#ffffff", "#ffffff", "grey60"),
                     values = scales::rescale(c(0,0.1, max(allCBs$norm_fitness)))) +
    theme_classic() + 
	facet_wrap(horizontal_drug ~ replicate, scales = "free", ncol = 6) +
  labs(x ="Combined drug (Âµg/ml)", y = "Nitroxoline (Âµg/ml)", fill="Normalised\nfitness") +
     theme(strip.placement = "outside",
        strip.background = element_blank(), 
        axis.text.x = element_text(angle = 60, hjust = 0.5,vjust = 0.5),
        axis.text.y = element_text(size = 10),
        panel.spacing = unit(2, "lines"),
        legend.position="right")

pdf(paste0(plotdir, "SuppFig.2_CB_heatmaps_antag_repssep.pdf"),  width=10.5, height=11)
shift_legend2 <- function(p) {
  # to grob
  gp <- ggplotGrob(p)
  facet.panels <- grep("^panel", gp[["layout"]][["name"]])
  empty.facet.panels <- sapply(facet.panels, function(i) "zeroGrob" %in% class(gp[["grobs"]][[i]]))
  empty.facet.panels <- facet.panels[empty.facet.panels]

  # establish name of empty panels
  empty.facet.panels <- gp[["layout"]][empty.facet.panels, ]
  names <- empty.facet.panels$name

  reposition_legend(p, 'center', panel=names, plot =F)
}

grid::grid.draw(shift_legend2(p))
dev.off()

pdf(paste0(plotdir, "SuppFig.2_CB_heatmaps_synergies_repssep.pdf"),  width=10.5, height=4.4)

p1

dev.off()


pdf(paste0(plotdir, "SuppFig.2_CB_heatmaps_neutrals_repssep.pdf"),  width=10.5, height=11)
grid::grid.draw(shift_legend2(p2))
dev.off()



```




## Fig. 4 - full growth curves Kpn resensitization
```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.S4.xlsx"))



toplot$facet = factor(toplot$strain, levels = c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1","E. coli J53 pKP2442_mcr-1",
                                                "E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)",
                                                "E. coli BW25113","E. coli BW25113 pGDP1_mcr1",
                                                "K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)", "K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)","K. pneumoniae pKP2442_mcr-1",
                                                "K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO","K. pneumoniae PRZ pTOPO_mcr-1", "K. pneumoniae PRZ pKP2442_mcr-1"))

toplot = toplot %>%
  mutate(row= case_when(strain %in% c("E. coli BW25113","E. coli BW25113 pGDP1_mcr1") ~ 1,
                        strain %in% c("E. coli MG1655","E. coli MG1655 pmrA (G53E)","E. coli MG1655 pmrB (V88E)") ~ 2,
                        strain %in% c("E. coli J53","E. coli J53 pTOPO", "E. coli J53 pTOPO_mcr-1", "E. coli J53 pKP2442_mcr-1") ~ 3,
                        strain %in% c("K. pneumoniae PRZ" , "K. pneumoniae PRZ pTOPO_mcr-1","K. pneumoniae PRZ pKP2442_mcr-1") ~ 5,
                        strain %in% c("K. pneumoniae CT-S", "K. pneumoniae CT-R3 pmrB (P95L)") ~ 6,
                        strain %in% c("K. pneumoniae LS107", "K. pneumoniae LS53 mgrB (IS1R)") ~ 7,
                        strain %in% c("K. pneumoniae pKP2442_mcr-1") ~ 8,
                        strain %in% c("K. pneumoniae PRZ pTOPO") ~ 9),
         resist = case_when(strain %in% c("E. coli BW25113","E. coli J53","E. coli MG1655",
                                          "K. pneumoniae CT-S","K. pneumoniae LS107","K. pneumoniae PRZ") ~ "sensitive",
                            TRUE ~ "resistant"))


fixtps = function(x) {
  x$condConc = interaction(x$condition, x$Concentration)
  x$Time_points = ave(x$Time, x$condConc, FUN = seq_along)
  return(x)
}
toplot = fixtps(toplot)


toplot = toplot %>%
  #decided not to include empty vector control in main fig, only supp 
  mutate(strain = gsub(strain, pattern  =" ",replacement = "~"),
        strain = gsub(strain, pattern  ="E.~coli",replacement = "italic('E. coli')"),
        strain = gsub(strain, pattern  ="K.~pneumoniae",replacement = "italic('K. pneumoniae')")) 


toplot = toplot %>%
  mutate(strain = case_when(strain =="italic('E. coli')~BW25113~pGDP1_mcr1" ~ "~ atop(paste(italic('E. coli'), ' BW25113'),paste('pGDP1_', italic('mcr-1')))",  
    strain =="italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
                            strain =="italic('E. coli')~MG1655~pmrA~(G53E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrA'),'(G53E)'))",
                            strain == "italic('E. coli')~MG1655~pmrB~(V88E)" ~ "~ atop(paste(italic('E. coli'), ' MG1655'),paste(italic('pmrB'),'(V88E)'))",
    strain == "italic('E. coli')~J53~pTOPO_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pTOPO_', italic('mcr-1')))",
    strain == "italic('E. coli')~J53~pKP2442_mcr-1" ~ "~ atop(paste(italic('E. coli'), ' J53'),paste('pKP2442_', italic('mcr-1')))",
    TRUE ~ strain))

toplot = toplot %>%
  dplyr::group_by(strain,Time,nitroxoline, Concentration, source) %>%
  dplyr::mutate(mean = mean(OD_no1st),
                SE = sqrt(var(OD_no1st)/length(OD_no1st)))



pdf(paste0(plotdir, "Fig.S4_colistin_resensitiz_growth_curves.pdf"), width=12, height=15)

ggplot(toplot%>%
         mutate(strainrep=interaction(strain,Replicate,nitroxoline,source)),
       aes(x = Time, y = mean,color = factor(nitroxoline), group=strainrep)) +
        geom_line(aes(group=strainrep)) +
  geom_ribbon(aes(ymin = mean - SE, ymax = mean + SE, fill = factor(nitroxoline)),alpha = 0.5) +
  scale_color_manual(labels = c("+ NX 0.75 Âµg/ml", "- NX"), values = c("turquoise4","darkgrey")) +
  scale_fill_manual(labels = c("+ NX 0.75 Âµg/ml", "- NX"), values = c("turquoise4","darkgrey")) +
      facet_grid(row + strain ~ Concentration, labeller = label_parsed) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]),color = "") +
  theme_bw() +
  guides(fill="none")+
  theme(legend.position="right")

dev.off()


```

## Fig. 5 - full growth curves LptD


```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.S5.xlsx"))

pdf(paste0(plotdir,"FigS5_fullgrowthcurves_lptD.pdf"),width = 10, height=2)
ggplot(toplot%>%
         mutate(strainrep=interaction(strain,Replicate)),
       aes(x = Time_h, y = mean,color = factor(strain), group=strain)) +
        geom_line(aes(group=strain)) +
  geom_ribbon(aes(ymin = mean - SE, ymax = mean + SE, fill = factor(strain)),alpha = 0.5) +
  scale_color_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
  scale_fill_manual(labels = c( "E. coli BW25113","E. coli lptD4213"), values = c("darkgrey", "turquoise4")) +
      facet_grid( ~ Concentration) +
      labs(x = "Time (h)", y = expression("OD"["595 nm"]),color = "",fill="") +
  theme_bw() +
  guides(color="none")+
  theme(legend.position="right")

dev.off()

```


## Fig. 6 - resp chain thermal profiles

```{r}
toplot =readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.4abc_EDFig4bf.xlsx"))

a <- plot2dTppFcHeatmap(
  df = toplot, name = "NuoB",
  drug_name = "Nitroxoline")
b <- plot2dTppFcHeatmap(
  df = toplot, name = "NuoF",
  drug_name = "Nitroxoline")
c <- plot2dTppFcHeatmap(
  df = toplot, name = "NuoI",
  drug_name = "Nitroxoline")
d <- plot2dTppFcHeatmap(
  df = toplot, name = "NarH",
  drug_name = "Nitroxoline")
e <- plot2dTppFcHeatmap(
  df = toplot, name = "NarG",
  drug_name = "Nitroxoline")
f <- plot2dTppFcHeatmap(
  df = toplot, name = "SdhA",
  drug_name = "Nitroxoline")
g <- plot2dTppFcHeatmap(
  df = toplot, name = "SdhB",
  drug_name = "Nitroxoline")
h <- plot2dTppFcHeatmap(
  df = toplot, name = "FdnG",
  drug_name = "Nitroxoline")
i <- plot2dTppFcHeatmap(
  df = toplot, name = "FdnH",
  drug_name = "Nitroxoline")
j <- plot2dTppFcHeatmap(
  df = toplot, name = "FdoG",
  drug_name = "Nitroxoline")
k <- plot2dTppFcHeatmap(
  df = toplot, name = "FdoH",
  drug_name = "Nitroxoline")



pdf(paste0(plotdir, "Fig.S6_thermal_profile_respiratory_chain.pdf"), width = 12, height = 7)

cowplot::plot_grid(a, b,c,d,e,f,g,h,i,j,k, ncol = 4)

dev.off()


```


## Fig. 9 - GO resistant strains

```{r}

final = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.S9.xslx"))


toplot = filter(final, source == "abau_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))


toplot = toplot[order(toplot$GO.term), ]

p1 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 75)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))


toplot = filter(final, source == "kpn_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))
                                            

toplot = filter(final, source == "ecoli_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p3 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('E. coli')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))
  

toplot = filter(final, source == "abau_downreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p4 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 70)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))



toplot = filter(final, source == "kpn_downreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p5 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(0,0.01),n.breaks = 2) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
  labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour),
        legend.position="right")


pdf(paste0(plotdir, "Fig.S9_GO_enrich_proteomics_ResistantStrains.pdf"), width = 12, height = 15)
design <- "
4455#
4455#
4455#
44556
44556
4455#
4455#
4455#
4422#
1133#
1133#
1133#
1133#
"
p1 + p2 + p3 + p4 + p5 + guide_area() + plot_layout(design=design, guides = "collect")


dev.off()



```


## Fig. 10 - GO sensitive strains

```{r}

final = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.S10.xslx"))
  

toplot = filter(final, source == "abau_upreg") %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p1 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('A. baumannii')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))

toplot = filter(final, source == "kpn_upreg"& !(GOs %in% c("GO:0072488", "GO:0022890", "GO:0043872"))) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p2 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  guides(alpha="none", colour = "none", fill= "none", size = "none")+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour))


toplot = filter(final, source == "kpn_downreg"& !(GOs %in% c("GO:0072488", "GO:0022890", "GO:0043872"))) %>%
  dplyr::mutate(GO.term = with(.,reorder(GO.term,fold_enrich)))
toplot = toplot[order(toplot$GO.term), ]

p3 = ggplot(toplot,aes(y = GO.term, x = fold_enrich)) +
  geom_segment(aes(xend=0, yend=GO.term), color= "darkgrey") +
  geom_point(aes(size=freq_foreground, colour=p_adj)) +
  scale_colour_gradientn(colours = c("#CCCCCC","#333333"),limits = c(0,0.01),n.breaks=2) +
  scale_size_continuous(breaks = c(2,4,6,7)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45)) +
   labs(size = "Number of proteins", x = "Fold enrichment", y = "",colour = "FDR", title = expression(paste(italic('K. pneumoniae')))) +
  theme_pubr()+
  theme(axis.text.y = element_text(lineheight = 0.8, size = 9, colour = toplot$colour),
        legend.position = "bottom",
        legend.direction = "horizontal",
        legend.key.width = unit(1, 'cm')) +
  guides(size = guide_legend(keywidth = 0.2))


pdf(paste0(plotdir, "Fig.S10_GO_enrich_SensitiveStrains.pdf"), width = 12, height = 17)

design <- "
12
12
12
12
32
32
42
"

p1 + p2 + p3 + guide_area() + plot_layout(design=design, guides = "collect")

dev.off()


```


## Fig. 9a - volcano sensitive strains proteomics

```{r}

toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig5a.xlsx"))

colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))

jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation

toplot = toplot %>% filter(Condition_treatment=="Nx")


pdf(paste0(plotdir, "Fig.S11a_volcano_sensitive.pdf"), width = 10, height = 1.9)

toplot = toplot %>%
  mutate(og = case_when(og == "BpeF/AdeG/OqxB" ~ "AdeG/OqxB",
                         og == "BpeE/AdeF/OqxA/BepF" ~ "AdeF/OqxA",
                         TRUE ~ og))

tolabel = c("BepF", "NfsA", "MarA", "GabT","FrdB","PlaP","OmpC","GarR","GarD", "OmpW", "MprA", "LamB", "TolC", "OprC","EmrR",
          "AcrA", "AcrB_1", "MdtK",
          "LamB","OmpC", "OmpD", "OmpF","BepA", "PlaP",
          "AdeG/OqxB","AdeF/OqxA","OmpR",
          "OprC","OprD","OmpW","EmrR","EmrA", "AcrB", "EntC","OmpD","LamB","OmpC","OmpN","OmpF","OmpW","OprD","OprM","MdtK","MarA","YedS", "ScrY")

ggplot(toplot,aes(logFC, -log10(adj.P.Val))) +
  geom_point(alpha = 0.3, size = 0.7,color = "grey") +
  geom_hline(yintercept = -log10(0.01), size = 0.2) +
  geom_vline(xintercept = 1, size = 0.2, alpha =0.5) +
  geom_vline(xintercept = -1, size = 0.2, alpha =0.5) +
  ylim(c(0,5)) +
  labs(x =expression('log'[2]~'(fold change)'),y= expression('-log'[10]~'(adjusted p-value)'), color = "") +
  geom_point(data = filter(toplot, hit == 1), aes(color = cross_species_conservation), size = 0.7,alpha=0.7) +
 ggrepel::geom_text_repel(data=filter(toplot, hit == 1 & og %in% tolabel),
                            aes(label = og, color = cross_species_conservation),size= 3,max.overlaps = 50) +
  scale_color_manual(values = jColors) +
  facet_wrap(~factor(Strain_treatment, levels=c("A. baumannii 2", "A. baumannii 4", "E. coli 1", "E. coli 2",
                                               "K. pneumoniae 2", "K. pneumoniae 3",
                                               "K. pneumoniae 5", "K. pneumoniae 9")),scales = "free",ncol=8) +
  theme_bw() +
  theme(legend.position="none",
        axis.text=element_text(size=8),
        axis.title =element_text(size=10))


dev.off()


```


## Fig. 9b - clustered hm only hits sensitive proteomics

```{r}

strains = c("A. baumannii 2","A. baumannii 4",
            "E. coli 2", "K. pneumoniae 10","K. pneumoniae 2","K. pneumoniae 3", "K. pneumoniae 5", "K. pneumoniae 9")


toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5b_EDFig.5b.xlsx")) %>%
  filter(hit==1 & og != "" & Strain_treatment %in% strains)

mymat = toplot %>% 
  distinct(og, Strain_treatment, logFC)

is.na.data.frame <- function(x)
  do.call(cbind, lapply(x, is.na))

fix_na <- function (x) {
  x[is.na(x)] <- 0
  return(x)
}


library(magrittr)

mat_data = mymat %>% 
  # only 9 doubles with same orthologous group annotation
  pivot_wider(names_from = Strain_treatment, values_from = logFC) %>%
  fix_na() %>%
  as.data.frame() %>%
  set_rownames(.$og) %>%
  dplyr::select(-og) %>%
  as.matrix()


#  dendrogram 
library(dendsort)
library(dendextend)
hc <- hclust(as.dist(1-cor(mat_data, method="pearson")), method="complete")
hr <- dendsort(hclust(as.dist(1-cor(t(mat_data), method="pearson")), method="complete"))

dend1 <- as.dendrogram(hr)
c_group <- 10 # number of clusters
dend1 <- set(dend1, "labels_cex", 0.2)


type <- gsub("s\\d+_", "", colnames(mat_data))

ha = HeatmapAnnotation(
  df = data.frame(Strain_treatment = type) %>%
  inner_join(toplot %>% select(Strain_treatment,species), by="Strain_treatment") %>%
    distinct() %>%
    select(-Strain_treatment),
   annotation_height = unit(4, "mm"),
  col = list(species = c("E. coli" = "#FFC107","K. pneumoniae" = "#D81B60", "A. baumannii" = "#1E88E5"))
  )



pdf(paste0(plotdir,"SuppFig.9b_clustered_proteomics_onlySensitiveStrains.pdf"), width=6, height=8)

Heatmap(mat_data,
        cluster_rows = dend1,
        row_names_gp = gpar(fontsize = 10),
         top_annotation = ha,
        show_heatmap_legend = F,
        row_split = 4, column_split = 3,
        use_raster = TRUE, raster_quality = 5)

dev.off()



```

## Fig. 9c - barplot conservation

```{r}
senstrains =c("A. baumannii 2", "A. baumannii 4", "E. coli 1","E. coli 2",
              "K. pneumoniae 2", "K. pneumoniae 3", "K. pneumoniae 5", "K. pneumoniae 9")


toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_EDFig5a.xlsx")) %>%
   dplyr::mutate(resorsens = case_when(Strain_treatment %in% senstrains ~ "Sensitive strains",
                                      TRUE ~ "Resistant strains")) %>%
  filter(hit==1)%>%
  distinct(og,cross_species_conservation, species, resorsens) %>%
  dplyr::group_by(resorsens, cross_species_conservation,species) %>%
  dplyr::summarise(count = n(), .groups = "keep")

toplot = toplot %>%
 dplyr::group_by(resorsens,species) %>%
  mutate(percent = count/sum(count))


  
colours = data.frame("cross_species_conservation" = c("conserved across species", "conserved within species","strain-specific", "no orthology group annotated"),
                     "values" = c("black", "#67a9cf","#ef8a62", "darkgrey"))
jColors <- as.character(colours$values)
names(jColors) <- colours$cross_species_conservation


pdf(paste0(plotdir, "barplot_conservation_resmut.pdf"), width = 5.5, height = 2.5)

ggplot(toplot, aes(x=species,y=percent*100, fill = cross_species_conservation)) +
  geom_bar(position = position_fill(),stat = "identity") +
  geom_text(aes(label = paste0(count)),
           # stat = "count",
            colour = "white",
            position = position_fill(vjust = 0.5)) +
  scale_fill_manual(values = jColors) +
  facet_wrap (~resorsens) +
  coord_flip() +
  labs(x = "", y = paste0("fraction of total (n = ", sum(toplot$count), ")"), fill = "") +
  theme_bw() +
  theme(legend.position = "bottom",
        axis.text.y = element_text(face="italic")) +
  guides(fill = guide_legend(nrow = 2))


dev.off()


```


# SUPP TABLES

## Supp Table 6 - proteomics

```{r}
toplot = readxl::read_xlsx(paste0(sourcedatadir,"source_Fig.5b_EDFig.5b.xlsx")) %>%
  filter(hit==1)

write_xlsx(toplot,paste0(supptablesdir,"Supplementary_Table_6.xlsx"))

```


